<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Identify clusters in droplet data using topic model</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Identify clusters in droplet data using topic model</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree605b540a24de563c843aac3d235d28ec1d14eb49targetblank605b540a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/605b540a24de563c843aac3d235d28ec1d14eb49" target="_blank">605b540</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree605b540a24de563c843aac3d235d28ec1d14eb49targetblank605b540a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/605b540a24de563c843aac3d235d28ec1d14eb49" target="_blank">605b540</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  analysis/clustering-droplet.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/clusters_droplet.Rmd</code>) and HTML (<code>docs/clusters_droplet.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/605b540a24de563c843aac3d235d28ec1d14eb49/analysis/clusters_droplet.Rmd" target="_blank">605b540</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/db6135c17587e1bb862ac005452ee934662b0c7b/docs/clusters_droplet.html" target="_blank">db6135c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
<td>
Added B+C cluster to clustering of droplet data, and added plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/69d1f0acfefba68a060ac2cf92be177a66c3e5e2/analysis/clusters_droplet.Rmd" target="_blank">69d1f0a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e2a807184032d94632431f14de8aeda1dd51637a/analysis/clusters_droplet.Rmd" target="_blank">e2a8071</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
Saved new clustering-droplet.rds.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b5e1a7e1b9f1357e79255b0d1fe780e995036c03/docs/clusters_droplet.html" target="_blank">b5e1a7e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
Fixed up Structure plot in clusters_droplet analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b7d1acccebb866981644a966589476316006edd0/analysis/clusters_droplet.Rmd" target="_blank">b7d1acc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/417202458ee4055297b95a2ea3994a867c13866f/docs/clusters_droplet.html" target="_blank">4172024</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
Identified H cluster in droplet data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/decefd48950eb8cd4a6b8d9a1e21f9876e54afaf/analysis/clusters_droplet.Rmd" target="_blank">decefd4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/5361fdfce874122596ccf2e97e7c00b8379f3e0b/docs/clusters_droplet.html" target="_blank">5361fdf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
Adjusted the plots in clusters_droplet analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7830b35c5dfc1d2d27060e5c2f85c7abaf784745/analysis/clusters_droplet.Rmd" target="_blank">7830b35</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/311b4e8f0de36c046e249e00f893c8a2b7721f81/docs/clusters_droplet.html" target="_blank">311b4e8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
Made a few minor improvements to the clusters_droplet analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ba90d80ea27ae477a88f163d39168e4899633622/analysis/clusters_droplet.Rmd" target="_blank">ba90d80</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b1cb82e4b0ce2e1db51ebb5f3a34a91178fe9dc3/docs/clusters_droplet.html" target="_blank">b1cb82e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
Added clustering from PCA plots to clusters_droplet analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/81e7fafbb5d3a2a5591bd61bab71ad38c0a96f2b/analysis/clusters_droplet.Rmd" target="_blank">81e7faf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
workflowr::wflow_publish(“clusters_droplet.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c8dd3afae9fa35c6d78ee6aff137c0034b6686db/analysis/clusters_droplet.Rmd" target="_blank">c8dd3af</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Implemented basic_pca_plot; improved labeled_pca_plot function.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform PCA on the topic proportions to identify clusters in the droplet data.</p>
<p>Load the packages used in the analysis below, as well as additional functions that we will use to generate some of the plots.</p>
<pre class="r"><code>library(dplyr)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/plots.R&quot;)</code></pre>
<div id="load-data-and-results" class="section level2">
<h2>Load data and results</h2>
<p>Load the droplet data.</p>
<pre class="r"><code>load(&quot;../data/droplet.RData&quot;)</code></pre>
<p>Load the <span class="math inline">\(k = 7\)</span> Poisson NMF model fit.</p>
<pre class="r"><code>fit &lt;- readRDS(&quot;../output/droplet/rds/fit-droplet-scd-ex-k=7.rds&quot;)$fit</code></pre>
</div>
<div id="identify-clusters-from-principal-components" class="section level2">
<h2>Identify clusters from principal components</h2>
<p>To identify clusters, we begin by plotting PCs computed from the topic proportions. (Note that only 6 PCs are needed for 7 topics.)</p>
<pre class="r"><code>p1 &lt;- basic_pca_plot(fit,1:2)
p2 &lt;- basic_pca_plot(fit,3:4)
p3 &lt;- basic_pca_plot(fit,5:6)
plot_grid(p1,p2,p3,nrow = 1,ncol = 3)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/clustering-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-1-1">
Past versions of clustering-1-1.png
</button>
</p>
<div id="fig-clustering-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b1cb82e4b0ce2e1db51ebb5f3a34a91178fe9dc3/docs/figure/clusters_droplet.Rmd/clustering-1-1.png" target="_blank">b1cb82e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Some of the structure is more evident from “hexbin” plots showing the density of the points.</p>
<pre class="r"><code>bins &lt;- c(0,1,5,10,100,Inf)
p4 &lt;- pca_hexbin_plot(fit,1:2,bins = bins) + guides(fill = &quot;none&quot;)
p5 &lt;- pca_hexbin_plot(fit,3:4,bins = bins) + guides(fill = &quot;none&quot;)
p6 &lt;- pca_hexbin_plot(fit,5:6,bins = bins) + guides(fill = &quot;none&quot;)
plot_grid(p4,p5,p6,nrow = 1,ncol = 3)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/clustering-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-2-1">
Past versions of clustering-2-1.png
</button>
</p>
<div id="fig-clustering-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b1cb82e4b0ce2e1db51ebb5f3a34a91178fe9dc3/docs/figure/clusters_droplet.Rmd/clustering-2-1.png" target="_blank">b1cb82e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From these PCA plots, we define 4 clusters, labeled A, Cil, G and T+N. (The reasoning behind these labels will become clear later.) Points that do not fit in any of these clusters are assigned to a “background cluster”, labeled U.</p>
<pre class="r"><code>pca &lt;- prcomp(poisson2multinom(fit)$L)$x
x   &lt;- rep(&quot;U&quot;,nrow(pca))
pc1 &lt;- pca[,1]
pc2 &lt;- pca[,2]
pc6 &lt;- pca[,6]
x[pc2 &gt; -0.15] &lt;- &quot;A&quot;
x[pc1 &gt; 0.3 &amp; pc2 &lt; -0.75] &lt;- &quot;Cil&quot;
x[pc1 &lt;= 0.3 &amp; pc2 &gt;= -0.75 &amp; pc2 &lt; -0.4] &lt;- &quot;T+N&quot;
x[pc6 &lt; -0.05] &lt;- &quot;G&quot;</code></pre>
<p>There is additional substructure in cluster A, which is more apparent in the projection onto the top 2 PCs computed from cluster A only.</p>
<pre class="r"><code>rows &lt;- which(x == &quot;A&quot;)
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p7   &lt;- basic_pca_plot(fit2,1:2)
p8   &lt;- pca_hexbin_plot(fit2,1:2,bins = bins) + guides(fill = &quot;none&quot;)
plot_grid(p7,p8)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/clustering-4-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-4-1">
Past versions of clustering-4-1.png
</button>
</p>
<div id="fig-clustering-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5361fdfce874122596ccf2e97e7c00b8379f3e0b/docs/figure/clusters_droplet.Rmd/clustering-4-1.png" target="_blank">5361fdf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/311b4e8f0de36c046e249e00f893c8a2b7721f81/docs/figure/clusters_droplet.Rmd/clustering-4-1.png" target="_blank">311b4e8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b1cb82e4b0ce2e1db51ebb5f3a34a91178fe9dc3/docs/figure/clusters_droplet.Rmd/clustering-4-1.png" target="_blank">b1cb82e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The variation in PCs 1 and 2 is mostly produced by topics 2, 4 and 5.</p>
<pre class="r"><code>p9 &lt;- pca_plot(fit2,pcs = 1:2,k = c(2,4,5))
print(p9)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/clustering-5-1.png" width="660" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-5-1">
Past versions of clustering-5-1.png
</button>
</p>
<div id="fig-clustering-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/db6135c17587e1bb862ac005452ee934662b0c7b/docs/figure/clusters_droplet.Rmd/clustering-5-1.png" target="_blank">db6135c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b5e1a7e1b9f1357e79255b0d1fe780e995036c03/docs/figure/clusters_droplet.Rmd/clustering-5-1.png" target="_blank">b5e1a7e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/417202458ee4055297b95a2ea3994a867c13866f/docs/figure/clusters_droplet.Rmd/clustering-5-1.png" target="_blank">4172024</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Topic 4 in particular corresponds closely to expression of <em>Krt13</em> which was identified as being uniquely expressed by transitional “hillock” cells.</p>
<pre class="r"><code>pca  &lt;- prcomp(fit2$L)$x
pc1  &lt;- pca[,1]
pc2  &lt;- pca[,2]
pdat &lt;- data.frame(PC1   = pc1,
                   PC2   = pc2,
                   Krt13 = log10(counts[rows,&quot;Krt13&quot;]))
p10 &lt;- ggplot(pdat,aes(x = PC1,y = PC2,fill = Krt13)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 1.25) +
  scale_fill_gradientn(colors = c(&quot;skyblue&quot;,&quot;gold&quot;,&quot;darkorange&quot;,&quot;magenta&quot;),
                        na.value = &quot;lightskyblue&quot;) +
  theme_cowplot(font_size = 10) +
  labs(fill = &quot;log10 expression level&quot;)
print(p10)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/clustering-6-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-6-1">
Past versions of clustering-6-1.png
</button>
</p>
<div id="fig-clustering-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/db6135c17587e1bb862ac005452ee934662b0c7b/docs/figure/clusters_droplet.Rmd/clustering-6-1.png" target="_blank">db6135c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/417202458ee4055297b95a2ea3994a867c13866f/docs/figure/clusters_droplet.Rmd/clustering-6-1.png" target="_blank">4172024</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5361fdfce874122596ccf2e97e7c00b8379f3e0b/docs/figure/clusters_droplet.Rmd/clustering-6-1.png" target="_blank">5361fdf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/311b4e8f0de36c046e249e00f893c8a2b7721f81/docs/figure/clusters_droplet.Rmd/clustering-6-1.png" target="_blank">311b4e8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We label the three more-or-less distinct subclusters as B, C and H, and assign the remaining “in between” data points to a new “background cluster”, B+C.</p>
<pre class="r"><code>y   &lt;- rep(&quot;B+C&quot;,nrow(pca))
y[pc1 &lt; 0.1] &lt;- &quot;B&quot;
y[pc1 &gt; 0.4 &amp; pc2 &lt; 0.45] &lt;- &quot;C&quot;
y[pc2 &gt; 0.55] &lt;- &quot;H&quot;
x[rows] &lt;- y</code></pre>
<p>In summary, we have subdivided the droplet data into 8 subsets.</p>
<pre class="r"><code>samples$cluster &lt;- factor(x,c(&quot;B&quot;,&quot;C&quot;,&quot;B+C&quot;,&quot;H&quot;,&quot;Cil&quot;,&quot;T+N&quot;,&quot;G&quot;,&quot;U&quot;))
cluster_colors &lt;- c(&quot;royalblue&quot;,&quot;slategray&quot;,&quot;forestgreen&quot;,&quot;firebrick&quot;,&quot;gold&quot;,
                    &quot;turquoise&quot;,&quot;darkorange&quot;,&quot;gainsboro&quot;)
p11 &lt;- labeled_pca_plot(fit,1:2,x,cluster_colors,font_size = 8)
p12 &lt;- labeled_pca_plot(fit,3:4,x,cluster_colors,font_size = 8)
p13 &lt;- labeled_pca_plot(fit,5:6,x,cluster_colors,font_size = 8)
plot_grid(p11,p12,p13,nrow = 1,ncol = 3)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/clustering-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-8-1">
Past versions of clustering-8-1.png
</button>
</p>
<div id="fig-clustering-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/db6135c17587e1bb862ac005452ee934662b0c7b/docs/figure/clusters_droplet.Rmd/clustering-8-1.png" target="_blank">db6135c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The clusters identified here correspond well to the Montoro <em>et al</em> (2018) clustering, with some exceptions (e.g., we do not identify an ionocytes cluster, and the neuroendocrine and tuft cells are included in the same cluster).</p>
<pre class="r"><code>with(samples,table(tissue,cluster))
#                 cluster
# tissue              B    C  B+C    H  Cil  T+N    G    U
#   Basal          3682   16  142    5    0    0    0    0
#   Ciliated          1   13    4    0  371    5    0   31
#   Club             93 1878  411  192    0    0    2    2
#   Goblet            2   20    1    0    0    0   42    0
#   Ionocyte          9    0    1    0    0    1    1   14
#   Neuroendocrine   27    4    6    0    0   51    0    8
#   Tuft             27    5    5    0    1  111    2    7</code></pre>
</div>
<div id="structure-plot" class="section level2">
<h2>Structure plot</h2>
<p>The structure plot summarizes the topic proportions in each of these 8 subsets:</p>
<pre class="r"><code>set.seed(1)
topic_colors &lt;- c(&quot;gold&quot;,&quot;royalblue&quot;,&quot;salmon&quot;,&quot;turquoise&quot;,&quot;olivedrab&quot;,
                  &quot;firebrick&quot;,&quot;forestgreen&quot;)
topics &lt;- c(3,4,5,1,7,2,6)
rows &lt;- sort(c(sample(which(samples$cluster == &quot;B&quot;),800),
               sample(which(samples$cluster == &quot;C&quot;),800),
               which(samples$cluster == &quot;B+C&quot;),
               which(samples$cluster == &quot;H&quot;),
               which(samples$cluster == &quot;Cil&quot;),
               which(samples$cluster == &quot;T+N&quot;),
               which(samples$cluster == &quot;G&quot;),
               which(samples$cluster == &quot;U&quot;)))
p14 &lt;- structure_plot(select(poisson2multinom(fit),loadings = rows),
                      grouping = samples[rows,&quot;cluster&quot;],
                      topics = topics,colors = topic_colors[topics],
                      perplexity = c(70,70,30,30,50,30,12,18),
                      n = Inf,gap = 30,num_threads = 4,verbose = FALSE)
print(p14)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/structure-plot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-1">
Past versions of structure-plot-1.png
</button>
</p>
<div id="fig-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/db6135c17587e1bb862ac005452ee934662b0c7b/docs/figure/clusters_droplet.Rmd/structure-plot-1.png" target="_blank">db6135c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b5e1a7e1b9f1357e79255b0d1fe780e995036c03/docs/figure/clusters_droplet.Rmd/structure-plot-1.png" target="_blank">b5e1a7e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/417202458ee4055297b95a2ea3994a867c13866f/docs/figure/clusters_droplet.Rmd/structure-plot-1.png" target="_blank">4172024</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Based on this structure plot, and the above results, we roughly subdivide the droplet data into two: (1) the Cil, T+N and G clusters that give rise to well-separated clusters, and (2) the B, C, B+C and H subsets that contain interesting substructure but much less distinct clustering. Therefore,the cluster labels B, C, B+C and H are useful as a guide but should be taken with a grain of salt as the boundaries between these clusters are somewhat arbitrary.</p>
<p>Note the distribution of the topics in cluster C that there is further substantial heterogeneity in these cells beyond what can be captured by hard clustering. In particular, there is additional continuous variation in gene expression primarily captured by topics 5 and 7:</p>
<pre class="r"><code>rows &lt;- which(is.element(x,c(&quot;B&quot;,&quot;C&quot;,&quot;H&quot;,&quot;B+C&quot;)))
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p15  &lt;- pca_plot(fit2,pcs = 2:3,k = c(4,5,7))
print(p15)</code></pre>
<p><img src="figure/clusters_droplet.Rmd/pca-plot-1.png" width="720" style="display: block; margin: auto;" /></p>
</div>
<div id="save-results" class="section level2">
<h2>Save results</h2>
<p>Save the clustering of the droplet data to an RDS file.</p>
<pre class="r"><code>saveRDS(samples,&quot;clustering-droplet.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.6
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0      ggplot2_3.3.0      fastTopics_0.3-175 dplyr_0.8.3       
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.0        Rcpp_1.0.5           lattice_0.20-38     
#  [4] tidyr_1.0.0          prettyunits_1.1.1    assertthat_0.2.1    
#  [7] zeallot_0.1.0        rprojroot_1.3-2      digest_0.6.23       
# [10] R6_2.4.1             backports_1.1.5      MatrixModels_0.4-1  
# [13] evaluate_0.14        coda_0.19-3          httr_1.4.1          
# [16] pillar_1.4.3         rlang_0.4.5          progress_1.2.2      
# [19] lazyeval_0.2.2       data.table_1.12.8    irlba_2.3.3         
# [22] SparseM_1.78         hexbin_1.28.0        whisker_0.4         
# [25] Matrix_1.2-18        rmarkdown_2.3        labeling_0.3        
# [28] Rtsne_0.15           stringr_1.4.0        htmlwidgets_1.5.1   
# [31] munsell_0.5.0        compiler_3.6.2       httpuv_1.5.2        
# [34] xfun_0.11            pkgconfig_2.0.3      mcmc_0.9-6          
# [37] htmltools_0.4.0      tidyselect_0.2.5     tibble_2.1.3        
# [40] workflowr_1.6.2.9000 quadprog_1.5-8       viridisLite_0.3.0   
# [43] crayon_1.3.4         withr_2.1.2          later_1.0.0         
# [46] MASS_7.3-51.4        grid_3.6.2           jsonlite_1.6        
# [49] gtable_0.3.0         lifecycle_0.1.0      git2r_0.26.1        
# [52] magrittr_1.5         scales_1.1.0         RcppParallel_4.4.2  
# [55] stringi_1.4.3        farver_2.0.1         fs_1.3.1            
# [58] promises_1.1.0       vctrs_0.2.1          tools_3.6.2         
# [61] glue_1.3.1           purrr_0.3.3          hms_0.5.2           
# [64] yaml_2.2.0           colorspace_1.4-1     plotly_4.9.2        
# [67] knitr_1.26           quantreg_5.54        MCMCpack_1.4-5</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
