<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Identify clusters in pulse-seq data using topic model</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Identify clusters in pulse-seq data using topic model</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree11192fcabef6dda119d09a09d52a29632f9eb4fdtargetblank11192fca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/11192fcabef6dda119d09a09d52a29632f9eb4fd" target="_blank">11192fc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree11192fcabef6dda119d09a09d52a29632f9eb4fdtargetblank11192fca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/11192fcabef6dda119d09a09d52a29632f9eb4fd" target="_blank">11192fc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  analysis/clustering-pulseseq.rds

Unstaged changes:
    Modified:   analysis/plots_tracheal_epithelium.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/clusters_pulseseq.Rmd</code>) and HTML (<code>docs/clusters_pulseseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/11192fcabef6dda119d09a09d52a29632f9eb4fd/analysis/clusters_pulseseq.Rmd" target="_blank">11192fc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7d78e71460e4f6a38c095b0d93d90d602d3f1d50/docs/clusters_pulseseq.html" target="_blank">7d78e71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
<td>
Resized a figure in clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a853b6048bb2ee7f9e1c62c6bc832088258d0b30/analysis/clusters_pulseseq.Rmd" target="_blank">a853b60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a67196abd388ee8c7e9ae4693bc40ed0894546e1/docs/clusters_pulseseq.html" target="_blank">a67196a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
<td>
Use pca_plot_with_counts in clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c2a357c1247b6ac924b5d351378c76acd3eb87e6/analysis/clusters_pulseseq.Rmd" target="_blank">c2a357c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/310ef96bc1c85e8a7cff055b0d0741e73d399a05/docs/clusters_pulseseq.html" target="_blank">310ef96</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
Fixed up figure dimensions in clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/eee6fed5a12ea294471f4f5bd9c44693442ceb36/analysis/clusters_pulseseq.Rmd" target="_blank">eee6fed</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/56d99a39b9cd2d6a09e725a08b714511e726cdc5/docs/clusters_pulseseq.html" target="_blank">56d99a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
Added Structure plot and PCA plot to clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ec4a70acf23792b039d388e44a6e8f0a0130f6e6/analysis/clusters_pulseseq.Rmd" target="_blank">ec4a70a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/6e4ccddaac9c949b976897c3787fc486d0405848/docs/clusters_pulseseq.html" target="_blank">6e4ccdd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
Made a few minor improvements to the clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/62fef5a31dfe22d31f12807031f5eed93ccfc0ec/analysis/clusters_pulseseq.Rmd" target="_blank">62fef5a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5af46f1322d6b97d127e70e4cf5e9a9060c77126/analysis/clusters_pulseseq.Rmd" target="_blank">5af46f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
Working on Structure plot for droplet data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a7b9cbe7890f456a0001e9a4423022a4f1c152f/analysis/clusters_pulseseq.Rmd" target="_blank">6a7b9cb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
Minor edit to clusters_pulseseq.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/04e1761611f3c352a77a9aaa99708daabb76042d/analysis/clusters_pulseseq.Rmd" target="_blank">04e1761</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
A couple minor edits to the R Markdown.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7489099fc83d655afc9690aedab75fa34e22d8e7/docs/clusters_pulseseq.html" target="_blank">7489099</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Re-built clusters_pulseseq page after merge conflict.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/942486b7eb0536b69a20ae4e4502c20ba58d235e/docs/clusters_pulseseq.html" target="_blank">942486b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Fixing merge issue.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/d052ec876b55baac46a0adce89aba7ccb64312c9/docs/clusters_pulseseq.html" target="_blank">d052ec8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/65f104e6f0eea164e0e5b11a0464720c88779997/docs/clusters_pulseseq.html" target="_blank">65f104e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Adjusted figure dimensions in clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/60fd1aedb06003bc4c30c267e7d1d8f5a37155db/analysis/clusters_pulseseq.Rmd" target="_blank">60fd1ae</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/775fb9104857c2be91d06aba90f45839945f1225/docs/clusters_pulseseq.html" target="_blank">775fb91</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Added step to save results in clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9cea9db035a5373d687e833e563834f49988f85e/analysis/clusters_pulseseq.Rmd" target="_blank">9cea9db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e62fb432fd8df5022e6e82f83e556085e0f197e7/docs/clusters_pulseseq.html" target="_blank">e62fb43</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Refined clustering of pulseseq data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ded741c00a8bc27526a768bb120f9fa72ab11c1f/analysis/clusters_pulseseq.Rmd" target="_blank">ded741c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/4b4b233a3e65d68dc2710d8d3ec618dfb84cb886/docs/clusters_pulseseq.html" target="_blank">4b4b233</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Make some improvements to clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/571b3119e6564d9a9d552355d7645e8592c22291/docs/clusters_pulseseq.html" target="_blank">571b311</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d2cb6029d0ec1f0bb7bb5bbed427fb9fe5b6ab21/analysis/clusters_pulseseq.Rmd" target="_blank">d2cb602</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/13a5956752a5210288b75522fd276ea980f1fed3/docs/clusters_pulseseq.html" target="_blank">13a5956</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Built clusters_pulseseq page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5e0ee236762430bff2a435ddfd8aaa5667938b16/analysis/clusters_pulseseq.Rmd" target="_blank">5e0ee23</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Completed first rough draft of clustering in clusters_pulseseq.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/337d6fcbd3315fcaf8d0738135b587b921da521c/analysis/clusters_pulseseq.Rmd" target="_blank">337d6fc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Added clusters identified in PCs 5 and 6 of k=11 pulse-seq fit.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c0a27bdeac0459c010524a7843f2624aed3e83d9/analysis/clusters_pulseseq.Rmd" target="_blank">c0a27bd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Added clustering of pulseseq data along PCs 3 and 4.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e7383b22650cf6e86b2686e8c1a705925702e332/docs/clusters_pulseseq.html" target="_blank">e7383b2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Produced first rendering of clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1dd20d47a7b865f26da7f4e1259218446376854c/analysis/clusters_pulseseq.Rmd" target="_blank">1dd20d4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
workflowr::wflow_publish(“clusters_pulseseq.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/da9ac094078cd4447fc3dea2b64856bb7fe6c9da/analysis/clusters_pulseseq.Rmd" target="_blank">da9ac09</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Added hexbin plots to clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/485639abc730beab3d2299e8bcc1bf7aab2986a3/analysis/clusters_pulseseq.Rmd" target="_blank">485639a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Working on clusters_pulseseq analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c8dd3afae9fa35c6d78ee6aff137c0034b6686db/analysis/clusters_pulseseq.Rmd" target="_blank">c8dd3af</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
<td>
Implemented basic_pca_plot; improved labeled_pca_plot function.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform PCA on the topic proportions to identify clusters in the pulse-seq data.</p>
<p>Load the packages used in the analysis below, as well as additional functions that we will use to generate some of the plots.</p>
<pre class="r"><code>library(Matrix)
library(dplyr)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/plots.R&quot;)</code></pre>
<div id="load-data-and-results" class="section level2">
<h2>Load data and results</h2>
<p>Load the pulse-seq data.</p>
<pre class="r"><code>load(&quot;../data/pulseseq.RData&quot;)
x &lt;- as.character(samples$tissue)
x[x == &quot;club (hillock-associated)&quot;] &lt;- &quot;club&quot;
x[x == &quot;goblet.1&quot; | x == &quot;goblet.2&quot; | x == &quot;goblet.progenitor&quot;] &lt;- &quot;goblet&quot;
x[x == &quot;tuft.1&quot; | x == &quot;tuft.2&quot; | x == &quot;tuft.progenitor&quot;] &lt;- &quot;tuft&quot;
samples$tissue &lt;- factor(x)</code></pre>
<p>Load the <span class="math inline">\(k = 11\)</span> Poisson NMF model fit.</p>
<pre class="r"><code>fit &lt;- readRDS(&quot;../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds&quot;)$fit</code></pre>
</div>
<div id="identify-clusters-from-principal-components" class="section level2">
<h2>Identify clusters from principal components</h2>
<p>To identify clusters, we begin by plotting PCs computed from the topic proportions. (Note that only 10 PCs are needed for 11 topics.)</p>
<pre class="r"><code>p1 &lt;- basic_pca_plot(fit,1:2)
p2 &lt;- basic_pca_plot(fit,3:4)
p3 &lt;- basic_pca_plot(fit,5:6)
p4 &lt;- basic_pca_plot(fit,7:8)
p5 &lt;- basic_pca_plot(fit,9:10)
plot_grid(p1,p2,p3,p4,p5,nrow = 2,ncol = 3)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/clustering-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-1-1">
Past versions of clustering-1-1.png
</button>
</p>
<div id="fig-clustering-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e7383b22650cf6e86b2686e8c1a705925702e332/docs/figure/clusters_pulseseq.Rmd/clustering-1-1.png" target="_blank">e7383b2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Some of the structure is more evident from “hexbin” plots showing the density of the points. For example, clear clusters emerge in the hexbin plots for PCs 3 and 4, and for PCs 5 and 6:</p>
<pre class="r"><code>p6  &lt;- pca_hexbin_plot(fit,1:2) + guides(fill = &quot;none&quot;)
p7  &lt;- pca_hexbin_plot(fit,3:4) + guides(fill = &quot;none&quot;)
p8  &lt;- pca_hexbin_plot(fit,5:6) + guides(fill = &quot;none&quot;)
p9  &lt;- pca_hexbin_plot(fit,7:8) + guides(fill = &quot;none&quot;)
p10 &lt;- pca_hexbin_plot(fit,9:10) + guides(fill = &quot;none&quot;)
plot_grid(p6,p7,p8,p9,p10,nrow = 2,ncol = 3)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/clustering-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-2-1">
Past versions of clustering-2-1.png
</button>
</p>
<div id="fig-clustering-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e7383b22650cf6e86b2686e8c1a705925702e332/docs/figure/clusters_pulseseq.Rmd/clustering-2-1.png" target="_blank">e7383b2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From these PCA plots, we define 4 clusters, labeled A, D, Cil and T+N. (The reasoning behind these cluster labels will become clear later.) Points that do not fit in any of these clusters are assigned to a “background cluster”, labeled U for “unknown”.</p>
<pre class="r"><code>pca &lt;- prcomp(poisson2multinom(fit)$L)$x
x   &lt;- rep(&quot;U&quot;,nrow(pca))
pc3 &lt;- pca[,3]
pc4 &lt;- pca[,4]
pc5 &lt;- pca[,5]
pc6 &lt;- pca[,6]
x[pc4 &lt; 5.5*pc3 + 0.5] &lt;- &quot;A&quot;
x[(pc3 + 0.7)^2 + (pc4 - 0.1)^2 &lt; 0.18^2] &lt;- &quot;Cil&quot;
x[x == &quot;A&quot; &amp; pc6 &gt; 1.3*pc5 + 0.87] &lt;- &quot;T+N&quot;
x[x == &quot;A&quot; &amp; pc5 &gt; -0.4 &amp; pc6 &lt; 1.3*pc5 + 0.49] &lt;- &quot;D&quot;</code></pre>
<p>We can refine the small cluster A somewhat by plotting PCs computed from cluster A only:</p>
<pre class="r"><code>rows &lt;- which(x == &quot;A&quot;)
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p11  &lt;- basic_pca_plot(fit2,1:2)
print(p11)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/clustering-4-1.png" width="360" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-4-1">
Past versions of clustering-4-1.png
</button>
</p>
<div id="fig-clustering-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/65f104e6f0eea164e0e5b11a0464720c88779997/docs/figure/clusters_pulseseq.Rmd/clustering-4-1.png" target="_blank">65f104e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/775fb9104857c2be91d06aba90f45839945f1225/docs/figure/clusters_pulseseq.Rmd/clustering-4-1.png" target="_blank">775fb91</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/13a5956752a5210288b75522fd276ea980f1fed3/docs/figure/clusters_pulseseq.Rmd/clustering-4-1.png" target="_blank">13a5956</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We label this refined cluster as “I”, and the rest are added back to the background cluster (U).</p>
<pre class="r"><code>pca  &lt;- prcomp(fit2$L)$x
y    &lt;- rep(&quot;U&quot;,nrow(pca))
pc1  &lt;- pca[,1]
y[pc1 &gt; -0.1] &lt;- &quot;I&quot;
x[rows] &lt;- y</code></pre>
<p>Similarly, we mine the much larger cluster D for substructure:</p>
<pre class="r"><code>rows &lt;- which(x == &quot;D&quot;)
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p12  &lt;- basic_pca_plot(fit2,1:2)
p13  &lt;- pca_hexbin_plot(fit2,1:2) + guides(fill = &quot;none&quot;)
plot_grid(p12,p13)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/clustering-6-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-6-1">
Past versions of clustering-6-1.png
</button>
</p>
<div id="fig-clustering-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/65f104e6f0eea164e0e5b11a0464720c88779997/docs/figure/clusters_pulseseq.Rmd/clustering-6-1.png" target="_blank">65f104e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/775fb9104857c2be91d06aba90f45839945f1225/docs/figure/clusters_pulseseq.Rmd/clustering-6-1.png" target="_blank">775fb91</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e62fb432fd8df5022e6e82f83e556085e0f197e7/docs/figure/clusters_pulseseq.Rmd/clustering-6-1.png" target="_blank">e62fb43</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/13a5956752a5210288b75522fd276ea980f1fed3/docs/figure/clusters_pulseseq.Rmd/clustering-6-1.png" target="_blank">13a5956</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The hexbin plot suggests two clusters. Although these clusters are not distinct, it may nonetheless be useful to subdivide these data points (somewhat arbitrarily) into two subsets, which we label as B and C.</p>
<pre class="r"><code>pca  &lt;- prcomp(fit2$L)$x
y    &lt;- rep(&quot;C&quot;,nrow(pca))
pc1  &lt;- pca[,1]
pc2  &lt;- pca[,2]
y[pc2 &gt; -pc1 - 0.15] &lt;- &quot;B&quot;
x[rows] &lt;- y</code></pre>
<p>Within cluster B, there is some interesting structure along PCs 5 and 6:</p>
<pre class="r"><code>rows &lt;- which(x == &quot;B&quot;)
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p14  &lt;- basic_pca_plot(fit2,5:6)
p15  &lt;- pca_hexbin_plot(fit2,5:6) + guides(fill = &quot;none&quot;)
plot_grid(p14,p15)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/clustering-8-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-8-1">
Past versions of clustering-8-1.png
</button>
</p>
<div id="fig-clustering-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/65f104e6f0eea164e0e5b11a0464720c88779997/docs/figure/clusters_pulseseq.Rmd/clustering-8-1.png" target="_blank">65f104e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e62fb432fd8df5022e6e82f83e556085e0f197e7/docs/figure/clusters_pulseseq.Rmd/clustering-8-1.png" target="_blank">e62fb43</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/13a5956752a5210288b75522fd276ea980f1fed3/docs/figure/clusters_pulseseq.Rmd/clustering-8-1.png" target="_blank">13a5956</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From PCs 5 and 6, define a new cluster, “P”, recognizing that this cluster is not particularly distinct.</p>
<pre class="r"><code>pca &lt;- prcomp(fit2$L)$x
y   &lt;- rep(&quot;B&quot;,nrow(pca))
pc5 &lt;- pca[,5]
pc6 &lt;- pca[,6]
y[pc5 &gt; 0.1 &amp; pc6 &lt; 0] &lt;- &quot;P&quot;
x[rows] &lt;- y</code></pre>
<p>In summary, we have subdivided the pulse-seq data into 7 subsets, which includes a background cluster (U).</p>
<pre class="r"><code>samples$cluster &lt;- factor(x,c(&quot;B&quot;,&quot;C&quot;,&quot;P&quot;,&quot;Cil&quot;,&quot;T+N&quot;,&quot;I&quot;,&quot;U&quot;))
cluster_colors  &lt;- c(&quot;royalblue&quot;,   # B
                     &quot;forestgreen&quot;, # C
                     &quot;peru&quot;,        # P
                     &quot;firebrick&quot;,   # Cil
                     &quot;darkorange&quot;,  # T+N
                     &quot;darkmagenta&quot;, # I
                     &quot;gainsboro&quot;)   # U
p16 &lt;- labeled_pca_plot(fit,1:2,samples$cluster,cluster_colors) +
       guides(fill = &quot;none&quot;)
p17 &lt;- labeled_pca_plot(fit,3:4,samples$cluster,cluster_colors) +
       guides(fill = &quot;none&quot;)
p18 &lt;- labeled_pca_plot(fit,5:6,samples$cluster,cluster_colors) +
       guides(fill = &quot;none&quot;)
p19 &lt;- labeled_pca_plot(fit,1:2,samples$cluster,cluster_colors,&quot;tissue&quot;) +
       xlim(0,0) + ylim(0,0)
plot_grid(p16,p17,p18,p19)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/clustering-10-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-10-1">
Past versions of clustering-10-1.png
</button>
</p>
<div id="fig-clustering-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56d99a39b9cd2d6a09e725a08b714511e726cdc5/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">56d99a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7489099fc83d655afc9690aedab75fa34e22d8e7/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">7489099</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/942486b7eb0536b69a20ae4e4502c20ba58d235e/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">942486b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/775fb9104857c2be91d06aba90f45839945f1225/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">775fb91</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e62fb432fd8df5022e6e82f83e556085e0f197e7/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">e62fb43</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/4b4b233a3e65d68dc2710d8d3ec618dfb84cb886/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">4b4b233</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/571b3119e6564d9a9d552355d7645e8592c22291/docs/figure/clusters_pulseseq.Rmd/clustering-10-1.png" target="_blank">571b311</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Comparing this to the Montoro <em>et al</em> (2018) clustering, we observe some close correspondence (e.g., B and “basal cells”, P and “proliferating cells”).</p>
<pre class="r"><code>with(samples,table(tissue,cluster))
#                 cluster
# tissue               B     C     P   Cil   T+N     I     U
#   basal          40389  1468   199     0     0     0    37
#   ciliated           0     0     6  2896     0     0   114
#   club            1766 15870    36     0     1     6    21
#   goblet             3   396     2     0     0     0     2
#   ionocyte           0    45    15     0     8   193    15
#   neuroendocrine     0     1     0     0   619     7     3
#   proliferating     61   194   914     9     0     4   231
#   tuft               0    18     4     0   691    12     9</code></pre>
</div>
<div id="structure-plot" class="section level2">
<h2>Structure plot</h2>
<p>The structure plot summarizes the topic proportions in each of the 7 subsets:</p>
<pre class="r"><code>set.seed(1)
topic_colors &lt;- c(&quot;turquoise&quot;,&quot;darkorange&quot;,&quot;dodgerblue&quot;,&quot;gold&quot;,&quot;peru&quot;,
                  &quot;greenyellow&quot;,&quot;firebrick&quot;,&quot;olivedrab&quot;,&quot;royalblue&quot;,
                  &quot;forestgreen&quot;,&quot;gray&quot;)
topics &lt;- c(11,1,3,4,5,6,8,10,9,2,7)
rows &lt;- sort(c(sample(which(samples$cluster == &quot;B&quot;),1000),
               sample(which(samples$cluster == &quot;C&quot;),1000),
               sample(which(samples$cluster == &quot;P&quot;),500),
               sample(which(samples$cluster == &quot;Cil&quot;),500),
               sample(which(samples$cluster == &quot;T+N&quot;),500),
               which(samples$cluster == &quot;I&quot;),
               which(samples$cluster == &quot;U&quot;)))
p20 &lt;- structure_plot(select(poisson2multinom(fit),loadings = rows),
                      grouping = samples[rows,&quot;cluster&quot;],
                      topics = topics,
                      colors = topic_colors[topics],
                      perplexity = c(70,70,30,50,50,30,70),
                      n = Inf,gap = 30,num_threads = 4,verbose = FALSE)
print(p20)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/structure-plot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-1">
Past versions of structure-plot-1.png
</button>
</p>
<div id="fig-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56d99a39b9cd2d6a09e725a08b714511e726cdc5/docs/figure/clusters_pulseseq.Rmd/structure-plot-1.png" target="_blank">56d99a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Based on this structure plot, and from the other results above, we roughly subdivide the pulse-seq data into two subsets: (1) the Cil, T+N and I clusters that give rise to fairly well-separated clusters, and (2) the B, C and P subsets that contain interesting substructure but much less distinct clusters. Therefore, the cluster labels B, C and P are useful as a guide but should be taken with a grain of salt as the boundaries between these clusters are somewhat arbitrary.</p>
<p>The structure plot suggests that there is substantial heterogeneity in the C cluster beyond what can be captured by “hard” clusters. In particular, three topics (<span class="math inline">\(k = 5, 6, 10\)</span>) are largely unique to the cells in this cluster, and two other topics (<span class="math inline">\(k = 4, 8\)</span>) primarily contribute to the cells in this cluster, although can be found in small proportions elsewhere.</p>
<p>These topics indeed pick up continuous variation in gene expression among the cells in this cluster:</p>
<pre class="r"><code>rows &lt;- which(x == &quot;C&quot;)
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p21 &lt;- pca_plot(fit2,pcs = 1:2,k = c(5,8,10))
print(p21)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/pca-plot-club-1-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-plot-club-1-1">
Past versions of pca-plot-club-1-1.png
</button>
</p>
<div id="fig-pca-plot-club-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/310ef96bc1c85e8a7cff055b0d0741e73d399a05/docs/figure/clusters_pulseseq.Rmd/pca-plot-club-1-1.png" target="_blank">310ef96</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56d99a39b9cd2d6a09e725a08b714511e726cdc5/docs/figure/clusters_pulseseq.Rmd/pca-plot-club-1-1.png" target="_blank">56d99a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And in PCs 3 and 4:</p>
<pre class="r"><code>p22 &lt;- pca_plot(fit2,pcs = 3:4,k = c(4,5,6))
print(p22)</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/pca-plot-club-2-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-plot-club-2-1">
Past versions of pca-plot-club-2-1.png
</button>
</p>
<div id="fig-pca-plot-club-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/310ef96bc1c85e8a7cff055b0d0741e73d399a05/docs/figure/clusters_pulseseq.Rmd/pca-plot-club-2-1.png" target="_blank">310ef96</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56d99a39b9cd2d6a09e725a08b714511e726cdc5/docs/figure/clusters_pulseseq.Rmd/pca-plot-club-2-1.png" target="_blank">56d99a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The B cell also shows a lot of heterogeneity. Topic 1 is of particular interest because of its close connection to expression of the “hillock” gene <em>Krt13</em>:</p>
<pre class="r"><code>rows &lt;- which(x == &quot;B&quot;)
fit2 &lt;- select(poisson2multinom(fit),loadings = rows)
p23  &lt;- pca_plot(fit2,pcs = 2:3,k = 1) + guides(fill = &quot;none&quot;)
p24  &lt;- pca_plot_with_counts(fit2,counts[rows,&quot;Krt13&quot;],2:3,log = TRUE)
plot_grid(p23,p24,rel_widths = c(4,5))</code></pre>
<p><img src="figure/clusters_pulseseq.Rmd/pca-plot-basal-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-plot-basal-1">
Past versions of pca-plot-basal-1.png
</button>
</p>
<div id="fig-pca-plot-basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7d78e71460e4f6a38c095b0d93d90d602d3f1d50/docs/figure/clusters_pulseseq.Rmd/pca-plot-basal-1.png" target="_blank">7d78e71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a67196abd388ee8c7e9ae4693bc40ed0894546e1/docs/figure/clusters_pulseseq.Rmd/pca-plot-basal-1.png" target="_blank">a67196a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/310ef96bc1c85e8a7cff055b0d0741e73d399a05/docs/figure/clusters_pulseseq.Rmd/pca-plot-basal-1.png" target="_blank">310ef96</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56d99a39b9cd2d6a09e725a08b714511e726cdc5/docs/figure/clusters_pulseseq.Rmd/pca-plot-basal-1.png" target="_blank">56d99a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-results" class="section level2">
<h2>Save results</h2>
<p>Save the clustering of the pulse-seq data to an RDS file.</p>
<pre class="r"><code>saveRDS(samples,&quot;clustering-pulseseq.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.6
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0      ggplot2_3.3.0      fastTopics_0.3-177 dplyr_0.8.3       
# [5] Matrix_1.2-18     
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.0        Rcpp_1.0.5           lattice_0.20-38     
#  [4] tidyr_1.0.0          prettyunits_1.1.1    assertthat_0.2.1    
#  [7] zeallot_0.1.0        rprojroot_1.3-2      digest_0.6.23       
# [10] R6_2.4.1             backports_1.1.5      MatrixModels_0.4-1  
# [13] evaluate_0.14        coda_0.19-3          httr_1.4.1          
# [16] pillar_1.4.3         rlang_0.4.5          progress_1.2.2      
# [19] lazyeval_0.2.2       data.table_1.12.8    irlba_2.3.3         
# [22] SparseM_1.78         hexbin_1.28.0        whisker_0.4         
# [25] rmarkdown_2.3        labeling_0.3         Rtsne_0.15          
# [28] stringr_1.4.0        htmlwidgets_1.5.1    munsell_0.5.0       
# [31] compiler_3.6.2       httpuv_1.5.2         xfun_0.11           
# [34] pkgconfig_2.0.3      mcmc_0.9-6           htmltools_0.4.0     
# [37] tidyselect_0.2.5     tibble_2.1.3         workflowr_1.6.2.9000
# [40] quadprog_1.5-8       viridisLite_0.3.0    crayon_1.3.4        
# [43] withr_2.1.2          later_1.0.0          MASS_7.3-51.4       
# [46] grid_3.6.2           jsonlite_1.6         gtable_0.3.0        
# [49] lifecycle_0.1.0      git2r_0.26.1         magrittr_1.5        
# [52] scales_1.1.0         RcppParallel_4.4.2   stringi_1.4.3       
# [55] farver_2.0.1         fs_1.3.1             promises_1.1.0      
# [58] vctrs_0.2.1          tools_3.6.2          glue_1.3.1          
# [61] purrr_0.3.3          hms_0.5.2            yaml_2.2.0          
# [64] colorspace_1.4-1     plotly_4.9.2         knitr_1.26          
# [67] quantreg_5.54        MCMCpack_1.4-5</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
