<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Comparison of fastTopics DE analysis with MAST and DESeq2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Comparison of fastTopics DE analysis with MAST and DESeq2</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree2a2d9b163fad86423b630733a1eaa2ca3dd250aftargetblank2a2d9b1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/2a2d9b163fad86423b630733a1eaa2ca3dd250af" target="_blank">2a2d9b1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree2a2d9b163fad86423b630733a1eaa2ca3dd250aftargetblank2a2d9b1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/2a2d9b163fad86423b630733a1eaa2ca3dd250af" target="_blank">2a2d9b1</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/
    Ignored:    output/sims/

Untracked files:
    Untracked:  analysis/de_analysis_detailed_look_cache/
    Untracked:  analysis/de_analysis_detailed_look_more_cache/
    Untracked:  plots/

Unstaged changes:
    Modified:   analysis/de_analysis_mast_deseq2.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_analysis_mast_deseq2.Rmd</code>) and HTML (<code>docs/de_analysis_mast_deseq2.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2a2d9b163fad86423b630733a1eaa2ca3dd250af/analysis/de_analysis_mast_deseq2.Rmd" target="_blank">2a2d9b1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
workflowr::wflow_publish(“de_analysis_mast_deseq2.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/10fc0dc9e8913ea6f028fc8e0b6d0c86603be3ec/docs/de_analysis_mast_deseq2.html" target="_blank">10fc0dc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
Added scatterplots to de_analysis_mast_deseq2 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/85eeed0e3c1f7c7e42f7a5e28cbba86f56ab85f9/analysis/de_analysis_mast_deseq2.Rmd" target="_blank">85eeed0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
workflowr::wflow_publish(“de_analysis_mast_deseq2.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/283c7ba7475e80054fdc5c01de8e3b89544c2c84/analysis/de_analysis_mast_deseq2.Rmd" target="_blank">283c7ba</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
Working on de_analysis_mast_deseq2 R Markdown.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/283c7ba7475e80054fdc5c01de8e3b89544c2c84/docs/de_analysis_mast_deseq2.html" target="_blank">283c7ba</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
Working on de_analysis_mast_deseq2 R Markdown.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/4ff98f43c8d478f04ed817713f5d5f3818857bfa/docs/de_analysis_mast_deseq2.html" target="_blank">4ff98f4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
Built de_analysis_mast_deseq2 page.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e164b8f9de0726673212a0b103e6f6351f9cbec1/docs/de_analysis_mast_deseq2.html" target="_blank">e164b8f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
Added link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3de0e0cdf12a4f1cb65280f7ef3dc7bea0c2ea38/analysis/de_analysis_mast_deseq2.Rmd" target="_blank">3de0e0c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This is a more systematic comparison of the new fastTopics DE analysis with MAST and DESeq2 based on an <a href="de_analysis_detailed_look.html">initial smaller simulation</a>. The results were generated using the <code>run_sims.R</code> script, and here we summarize the results of these simulations.</p>
<p>Load the packages needed for this analysis, and some additional functions used to compile the results and generate the plots.</p>
<pre class="r"><code>library(Matrix)
library(Seurat)
library(DESeq2)
library(MAST)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/de_analysis_functions.R&quot;)</code></pre>
<p>Load the results of the simulations.</p>
<pre class="r"><code>load(&quot;../output/sims/sims-k=2-alpha=0.01.RData&quot;)</code></pre>
<p>These were the R packages used to run the simulations:</p>
<pre class="r"><code>res$session.info
# R version 4.1.0 (2021-05-18)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats4    tools     stats     graphics  grDevices utils     datasets 
# [8] methods   base     
# 
# other attached packages:
#  [1] fastTopics_0.6-97           MAST_1.20.0                
#  [3] SeuratObject_4.0.2          Seurat_4.0.3               
#  [5] DESeq2_1.34.0               scran_1.22.1               
#  [7] scuttle_1.4.0               SingleCellExperiment_1.16.0
#  [9] SummarizedExperiment_1.24.0 Biobase_2.54.0             
# [11] GenomicRanges_1.46.0        GenomeInfoDb_1.30.0        
# [13] IRanges_2.28.0              S4Vectors_0.32.0           
# [15] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
# [17] matrixStats_0.60.1          Matrix_1.3-3               
# 
# loaded via a namespace (and not attached):
#   [1] utf8_1.2.1                reticulate_1.20          
#   [3] tidyselect_1.1.1          RSQLite_2.2.8            
#   [5] AnnotationDbi_1.56.1      htmlwidgets_1.5.3        
#   [7] grid_4.1.0                BiocParallel_1.28.0      
#   [9] Rtsne_0.15                munsell_0.5.0            
#  [11] ScaledMatrix_1.2.0        codetools_0.2-18         
#  [13] ica_1.0-2                 statmod_1.4.36           
#  [15] future_1.21.0             miniUI_0.1.1.1           
#  [17] colorspace_2.0-2          rstudioapi_0.13          
#  [19] ROCR_1.0-11               tensor_1.5               
#  [21] listenv_0.8.0             GenomeInfoDbData_1.2.7   
#  [23] mixsqp_0.3-43             polyclip_1.10-0          
#  [25] MCMCpack_1.6-0            bit64_4.0.5              
#  [27] coda_0.19-4               parallelly_1.26.1        
#  [29] vctrs_0.3.8               generics_0.1.0           
#  [31] R6_2.5.0                  rsvd_1.0.5               
#  [33] invgamma_1.1              locfit_1.5-9.4           
#  [35] bitops_1.0-7              spatstat.utils_2.2-0     
#  [37] cachem_1.0.5              DelayedArray_0.20.0      
#  [39] assertthat_0.2.1          promises_1.2.0.1         
#  [41] scales_1.1.1              gtable_0.3.0             
#  [43] beachmat_2.10.0           globals_0.14.0           
#  [45] conquer_1.0.2             goftest_1.2-2            
#  [47] mcmc_0.9-7                rlang_0.4.11             
#  [49] MatrixModels_0.5-0        genefilter_1.76.0        
#  [51] splines_4.1.0             lazyeval_0.2.2           
#  [53] spatstat.geom_2.2-0       reshape2_1.4.4           
#  [55] abind_1.4-5               httpuv_1.6.1             
#  [57] ggplot2_3.3.5             ellipsis_0.3.2           
#  [59] spatstat.core_2.2-0       RColorBrewer_1.1-2       
#  [61] ggridges_0.5.3            Rcpp_1.0.7               
#  [63] plyr_1.8.6                sparseMatrixStats_1.6.0  
#  [65] progress_1.2.2            zlibbioc_1.40.0          
#  [67] purrr_0.3.4               RCurl_1.98-1.5           
#  [69] prettyunits_1.1.1         rpart_4.1-15             
#  [71] deldir_0.2-10             pbapply_1.4-3            
#  [73] ashr_2.2-51               cowplot_1.1.1            
#  [75] zoo_1.8-9                 ggrepel_0.9.1            
#  [77] cluster_2.1.2             magrittr_2.0.1           
#  [79] data.table_1.14.0         scattermore_0.7          
#  [81] SparseM_1.81              lmtest_0.9-38            
#  [83] RANN_2.6.1                truncnorm_1.0-8          
#  [85] SQUAREM_2021.1            fitdistrplus_1.1-5       
#  [87] hms_1.1.0                 patchwork_1.1.1          
#  [89] mime_0.11                 xtable_1.8-4             
#  [91] XML_3.99-0.6              gridExtra_2.3            
#  [93] compiler_4.1.0            tibble_3.1.2             
#  [95] KernSmooth_2.23-20        crayon_1.4.1             
#  [97] htmltools_0.5.1.1         mgcv_1.8-35              
#  [99] later_1.2.0               tidyr_1.1.3              
# [101] geneplotter_1.72.0        RcppParallel_5.1.4       
# [103] DBI_1.1.1                 MASS_7.3-54              
# [105] quadprog_1.5-8            parallel_4.1.0           
# [107] metapod_1.2.0             igraph_1.2.6             
# [109] pkgconfig_2.0.3           plotly_4.9.4.1           
# [111] spatstat.sparse_2.0-0     annotate_1.72.0          
# [113] dqrng_0.3.0               XVector_0.34.0           
# [115] stringr_1.4.0             digest_0.6.27            
# [117] sctransform_0.3.2         RcppAnnoy_0.0.18         
# [119] spatstat.data_2.1-0       Biostrings_2.62.0        
# [121] leiden_0.3.8              uwot_0.1.10              
# [123] edgeR_3.36.0              DelayedMatrixStats_1.16.0
# [125] shiny_1.6.0               quantreg_5.86            
# [127] lifecycle_1.0.0           nlme_3.1-152             
# [129] jsonlite_1.7.2            BiocNeighbors_1.12.0     
# [131] viridisLite_0.4.0         limma_3.50.0             
# [133] fansi_0.5.0               pillar_1.6.1             
# [135] lattice_0.20-44           KEGGREST_1.34.0          
# [137] fastmap_1.1.0             httr_1.4.2               
# [139] survival_3.2-11           glue_1.4.2               
# [141] png_0.1-7                 bluster_1.4.0            
# [143] bit_4.0.4                 stringi_1.6.2            
# [145] blob_1.2.1                BiocSingular_1.10.0      
# [147] memoise_2.0.0             dplyr_1.0.7              
# [149] irlba_2.3.3               future.apply_1.7.0</code></pre>
<p>Since we are comparing with MAST and DESeq2, two methods that do not allow for partial membership to groups, we have simulated count data from a topic model in which the true topic proportions are all 0 or 1, or mostly very close to 0 or 1.</p>
<pre class="r"><code>res[&quot;session.info&quot;] &lt;- NULL
x &lt;- combine_sim_res(res,function (x) apply(x$dat$L,1,max))
mean(x &gt; 0.99)
# [1] 0.98625</code></pre>
<p>Before comparing the methods, first we assess accuracy of the Monte Carlo computations by comparing the estimates from two independent MCMC simulations. The estimates from the two independent simulations are largely consistent.</p>
<pre class="r"><code>pdat &lt;- data.frame(lfc1 = combine_sim_res(res,function (x) x$de1$postmean[,2]),
                   lfc2 = combine_sim_res(res,function (x) x$de2$postmean[,2]),
                   z1   = combine_sim_res(res,function (x) x$de1$z[,2]),
           z2   = combine_sim_res(res,function (x) x$de2$z[,2]))
p1 &lt;- ggplot(pdat,aes(x = lfc1,y = lfc2)) +
  geom_point(color = &quot;dodgerblue&quot;,shape = 4,size = 0.75) +
  geom_abline(intercept = 0,slope = 1,linetype = &quot;dotted&quot;) +
  labs(x = &quot;first posterior mean&quot;,y = &quot;second posterior mean&quot;) +
  theme_cowplot(font_size = 12)
p2 &lt;- ggplot(pdat,aes(x = z1,y = z2)) +
  geom_point(color = &quot;dodgerblue&quot;,shape = 4,size = 0.75) +
  geom_abline(intercept = 0,slope = 1,linetype = &quot;dotted&quot;) +
  labs(x = &quot;first z-score estimate&quot;,y = &quot;second z-score estimate&quot;) +
  xlim(-45,42) + 
  ylim(-45,42) + 
  theme_cowplot(font_size = 12)
plot_grid(p1,p2)</code></pre>
<p><img src="figure/de_analysis_mast_deseq2.Rmd/mcmc-scatterplots-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-mcmc-scatterplots-1">
Past versions of mcmc-scatterplots-1.png
</button>
</p>
<div id="fig-mcmc-scatterplots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/10fc0dc9e8913ea6f028fc8e0b6d0c86603be3ec/docs/figure/de_analysis_mast_deseq2.Rmd/mcmc-scatterplots-1.png" target="_blank">10fc0dc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now we compare LFC estimates for the second topic:</p>
<pre class="r"><code>j  &lt;- paste0(&quot;g&quot;,1:10000)
lfc.deseq &lt;- combine_sim_res(res,function (x) x$deseq$log2FoldChange)
lfc.fasttopics &lt;- combine_sim_res(res,function (x) x$de1$postmean[,2])
lfc.mast &lt;- combine_sim_res(res,function (x) x$mast[j,&quot;avg_log2FC&quot;])
z.deseq &lt;- combine_sim_res(res,function (x) with(x$deseq,log2FoldChange/lfcSE))
z.fasttopics &lt;- combine_sim_res(res,function (x) x$de1$z[,2])
pdat &lt;- data.frame(lfc.deseq      = lfc.deseq,
                   lfc.fasttopics = lfc.fasttopics,
                   lfc.mast       = clamp(lfc.mast,-6,+6),
                   z.deseq        = z.deseq,
                   z.fasttopics   = z.fasttopics)
p1 &lt;- ggplot(pdat,aes(x = lfc.mast,y = lfc.fasttopics)) +
  geom_point(color = &quot;dodgerblue&quot;,shape = 4,size = 0.75) +
  geom_abline(intercept = 0,slope = 1,linetype = &quot;dotted&quot;) +
  labs(x = &quot;MAST&quot;,y = &quot;fastTopics&quot;,title = &quot;LFC estimates&quot;) +
  theme_cowplot(font_size = 12)
p2 &lt;- ggplot(pdat,aes(x = lfc.deseq,y = lfc.fasttopics)) +
  geom_point(color = &quot;dodgerblue&quot;,shape = 4,size = 0.75) +
  geom_abline(intercept = 0,slope = 1,linetype = &quot;dotted&quot;) +
  labs(x = &quot;DESeq2&quot;,y = &quot;fastTopics&quot;,title = &quot;LFC estimates&quot;) +
  theme_cowplot(font_size = 12)
p3 &lt;- ggplot(pdat,aes(x = z.deseq,y = z.fasttopics)) +
  geom_point(color = &quot;dodgerblue&quot;,shape = 4,size = 0.75) +
  geom_abline(intercept = 0,slope = 1,linetype = &quot;dotted&quot;) +
  labs(x = &quot;DESeq2&quot;,y = &quot;fastTopics&quot;,title = &quot;z-scores&quot;) +
  theme_cowplot(font_size = 12)
plot_grid(p1,p2,p3,nrow = 1,ncol = 3)</code></pre>
<p><img src="figure/de_analysis_mast_deseq2.Rmd/lfc-scatterplots-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lfc-scatterplots-1">
Past versions of lfc-scatterplots-1.png
</button>
</p>
<div id="fig-lfc-scatterplots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/10fc0dc9e8913ea6f028fc8e0b6d0c86603be3ec/docs/figure/de_analysis_mast_deseq2.Rmd/lfc-scatterplots-1.png" target="_blank">10fc0dc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-01-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The fastTopics and DESeq2 estimates are very similar across the board, whereas the MAST estimare are broadly correlated, but show a number of large differences.</p>
<p>Compare the MAST, DESeq2 and fastTopics <em>p</em>-values (or <em>s</em>-values), separately for all true positives (dark blue) and true negatives (orange). Not surprisingly, the DESeq2 and fastTopics <em>s</em>-value distributions are also very similar.</p>
<pre class="r"><code>deseq       &lt;- combine_sim_res(res,function (x) x$deseq$svalue)
fasttopics  &lt;- combine_sim_res(res,function (x) x$de1$svalue[,2])
mast        &lt;- combine_sim_res(res,function (x) x$mast[j,&quot;p_val&quot;])
nonzero_lfc &lt;-
  combine_sim_res(res,function (x) with(x$dat,abs(F[,1] - F[,2]) &gt; 1e-8))
pdat &lt;- data.frame(deseq       = deseq,
                   fasttopics  = fasttopics,
                   mast        = mast,
                   nonzero_lfc = factor(nonzero_lfc))
p4 &lt;- ggplot(pdat,aes(x = mast,color = nonzero_lfc,fill = nonzero_lfc)) +
  geom_histogram(bins = 64,show.legend = FALSE) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;MAST&quot;) +
  theme_cowplot(font_size = 12)
p5 &lt;- ggplot(pdat,aes(x = deseq,color = nonzero_lfc,fill = nonzero_lfc)) +
  geom_histogram(bins = 64,show.legend = FALSE) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;s-value&quot;,y = &quot;genes&quot;,title = &quot;DESeq2&quot;) +
  theme_cowplot(font_size = 12)
p6 &lt;- ggplot(pdat,aes(x = fasttopics,color = nonzero_lfc,fill = nonzero_lfc)) +
  geom_histogram(bins = 64,show.legend = FALSE) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;s-value&quot;,y = &quot;genes&quot;,title = &quot;fastTopics&quot;) +
  theme_cowplot(font_size = 12)
plot_grid(p4,p5,p6,nrow = 1,ncol = 3)</code></pre>
<p><img src="figure/de_analysis_mast_deseq2.Rmd/pval-histograms-1.png" width="840" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
# [8] methods   base     
# 
# other attached packages:
#  [1] cowplot_1.0.0               ggplot2_3.3.5              
#  [3] fastTopics_0.6-96           MAST_1.12.0                
#  [5] SingleCellExperiment_1.8.0  DESeq2_1.33.5              
#  [7] SummarizedExperiment_1.16.1 DelayedArray_0.12.3        
#  [9] BiocParallel_1.18.1         matrixStats_0.61.0         
# [11] Biobase_2.46.0              GenomicRanges_1.38.0       
# [13] GenomeInfoDb_1.22.1         IRanges_2.20.2             
# [15] S4Vectors_0.24.4            BiocGenerics_0.32.0        
# [17] Seurat_3.2.3                Matrix_1.2-18              
# 
# loaded via a namespace (and not attached):
#   [1] backports_1.1.5        workflowr_1.6.2        systemfonts_1.0.2     
#   [4] plyr_1.8.5             igraph_1.2.5           lazyeval_0.2.2        
#   [7] splines_3.6.2          listenv_0.8.0          scattermore_0.7       
#  [10] digest_0.6.23          invgamma_1.1           htmltools_0.4.0       
#  [13] SQUAREM_2017.10-1      fansi_0.4.0            magrittr_2.0.1        
#  [16] memoise_1.1.0          tensor_1.5             cluster_2.1.0         
#  [19] ROCR_1.0-11            globals_0.13.0         annotate_1.64.0       
#  [22] RcppParallel_4.4.2     MCMCpack_1.4-5         prettyunits_1.1.1     
#  [25] colorspace_1.4-1       blob_1.2.1             ggrepel_0.9.1         
#  [28] xfun_0.11              dplyr_1.0.7            crayon_1.4.1          
#  [31] RCurl_1.98-1.2         jsonlite_1.7.2         genefilter_1.68.0     
#  [34] spatstat_1.64-1        spatstat.data_1.4-3    survival_3.1-8        
#  [37] zoo_1.8-7              glue_1.4.2             polyclip_1.10-0       
#  [40] gtable_0.3.0           zlibbioc_1.32.0        XVector_0.26.0        
#  [43] MatrixModels_0.4-1     leiden_0.3.3           future.apply_1.6.0    
#  [46] SparseM_1.78           abind_1.4-5            scales_1.1.0          
#  [49] DBI_1.1.0              miniUI_0.1.1.1         Rcpp_1.0.7            
#  [52] progress_1.2.2         viridisLite_0.3.0      xtable_1.8-4          
#  [55] reticulate_1.16        bit_1.1-15.2           rsvd_1.0.2            
#  [58] truncnorm_1.0-8        htmlwidgets_1.5.1      httr_1.4.2            
#  [61] RColorBrewer_1.1-2     ellipsis_0.3.2         ica_1.0-2             
#  [64] farver_2.0.1           pkgconfig_2.0.3        XML_3.99-0.3          
#  [67] uwot_0.1.10            deldir_0.1-29          locfit_1.5-9.4        
#  [70] utf8_1.1.4             labeling_0.3           tidyselect_1.1.1      
#  [73] rlang_0.4.11           reshape2_1.4.3         later_1.0.0           
#  [76] AnnotationDbi_1.48.0   munsell_0.5.0          tools_3.6.2           
#  [79] generics_0.0.2         RSQLite_2.2.0          ggridges_0.5.2        
#  [82] evaluate_0.14          stringr_1.4.0          fastmap_1.0.1         
#  [85] ragg_0.3.1             yaml_2.2.0             goftest_1.2-2         
#  [88] mcmc_0.9-6             knitr_1.26             bit64_0.9-7           
#  [91] fs_1.3.1               fitdistrplus_1.1-1     purrr_0.3.4           
#  [94] RANN_2.6.1             pbapply_1.5-1          future_1.18.0         
#  [97] nlme_3.1-142           quantreg_5.54          whisker_0.4           
# [100] mime_0.8               compiler_3.6.2         plotly_4.9.2          
# [103] png_0.1-7              spatstat.utils_1.17-0  tibble_3.1.3          
# [106] geneplotter_1.64.0     stringi_1.4.3          lattice_0.20-38       
# [109] vctrs_0.3.8            pillar_1.6.2           lifecycle_1.0.0       
# [112] lmtest_0.9-38          RcppAnnoy_0.0.18       data.table_1.12.8     
# [115] bitops_1.0-6           irlba_2.3.3            httpuv_1.5.2          
# [118] patchwork_1.0.1        R6_2.4.1               promises_1.1.0        
# [121] KernSmooth_2.23-16     gridExtra_2.3          codetools_0.2-16      
# [124] MASS_7.3-51.4          assertthat_0.2.1       rprojroot_1.3-2       
# [127] withr_2.4.2            sctransform_0.3.2      GenomeInfoDbData_1.2.2
# [130] hms_1.1.0              mgcv_1.8-31            quadprog_1.5-8        
# [133] grid_3.6.2             rpart_4.1-15           coda_0.19-3           
# [136] tidyr_1.1.3            rmarkdown_2.3          ashr_2.2-51           
# [139] Rtsne_0.15             git2r_0.26.1           mixsqp_0.3-46         
# [142] shiny_1.4.0</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
