<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>DE analysis using a topic model: evaluation using simulated data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DE analysis using a topic model: evaluation using simulated data</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
assess-monte-carlo-3
</li>
<li>
de-noshrink
</li>
<li>
de-with-ash
</li>
<li>
fit-poisson-nmf
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>de_analysis_detailed_look_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree473b08590f0282b55f223df7199837fb44e9f39btargetblank473b085a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/473b08590f0282b55f223df7199837fb44e9f39b" target="_blank">473b085</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree473b08590f0282b55f223df7199837fb44e9f39btargetblank473b085a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/473b08590f0282b55f223df7199837fb44e9f39b" target="_blank">473b085</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  analysis/de_analysis_detailed_look_cache/
    Untracked:  analysis/de_analysis_detailed_look_more_cache/
    Untracked:  plots/

Unstaged changes:
    Modified:   analysis/de_analysis_detailed_look_more.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_analysis_detailed_look.Rmd</code>) and HTML (<code>docs/de_analysis_detailed_look.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/473b08590f0282b55f223df7199837fb44e9f39b/analysis/de_analysis_detailed_look.Rmd" target="_blank">473b085</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b7503caa46b8ec7259f7fadd1b7ed59cd2c57864/docs/de_analysis_detailed_look.html" target="_blank">b7503ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
<td>
Added ROC curves to the de_analysis_detailed_look analysis, and made
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/caa23002bf964a3122bc83e09da0303fd26d7c12/analysis/de_analysis_detailed_look.Rmd" target="_blank">caa2300</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f61736275ff234e1a6d2e90ffff39650bcc6982e/analysis/de_analysis_detailed_look.Rmd" target="_blank">f617362</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
<td>
Revised text accompanying power vs. fdr plot in de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/01ba9892207b9e79aba461be6863d5cb9e313f46/analysis/de_analysis_detailed_look.Rmd" target="_blank">01ba989</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
<td>
Revised the KL density plot and first fdr-vs-power curve in the analysis/de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2b5780a574f73a75892b238be9e7db51e5126f63/analysis/de_analysis_detailed_look.Rmd" target="_blank">2b5780a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-25
</td>
<td>
Made some improvements to the detailed DE investigations.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fc5c27a1354254ab752224dbad56770cfa0ff0ce/analysis/de_analysis_detailed_look.Rmd" target="_blank">fc5c27a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-25
</td>
<td>
Added initial code for ROC plot.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/297d0d1382ae7a22a1cbd9ec5959381bce465831/docs/de_analysis_detailed_look.html" target="_blank">297d0d1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Fixed link in de_analysis_detailed_look page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/53da33b0b77d5ead706041b653e3181fd265fd95/analysis/de_analysis_detailed_look.Rmd" target="_blank">53da33b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bd91aec3383cd4086332a613ffaea2b9220d7942/analysis/de_analysis_detailed_look.Rmd" target="_blank">bd91aec</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Fixed small typo in text.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/fe1290c070b0699230cea42826ab568291caba1e/docs/de_analysis_detailed_look.html" target="_blank">fe1290c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Revised accompanying text in de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/71a19efd9035cc73f7549163a4e14b90711616e6/docs/de_analysis_detailed_look.html" target="_blank">71a19ef</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Revised accompanying text in de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/32d547965b92f4dab3300ba2a2f01ed62380016f/analysis/de_analysis_detailed_look.Rmd" target="_blank">32d5479</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d9557e4e60e5692593648fae88a2271943adf843/analysis/de_analysis_detailed_look.Rmd" target="_blank">d9557e4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
Fixed typo.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/6a25fa860a3a6e9f4b90d11f8281aff552d032bf/docs/de_analysis_detailed_look.html" target="_blank">6a25fa8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
Added more detailed analysis of Monte Carlo estimates to
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/825dad09d9284253a368e453a205c0c7efcba1d0/analysis/de_analysis_detailed_look.Rmd" target="_blank">825dad0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7682717241c7257148bd41e88a3799fd70e69a5b/docs/de_analysis_detailed_look.html" target="_blank">7682717</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/63c16ca8c9b3353e0ce00321ed0d274d4b8c9e25/analysis/de_analysis_detailed_look.Rmd" target="_blank">63c16ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/67b97a814f316d28162d2d01a3f03d88401d9e32/docs/de_analysis_detailed_look.html" target="_blank">67b97a8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
Re-built de_analysis_detailed_look with fastTopics 0.6.74.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/633212a89bedd89b14041302363440e89ce8a7fd/analysis/de_analysis_detailed_look.Rmd" target="_blank">633212a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/90e13523c4f15b31376b40fe1d0754435a4f8aa8/docs/de_analysis_detailed_look.html" target="_blank">90e1352</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
Added summary to top of de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ff01162d1a1d6088cf9a0fdde01e126ad73c5a38/analysis/de_analysis_detailed_look.Rmd" target="_blank">ff01162</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/67e137efb2944b8d69ac8a25cbc7b16efecb868a/analysis/de_analysis_detailed_look.Rmd" target="_blank">67e137e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
Renamed de_eval_functions.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c483c156c4d26f1174108f5ae5bc8bfa819ed111/docs/de_analysis_detailed_look.html" target="_blank">c483c15</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
A few refinements to the de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/35cee4ebe05a380360d1809a7805be3068faae67/analysis/de_analysis_detailed_look.Rmd" target="_blank">35cee4e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/25798e0533999ba6a12349de8e738ee343388668/docs/de_analysis_detailed_look.html" target="_blank">25798e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
Added plots comparing deseq2 vs fastTopics in de_analysis_detailed_look.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1a55a9bbfdb0c09d6cc948bf2b816779055bb8a9/analysis/de_analysis_detailed_look.Rmd" target="_blank">1a55a9b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/365563fdd0769e961df1024180575cd09d56ec07/analysis/de_analysis_detailed_look.Rmd" target="_blank">365563f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
Working on comparison with DESeq2 in de_analysis_detailed_look.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/365563fdd0769e961df1024180575cd09d56ec07/docs/de_analysis_detailed_look.html" target="_blank">365563f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
Working on comparison with DESeq2 in de_analysis_detailed_look.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2e9deef49419c2cce2a5525f99590037cde925c6/analysis/de_analysis_detailed_look.Rmd" target="_blank">2e9deef</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
Made a few small improvements to the de_analysis_detailed_look analysis; mostly revised the accompanying text.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/090a487cb2ec24118864979753d9f0340a892bb8/docs/de_analysis_detailed_look.html" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
Rebuilt de_analysis_detailed_look with a new version of fastTopics
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/afaa55d75673dc33e791dc3dd1146009aed77613/analysis/de_analysis_detailed_look.Rmd" target="_blank">afaa55d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/13d2f7daf527bae566bdf5d91ac8520f2123e988/analysis/de_analysis_detailed_look.Rmd" target="_blank">13d2f7d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
Added pvalue/lfsr histograms for deseq2 and fastTopics.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/09089ff5307dccd9302c4ad67f64eb234fa8933c/analysis/de_analysis_detailed_look.Rmd" target="_blank">09089ff</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
Revising de_analysis_detailed_look after fixing z-score calculations in de_analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/f61da64befacd98d80cd829e9a8b387b810c5782/docs/de_analysis_detailed_look.html" target="_blank">f61da64</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
Fixed a bug in the power vs fdr plot in de_analysis_detailed_look.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/cb60f4c9bd3d3929348dbe440fe2d8800f1ba2a6/analysis/de_analysis_detailed_look.Rmd" target="_blank">cb60f4c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/3d62d9e57a5e422d2f42f9100c220bc6f5aedf46/docs/de_analysis_detailed_look.html" target="_blank">3d62d9e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
Revised plots and explanatory text in de_analysis_detailed_look.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5d5c6a6c0cbd79d51a68c626f6dabab1fa6ea1af/analysis/de_analysis_detailed_look.Rmd" target="_blank">5d5c6a6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/468f4d9fa4d881e4076c0ba5e16f47fde71ff731/docs/de_analysis_detailed_look.html" target="_blank">468f4d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
Added draft plots to de_analysis_detailed_look investigation.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/eceb899ccb4524fe9b3afc10b3a2f9f86ae9e2d0/analysis/de_analysis_detailed_look.Rmd" target="_blank">eceb899</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/de_analysis_detailed_look.html" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
Revised the data simulation and added some “shrink vs. noshrink” plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6d639935d7369c9e29089d07777877134d21ce22/analysis/de_analysis_detailed_look.Rmd" target="_blank">6d63993</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
Made another simplification to simulate_twotopic_umi_data and added plots to de_analysis_detailed_look analysis comparing results with and without shrinkage.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/157f7d3a004b5b065d83bbfaae50db2e4eff3804/analysis/de_analysis_detailed_look.Rmd" target="_blank">157f7d3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
<td>
Working on improved script driving_genes_better.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/af856a2297d146d651175053b4f585eed647d894/docs/de_analysis_detailed_look.html" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
<td>
Fixed the plots showing properties of the simulated data in the
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2726a8651c8e07596cf2de0d4bbff1fe60d8d21d/analysis/de_analysis_detailed_look.Rmd" target="_blank">2726a86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/de_analysis_detailed_look.html" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
<td>
Added some basic histograms to the de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e2577f16eb889cb6059447c4e46224282d9f28d4/analysis/de_analysis_detailed_look.Rmd" target="_blank">e2577f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
<td>
Working on the data simulation step in the de_analysis_detailed_look
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/cd17ccf4bfcf90a1a9844f5daa7ee3bf22f9f1e0/docs/de_analysis_detailed_look.html" target="_blank">cd17ccf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-09
</td>
<td>
Initial build of the de_analysis_detailed_look workflowr page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8e1fe8504c6d408234a97f2fb8bdcd5727c6bff/analysis/de_analysis_detailed_look.Rmd" target="_blank">b8e1fe8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-09
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/888ea7d132ef94726f5db395a8b089e052ee1295/analysis/de_analysis_detailed_look.Rmd" target="_blank">888ea7d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-08
</td>
<td>
I have an initial implementation of function simulate_twotopic_scrnaseq_data used to evaluate the de_analysis methods in fastTopics.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we evaluate the new topic-model-based DE methods implemented in the fastTopics package using simulated data. We show the benefits of shrinking the estimates using adaptive shrinkage and quantifying uncertainty. We also compare with DESeq2 and show that the results are very similar when the cells separate into distinct clusters. In this first analysis, we focus on the case of <span class="math inline">\(K = 2\)</span> topics, and in the <a href="de_analysis_detailed_look_more.html">second analysis</a> we evaluate the new DE methods with more than two topics.</p>
<p>Load the packages used in the analysis below, and some additional functions for simulating the data.</p>
<pre class="r"><code>library(Matrix)
library(scran)
library(DESeq2)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/de_analysis_functions.R&quot;)</code></pre>
<div id="simulate-umi-counts" class="section level2">
<h2>Simulate UMI counts</h2>
<p>We begin by simulating counts from a rank-2 Poisson NMF model with parameters chosen to roughly mimic the UMI counts from a single-cell RNA sequencing experiment. For now, we focus our attention on the rank-2 (<em>i.e.</em>, two topics) case to simplify the evaluation of the DE methods; comparing gene expression between three or topics brings some additional complications which aren’t necessary for assessing basic properties of the new DE methods. For more details on how the data are simulated, see function <code>simulate_twotopic_umi_data</code>.</p>
<pre class="r"><code>set.seed(1)
dat &lt;- simulate_twotopic_umi_data()
X &lt;- dat$X
F &lt;- dat$F
L &lt;- dat$L</code></pre>
<p>Before fitting a topic model and running a DE analysis, we first inspect some basic properties of the simulated data.</p>
<p>The sample sizes (total counts for each cell) are roughly normal on the (base-10) log scale:</p>
<pre class="r"><code>s &lt;- rowSums(X)
ggplot(data.frame(s = log10(s)),aes(x = s)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 32) +
  labs(x = &quot;log10 size&quot;,y = &quot;cells&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/sizes-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sizes-1">
Past versions of sizes-1.png
</button>
</p>
<div id="fig-sizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/365563fdd0769e961df1024180575cd09d56ec07/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">365563f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/af856a2297d146d651175053b4f585eed647d894/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The expression rates were simulated so that they are normally distributed on the log scale:</p>
<pre class="r"><code>ggplot(data.frame(f = as.vector(F)),aes(x = log10(f))) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 32) +
  labs(x = &quot;log10 expression rate&quot;,y = &quot;genes&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gene-expression-rates-1">
Past versions of gene-expression-rates-1.png
</button>
</p>
<div id="fig-gene-expression-rates-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/365563fdd0769e961df1024180575cd09d56ec07/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">365563f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/af856a2297d146d651175053b4f585eed647d894/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>About half of the genes have nonzero differences in expression between the two topics. Among the nonzero gene expression differences, the log-fold changes (LFCs) were simulated from the normal distribution:</p>
<pre class="r"><code>nonzero_lfc &lt;- abs(F[,1] - F[,2]) &gt; 1e-8
lfc &lt;- log2(F[,2]/F[,1])
ggplot(data.frame(lfc = lfc[nonzero_lfc]),aes(x = lfc)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 32) +
  labs(x = &quot;LFC&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
mean(nonzero_lfc)
# [1] 0.496</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lfc-density-1">
Past versions of lfc-density-1.png
</button>
</p>
<div id="fig-lfc-density-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/365563fdd0769e961df1024180575cd09d56ec07/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">365563f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/af856a2297d146d651175053b4f585eed647d894/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fit-multinomial-topic-model-to-umi-counts" class="section level2">
<h2>Fit multinomial topic model to UMI counts</h2>
<p>Now we fit a multinomial topic model to the simulated UMI count data. To simplify evaluation, we assume that the topic proportions are known, and fix them to their ground-truth values. In this way, the only error that can arise is in the estimates of the expression rates <span class="math inline">\(f_{ij}\)</span>.</p>
<pre class="r"><code>fit0 &lt;- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit &lt;- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = &quot;scd&quot;,
                       update.loadings = NULL,verbose = &quot;none&quot;)
fit &lt;- poisson2multinom(fit)
summary(fit)
# Model overview:
#   Number of data rows, n: 200
#   Number of data cols, m: 10000
#   Rank/number of topics, k: 2
# Evaluation of model fit (40 updates performed):
#   Poisson NMF log-likelihood: -7.169037374429e+05
#   Multinomial topic model log-likelihood: -7.158926302076e+05
#   Poisson NMF deviance: +8.564731575439e+05
#   Max KKT residual: +4.123695e-06</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="de-analysis-with-and-without-shrinkage" class="section level2">
<h2>DE analysis with and without shrinkage</h2>
<p>We perform a DE analysis <em>without</em> shrinking the LFC estimates,</p>
<pre class="r"><code>set.seed(1)
de.noshrink &lt;- de_analysis(fit,X,shrink.method = &quot;none&quot;,
                           control = list(ns = 10000,nc = 2))
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>and we perform a second DE analysis using adaptive shrinkage to shrink (and hopefully improve accuracy of) the LFC estimates,</p>
<pre class="r"><code>set.seed(1)
de &lt;- de_analysis(fit,X,shrink.method = &quot;ash&quot;,control = list(ns = 1e4,nc = 2))
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>This next scatterplot compares the posterior mean LFC estimates with and without performing the shrinkage step. As expected, the smaller (in magnitude) LFC estimates and the LFC estimates corresponding to genes with lower expression tend to be more strongly shrunk toward zero.</p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink  = de.noshrink$postmean[,2],
                   shrink    = de$postmean[,2],
                   log10mean = log10(de$f0))
ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = -4) +
  labs(x = &quot;original LFC estimates&quot;,
       y = &quot;shrunken LFC estimates&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-scatterplot-1">
Past versions of shrink-vs-noshrink-scatterplot-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-scatterplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3d62d9e57a5e422d2f42f9100c220bc6f5aedf46/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">3d62d9e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/468f4d9fa4d881e4076c0ba5e16f47fde71ff731/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">468f4d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The next two bar charts show the overall impact of the shrinkage on the distribution of the <span class="math inline">\(z\)</span>-scores; dark blue bars are for diifferentially expressed genes, and orange bars are for non-differentially expressed genes. (For better visualizataion the <span class="math inline">\(z\)</span>-scores have been projected onto the interval <span class="math inline">\([-8,+8]\)</span>.)</p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink = clamp(de.noshrink$z[,2],-8,+8),
                   shrink   = clamp(de$z[,2],-8,+8),
                   de       = factor(nonzero_lfc))
p1 &lt;- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-zscores-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-zscores-1">
Past versions of shrink-vs-noshrink-zscores-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-zscores-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-zscores-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/468f4d9fa4d881e4076c0ba5e16f47fde71ff731/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-zscores-1.png" target="_blank">468f4d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-zscores-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And the next two bar charts show the overall impact of the shrinkage on the <span class="math inline">\(p\)</span>-values.</p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink = 10^(-de.noshrink$lpval[,2]),
                   shrink   = 10^(-de$lpval[,2]),
                   de       = factor(nonzero_lfc))
p1 &lt;- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-pvalues-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-pvalues-1">
Past versions of shrink-vs-noshrink-pvalues-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-pvalues-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-pvalues-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/468f4d9fa4d881e4076c0ba5e16f47fde71ff731/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-pvalues-1.png" target="_blank">468f4d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-pvalues-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Both the <span class="math inline">\(z\)</span>-score and <span class="math inline">\(p\)</span>-value plots show that the adaptive shrinkage step more strongly encourages the LFC estimates for the non-DE genes toward zero.</p>
</div>
<div id="quantifying-uncertainty-in-the-lfc-estimates" class="section level2">
<h2>Quantifying uncertainty in the LFC estimates</h2>
<p>Above, we assessed the usefulness of applying the adaptive shrinkage method to improve the accuracy of the LFC estimates. This idea of shrinking the LFC estimates is not new, and is already implemented in DESeq2 (also using adaptive shrinkage).</p>
<p>A more important aspect of the DE analysis methods in fastTopics is that the methods also quantify uncertainty in the LFC estimates by computing posterior HPD intervals. To see why this is important, consider this plot comparing the MLEs of the LFCs against the shrunken posterior mean estimates of the LFCs. What is striking is that even very large (in magnitude) point estimates are sometimes shrunk to zero; this happens because although the MLEs can be far from zero, the HPD intervals are also very large.</p>
<pre class="r"><code>pdat &lt;- data.frame(est       = de$est[,2],
                   shrink    = de$postmean[,2],
                   log10mean = log10(de$f0))
ggplot(pdat,aes(x = est,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = -4) +
  labs(x = &quot;point estimates&quot;,
       y = &quot;shrunken estimates&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/point-vs-shrunken-scatterplot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-point-vs-shrunken-scatterplot-1">
Past versions of point-vs-shrunken-scatterplot-1.png
</button>
</p>
<div id="fig-point-vs-shrunken-scatterplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/point-vs-shrunken-scatterplot-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f61da64befacd98d80cd829e9a8b387b810c5782/docs/figure/de_analysis_detailed_look.Rmd/point-vs-shrunken-scatterplot-1.png" target="_blank">f61da64</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3d62d9e57a5e422d2f42f9100c220bc6f5aedf46/docs/figure/de_analysis_detailed_look.Rmd/point-vs-shrunken-scatterplot-1.png" target="_blank">3d62d9e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To illustrate the impact of this on the DE analysis, consider the K-L divergence measure used in Dey, Hsiao &amp; Stephens (2017) to rank genes, a measure which doesn’t fully account for uncertainty in the LFCs. Since the K-L divergence is not a signed ranking measure like the <span class="math inline">\(z\)</span>-score, here we focus only on genes that are estimated to have greater expression in topic 2 compared to topic 1. (For better visualization, we show the distribution of the K-L divergences on the log scale, and any <span class="math inline">\(z\)</span>-scores larger than 4 are shown as 4.)</p>
<pre class="r"><code>i     &lt;- which(fit$F[,2] - fit$F[,1] &gt; -1e-8)
D     &lt;- fastTopics:::min_kl_poisson(fit$F)
pdat1 &lt;- data.frame(lkl = log10(D[,2] + 1e-8),de = factor(nonzero_lfc))
pdat2 &lt;- data.frame(z = clamp(de$z[,2],-4,+4),de = factor(nonzero_lfc))
pdat1 &lt;- pdat1[i,]
pdat2 &lt;- pdat2[i,]
p1 &lt;- ggplot(pdat1,aes(x = lkl,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;log10 K-L divergence&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(pdat2,aes(x = z,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kl-vs-fasttopics-1-1">
Past versions of kl-vs-fasttopics-1-1.png
</button>
</p>
<div id="fig-kl-vs-fasttopics-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b7503caa46b8ec7259f7fadd1b7ed59cd2c57864/docs/figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-1-1.png" target="_blank">b7503ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/25798e0533999ba6a12349de8e738ee343388668/docs/figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-1-1.png" target="_blank">25798e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Without fully accounting for uncertainty, sometimes genes with very large LFCs get highly ranked by the K-L divergence measure (while also being uncertain about these estimates), whereas this happens less often for the <span class="math inline">\(z\)</span>-scores.</p>
<p>To underscore the importance of accounting for uncertainty, suppose we used the KL-divergences and the <em>p</em>-values returned by <code>de_analysis(shrink.method = &quot;ash&quot;)</code> to discover differentially expressed genes. The power vs. FDR curves show that, at low false discovery rates (say, FDR &lt; 0.2), the rankings accounting for uncertainty in the LFC estimates (with or without shrinkage) perform much better than the ranking based on the KL divergence measure. We also see that using adaptive shrinkage produces a modest improvement. As before, to draw a fair comparison with the K-L divergence-based ranking, we focus only on genes that are estimated to have greater expression in topic 2 compared to topic 1.</p>
<pre class="r"><code>pdat1 &lt;- create_fdr_vs_power_curve(-D[i,2],nonzero_lfc[i],length.out = 140)
pdat2 &lt;- create_fdr_vs_power_curve(-de.noshrink$lpval[i,2],nonzero_lfc[i],
                                   length.out = 140)
pdat3 &lt;- create_fdr_vs_power_curve(-de$lpval[i,2],nonzero_lfc[i],
                                   length.out = 140)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;kl&quot;),
               cbind(pdat2,method = &quot;noshrink&quot;),
               cbind(pdat3,method = &quot;shrink&quot;))
ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_line(size = 0.65)  +
  scale_color_manual(values = c(&quot;royalblue&quot;,&quot;limegreen&quot;,&quot;darkorange&quot;)) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-2-1.png" width="510" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kl-vs-fasttopics-2-1">
Past versions of kl-vs-fasttopics-2-1.png
</button>
</p>
<div id="fig-kl-vs-fasttopics-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b7503caa46b8ec7259f7fadd1b7ed59cd2c57864/docs/figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-2-1.png" target="_blank">b7503ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For a different perspective we also show the ROC curve comparinig the true positive rate (TPR) against the false positive rate (FPR):</p>
<pre class="r"><code>pdat1 &lt;- create_roc_curve(-D[,2],nonzero_lfc,length.out = 100)
pdat2 &lt;- create_roc_curve(-de.noshrink$lpval[,2],nonzero_lfc,length.out = 100)
pdat3 &lt;- create_roc_curve(-de$lpval[,2],nonzero_lfc,length.out = 100)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;kl&quot;),
               cbind(pdat2,method = &quot;noshrink&quot;),
               cbind(pdat3,method = &quot;shrink&quot;))
ggplot(pdat,aes(x = fpr,y = tpr,color = method)) +
  geom_line(size = 0.65) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_color_manual(values = c(&quot;royalblue&quot;,&quot;limegreen&quot;,&quot;darkorange&quot;)) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-3-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kl-vs-fasttopics-3-1">
Past versions of kl-vs-fasttopics-3-1.png
</button>
</p>
<div id="fig-kl-vs-fasttopics-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b7503caa46b8ec7259f7fadd1b7ed59cd2c57864/docs/figure/de_analysis_detailed_look.Rmd/kl-vs-fasttopics-3-1.png" target="_blank">b7503ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="assessing-accuracy-of-the-monte-carlo-estimates" class="section level2">
<h2>Assessing accuracy of the Monte Carlo estimates</h2>
<p>We use a simple MCMC algorithm to estimate LFC posterior statistics. To assess accuracy of the posterior calculations, we erform the DE analysis twice, and compare the <span class="math inline">\(z\)</span>-scores returned by the two <code>de_analysis</code> runs. If the <span class="math inline">\(z\)</span>-scores are very consistent between the two runs this suggests that they are mostly accurate.</p>
<p>To illustrate, we first compute the posteriors using 1,000 Monte Carlo samples:</p>
<pre class="r"><code>set.seed(1); de1 &lt;- de_analysis(fit,X,control = list(ns = 1000,nc = 2))
# Using 2 RcppParallel threads.
# Using 2 threads.
set.seed(2); de2 &lt;- de_analysis(fit,X,control = list(ns = 1000,nc = 2))
# Using 2 RcppParallel threads.
# Using 2 threads.
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.</code></pre>
<p>Now we compare the <span class="math inline">\(z\)</span>-scores estimated by these two runs in a simple scatterplot:</p>
<pre class="r"><code>pdat &lt;- data.frame(x = de1$z[,2],y = de2$z[,2])
ggplot(pdat,aes(x = x,y = y,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,fill = &quot;royalblue&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  labs(x = &quot;first z-score estimate&quot;,
       y = &quot;second z-score estimate&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/assess-monte-carlo-2-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-assess-monte-carlo-2-1">
Past versions of assess-monte-carlo-2-1.png
</button>
</p>
<div id="fig-assess-monte-carlo-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7682717241c7257148bd41e88a3799fd70e69a5b/docs/figure/de_analysis_detailed_look.Rmd/assess-monte-carlo-2-1.png" target="_blank">7682717</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here we see that the concordance is quite strong, but we should be able to improve the accuracy by running a longer Markov chain. Let’s try simulating a longer Markov chain with 10,000 iterations:</p>
<pre class="r"><code>set.seed(1); de1 &lt;- de_analysis(fit,X,control = list(ns = 10000,nc = 2))
# Using 2 RcppParallel threads.
# Using 2 threads.
set.seed(2); de2 &lt;- de_analysis(fit,X,control = list(ns = 10000,nc = 2))
# Using 2 RcppParallel threads.
# Using 2 threads.
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Indeed, with 10,000 Monte Carlo samples the two sets of <span class="math inline">\(z\)</span>-score estimates are now very similar:</p>
<pre class="r"><code>pdat &lt;- data.frame(x = de1$z[,2],y = de2$z[,2])
ggplot(pdat,aes(x = x,y = y,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,fill = &quot;royalblue&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  labs(x = &quot;first z-score estimate&quot;,
       y = &quot;second z-score estimate&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/assess-monte-carlo-4-1.png" width="390" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-assess-monte-carlo-4-1">
Past versions of assess-monte-carlo-4-1.png
</button>
</p>
<div id="fig-assess-monte-carlo-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a25fa860a3a6e9f4b90d11f8281aff552d032bf/docs/figure/de_analysis_detailed_look.Rmd/assess-monte-carlo-4-1.png" target="_blank">6a25fa8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7682717241c7257148bd41e88a3799fd70e69a5b/docs/figure/de_analysis_detailed_look.Rmd/assess-monte-carlo-4-1.png" target="_blank">7682717</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="comparison-with-deseq2" class="section level2">
<h2>Comparison with DESeq2</h2>
<p>DESeq2 is a leading method for DE analysis in single-cell RNA sequencing data. Here we apply DESeq2 to these same UMI count data, and show that the fastTopics DE results are very similar to DESeq2. Since DESeq2 assumes discrete cell types or conditions, we apply DESeq2 to the subset of cells that were are almost entirely (&gt;99%) simulated from one topic; in this simulation, approximatmely 2/3 of the cells were simulated from one topic (entirely topic 1 or entirely topic 2), and the remaining 1/3 were simulated from a mixture of the two topics (with the mixture proportion being uniformly distributed on <span class="math inline">\([0,1]\)</span>). There may be a small loss of accuracy in the DESeq2 results since DESeq2 is using only 2/3 of the data, but overall we expect that the fastTopics DE analysis should closely replicate the DESeq2 analysis.</p>
<pre class="r"><code>single_topic &lt;- apply(L,1,max) &gt; 0.99
mean(single_topic)
# [1] 0.655</code></pre>
<p>These next few lines of code prepare the UMI count data for analysis with DESeq2.</p>
<pre class="r"><code>i &lt;- which(single_topic)
Y &lt;- t(X[i,])
cluster &lt;- factor(apply(L,1,which.max))
deseq &lt;- DESeqDataSetFromMatrix(Y,data.frame(cluster = cluster[i]),~cluster)
# converting counts to integer mode</code></pre>
<p>Now we perform the DE analysis using DESeq2, using the settings recommended for single-cell RNA-seq data (see the main DESeq2 vignette for details). To replicate the fastTopics analysis as closely as possible, we also shrink the LFC estimates using adaptive shrinkage.</p>
<pre class="r"><code>sizeFactors(deseq) &lt;- calculateSumFactors(Y)
deseq &lt;- DESeq(deseq,test = &quot;LRT&quot;,reduced=~1,useT = TRUE,minmu = 1e-6,
               minReplicatesForReplace = Inf)
# using pre-existing size factors
# estimating dispersions
# gene-wise dispersion estimates
# mean-dispersion relationship
# -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
#    function: y = a/x + b, and a local regression fit was automatically substituted.
#    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.
# final dispersion estimates
# fitting model and testing
deseq &lt;- lfcShrink(deseq,coef = &quot;cluster_2_vs_1&quot;,type = &quot;ashr&quot;)
# using &#39;ashr&#39; for LFC shrinkage. If used in published research, please cite:
#     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.
#     https://doi.org/10.1093/biostatistics/kxw041</code></pre>
<p>The posterior mean estimates from the fastTopics DE analysis closely align with the DESeq2 estimates:</p>
<pre class="r"><code>pdat &lt;- data.frame(DESeq2     = deseq$log2FoldChange,
                   fastTopics = de$postmean[,2])
ggplot(pdat,aes(x = DESeq2,y = fastTopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = &quot;skyblue&quot;,linetype = &quot;dotted&quot;) +
  ggtitle(&quot;LFC estimates&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-lfc-1.png" width="360" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fasttopics-vs-deseq2-lfc-1">
Past versions of fasttopics-vs-deseq2-lfc-1.png
</button>
</p>
<div id="fig-fasttopics-vs-deseq2-lfc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/25798e0533999ba6a12349de8e738ee343388668/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-lfc-1.png" target="_blank">25798e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And likewise the <span class="math inline">\(z\)</span>-scores:</p>
<pre class="r"><code>deseq$z &lt;- with(deseq,log2FoldChange/lfcSE)
pdat &lt;- data.frame(DESeq2     = deseq$z,
                   fastTopics = de$z[,2])
ggplot(pdat,aes(x = DESeq2,y = fastTopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = &quot;skyblue&quot;,linetype = &quot;dotted&quot;) +
  ggtitle(&quot;z-scores&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-zscores-1.png" width="360" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fasttopics-vs-deseq2-zscores-1">
Past versions of fasttopics-vs-deseq2-zscores-1.png
</button>
</p>
<div id="fig-fasttopics-vs-deseq2-zscores-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/25798e0533999ba6a12349de8e738ee343388668/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-zscores-1.png" target="_blank">25798e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-zscores-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <span class="math inline">\(p\)</span>-value and LFSR distributions are also correspondingly similar in shape:</p>
<pre class="r"><code>pdat &lt;- data.frame(deseq2 = deseq$padj,
                   shrink = de$lfsr[,2],
                   de     = factor(nonzero_lfc))
p1 &lt;- ggplot(pdat,aes(x = deseq2,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;DESeq2&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;lfsr&quot;,y = &quot;genes&quot;,title = &quot;fastTopics&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-pvalues-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fasttopics-vs-deseq2-pvalues-1">
Past versions of fasttopics-vs-deseq2-pvalues-1.png
</button>
</p>
<div id="fig-fasttopics-vs-deseq2-pvalues-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/25798e0533999ba6a12349de8e738ee343388668/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-pvalues-1.png" target="_blank">25798e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/090a487cb2ec24118864979753d9f0340a892bb8/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-pvalues-1.png" target="_blank">090a487</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Based on these results, we would expect that the performance of fastTopics and DESeq2 for discovering differentially expressed genes is very similar, with a slight advantage given to fastTopics (although consider it is using 1/3 more data):</p>
<pre class="r"><code>pdat1 &lt;- create_fdr_vs_power_curve(deseq$padj,nonzero_lfc,length.out = 200)
pdat2 &lt;- create_fdr_vs_power_curve(de$lfsr[,2],nonzero_lfc,length.out = 200)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;DESeq2&quot;),
               cbind(pdat2,method = &quot;fastTopics&quot;))
ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_line(size = 0.65)  +
  scale_color_manual(values = c(&quot;darkblue&quot;,&quot;tomato&quot;)) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-power-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fasttopics-vs-deseq2-power-1">
Past versions of fasttopics-vs-deseq2-power-1.png
</button>
</p>
<div id="fig-fasttopics-vs-deseq2-power-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b7503caa46b8ec7259f7fadd1b7ed59cd2c57864/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-power-1.png" target="_blank">b7503ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c483c156c4d26f1174108f5ae5bc8bfa819ed111/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-power-1.png" target="_blank">c483c15</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/25798e0533999ba6a12349de8e738ee343388668/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-power-1.png" target="_blank">25798e0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And here is the ROC curve comparing DESeq2 and fastTopics:</p>
<pre class="r"><code>pdat1 &lt;- create_roc_curve(deseq$padj,nonzero_lfc,length.out = 100)
pdat2 &lt;- create_roc_curve(de$lfsr[,2],nonzero_lfc,length.out = 100)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;DESeq2&quot;),
               cbind(pdat2,method = &quot;fastTopics&quot;))
ggplot(pdat,aes(x = fpr,y = tpr,color = method)) +
  geom_line(size = 0.65)  +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_color_manual(values = c(&quot;darkblue&quot;,&quot;tomato&quot;)) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-roc-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fasttopics-vs-deseq2-roc-1">
Past versions of fasttopics-vs-deseq2-roc-1.png
</button>
</p>
<div id="fig-fasttopics-vs-deseq2-roc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b7503caa46b8ec7259f7fadd1b7ed59cd2c57864/docs/figure/de_analysis_detailed_look.Rmd/fasttopics-vs-deseq2-roc-1.png" target="_blank">b7503ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
# [8] methods   base     
# 
# other attached packages:
#  [1] cowplot_1.0.0               ggplot2_3.3.5              
#  [3] fastTopics_0.6-74           DESeq2_1.33.5              
#  [5] scran_1.14.6                SingleCellExperiment_1.8.0 
#  [7] SummarizedExperiment_1.16.1 DelayedArray_0.12.3        
#  [9] BiocParallel_1.18.1         matrixStats_0.61.0         
# [11] Biobase_2.46.0              GenomicRanges_1.38.0       
# [13] GenomeInfoDb_1.22.1         IRanges_2.20.2             
# [15] S4Vectors_0.24.4            BiocGenerics_0.32.0        
# [17] Matrix_1.2-18              
# 
# loaded via a namespace (and not attached):
#   [1] Rtsne_0.15               ggbeeswarm_0.6.0         colorspace_1.4-1        
#   [4] ellipsis_0.3.2           rprojroot_1.3-2          XVector_0.26.0          
#   [7] BiocNeighbors_1.2.0      fs_1.3.1                 farver_2.0.1            
#  [10] MatrixModels_0.4-1       ggrepel_0.9.1            bit64_0.9-7             
#  [13] AnnotationDbi_1.48.0     fansi_0.4.0              splines_3.6.2           
#  [16] geneplotter_1.64.0       knitr_1.26               scater_1.14.6           
#  [19] jsonlite_1.7.2           workflowr_1.6.2          mcmc_0.9-6              
#  [22] annotate_1.64.0          ashr_2.2-51              httr_1.4.2              
#  [25] compiler_3.6.2           dqrng_0.2.1              backports_1.1.5         
#  [28] lazyeval_0.2.2           assertthat_0.2.1         limma_3.42.2            
#  [31] later_1.0.0              BiocSingular_1.2.2       prettyunits_1.1.1       
#  [34] htmltools_0.4.0          quantreg_5.54            tools_3.6.2             
#  [37] rsvd_1.0.2               igraph_1.2.5             coda_0.19-3             
#  [40] gtable_0.3.0             glue_1.4.2               GenomeInfoDbData_1.2.2  
#  [43] dplyr_1.0.7              Rcpp_1.0.7               vctrs_0.3.8             
#  [46] DelayedMatrixStats_1.6.1 xfun_0.11                stringr_1.4.0           
#  [49] lifecycle_1.0.0          irlba_2.3.3              statmod_1.4.34          
#  [52] XML_3.99-0.3             edgeR_3.28.1             zlibbioc_1.32.0         
#  [55] MASS_7.3-51.4            scales_1.1.0             hms_1.1.0               
#  [58] promises_1.1.0           SparseM_1.78             RColorBrewer_1.1-2      
#  [61] yaml_2.2.0               pbapply_1.5-1            memoise_1.1.0           
#  [64] gridExtra_2.3            stringi_1.4.3            RSQLite_2.2.0           
#  [67] SQUAREM_2017.10-1        genefilter_1.68.0        truncnorm_1.0-8         
#  [70] rlang_0.4.11             pkgconfig_2.0.3          bitops_1.0-6            
#  [73] invgamma_1.1             evaluate_0.14            lattice_0.20-38         
#  [76] purrr_0.3.4              labeling_0.3             htmlwidgets_1.5.1       
#  [79] bit_1.1-15.2             tidyselect_1.1.1         magrittr_2.0.1          
#  [82] R6_2.4.1                 generics_0.0.2           DBI_1.1.0               
#  [85] withr_2.4.2              pillar_1.6.2             whisker_0.4             
#  [88] mixsqp_0.3-46            survival_3.1-8           RCurl_1.98-1.2          
#  [91] tibble_3.1.3             crayon_1.4.1             utf8_1.1.4              
#  [94] plotly_4.9.2             rmarkdown_2.3            progress_1.2.2          
#  [97] viridis_0.5.1            locfit_1.5-9.4           grid_3.6.2              
# [100] data.table_1.12.8        blob_1.2.1               git2r_0.26.1            
# [103] digest_0.6.23            xtable_1.8-4             tidyr_1.1.3             
# [106] httpuv_1.5.2             MCMCpack_1.4-5           RcppParallel_4.4.2      
# [109] munsell_0.5.0            beeswarm_0.2.3           viridisLite_0.3.0       
# [112] vipor_0.4.5              quadprog_1.5-8</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
