<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>DE analysis using a topic model: evaluation using simulated data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DE analysis using a topic model: evaluation using simulated data</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreeeceb899ccb4524fe9b3afc10b3a2f9f86ae9e2d0targetblankeceb899a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/eceb899ccb4524fe9b3afc10b3a2f9f86ae9e2d0" target="_blank">eceb899</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreeeceb899ccb4524fe9b3afc10b3a2f9f86ae9e2d0targetblankeceb899a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/eceb899ccb4524fe9b3afc10b3a2f9f86ae9e2d0" target="_blank">eceb899</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  plots/

Unstaged changes:
    Modified:   code/de_eval_functions.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_analysis_detailed_look.Rmd</code>) and HTML (<code>docs/de_analysis_detailed_look.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/eceb899ccb4524fe9b3afc10b3a2f9f86ae9e2d0/analysis/de_analysis_detailed_look.Rmd" target="_blank">eceb899</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/de_analysis_detailed_look.html" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
Revised the data simulation and added some “shrink vs. noshrink” plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6d639935d7369c9e29089d07777877134d21ce22/analysis/de_analysis_detailed_look.Rmd" target="_blank">6d63993</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
<td>
Made another simplification to simulate_twotopic_umi_data and added plots to de_analysis_detailed_look analysis comparing results with and without shrinkage.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/157f7d3a004b5b065d83bbfaae50db2e4eff3804/analysis/de_analysis_detailed_look.Rmd" target="_blank">157f7d3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
<td>
Working on improved script driving_genes_better.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/af856a2297d146d651175053b4f585eed647d894/docs/de_analysis_detailed_look.html" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
<td>
Fixed the plots showing properties of the simulated data in the
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2726a8651c8e07596cf2de0d4bbff1fe60d8d21d/analysis/de_analysis_detailed_look.Rmd" target="_blank">2726a86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/de_analysis_detailed_look.html" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
<td>
Added some basic histograms to the de_analysis_detailed_look analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e2577f16eb889cb6059447c4e46224282d9f28d4/analysis/de_analysis_detailed_look.Rmd" target="_blank">e2577f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
<td>
Working on the data simulation step in the de_analysis_detailed_look
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/cd17ccf4bfcf90a1a9844f5daa7ee3bf22f9f1e0/docs/de_analysis_detailed_look.html" target="_blank">cd17ccf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-09
</td>
<td>
Initial build of the de_analysis_detailed_look workflowr page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8e1fe8504c6d408234a97f2fb8bdcd5727c6bff/analysis/de_analysis_detailed_look.Rmd" target="_blank">b8e1fe8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-09
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/888ea7d132ef94726f5db395a8b089e052ee1295/analysis/de_analysis_detailed_look.Rmd" target="_blank">888ea7d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-08
</td>
<td>
I have an initial implementation of function simulate_twotopic_scrnaseq_data used to evaluate the de_analysis methods in fastTopics.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><em>Add summary of the analysis here.</em></p>
<p>Load the packages used in the analysis below, and some additional functions for simulating the data.</p>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/de_eval_functions.R&quot;)</code></pre>
<div id="simulate-umi-counts" class="section level2">
<h2>Simulate UMI counts</h2>
<p>We begin by simulating counts from a rank-2 Poisson NMF model with parameters chosen to roughly mimic the UMI counts from a single-cell RNA sequencing experiment. In particular, we simulate counts <span class="math inline">\(x_{ij} \sim \mathrm{Poisson}(\lambda_{ij})\)</span> such that <span class="math inline">\(\lambda_{ij} = \sum_{k=1}^K s_i l_{ik} f_{jk}\)</span>, with <span class="math inline">\(K = 2\)</span>. For now, we focus our attention on the <span class="math inline">\(K = 2\)</span> case to simplify the evaluation of the DE methods; comparing gene expression between three or topics brings some additional complications which aren’t necessary for assessing basic properties of the new DE methods. For more details on how the data are simulated, see function <code>simulate_twotopic_umi_data</code>.</p>
<pre class="r"><code>set.seed(1)
dat &lt;- simulate_twotopic_umi_data()
X &lt;- dat$X
F &lt;- dat$F
L &lt;- dat$L</code></pre>
<p>Before fitting a topic model and running a DE analysis, we first inspect some basic properties of the simulated data.</p>
<p>The sample sizes (total counts for each cell) are roughly normal on the (base-10) log scale:</p>
<pre class="r"><code>s &lt;- rowSums(X)
p1 &lt;- ggplot(data.frame(s = log10(s)),aes(x = s)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 32) +
  labs(x = &quot;log10 size&quot;,y = &quot;cells&quot;) +
  theme_cowplot()
print(p1)</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/sizes-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-sizes-1">
Past versions of sizes-1.png
</button>
</p>
<div id="fig-sizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/af856a2297d146d651175053b4f585eed647d894/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/figure/de_analysis_detailed_look.Rmd/sizes-1.png" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The expression rates were simulated so that they are normally distributed on the log scale:</p>
<pre class="r"><code>p2 &lt;- ggplot(data.frame(f = as.vector(F)),aes(x = log10(f))) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 32) +
  labs(x = &quot;log10 expression rate&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
print(p2)</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-gene-expression-rates-1">
Past versions of gene-expression-rates-1.png
</button>
</p>
<div id="fig-gene-expression-rates-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/af856a2297d146d651175053b4f585eed647d894/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/figure/de_analysis_detailed_look.Rmd/gene-expression-rates-1.png" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>About half of the genes have nonzero differences in expression between the two topics. Among the nonzero gene expression differences, the log-fold changes (LFCs) were simulated from the normal distribution:</p>
<pre class="r"><code>nonzero_lfc &lt;- abs(F[,1] - F[,2]) &gt; 1e-8
lfc &lt;- log2(F[,1]/F[,2])
p3 &lt;- ggplot(data.frame(lfc = lfc[nonzero_lfc]),aes(x = lfc)) +
  geom_histogram(color = &quot;white&quot;,fill = &quot;black&quot;,bins = 32) +
  labs(x = &quot;LFC&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
print(p3)
mean(nonzero_lfc)
# [1] 0.496</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lfc-density-1">
Past versions of lfc-density-1.png
</button>
</p>
<div id="fig-lfc-density-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/af856a2297d146d651175053b4f585eed647d894/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">af856a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b8eadee74bec2b96540bc9b4ca80482e3ef22942/docs/figure/de_analysis_detailed_look.Rmd/lfc-density-1.png" target="_blank">b8eadee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fit-multinomial-topic-model-to-umi-counts" class="section level2">
<h2>Fit multinomial topic model to UMI counts</h2>
<p>Here we fit a multinomial topic model to the simulated UMI count data. To simplify evaluation of the DE methods, we assume that the topic proportions are known, and set them to their ground truth values (that is, the values used to simulate the data), so that the only error that can arise is in the estimates of the expression rates <span class="math inline">\(f_{ij}\)</span> in the multinomial topic model.</p>
<pre class="r"><code>fit0 &lt;- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit &lt;- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = &quot;scd&quot;,
                       update.loadings = NULL,verbose = &quot;none&quot;)
fit &lt;- poisson2multinom(fit)
summary(fit)
# Model overview:
#   Number of data rows, n: 200
#   Number of data cols, m: 10000
#   Rank/number of topics, k: 2
# Evaluation of model fit (40 updates performed):
#   Poisson NMF log-likelihood: -7.169037374429e+05
#   Multinomial topic model log-likelihood: -7.158926302076e+05
#   Poisson NMF deviance: +8.564731575439e+05
#   Max KKT residual: +4.123695e-06</code></pre>
</div>
<div id="de-analysis-with-and-without-shrinkage" class="section level2">
<h2>DE analysis with and without shrinkage</h2>
<p>First we perform a DE analysis <em>without</em> shrinking the LFC estimates:</p>
<pre class="r"><code>set.seed(1)
de.noshrink &lt;- de_analysis(fit,X,shrink.method = &quot;none&quot;,
                           control = list(ns = 1e4,nc = 4))
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.</code></pre>
<p>Now we perform a second DE analysis using adaptive shrinkage to shrink (and hopefully improve accuracy of) the LFC estimates:</p>
<pre class="r"><code>set.seed(1)
de &lt;- de_analysis(fit,X,shrink.method = &quot;ash&quot;,control = list(ns = 1e4,nc = 4))
# Fitting 10000 Poisson models with k=2 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=2.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.</code></pre>
<p><em>Add text here.</em></p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink  = de.noshrink$postmean[,1],
                   shrink    = de$postmean[,1],
                   log10mean = log10(de$f0))
p4 &lt;- ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = -4) +
  labs(x = &quot;original LFC estimates&quot;,
       y = &quot;shrunken LFC estimates&quot;) +
  theme_cowplot()
print(p4)</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-scatterplot-1">
Past versions of shrink-vs-noshrink-scatterplot-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-scatterplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em>Add text here.</em></p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink = clamp(de.noshrink$z[,1],-4,+4),
                   shrink   = clamp(de$z[,1],-4,+4),
                   de       = factor(nonzero_lfc))
p5 &lt;- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p6 &lt;- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
print(plot_grid(p5,p6))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-zscores-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-zscores-1">
Past versions of shrink-vs-noshrink-zscores-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-zscores-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-zscores-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em>Add text here.</em></p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink = 10^(-de.noshrink$lpval[,1]),
                   shrink   = 10^(-de$lpval[,1]),
                   de       = factor(nonzero_lfc))
p7 &lt;- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p8 &lt;- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
print(plot_grid(p7,p8))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-pvalues-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-pvalues-1">
Past versions of shrink-vs-noshrink-pvalues-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-pvalues-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7236187363d75fb8477a6a6c7c2b726f371083e0/docs/figure/de_analysis_detailed_look.Rmd/shrink-vs-noshrink-pvalues-1.png" target="_blank">7236187</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em>Add text here.</em></p>
<pre class="r"><code>i &lt;- which(fit$F[,1] - fit$F[,2] &gt; -1e-8)
D &lt;- fastTopics:::min_kl_poisson(fit$F)
pdat &lt;- data.frame(kl = pmin(D[,1],0.00025),
                   de = factor(nonzero_lfc))
pdat1 &lt;- pdat[i,]
p9 &lt;- ggplot(pdat1,aes(x = kl,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;K-L divergence&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
pdat2 &lt;- data.frame(z  = clamp(de$z[,1],-4,+4),
                    de = factor(nonzero_lfc))
pdat2 &lt;- pdat2[i,]
p10 &lt;- ggplot(pdat2,aes(x = z,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
print(plot_grid(p9,p10))</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/etc-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdat &lt;- data.frame(est       = de$est[,1],
                   shrink    = de$postmean[,1],
                   log10mean = log10(de$f0))
p11 &lt;- ggplot(pdat,aes(x = est,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = -4) +
  labs(x = &quot;point estimates&quot;,
       y = &quot;shrunken posterior estimates&quot;) +
  theme_cowplot()
print(p11)</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/etc-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdat1 &lt;- create_fdr_vs_power_curve(-D[,1],nonzero_lfc)
pdat2 &lt;- create_fdr_vs_power_curve(-de.noshrink$lpval,nonzero_lfc)
pdat3 &lt;- create_fdr_vs_power_curve(-de$lpval,nonzero_lfc)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;kl&quot;),
               cbind(pdat2,method = &quot;noshrink&quot;),
               cbind(pdat3,method = &quot;shrink&quot;))
p12 &lt;- ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_point(size = 0.75)  +
  scale_color_manual(values = c(&quot;royalblue&quot;,&quot;limegreen&quot;,&quot;darkorange&quot;)) +
  theme_cowplot()
print(p12)</code></pre>
<p><img src="figure/de_analysis_detailed_look.Rmd/etc-3,%20fig.height-1.png" width="540" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0     ggplot2_3.3.5     fastTopics_0.6-70 Matrix_1.2-18    
# 
# loaded via a namespace (and not attached):
#  [1] httr_1.4.2         tidyr_1.1.3        jsonlite_1.7.2     viridisLite_0.3.0 
#  [5] RcppParallel_4.4.2 assertthat_0.2.1   mixsqp_0.3-46      yaml_2.2.0        
#  [9] progress_1.2.2     ggrepel_0.9.1      pillar_1.6.2       backports_1.1.5   
# [13] lattice_0.20-38    quantreg_5.54      glue_1.4.2         quadprog_1.5-8    
# [17] digest_0.6.23      promises_1.1.0     colorspace_1.4-1   htmltools_0.4.0   
# [21] httpuv_1.5.2       pkgconfig_2.0.3    invgamma_1.1       SparseM_1.78      
# [25] purrr_0.3.4        scales_1.1.0       whisker_0.4        later_1.0.0       
# [29] Rtsne_0.15         MatrixModels_0.4-1 git2r_0.26.1       tibble_3.1.3      
# [33] farver_2.0.1       generics_0.0.2     ellipsis_0.3.2     withr_2.4.2       
# [37] ashr_2.2-51        lazyeval_0.2.2     magrittr_2.0.1     crayon_1.4.1      
# [41] mcmc_0.9-6         evaluate_0.14      fs_1.3.1           fansi_0.4.0       
# [45] MASS_7.3-51.4      truncnorm_1.0-8    tools_3.6.2        data.table_1.12.8 
# [49] prettyunits_1.1.1  hms_1.1.0          lifecycle_1.0.0    stringr_1.4.0     
# [53] MCMCpack_1.4-5     plotly_4.9.2       munsell_0.5.0      irlba_2.3.3       
# [57] compiler_3.6.2     rlang_0.4.11       grid_3.6.2         htmlwidgets_1.5.1 
# [61] labeling_0.3       rmarkdown_2.3      gtable_0.3.0       DBI_1.1.0         
# [65] R6_2.4.1           knitr_1.26         dplyr_1.0.7        utf8_1.1.4        
# [69] workflowr_1.6.2    rprojroot_1.3-2    stringi_1.4.3      parallel_3.6.2    
# [73] SQUAREM_2017.10-1  Rcpp_1.0.7         vctrs_0.3.8        tidyselect_1.1.1  
# [77] xfun_0.11          coda_0.19-3</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
