<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Differential expression analysis using a topic model: illustration in mixture of FACS-purified PBMC data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential expression analysis using a topic model: illustration in mixture of FACS-purified PBMC data</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree9fcc15c96215475bec2005681bada954265a3fd2targetblank9fcc15ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/9fcc15c96215475bec2005681bada954265a3fd2" target="_blank">9fcc15c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree9fcc15c96215475bec2005681bada954265a3fd2targetblank9fcc15ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/9fcc15c96215475bec2005681bada954265a3fd2" target="_blank">9fcc15c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  analysis/de_analysis_detailed_look_cache/
    Untracked:  analysis/de_analysis_detailed_look_more_cache/
    Untracked:  plots/

Unstaged changes:
    Modified:   analysis/temp6.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_analysis_purified_pbmc.Rmd</code>) and HTML (<code>docs/de_analysis_purified_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9fcc15c96215475bec2005681bada954265a3fd2/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">9fcc15c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-24
</td>
<td>
workflowr::wflow_publish(“de_analysis_purified_pbmc.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7b343625d0cd1703c01264f3e02fa9aac173c620/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">7b34362</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-24
</td>
<td>
Working on comparison of deseq2 and fasttopics in B cells in de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6fbc554ad02c880ae6f30ebdb757663763551df1/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">6fbc554</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-24
</td>
<td>
Working on deseq2 vs fasttopics comparison in de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c1e09319e31e07d33c7987d22effa810f529a461/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">c1e0931</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Revised explanatory test in de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fb81385c61542066eac866572060790b8bfca4a0/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">fb81385</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Made a few improvements to the volcano plot in de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a11826f6195b1ef2317b667c7da4dce9469f02bd/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">a11826f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Added scatterplot to de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/161d0e92a73270a74ffcc18886f76c67828872f4/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">161d0e9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Made a few improvements the b-cells scatterplot in the de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/760f6ad2b614361a13c0833ae4e07c6814c51e51/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">760f6ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Added a z-score q-q plot to de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5fa98f8d53a7df09c136d86b584428e632396055/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">5fa98f8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Working on de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/5fa98f8d53a7df09c136d86b584428e632396055/docs/de_analysis_purified_pbmc.html" target="_blank">5fa98f8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
<td>
Working on de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c30816e5db07a8b957f8eea91318412be3791378/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">c30816e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
<td>
A few small edits.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0d0c72069543e91cacb46f4034c361e1a4c8f50f/docs/de_analysis_purified_pbmc.html" target="_blank">0d0c720</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
<td>
Added scatterplots assessing accuracy of Monte Carlo estimates to
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f7a5a86cbff1d6abde9046f9d263ee261b03ff65/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">f7a5a86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
<td>
workflowr::wflow_publish(“de_analysis_purified_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/1917832caf043fcd580e128dc4915c404d95cd3f/docs/de_analysis_purified_pbmc.html" target="_blank">1917832</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
<td>
Working on de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/90c6584edade80ea1f4392583675771808dd6951/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">90c6584</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
<td>
workflowr::wflow_publish(“de_analysis_purified_pbmc.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3143481a5759eb096693c41d872670d9b4d836a7/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">3143481</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
<td>
Working on the de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a16fdf93ae172583a248a13b476c9981f2fe3775/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">a16fdf9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-20
</td>
<td>
Added structure plot to de_analysis_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e1ab3a0a4cba6e154daf48f9fa51fe7f30e28e1a/docs/de_analysis_purified_pbmc.html" target="_blank">e1ab3a0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-08
</td>
<td>
Built the initial de_analysis_purified_pbmc analysis page.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/2befadb7655acd2fd1be173572effe9c7480df6c/docs/de_analysis_purified_pbmc.html" target="_blank">2befadb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-08
</td>
<td>
Added link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/71e267dff418062e1061ad83c5a54f78322c2a01/analysis/de_analysis_purified_pbmc.Rmd" target="_blank">71e267d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-08
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>The aim of this analysis is to understand by way of illustration the differences between a “classical” differential expresion analysis comparing expression among cell types (here we implement this “classical” DE analysis using <a href="https://github.com/mikelove/DESeq2">DESEq2</a>), and a differential expression analysis using the topic model, which allows for grades of membership to cell types (or more generally cellular expression factors).</p>
<p>Begin by loading the packages and some function definitions used in the analysis.</p>
<pre class="r"><code>library(Matrix)
library(DESeq2)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source(&quot;../code/de_analysis_functions.R&quot;)</code></pre>
<div id="load-count-data-and-topic-model-fit" class="section level2">
<h2>Load count data and topic model fit</h2>
<p>Load the UMI count data for 94,655 cells and 21,952 genes.</p>
<pre class="r"><code>load(&quot;../data/pbmc_purified.RData&quot;)
dim(counts)
# [1] 94655 21952</code></pre>
<p>Load the <span class="math inline">\(K = 6\)</span> Poisson NMF model fit to these data, and convert the Poisson NMF model fit to a multinomial topic model fit.</p>
<pre class="r"><code>fit &lt;- readRDS(file.path(&quot;../output/pbmc-purified/rds&quot;,
                         &quot;fit-pbmc-purified-scd-ex-k=6.rds&quot;))$fit
fit &lt;- poisson2multinom(fit)</code></pre>
<p>The cells are subdivided, based on FACS sorting, into 10 “cell types”. Several of the cell types are virtually indistinguishable based on their gene expression profiles alone, so we combine these indistinguishable cell types into a single “T cell” cell type. This results in 5 predefined cell types, the majority of which are T cells:</p>
<pre class="r"><code>set.seed(1)
celltype &lt;- as.character(samples$celltype)
celltype[celltype == &quot;CD4+ T Helper2&quot; |
         celltype == &quot;CD4+/CD45RO+ Memory&quot; |
         celltype == &quot;CD8+/CD45RA+ Naive Cytotoxic&quot; |
         celltype == &quot;CD4+/CD45RA+/CD25- Naive T&quot; |
         celltype == &quot;CD8+ Cytotoxic T&quot; |
         celltype == &quot;CD4+/CD25 T Reg&quot;] &lt;- &quot;T cell&quot;
celltype &lt;- factor(celltype,
                   c(&quot;CD19+ B&quot;,&quot;CD14+ Monocyte&quot;,&quot;CD34+&quot;,&quot;CD56+ NK&quot;,&quot;T cell&quot;))
table(celltype)
# celltype
#        CD19+ B CD14+ Monocyte          CD34+       CD56+ NK         T cell 
#          10085           2612           9232           8385          64341</code></pre>
</div>
<div id="structure-plot" class="section level2">
<h2>Structure plot</h2>
<p>Next we visualize the structure inferred by the $K = $ topic model using a “structure plot”. The cells in this plot are arranged horizontally according to their predefined cell type to relate the topics to these predefinend cell types:</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;gold&quot;,&quot;forestgreen&quot;,&quot;dodgerblue&quot;,&quot;gray&quot;,
                  &quot;darkmagenta&quot;,&quot;violet&quot;)
topics &lt;- c(5,3,2,4,1,6)
rows &lt;- sort(c(sample(which(celltype == &quot;CD19+ B&quot;),500),
               sample(which(celltype == &quot;CD14+ Monocyte&quot;),250),
               sample(which(celltype == &quot;CD34+&quot;),500),
               sample(which(celltype == &quot;CD56+ NK&quot;),400),
               sample(which(celltype == &quot;T cell&quot;),1000)))
p1 &lt;- structure_plot(select_loadings(fit,loadings = rows),
                     grouping = celltype[rows],
                     topics = topics,colors = topic_colors[topics],
                     perplexity = c(70,30,30,30,70),n = Inf,gap = 30,
                     num_threads = 4,verbose = FALSE)
print(p1)</code></pre>
<p><img src="figure/de_analysis_purified_pbmc.Rmd/structure-plot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-1">
Past versions of structure-plot-1.png
</button>
</p>
<div id="fig-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1917832caf043fcd580e128dc4915c404d95cd3f/docs/figure/de_analysis_purified_pbmc.Rmd/structure-plot-1.png" target="_blank">1917832</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Some of the topics correspond very closely to the predefined cell types. In particular, topics 1 through 4 closely correspond, respectively, to T cells, CD14+ monocytes (myeloid cells), B cells and natural killer (NK) cells.</p>
<p>Topic 5 (violet) closely corresponds to the “CD34+” FACS cell type, but from the structure plot we observe many cells labeled as “CD34+” with little to no contribution from topic 5, which suggests mislabeling of the CD34+ cells.</p>
<p>Topic 6 (magenta) does not correspond to any FACS cell type and as we will see it captures a different characteristic of the cells—specifically, abundance of ribosomal protein genes. Therefore, the DE results for topics 1–4 are most comparable to a classical DE analysis, and we begin with these comparisons. But before doing this we first assess accuracy of the MCMC computations used in the topic-model-based DE analysis.</p>
</div>
<div id="assessing-accuracy-of-the-monte-carlo-estimates" class="section level2">
<h2>Assessing accuracy of the Monte Carlo estimates</h2>
<p>The topic-model-based DE analysis was performed previously by simulating the posterior distribution of the LFC statistics via MCMC. Here we assess accuracy of the MCMC calculations. We performed the DE analysis twice (using different seeds, each with 100,000 Monte Carlo samples), so we can compare the posterior mean estimates and <em>z</em>-scores returned by the two <code>de_analysis</code> runs.</p>
<pre class="r"><code>load(&quot;../output/pbmc-purified/de-pbmc-purified-seed=1.RData&quot;)
de1 &lt;- de
load(&quot;../output/pbmc-purified/de-pbmc-purified-seed=2.RData&quot;)
de2 &lt;- de
rm(de)</code></pre>
<p>The MCMC estimates of the posterior mean log-fold change (LFC) estimates are largely consistent:</p>
<pre class="r"><code>pdat &lt;- data.frame(postmean1 = as.vector(de1$postmean),
                   postmean2 = as.vector(de2$postmean))
ggplot(pdat,aes(x = postmean1,y = postmean2)) +
  geom_point(shape = 21,color = &quot;white&quot;,fill = &quot;black&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;magenta&quot;,linetype = &quot;dashed&quot;) +
  labs(x = &quot;first posterior mean&quot;,y = &quot;second posterior mean&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_purified_pbmc.Rmd/assess-monte-carlo-1-1.png" width="390" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-assess-monte-carlo-1-1">
Past versions of assess-monte-carlo-1-1.png
</button>
</p>
<div id="fig-assess-monte-carlo-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5fa98f8d53a7df09c136d86b584428e632396055/docs/figure/de_analysis_purified_pbmc.Rmd/assess-monte-carlo-1-1.png" target="_blank">5fa98f8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0d0c72069543e91cacb46f4034c361e1a4c8f50f/docs/figure/de_analysis_purified_pbmc.Rmd/assess-monte-carlo-1-1.png" target="_blank">0d0c720</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <span class="math inline">\(z\)</span>-scores on the other hand are estimated less consistently, presumably because accurately estimating uncertainty is harder. Still, the <span class="math inline">\(z\)</span>-scores are still are consistent enough in that it is rare for an LFC to have an lfsr less than 0.05 in one MCMC simulation and not the other (these are the red points in the scatterplot). Note for better visualization <span class="math inline">\(z\)</span>-scores larger than 100 (or smaller than -100) are shown as 100 (or -100) in this plot.</p>
<pre class="r"><code>pdat &lt;- data.frame(z1 = clamp(as.vector(de1$z),-100,+100),
                   z2 = clamp(as.vector(de2$z),-100,+100),
                   lfsr = factor((de1$lfsr &lt; 0.05) + (de2$lfsr &lt; 0.05)))
ggplot(pdat,aes(x = z1,y = z2,fill = lfsr)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;magenta&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_manual(values = c(&quot;darkblue&quot;,&quot;tomato&quot;,&quot;dodgerblue&quot;),
                    na.value = &quot;white&quot;) +
  labs(x = &quot;first z-score&quot;,y = &quot;second z-score&quot;,fill = &quot;lfsr &lt; 0.05&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_purified_pbmc.Rmd/assess-monte-carlo-3-1.png" width="510" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-assess-monte-carlo-3-1">
Past versions of assess-monte-carlo-3-1.png
</button>
</p>
<div id="fig-assess-monte-carlo-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5fa98f8d53a7df09c136d86b584428e632396055/docs/figure/de_analysis_purified_pbmc.Rmd/assess-monte-carlo-3-1.png" target="_blank">5fa98f8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0d0c72069543e91cacb46f4034c361e1a4c8f50f/docs/figure/de_analysis_purified_pbmc.Rmd/assess-monte-carlo-3-1.png" target="_blank">0d0c720</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Moving forward, when the two <em>z</em>-scores disagree, we use the one that is nearer to zero.</p>
<pre class="r"><code>de             &lt;- de1[c(&quot;f0&quot;,&quot;postmean&quot;,&quot;z&quot;,&quot;lfsr&quot;)]
class(de)      &lt;- c(&quot;topic_model_de_analysis&quot;,&quot;list&quot;)
i              &lt;- which(abs(de2$z) &lt; abs(de1$z))
de$postmean[i] &lt;- de2$postmean[i]
de$z[i]        &lt;- de2$z[i]
de$lfsr[i]     &lt;- de$lfsr[i]</code></pre>
</div>
<div id="load-deseq2-results-for-all-cell-types" class="section level2">
<h2>Load DESeq2 results for all cell types</h2>
<p>We load the results of the DESeq2 analyes, and combine them into two data frames: a data frame for the posterior mean LFC estimates, and a data frame for the <em>z</em>-scores.</p>
<pre class="r"><code>load(&quot;../output/pbmc-purified/deseq2-pbmc-purified.RData&quot;)
celltypes &lt;- names(deseq)
n &lt;- length(celltypes)
p &lt;- nrow(genes)
deseq2 &lt;- list(postmean = matrix(0,p,n),
               z        = matrix(0,p,n))
rownames(deseq2$postmean) &lt;- genes$ensembl
rownames(deseq2$z)        &lt;- genes$ensembl
colnames(deseq2$postmean) &lt;- celltypes
colnames(deseq2$z)        &lt;- celltypes
for (i in 1:n) {
  deseq2$postmean[,i] &lt;- deseq[[i]]$log2FoldChange
  deseq2$z[,i]        &lt;- with(deseq[[i]],log2FoldChange/lfcSE)
}
deseq &lt;- deseq2
rm(deseq2)</code></pre>
<p>Since we filtered out a few lowly expressed genes before running DESeq2, we subset the fastTopics results to match up with DESeq2.</p>
<pre class="r"><code>rows        &lt;- match(rownames(deseq$z),rownames(de$z))
de$f0       &lt;- de$f0[rows]
de$postmean &lt;- de$postmean[rows,]
de$z        &lt;- de$z[rows,]
de$lfsr     &lt;- de$lfsr[rows,]</code></pre>
</div>
<div id="deseq2-vs.fasttopics" class="section level2">
<h2>DESeq2 vs. fastTopics</h2>
<p>Comparing the distributions of all <span class="math inline">\(z\)</span>-scores, we see that the DE analysis allowing for grades of membership has many more <span class="math inline">\(z\)</span>-scores near zero, yet still has (Because there are a few extremely large and extremely small <span class="math inline">\(z\)</span>-scores, for better visualization z-scores larger than 100 in magnitude are shown as 100 or -100.)</p>
<pre class="r"><code>pdat &lt;-
  data.frame(deseq     =quantile(deseq$z,seq(0,1,length.out=1000),na.rm=TRUE),
             fasttopics=quantile(de$z,seq(0,1,length.out=1000),na.rm=TRUE))
pdat &lt;- transform(pdat,
                  deseq      = clamp(deseq,-100,+100),
                  fasttopics = clamp(fasttopics,-100,+100))
ggplot(pdat,aes(x = deseq,y = fasttopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = &quot;magenta&quot;,linetype = &quot;dotted&quot;) +
  labs(x = &quot;DESeq2&quot;,y = &quot;fastTopics&quot;,title = &quot;z-score quantiles&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_purified_pbmc.Rmd/z-score-scatterplot-1.png" width="390" style="display: block; margin: auto;" /></p>
<p>Now let’s look closely at the DESeq2 and fastTopics results for B cells, one of the cell types that closely corresponds to a topic (topic 3). Here we focus on genes for which the <span class="math inline">\(z\)</span>-score is greater than 2 in at least one of the analyses. Genes are colored according to the LFSR estimated in the fastTopics analysis ().</p>
<pre class="r"><code>i &lt;- &quot;CD19+ B&quot;
k &lt;- &quot;k3&quot;
pdat &lt;- data.frame(gene                = genes$symbol,
                   postmean.deseq      = deseq$postmean[,i],
                   postmean.fasttopics = de$postmean[,k],
                   z.deseq             = deseq$z[,i],
                   z.fasttopics        = de$z[,k],
                   lfsr = cut(de$lfsr[,k],c(-1,0.001,0.01,0.05,Inf)),
                   stringsAsFactors = FALSE)
j &lt;- which(pdat$postmean.fasttopics &lt; 8)
pdat[j,&quot;gene&quot;] &lt;- &quot;&quot;
pdat[genes$symbol == &quot;HOPX&quot;,&quot;gene&quot;] &lt;- &quot;HOPX&quot;
pdat &lt;- subset(pdat,abs(z.deseq) &gt; 2 | abs(z.fasttopics) &gt; 2)
ggplot(pdat,aes(x = postmean.deseq,y = postmean.fasttopics,
                fill = lfsr,label = gene)) +
  geom_point(shape = 21,color = &quot;white&quot;) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  geom_text_repel(color = &quot;darkgray&quot;,size = 2.25,fontface = &quot;italic&quot;,
                  segment.color = &quot;darkgray&quot;,segment.size = 0.25,
                  min.segment.length = 0,max.overlaps = Inf,na.rm = TRUE) +
  scale_fill_manual(values = c(&quot;deepskyblue&quot;,&quot;gold&quot;,&quot;orange&quot;,&quot;coral&quot;),
                    na.value = &quot;gainsboro&quot;) +
  xlim(-12.3,13.1) +
  ylim(-10,13.1) +
  labs(x = &quot;DESeq2&quot;,y = &quot;fastTopics&quot;,
       title = &quot;LFC in B cells&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_purified_pbmc.Rmd/bcells-z-scores-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>A few interesting themes emerge from this plot:</p>
<ul>
<li><p>Many genes with large LFC in the DESeq2, particularly genes with very negative LFCs (genes with large decreases in expression in B cells) have zero LFC in the fastTopics analysis.</p></li>
<li><p>Most of the LFC estimates in the fastTopics DE analysis agree with, but are lower than, the LFC estimates in the DESeq2 analysis. The one exception is the genes with the largest expression increases in B cells: both DESeq2 and fastTopics largely identifies the same genes with the largest increases in expression in B cells, but the LFC is higher for most of these genes in the fastTopics analysis.</p></li>
<li><p>I highlighted here the <em>HOPX</em> gene because this is a gene that was identified in the fastTopics analysis, but not DESeq2 analysis, presumably because this gene is more strongly expressed in cells that are not actually labeled as “B cell” by FACS.</p></li>
</ul>
<p>The fastTopics DE analysis for the “B cells” topic is summarized in the following volcano plot:</p>
<pre class="r"><code>pdat &lt;- data.frame(gene     = genes$symbol,
                   postmean = de$postmean[,k],
                   z        = pmin(150,abs(de$z[,k])),
                   lfsr     = cut(de$lfsr[,k],c(-1,0.001,0.01,0.05,Inf)),
                   stringsAsFactors = FALSE)
j &lt;- which(with(pdat,!(postmean &gt; 8 | z &gt; 75)))
pdat[j,&quot;gene&quot;] &lt;- &quot;&quot;
ggplot(pdat,aes(x = postmean,y = z,fill = lfsr,label = gene)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 1.5) +
  geom_text_repel(color = &quot;darkgray&quot;,size = 2.25,fontface = &quot;italic&quot;,
                  segment.color = &quot;darkgray&quot;,segment.size = 0.25,
                  min.segment.length = 0,max.overlaps = Inf,na.rm = TRUE) +
  scale_y_continuous(trans = &quot;sqrt&quot;,breaks = c(1,2,5,10,20,50,100)) +
  scale_fill_manual(values = c(&quot;deepskyblue&quot;,&quot;gold&quot;,&quot;orange&quot;,&quot;tomato&quot;),
                    na.value = &quot;gainsboro&quot;) +
  labs(x = &quot;posterior mean LFC&quot;,y = &quot;|z-score|&quot;) +
  theme_cowplot()
# Warning: Removed 323 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/de_analysis_purified_pbmc.Rmd/bcells-volcano-plot-1.png" width="540" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
# [8] methods   base     
# 
# other attached packages:
#  [1] cowplot_1.0.0               ggrepel_0.9.1              
#  [3] ggplot2_3.3.5               fastTopics_0.6-79          
#  [5] DESeq2_1.33.5               SummarizedExperiment_1.16.1
#  [7] DelayedArray_0.12.3         BiocParallel_1.18.1        
#  [9] matrixStats_0.61.0          Biobase_2.46.0             
# [11] GenomicRanges_1.38.0        GenomeInfoDb_1.22.1        
# [13] IRanges_2.20.2              S4Vectors_0.24.4           
# [15] BiocGenerics_0.32.0         Matrix_1.2-18              
# 
# loaded via a namespace (and not attached):
#  [1] Rtsne_0.15             colorspace_1.4-1       ellipsis_0.3.2        
#  [4] rprojroot_1.3-2        XVector_0.26.0         fs_1.3.1              
#  [7] farver_2.0.1           MatrixModels_0.4-1     bit64_0.9-7           
# [10] AnnotationDbi_1.48.0   fansi_0.4.0            splines_3.6.2         
# [13] geneplotter_1.64.0     knitr_1.26             jsonlite_1.7.2        
# [16] workflowr_1.6.2        mcmc_0.9-6             annotate_1.64.0       
# [19] ashr_2.2-51            compiler_3.6.2         httr_1.4.2            
# [22] backports_1.1.5        assertthat_0.2.1       lazyeval_0.2.2        
# [25] later_1.0.0            htmltools_0.4.0        quantreg_5.54         
# [28] prettyunits_1.1.1      tools_3.6.2            coda_0.19-3           
# [31] gtable_0.3.0           glue_1.4.2             GenomeInfoDbData_1.2.2
# [34] dplyr_1.0.7            Rcpp_1.0.7             vctrs_0.3.8           
# [37] xfun_0.11              stringr_1.4.0          lifecycle_1.0.0       
# [40] irlba_2.3.3            XML_3.99-0.3           zlibbioc_1.32.0       
# [43] MASS_7.3-51.4          scales_1.1.0           hms_1.1.0             
# [46] promises_1.1.0         SparseM_1.78           RColorBrewer_1.1-2    
# [49] yaml_2.2.0             memoise_1.1.0          pbapply_1.5-1         
# [52] stringi_1.4.3          RSQLite_2.2.0          SQUAREM_2017.10-1     
# [55] genefilter_1.68.0      truncnorm_1.0-8        rlang_0.4.11          
# [58] pkgconfig_2.0.3        bitops_1.0-6           evaluate_0.14         
# [61] lattice_0.20-38        invgamma_1.1           purrr_0.3.4           
# [64] htmlwidgets_1.5.1      labeling_0.3           bit_1.1-15.2          
# [67] tidyselect_1.1.1       magrittr_2.0.1         R6_2.4.1              
# [70] generics_0.0.2         DBI_1.1.0              pillar_1.6.2          
# [73] whisker_0.4            withr_2.4.2            survival_3.1-8        
# [76] RCurl_1.98-1.2         mixsqp_0.3-46          tibble_3.1.3          
# [79] crayon_1.4.1           utf8_1.1.4             plotly_4.9.2          
# [82] rmarkdown_2.3          progress_1.2.2         locfit_1.5-9.4        
# [85] grid_3.6.2             data.table_1.12.8      blob_1.2.1            
# [88] git2r_0.26.1           digest_0.6.23          xtable_1.8-4          
# [91] tidyr_1.1.3            httpuv_1.5.2           MCMCpack_1.4-5        
# [94] RcppParallel_4.4.2     munsell_0.5.0          viridisLite_0.3.0     
# [97] quadprog_1.5-8</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
