<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Evaluation of DE analysis with more than two topics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Evaluation of DE analysis with more than two topics</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
de-noshrink
</li>
<li>
de-with-ash
</li>
<li>
fit-poisson-nmf
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>de_analysis_detailed_look_more_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree17d0249a6e011bac52f9e66597beb7dd8d939b60targetblank17d0249a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/17d0249a6e011bac52f9e66597beb7dd8d939b60" target="_blank">17d0249</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree17d0249a6e011bac52f9e66597beb7dd8d939b60targetblank17d0249a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/17d0249a6e011bac52f9e66597beb7dd8d939b60" target="_blank">17d0249</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  analysis/de_analysis_detailed_look_cache/
    Untracked:  analysis/de_analysis_detailed_look_more_cache/
    Untracked:  plots/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/de_analysis_detailed_look_more.Rmd</code>) and HTML (<code>docs/de_analysis_detailed_look_more.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/17d0249a6e011bac52f9e66597beb7dd8d939b60/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">17d0249</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-26
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look_more.Rmd”,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2b5780a574f73a75892b238be9e7db51e5126f63/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">2b5780a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-25
</td>
<td>
Made some improvements to the detailed DE investigations.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9a8de60be67463472ddbb15e492237be00ed27bc/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">9a8de60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-25
</td>
<td>
A few revisions to de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/8445a1c125f93aace463d66d46d2a58165277cd3/docs/de_analysis_detailed_look_more.html" target="_blank">8445a1c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
One more small edit to the de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a65de9e7c10a6194c0696675c507a8056c27499d/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">a65de9e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look_more.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e3cc4a3acdf723a08f706bf02be25b3d666c4bb9/docs/de_analysis_detailed_look_more.html" target="_blank">e3cc4a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Fixed plots and added text to de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f746cb38c01345741ac11b701c409bf66da53902/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">f746cb3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look_more.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/9d6104692d58c818b5d33f9f15a09630b59e7261/docs/de_analysis_detailed_look_more.html" target="_blank">9d61046</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Revised text and plots in de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/503cca366a85e331949a323913caa7f4fd0f6515/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">503cca3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look_more.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/36acb71112107faba6b3030aa762bc469fb2d51e/docs/de_analysis_detailed_look_more.html" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Added some text to the de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a7f4b810df1dde08c7b24c9a361c349a0945b25a/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">a7f4b81</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look_more.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1e13200a73f548ed4417bdd9b7e52a337791b511/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">1e13200</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
Added plots to de_analysis_detailed_look_more.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0b25553d9322d42352ba22e9bf405127b78203dc/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">0b25553</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-19
</td>
<td>
Working on new de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bb56afa5d01752131109b14d1283a2edd7a9c944/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">bb56afa</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
Implemented function simulate_manytopic_umi_data.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/74088201a4bcf71d17ef605d2b717feb8a601d96/docs/de_analysis_detailed_look_more.html" target="_blank">7408820</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
First build of de_analysis_detailed_look_more analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d49ec084e52986c2af1bd0a36d85beb24eb73d00/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">d49ec08</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look_more.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7484c1447565f73781ff375cecfa281c3f6e2b40/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">7484c14</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
Added rmd for new analysis de_analysis_detailed_look_more.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/cb60f4c9bd3d3929348dbe440fe2d8800f1ba2a6/analysis/de_analysis_detailed_look_more.Rmd" target="_blank">cb60f4c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-14
</td>
<td>
workflowr::wflow_publish(“de_analysis_detailed_look.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we expand the evaluation of the new topic-model-based DE analysis methods to consider the case of more than two topics. When comparing multiple groups, clusters or topics it is less clear what it means for a gene to be “differentially expressed”. So to simplify the evaluation we simulate the data so that each gene is differentially expressed in at most one topic.</p>
<p>In this evaluation can no longer compare to DESeq2 because DESeq2 is designed to compare gene expression across two groups. Still, we can closely examine the benefit of accounting for uncertainty.</p>
<p>Load the packages used in the analysis below, and some additional functions for simulating the data.</p>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/de_analysis_functions.R&quot;)</code></pre>
<div id="simulate-umi-counts" class="section level2">
<h2>Simulate UMI counts</h2>
<p>We begin by simulating UMI count data from a rank-6 Poisson NMF model such that roughly half of the genes have the same expression rate across all topics, and the other half have a different rate of expression in exactly one out of the six topics. The details of the simulation are similar to the two-topic case.</p>
<pre class="r"><code>set.seed(1)
m &lt;- 10000
k &lt;- 6
dat &lt;- simulate_manytopic_umi_data(m = m,k = k)
X &lt;- dat$X
F &lt;- dat$F
L &lt;- dat$L</code></pre>
</div>
<div id="fit-multinomial-topic-model-to-umi-counts" class="section level2">
<h2>Fit multinomial topic model to UMI counts</h2>
<p>Now we fit a multinomial topic model to the simulated UMI count data. To simplify evaluation, we assume that the topic proportions are known, and fix them to their ground-truth values. In this way, the only error that can arise is in the estimates of the expression rates <span class="math inline">\(f_{ij}\)</span>.</p>
<pre class="r"><code>fit0 &lt;- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit &lt;- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = &quot;scd&quot;,
                       update.loadings = NULL,verbose = &quot;none&quot;)
fit &lt;- poisson2multinom(fit)
summary(fit)
# Model overview:
#   Number of data rows, n: 1000
#   Number of data cols, m: 10000
#   Rank/number of topics, k: 6
# Evaluation of model fit (40 updates performed):
#   Poisson NMF log-likelihood: -3.521271656222e+06
#   Multinomial topic model log-likelihood: -3.516218632828e+06
#   Poisson NMF deviance: +4.250115181654e+06
#   Max KKT residual: +4.432856e-04</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="de-analysis-with-and-without-shrinkage" class="section level2">
<h2>DE analysis with and without shrinkage</h2>
<p>First we perform a DE analysis <em>without</em> shrinking the LFC estimates:</p>
<pre class="r"><code>set.seed(1)
de.noshrink &lt;- de_analysis(fit,X,shrink.method = &quot;none&quot;,
                           control = list(ns = 10000,nc = 2))
# Fitting 10000 Poisson models with k=6 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=6.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Next we perform a second DE analysis using adaptive shrinkage to shrink the LFC estimates:</p>
<pre class="r"><code>set.seed(1)
de &lt;- de_analysis(fit,X,shrink.method = &quot;ash&quot;,control = list(ns = 1e4,nc = 2))
# Fitting 10000 Poisson models with k=6 using method=&quot;scd&quot;.
# Computing log-fold change statistics from 10000 Poisson models with k=6.
# Stabilizing posterior log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>To measure differential expression across multiple topics, we use the “least extreme” log-fold change, which, for topic <span class="math inline">\(k\)</span>, is defined as the LFC between topic <span class="math inline">\(k\)</span> and <span class="math inline">\(j \neq k\)</span> such that the expression ratio is closest to 1. Then we compare estimates of the “least extreme” LFCs with and without shrinkage.</p>
<p>As expected, smaller estimates, and estimates corresponding to genes with lower expression, tend to be more strongly shrunk toward zero.</p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink  = c(de.noshrink$postmean),
                   shrink    = c(de$postmean),
                   log10mean = rep(log10(de$f0),k))
ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = -4) +
  labs(x = &quot;original LFC estimates&quot;,
       y = &quot;shrunken LFC estimates&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-scatterplot-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-scatterplot-1">
Past versions of shrink-vs-noshrink-scatterplot-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-scatterplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9d6104692d58c818b5d33f9f15a09630b59e7261/docs/figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">9d61046</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/36acb71112107faba6b3030aa762bc469fb2d51e/docs/figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-scatterplot-1.png" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The next two bar charts show the overall impact of the shrinkage on the distribution of the <span class="math inline">\(z\)</span>-scores; dark blue bars are for diifferentially expressed genes, and orange bars are for non-differentially expressed genes. Note that for visualization purposes the <span class="math inline">\(z\)</span>-scores are projected onto the interval <span class="math inline">\([-4,+4]\)</span>.</p>
<pre class="r"><code>nonzero_lfc &lt;- matrix(FALSE,m,k)
for (i in 1:m) {
  y &lt;- dat$F[i,]
  if (max(y) - min(y) &gt; 1e-8) {
    j &lt;- which.max((y - mean(y))^2)
    nonzero_lfc[i,j] &lt;- TRUE
  }
}
pdat &lt;- data.frame(noshrink = clamp(c(de.noshrink$z),-4,+4),
                   shrink   = clamp(c(de$z),-4,+4),
                   de       = c(nonzero_lfc))
p1 &lt;- ggplot(subset(pdat,de),aes(x = noshrink,color = de)) +
  geom_histogram(color = &quot;darkblue&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  coord_cartesian(xlim = c(-4,+4),ylim = c(0,400)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(subset(pdat,de),aes(x = shrink,color = de)) +
  geom_histogram(color = &quot;darkblue&quot;,fill = &quot;darkblue&quot;,bins = 64) +
  coord_cartesian(xlim = c(-4,+4),ylim = c(0,400)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
p3 &lt;- ggplot(subset(pdat,!de),aes(x = noshrink,color = de)) +
  geom_histogram(color = &quot;darkorange&quot;,fill = &quot;darkorange&quot;,bins = 64) +
  coord_cartesian(xlim = c(-4,+4)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p4 &lt;- ggplot(subset(pdat,!de),aes(x = shrink,color = de)) +
  geom_histogram(color = &quot;darkorange&quot;,fill = &quot;darkorange&quot;,bins = 64) +
  coord_cartesian(xlim = c(-4,+4)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2,p3,p4,nrow = 2,ncol = 2))</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-zscores-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-zscores-1">
Past versions of shrink-vs-noshrink-zscores-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-zscores-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9d6104692d58c818b5d33f9f15a09630b59e7261/docs/figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-zscores-1.png" target="_blank">9d61046</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/36acb71112107faba6b3030aa762bc469fb2d51e/docs/figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-zscores-1.png" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And the next two bar charts show the overall impact of the shrinkage on the <span class="math inline">\(p\)</span>-values. (The Y axis in the “with shrinkage” plot has been cut off at 2,500 genes to show in more detail the smaller <span class="math inline">\(p\)</span>-values identifying true positives.)</p>
<pre class="r"><code>pdat &lt;- data.frame(noshrink = 10^(-c(de.noshrink$lpval)),
                   shrink   = 10^(-c(de$lpval)),
                   de       = factor(c(nonzero_lfc)))
p1 &lt;- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  coord_cartesian(ylim = c(0,2500)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;without shrinkage&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  coord_cartesian(ylim = c(0,2500)) +
  labs(x = &quot;p-value&quot;,y = &quot;genes&quot;,title = &quot;with shrinkage&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2))</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-pvalues-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-shrink-vs-noshrink-pvalues-1">
Past versions of shrink-vs-noshrink-pvalues-1.png
</button>
</p>
<div id="fig-shrink-vs-noshrink-pvalues-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9d6104692d58c818b5d33f9f15a09630b59e7261/docs/figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-pvalues-1.png" target="_blank">9d61046</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/36acb71112107faba6b3030aa762bc469fb2d51e/docs/figure/de_analysis_detailed_look_more.Rmd/shrink-vs-noshrink-pvalues-1.png" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Since most of the LFCs are zero, the adaptive shrinkage strongly shrinks most of the LFC estimates toward zero. Still, because it is adaptive, the DE genes with stronger support are shrunk only a little or not at all.</p>
</div>
<div id="quantifying-uncertainty-in-the-lfc-estimates" class="section level2">
<h2>Quantifying uncertainty in the LFC estimates</h2>
<p>Here we briefly assess the benefits of accounting for uncertainty in the LFC estimates. The benefits are more pronounced here because more topics brings more uncertainty to the estimates.</p>
<p>First we compare the point estimates to the (shrunk) posterior mean estimates:</p>
<pre class="r"><code>pdat &lt;- data.frame(est       = c(de$est),
                   shrink    = c(de$postmean),
                   log10mean = rep(log10(de$f0),k))
ggplot(pdat,aes(x = est,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = -4) +
  labs(x = &quot;point estimates&quot;,
       y = &quot;shrunken estimates&quot;) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/point-vs-shrunken-scatterplot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-point-vs-shrunken-scatterplot-1">
Past versions of point-vs-shrunken-scatterplot-1.png
</button>
</p>
<div id="fig-point-vs-shrunken-scatterplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/36acb71112107faba6b3030aa762bc469fb2d51e/docs/figure/de_analysis_detailed_look_more.Rmd/point-vs-shrunken-scatterplot-1.png" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As before, even very large (in magnitude) point estimates are sometimes shrunk to zero; this happens because although the MLEs can be far from zero, the HPD intervals can also be very large.</p>
<p>Now we compare a ranking of the LFCs based on the K-L divergence measure used in Dey, Hsiao &amp; Stephens (2017), which doesn’t fully account for uncertainty in the LFC estimates, to a ranking based on the <span class="math inline">\(z\)</span>-scores. Since the K-L divergence is not a signed ranking, here we restrict our attention to LFCs estimated to be positive. (Note that the Y axis in the right-hand plot has been cut off at 2,000 genes to give more detail to the distribution of the true positives. Also, for better visualization the K-L divergences are shown on the log scale.)</p>
<pre class="r"><code>D     &lt;- fastTopics:::min_kl_poisson(fit$F)
i     &lt;- which(de$est &gt; 0)
pdat1 &lt;- data.frame(lkl = log10(D[i] + 1e-8),de = factor(nonzero_lfc[i]))
pdat2 &lt;- data.frame(z = clamp(de$z[i],-4,+4),de = factor(nonzero_lfc[i]))
p1 &lt;- ggplot(pdat1,aes(x = lkl,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  labs(x = &quot;log10 K-L divergence&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
p2 &lt;- ggplot(pdat2,aes(x = z,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  scale_fill_manual(values = c(&quot;darkorange&quot;,&quot;darkblue&quot;)) +
  coord_cartesian(ylim = c(0,2000)) +
  labs(x = &quot;z-score&quot;,y = &quot;genes&quot;) +
  theme_cowplot()
print(plot_grid(p1,p2))</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kl-vs-fasttopics-1-1">
Past versions of kl-vs-fasttopics-1-1.png
</button>
</p>
<div id="fig-kl-vs-fasttopics-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e3cc4a3acdf723a08f706bf02be25b3d666c4bb9/docs/figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-1-1.png" target="_blank">e3cc4a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/36acb71112107faba6b3030aa762bc469fb2d51e/docs/figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-1-1.png" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Sometimes genes with very large LFCs get highly ranked by the K-L divergence measure (while being uncertain about these estimates), whereas this happens less often for the <span class="math inline">\(z\)</span>-scores. As a result the fastTopics <span class="math inline">\(z\)</span>-scores yield improved power at low false discovery rates:</p>
<pre class="r"><code>pdat1 &lt;- create_fdr_vs_power_curve(-D[i],nonzero_lfc[i],length.out = 400)
pdat2 &lt;- create_fdr_vs_power_curve(de$lfsr[i],nonzero_lfc[i],length.out = 400)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;kl&quot;),
               cbind(pdat2,method = &quot;fastTopics&quot;))
ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_line(size = 0.65)  +
  scale_color_manual(values = c(&quot;royalblue&quot;,&quot;darkorange&quot;)) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-2-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-kl-vs-fasttopics-2-1">
Past versions of kl-vs-fasttopics-2-1.png
</button>
</p>
<div id="fig-kl-vs-fasttopics-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e3cc4a3acdf723a08f706bf02be25b3d666c4bb9/docs/figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-2-1.png" target="_blank">e3cc4a3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/36acb71112107faba6b3030aa762bc469fb2d51e/docs/figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-2-1.png" target="_blank">36acb71</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Alternatively we can compare the two rankings in an ROC plot:</p>
<pre class="r"><code>pdat1 &lt;- create_roc_curve(-log(D[i]),nonzero_lfc[i],length.out = 100)
pdat2 &lt;- create_roc_curve(de$lfsr[i],nonzero_lfc[i],length.out = 100)
pdat  &lt;- rbind(cbind(pdat1,method = &quot;kl&quot;),
               cbind(pdat2,method = &quot;fastTopics&quot;))
ggplot(pdat,aes(x = fpr,y = tpr,color = method)) +
  geom_line(size = 0.75) +
  geom_abline(intercept = 0,slope = 1,color = &quot;black&quot;,linetype = &quot;dotted&quot;) +
  scale_color_manual(values = c(&quot;royalblue&quot;,&quot;darkorange&quot;)) +
  theme_cowplot()</code></pre>
<p><img src="figure/de_analysis_detailed_look_more.Rmd/kl-vs-fasttopics-3-1.png" width="540" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0     ggplot2_3.3.5     fastTopics_0.6-74 Matrix_1.2-18    
# 
# loaded via a namespace (and not attached):
#  [1] httr_1.4.2         tidyr_1.1.3        jsonlite_1.7.2     viridisLite_0.3.0 
#  [5] RcppParallel_4.4.2 assertthat_0.2.1   mixsqp_0.3-46      yaml_2.2.0        
#  [9] progress_1.2.2     ggrepel_0.9.1      pillar_1.6.2       backports_1.1.5   
# [13] lattice_0.20-38    quantreg_5.54      glue_1.4.2         quadprog_1.5-8    
# [17] digest_0.6.23      promises_1.1.0     colorspace_1.4-1   htmltools_0.4.0   
# [21] httpuv_1.5.2       pkgconfig_2.0.3    invgamma_1.1       SparseM_1.78      
# [25] purrr_0.3.4        scales_1.1.0       whisker_0.4        later_1.0.0       
# [29] Rtsne_0.15         MatrixModels_0.4-1 git2r_0.26.1       tibble_3.1.3      
# [33] farver_2.0.1       generics_0.0.2     ellipsis_0.3.2     withr_2.4.2       
# [37] ashr_2.2-51        pbapply_1.5-1      lazyeval_0.2.2     magrittr_2.0.1    
# [41] crayon_1.4.1       mcmc_0.9-6         evaluate_0.14      fs_1.3.1          
# [45] fansi_0.4.0        MASS_7.3-51.4      truncnorm_1.0-8    tools_3.6.2       
# [49] data.table_1.12.8  prettyunits_1.1.1  hms_1.1.0          lifecycle_1.0.0   
# [53] stringr_1.4.0      MCMCpack_1.4-5     plotly_4.9.2       munsell_0.5.0     
# [57] irlba_2.3.3        compiler_3.6.2     rlang_0.4.11       grid_3.6.2        
# [61] htmlwidgets_1.5.1  labeling_0.3       rmarkdown_2.3      gtable_0.3.0      
# [65] DBI_1.1.0          R6_2.4.1           knitr_1.26         dplyr_1.0.7       
# [69] utf8_1.1.4         workflowr_1.6.2    rprojroot_1.3-2    stringi_1.4.3     
# [73] parallel_3.6.2     SQUAREM_2017.10-1  Rcpp_1.0.7         vctrs_0.3.8       
# [77] tidyselect_1.1.1   xfun_0.11          coda_0.19-3</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
