<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Analysis of rare epithelial cell types in droplet data</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of rare epithelial cell types in droplet data</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreef05a9a0787b8b5d2ce5a05a955133558b83ae62etargetblankf05a9a0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/f05a9a0787b8b5d2ce5a05a955133558b83ae62e" target="_blank">f05a9a0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreef05a9a0787b8b5d2ce5a05a955133558b83ae62etargetblankf05a9a0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/f05a9a0787b8b5d2ce5a05a955133558b83ae62e" target="_blank">f05a9a0</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/
    Ignored:    output/sims/

Untracked files:
    Untracked:  analysis/de_analysis_detailed_look_cache/
    Untracked:  analysis/de_analysis_detailed_look_more_cache/
    Untracked:  analysis/structure-plot-pulseseq-merged.png
    Untracked:  plots/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/droplet_rare.Rmd</code>) and HTML (<code>docs/droplet_rare.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f05a9a0787b8b5d2ce5a05a955133558b83ae62e/analysis/droplet_rare.Rmd" target="_blank">f05a9a0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-12
</td>
<td>
workflowr::wflow_publish(“droplet_rare.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/323f198e6a987a8a97c55e4c85978e9dd437f3d7/docs/droplet_rare.html" target="_blank">323f198</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
Added volcano plots to droplet_rare analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1f5b204b24f19174e19f2070a83339ef6daa285b/analysis/droplet_rare.Rmd" target="_blank">1f5b204</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
workflowr::wflow_publish(“droplet_rare.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0ed33291c8a8b894fea4c4b927fb1ba7d379d317/docs/droplet_rare.html" target="_blank">0ed3329</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
Added structure plot to droplet_rare analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/096729f8478ca31aa51b65e9bdc97db7a7af71f5/analysis/droplet_rare.Rmd" target="_blank">096729f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
workflowr::wflow_publish(“droplet_rare.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a64f3eaa07942a5c302ebab68983fd663e16e6c6/docs/droplet_rare.html" target="_blank">a64f3ea</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
Added loglik-vs-k plot to droplet_rare analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1bd5cef02952634d8bc281e66e4926d5be4824af/analysis/droplet_rare.Rmd" target="_blank">1bd5cef</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
workflowr::wflow_publish(“droplet_rare.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b4b6ab07031228123d616e2865dbe6c535639c2e/analysis/droplet_rare.Rmd" target="_blank">b4b6ab0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
<td>
Working on droplet_rare analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0301b6ea0cff73d1eaf48dc34229fc96b14e6afd/docs/droplet_rare.html" target="_blank">0301b6e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-10
</td>
<td>
First build of droplet_rare analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/687fadc1c4352526ed9d486ac1fbd26fee6edbc3/analysis/droplet_rare.Rmd" target="_blank">687fadc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-10
</td>
<td>
workflowr::wflow_publish(“droplet_rare.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/271518c8015da204bd2e8b7ff7ab953bf8b5b804/docs/droplet_rare.html" target="_blank">271518c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-10
</td>
<td>
Added another link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ad0daeecd17120c05d791439b57a7292bf9d33ff/analysis/droplet_rare.Rmd" target="_blank">ad0daee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-10
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Based on the topic modeling results in the droplet data, with <span class="math inline">\(K = 7\)</span> topics, one of the topics, topic 6, appears to be capturing rare, specialized epithelial cells, including ionocyte, tuft and neuroendocrine cells. Therefore, here we reanalyze the cells with membership to this topic.</p>
<p>Load the packages used in the analysis.</p>
<pre class="r"><code>library(fastTopics)
library(ggplot2)
library(cowplot)</code></pre>
<p>Initialize the sequence of pseudorandom numbers.</p>
<pre class="r"><code>set.seed(1)</code></pre>
<p>Load the previously prepared count data.</p>
<pre class="r"><code>load(&quot;../data/droplet.RData&quot;)</code></pre>
<p>Load the results on the rare cell types.</p>
<pre class="r"><code>out       &lt;- readRDS(&quot;../output/droplet/refit-droplet.rds&quot;)
fits      &lt;- out$fits
de        &lt;- out$de
de_merged &lt;- out$de_merged</code></pre>
<p>Below we will look more closely at the topic model with <span class="math inline">\(K = 5\)</span> topics:</p>
<pre class="r"><code>fit &lt;- poisson2multinom(fits$k5)</code></pre>
<p>Find the subset of genes and samples that were used to fit the model:</p>
<pre class="r"><code>rownames(samples) &lt;- with(samples,paste(mouse.id,barcode,sep = &quot;_&quot;))
samples &lt;- samples[rownames(fit$L),]
counts  &lt;- counts[rownames(fit$L),rownames(fit$F)]
dim(counts)
# [1]   637 15594</code></pre>
<p>This plot shows the improvement in the log-likelihood as the rank, <span class="math inline">\(K\)</span>, is increased. The log-likelihoods are shown relative to the log-likelihood at <span class="math inline">\(K = 2\)</span>.</p>
<pre class="r"><code>plot_loglik_vs_rank(fits) +
  theme_cowplot(font_size = 12)</code></pre>
<p><img src="figure/droplet_rare.Rmd/plot-loglik-vs-k-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-loglik-vs-k-1">
Past versions of plot-loglik-vs-k-1.png
</button>
</p>
<div id="fig-plot-loglik-vs-k-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed33291c8a8b894fea4c4b927fb1ba7d379d317/docs/figure/droplet_rare.Rmd/plot-loglik-vs-k-1.png" target="_blank">0ed3329</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a64f3eaa07942a5c302ebab68983fd663e16e6c6/docs/figure/droplet_rare.Rmd/plot-loglik-vs-k-1.png" target="_blank">a64f3ea</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This PCA plot showing the top two PCs of the topic proportions shows that the topics correspond well with the clusters identified by Montoro <em>et al</em> (2018):</p>
<pre class="r"><code>clusters &lt;- as.character(samples$tissue)
clusters[clusters == &quot;Basal&quot; |
         clusters == &quot;Club&quot; |
         clusters == &quot;Goblet&quot;] &lt;- &quot;Other&quot;
clusters &lt;- factor(clusters)
tissue_colors &lt;- c(&quot;firebrick&quot;,   # ciliated
                   &quot;darkmagenta&quot;, # ionocyte
                   &quot;darkorange&quot;,  # neuroendocrine
                   &quot;gainsboro&quot;,   # other
                   &quot;skyblue&quot;)     # tuft
p &lt;- pca_plot(fit,pcs = 1:2,fill = clusters) +
  scale_fill_manual(values = tissue_colors)
# Scale for &#39;fill&#39; is already present. Adding another scale for &#39;fill&#39;, which
# will replace the existing scale.
print(p)</code></pre>
<p><img src="figure/droplet_rare.Rmd/pca-plot-1.png" width="510" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-plot-1">
Past versions of pca-plot-1.png
</button>
</p>
<div id="fig-pca-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed33291c8a8b894fea4c4b927fb1ba7d379d317/docs/figure/droplet_rare.Rmd/pca-plot-1.png" target="_blank">0ed3329</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(&quot;../plots/pca_cluster_droplet_rare.png&quot;,p,
       height = 3,width = 4,dpi = 450,bg = &quot;white&quot;)</code></pre>
<p>The structure plot summarizes the estimated topic proportions:</p>
<pre class="r"><code>set.seed(1)
topic_colors &lt;- c(&quot;skyblue&quot;,&quot;darkorange&quot;,&quot;red&quot;,&quot;violet&quot;,&quot;gainsboro&quot;)
topics &lt;- c(5,2,1,3,4)
p &lt;- structure_plot(fit,topics = topics,colors = topic_colors,
                    perplexity = 70,num_threads = 2,verbose = FALSE)
print(p)</code></pre>
<p><img src="figure/droplet_rare.Rmd/structure-plot-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-1">
Past versions of structure-plot-1.png
</button>
</p>
<div id="fig-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed33291c8a8b894fea4c4b927fb1ba7d379d317/docs/figure/droplet_rare.Rmd/structure-plot-1.png" target="_blank">0ed3329</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These volcano plots summarize the results of the GoM DE analysis after merging topics 3 and 4:</p>
<pre class="r"><code>p1 &lt;- volcano_plot(de_merged,k = &quot;k1&quot;,ymax = 50)
p2 &lt;- volcano_plot(de_merged,k = &quot;k2&quot;,ymax = 100)
p3 &lt;- volcano_plot(de_merged,k = &quot;k3+k4&quot;,ymax = 100)
p4 &lt;- volcano_plot(de_merged,k = &quot;k5&quot;,ymax = 100)
plot_grid(p1,p2,p3,p4)</code></pre>
<p><img src="figure/droplet_rare.Rmd/volcano-plots-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-1-1">
Past versions of volcano-plots-1-1.png
</button>
</p>
<div id="fig-volcano-plots-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/323f198e6a987a8a97c55e4c85978e9dd437f3d7/docs/figure/droplet_rare.Rmd/volcano-plots-1-1.png" target="_blank">323f198</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Judging by the most distinctive genes in these volcano plots, the topics capture tuft cells (topic 1), neuroendocrine cells (topic 2) and ciliated cells (topics 3 + 4).</p>
<p>Meanwhile, from the structure plot above, we see that topics 3 and 4 capture continuous variation in the ciliated cells. The most distinctive genes in these two topics suggest…</p>
<pre class="r"><code>p5 &lt;- volcano_plot(de,k = &quot;k3&quot;,ymax = 50)
p6 &lt;- volcano_plot(de,k = &quot;k4&quot;,ymax = 200)
plot_grid(p5,p6)</code></pre>
<p><img src="figure/droplet_rare.Rmd/volcano-plots-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-2-1">
Past versions of volcano-plots-2-1.png
</button>
</p>
<div id="fig-volcano-plots-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/323f198e6a987a8a97c55e4c85978e9dd437f3d7/docs/figure/droplet_rare.Rmd/volcano-plots-2-1.png" target="_blank">323f198</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>TO DO: Create interactive volcano plots.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0     ggplot2_3.3.5     fastTopics_0.6-98
# 
# loaded via a namespace (and not attached):
#  [1] httr_1.4.2         tidyr_1.1.3        jsonlite_1.7.2     viridisLite_0.3.0 
#  [5] RcppParallel_4.4.2 assertthat_0.2.1   highr_0.8          mixsqp_0.3-46     
#  [9] yaml_2.2.0         progress_1.2.2     ggrepel_0.9.1      pillar_1.6.2      
# [13] backports_1.1.5    lattice_0.20-38    quantreg_5.54      glue_1.4.2        
# [17] quadprog_1.5-8     digest_0.6.23      promises_1.1.0     colorspace_1.4-1  
# [21] htmltools_0.4.0    httpuv_1.5.2       Matrix_1.2-18      pkgconfig_2.0.3   
# [25] invgamma_1.1       SparseM_1.78       purrr_0.3.4        scales_1.1.0      
# [29] whisker_0.4        later_1.0.0        Rtsne_0.15         MatrixModels_0.4-1
# [33] git2r_0.26.1       tibble_3.1.3       farver_2.0.1       generics_0.0.2    
# [37] ellipsis_0.3.2     withr_2.4.2        ashr_2.2-51        pbapply_1.5-1     
# [41] lazyeval_0.2.2     magrittr_2.0.1     crayon_1.4.1       mcmc_0.9-6        
# [45] evaluate_0.14      fs_1.3.1           fansi_0.4.0        MASS_7.3-51.4     
# [49] truncnorm_1.0-8    tools_3.6.2        data.table_1.12.8  prettyunits_1.1.1 
# [53] hms_1.1.0          lifecycle_1.0.0    stringr_1.4.0      MCMCpack_1.4-5    
# [57] plotly_4.9.2       munsell_0.5.0      irlba_2.3.3        compiler_3.6.2    
# [61] jquerylib_0.1.4    systemfonts_1.0.2  rlang_0.4.11       grid_3.6.2        
# [65] htmlwidgets_1.5.1  labeling_0.3       rmarkdown_2.11     gtable_0.3.0      
# [69] DBI_1.1.0          R6_2.4.1           knitr_1.37         dplyr_1.0.7       
# [73] uwot_0.1.10        utf8_1.1.4         workflowr_1.7.0    rprojroot_1.3-2   
# [77] ragg_0.3.1         stringi_1.4.3      parallel_3.6.2     SQUAREM_2017.10-1 
# [81] Rcpp_1.0.7         vctrs_0.3.8        tidyselect_1.1.1   xfun_0.29         
# [85] coda_0.19-3</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
