<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Analysis of single-cell RNA-seq data using topic models</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis of single-cell RNA-seq data using topic models</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree71e267dff418062e1061ad83c5a54f78322c2a01targetblank71e267da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/71e267dff418062e1061ad83c5a54f78322c2a01" target="_blank">71e267d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree71e267dff418062e1061ad83c5a54f78322c2a01targetblank71e267da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/71e267dff418062e1061ad83c5a54f78322c2a01" target="_blank">71e267d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  analysis/de_analysis_detailed_look_cache/
    Untracked:  analysis/de_analysis_detailed_look_more_cache/
    Untracked:  plots/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/71e267dff418062e1061ad83c5a54f78322c2a01/analysis/index.Rmd" target="_blank">71e267d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-11-08
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/f2cd3c74bd3ede52bebb0f56f8d94daba52c3bdf/docs/index.html" target="_blank">f2cd3c7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
Fixed link in overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/27b2d0876755eb53bc38c90bdcd6b5cce52a1800/analysis/index.Rmd" target="_blank">27b2d08</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-20
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/ed35cd92428e9f9c583d0f5c873b52c7641e0ce9/docs/index.html" target="_blank">ed35cd9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
Added link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/681296e07a1d852cd77eeb31888a9c463a66d16a/analysis/index.Rmd" target="_blank">681296e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-18
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/da1a3c60446237d7222470144da5a03aaaf36a04/docs/index.html" target="_blank">da1a3c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-09
</td>
<td>
Added link to home page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3be203561dd6390b43715193384a0b689fd62419/analysis/index.Rmd" target="_blank">3be2035</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-10-09
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/4e298d9134adcb892218656f8045b75f77563952/analysis/index.Rmd" target="_blank">4e298d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
Small edit to overview page.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/5303229d47c0c3abed05073545425a480c5d9d42/docs/index.html" target="_blank">5303229</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-04
</td>
<td>
Made a few revisions to the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b6f77c1bdcae0d87d0424a7760320004a6078148/analysis/index.Rmd" target="_blank">b6f77c1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-04
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5b1983f161f13b6df7666ab20008c9beeb5f1665/analysis/index.Rmd" target="_blank">5b1983f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-03
</td>
<td>
A few minor changes to the analysis of the purified PBMC data.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/134ea7843b66d02293a3393a193162083db7599f/docs/index.html" target="_blank">134ea78</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-03
</td>
<td>
Updated the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/83151f0521db924cff0c2b46194273d7aa607c27/analysis/index.Rmd" target="_blank">83151f0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-03
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56b968471026cb454ef025de13c362141db586e5/analysis/index.Rmd" target="_blank">56b9684</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-12-29
</td>
<td>
Added new section to overview page on single-cell likelihoods.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ef8a20a28d5c1f0968ff572f67e6cbc3ad4453c/analysis/index.Rmd" target="_blank">0ef8a20</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-11-26
</td>
<td>
Working on solving purified PBMC ‘mysteries’.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a9eece06c120d7dfc9ef400202b633ec28e9013d/docs/index.html" target="_blank">a9eece0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-11-13
</td>
<td>
Updated the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/672e1e31e76683806b282ef4a1394063899c76a7/analysis/index.Rmd" target="_blank">672e1e3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-11-13
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/dc984494918d3f593c952855cb094ba1bc0ad0f6/analysis/index.Rmd" target="_blank">dc98449</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-11-11
</td>
<td>
Added B-cells cluster to clusters_purified_pbmc.R.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5e57ced077fc483c6e9040dc834eaa62e1e783d8/analysis/index.Rmd" target="_blank">5e57ced</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-10-03
</td>
<td>
Working on plots highlighting substructure in T+N cluster.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/29e727ce49a82c1632e178214e71f21939da7491/analysis/index.Rmd" target="_blank">29e727c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-10-01
</td>
<td>
Minor edits.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a9ae3b4f3c20392a21b4bd77d5d3d9be91f62dd6/analysis/index.Rmd" target="_blank">a9ae3b4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
Working on new analysis, diff_count_droplet.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/71300aef4f7d33949724f01c1c7165afd2530397/docs/index.html" target="_blank">71300ae</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
Added links to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ab228245aad3d56143219be9797028fcfe9bb6b8/analysis/index.Rmd" target="_blank">ab22824</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/485628f58e0c985b706faab9357ffe61d3101ac6/docs/index.html" target="_blank">485628f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
<td>
Revised overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/dd9d050507778102679e0ecaeaa6f96c800f1ffd/analysis/index.Rmd" target="_blank">dd9d050</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-21
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/04e1761611f3c352a77a9aaa99708daabb76042d/analysis/index.Rmd" target="_blank">04e1761</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
A couple minor edits to the R Markdown.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/48209624d34951f447be3677c7302d91191ac223/docs/index.html" target="_blank">4820962</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Added a couple links to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/979b21706060487fad7e34d9c6c3a47b5575f24b/analysis/index.Rmd" target="_blank">979b217</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/279f62d03a1b5042fbc0060a36206b7c171b9635/docs/index.html" target="_blank">279f62d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-11
</td>
<td>
Fixed link in overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/19f5623c40193219551896b9854da210b02afc8c/analysis/index.Rmd" target="_blank">19f5623</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-11
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0208a0832efa1c70bbb5861a75d8dc1a8288a8f1/docs/index.html" target="_blank">0208a08</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-11
</td>
<td>
Added link to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d32e14c8d2d32352b446dc89c0994e09a6fe67fa/analysis/index.Rmd" target="_blank">d32e14c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-11
</td>
<td>
workflowr::wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/00e800241f9154c9283dac0bf27e056d282a2705/analysis/index.Rmd" target="_blank">00e8002</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
Implemented new function pca_hex_plot in plots.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e77420b3227da196445b482796b110350c646eb7/docs/index.html" target="_blank">e77420b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added “source” button to navbar.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/66bbdf45e7ac8c5cf4f960fd9187e5df90cdf688/analysis/index.Rmd" target="_blank">66bbdf4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Implemented functions basic_pca_plot and pca_plot_with_labels in plots.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/64b881d029ba8534aff6e8c6b85de9213edfc132/docs/index.html" target="_blank">64b881d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Updated links in overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/61917ad82d0229261c265c7546d2cb48e3ce2a67/analysis/index.Rmd" target="_blank">61917ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Working on new analysis, plots_tracheal_epithelium.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7b10d810dd41b920d835f074fa224af1325d1a69/docs/index.html" target="_blank">7b10d81</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-10
</td>
<td>
A couple minor edits to the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2b9f856ad1d2beaed80b0703d38789a376e3a0c7/analysis/index.Rmd" target="_blank">2b9f856</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-10
</td>
<td>
wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/94e55aeb00f1ce339ab858bd01a1dd2a4197d981/docs/index.html" target="_blank">94e55ae</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-10
</td>
<td>
Updated overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bd72e7de1d71bafc1016f6aec37d485cc3846523/analysis/index.Rmd" target="_blank">bd72e7d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-10
</td>
<td>
wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c6520997658f6d0bf93528d7a6bfde86a01a679d/docs/index.html" target="_blank">c652099</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
Added some explanatory text to the overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7d5e9c8376dfe25f61b8d48bd563a96ea9d67ef0/analysis/index.Rmd" target="_blank">7d5e9c8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
wflow_publish(“index.Rmd”, view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e8da7669cc56e5ba2ac478b7d51fad09c249c96f/docs/index.html" target="_blank">e8da766</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
Added more links to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3136222f56b79c5ca8a2478f8c1f0eaa9514e191/analysis/index.Rmd" target="_blank">3136222</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a42d6cc9fd46abae669e5eec53576796bcfe69a5/docs/index.html" target="_blank">a42d6cc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
Fixed links in overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/67d1667f23c0fec070145eb1c4d5a2f6af0d5a6d/analysis/index.Rmd" target="_blank">67d1667</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/15b3a5713dc47ef4e6ea85e18c5baa18872eb89f/docs/index.html" target="_blank">15b3a57</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
Added links to overview page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/be55916d0f44b2dd9ffabdcb7a13206262afaaae/analysis/index.Rmd" target="_blank">be55916</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c46e2fe483ac964aebbb11fed1029796cafb6e9f/docs/index.html" target="_blank">c46e2fe</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
Build home page and populated docs folder.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8890f6b09c74d8975bc4c4d41f4500d4fc6c96aa/analysis/index.Rmd" target="_blank">8890f6b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
wflow_publish(“index.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/82378421d5f51066577890de30d10fd131d9130d/analysis/index.Rmd" target="_blank">8237842</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-26
</td>
<td>
Added workflowr files.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="evaluation-of-de-analysis-using-a-topic-model" class="section level2">
<h2>Evaluation of DE analysis using a topic model</h2>
<p>In these pages, we assess the new topic-model-based methods for differential expression analysis, and compare with DESEq2.</p>
<ul>
<li><p><a href="de_analysis_detailed_look.html">Evaluation of DE methods using simulated data</a></p></li>
<li><p><a href="de_analysis_detailed_look_more.html">Evaluation of DE analysis with more than two topics</a></p></li>
<li><p><a href="de_analysis_purified_pbmc.html">Illustration in mixture of FACS-purified PBMC data</a></p></li>
</ul>
</div>
<div id="data-preparation" class="section level2">
<h2>Data preparation</h2>
<p>These pages detail the steps taken to prepare the single-cell RNA-seq data sets for topic modeling analysis.</p>
<ul>
<li><p><a href="prepare_purified_pbmc.html">Mixture of FACS-purified PBMCs</a></p></li>
<li><p><a href="prepare_68k_pbmc.html">PBMC 68k</a></p></li>
<li><p><a href="prepare_droplet.html">Droplet</a></p></li>
<li><p><a href="prepare_pulseseq.html">Pulse-seq</a></p></li>
</ul>
</div>
<div id="assess-poisson-nmf-model-fits" class="section level2">
<h2>Assess Poisson NMF model fits</h2>
<p>In these pages, we compare the quality of the Poisson NMF fits with different updates (EM, CCD, SCD, with and without extrapolation), and with different settings for <span class="math inline">\(K\)</span>.</p>
<ul>
<li><p><a href="assess_fits_purified_pbmc.html">Mixture of FACS-purified PBMCs</a></p></li>
<li><p><a href="assess_fits_68k_pbmc.html">PBMC 68k</a></p></li>
<li><p><a href="assess_fits_droplet.html">Droplet</a></p></li>
<li><p><a href="assess_fits_pulseseq.html">Pulse-seq</a></p></li>
</ul>
</div>
<div id="identify-clusters" class="section level2">
<h2>Identify clusters</h2>
<p>In these pages, we identify clusters of cells from the mixture proportions.</p>
<ul>
<li><p><a href="clusters_purified_pbmc.html">Mixture of FACS-purified PBMCs</a></p></li>
<li><p><a href="clusters_68k_pbmc.html">PBMC 68k</a></p></li>
<li><p><a href="clusters_droplet.html">Droplet</a></p></li>
<li><p><a href="clusters_pulseseq.html">Pulse-seq</a></p></li>
</ul>
</div>
<div id="visualize-and-annotate-topics" class="section level2">
<h2>Visualize and annotate topics</h2>
<p>In these pages, we annotate topics and substructures to determine their biological relevance. This includes differential expression analysis using the topic model.</p>
<ul>
<li><p><a href="plots_purified_pbmc.html">Mixture of FACS-purified PBMCs</a></p></li>
<li><p><a href="plots_68k_pbmc.html">PBMC 68k</a></p></li>
<li><p><a href="plots_tracheal_epithelium.html">Droplet and pulse-seq</a></p></li>
</ul>
</div>
<div id="examine-single-cell-likelihoods" class="section level2">
<h2>Examine single-cell likelihoods</h2>
<p>In these pages, we calculate single-cell likelihoods to assess how well the multinomial topic model captures expression in different cells and cell types.</p>
<ul>
<li><p><a href="loglik_purified_pbmc.html">Mixture of FACS-purified PBMCs</a></p></li>
<li><p><a href="loglik_68k_pbmc.html">PBMC 68k</a></p></li>
<li><p><a href="loglik_droplet.html">Droplet</a></p></li>
<li><p><a href="loglik_pulseseq.html">Pulse-seq</a></p></li>
</ul>
</div>
<div id="other-analyses" class="section level2">
<h2>Other analyses</h2>
<ul>
<li><a href="pca_vs_tsne.html">PCA vs. t-SNE and UMAP: an illustration</a></li>
</ul>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
