<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Visualize and interpret droplet and pulse-seq topics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Visualize and interpret droplet and pulse-seq topics</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreee11855bf04b544e1fbfd4ef90b0162a8aa8d6df8targetblanke11855ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8" target="_blank">e11855b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreee11855bf04b544e1fbfd4ef90b0162a8aa8d6df8targetblanke11855ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8" target="_blank">e11855b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plots_tracheal_epithelium.Rmd</code>) and HTML (<code>docs/plots_tracheal_epithelium.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8/analysis/plots_tracheal_epithelium.Rmd" target="_blank">e11855b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Working on revised analysis of droplet and pulse-seq data sets.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8/docs/plots_tracheal_epithelium.html" target="_blank">e11855b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Working on revised analysis of droplet and pulse-seq data sets.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bf23ca096435ce5829d49f79624deb0e64846fb0/analysis/plots_tracheal_epithelium.Rmd" target="_blank">bf23ca0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added manual labeling of purified PBMC data to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/077d3d55ca6da2879de67f6d1e866f0b5ef5ac99/analysis/plots_tracheal_epithelium.Rmd" target="_blank">077d3d5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added k=9 and k=11 pulseseq fits to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0ce960489da333344f062deaac7bffdbaf8b5838/docs/plots_tracheal_epithelium.html" target="_blank">0ce9604</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Re-built plots_tracheal_epithelium with fastTopics 0.3-162.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/961570e0d6a33b7b3908c6feb3beca085c401268/analysis/plots_tracheal_epithelium.Rmd" target="_blank">961570e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b17bfa4f7de6a68e6b6cee8ed7620188a8411230/docs/plots_tracheal_epithelium.html" target="_blank">b17bfa4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added pulseseq PCA plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/76dc0c6859e954265dc836ca481a7c06cc174866/analysis/plots_tracheal_epithelium.Rmd" target="_blank">76dc0c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c70612f1d996cac0442f6b5346aa812e5e51cfa4/analysis/plots_tracheal_epithelium.Rmd" target="_blank">c70612f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Revised structure plot settings for abundant droplet samples in plots_tracheal_epithelium.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/adda33faa241fa7ea1abe8bf5eff706d459f5547/docs/plots_tracheal_epithelium.html" target="_blank">adda33f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Fixed another structure plot in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/29a9258d418e4df1553836d2e1a3de7073ce2e3e/analysis/plots_tracheal_epithelium.Rmd" target="_blank">29a9258</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0a16b60e3247e5a35fa4637e0133965db35eab0b/docs/plots_tracheal_epithelium.html" target="_blank">0a16b60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Fixed structure plot in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3a7bd7465276742c305377eb2737c52e0e031abe/analysis/plots_tracheal_epithelium.Rmd" target="_blank">3a7bd74</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/f4bdf191a428ef8917fdd6f9e386653350125a0d/docs/plots_tracheal_epithelium.html" target="_blank">f4bdf19</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added explanatory text and improved PC-based manual clustering of
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c7b77ee2b8b9d0ac075a4cfcf78a1a755ca868d6/analysis/plots_tracheal_epithelium.Rmd" target="_blank">c7b77ee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/70a4a60fe8655bd8d5609ed02a7f3718c41ae6fa/analysis/plots_tracheal_epithelium.Rmd" target="_blank">70a4a60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added note to plots_tracheal_epithelium.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/fb21b3b781ba22db8b89d1fb531cf91e63e4e1d5/docs/plots_tracheal_epithelium.html" target="_blank">fb21b3b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added very initial Structure plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d35cb03fade4ee2fc5da4880b31c21d1c2bb9dda/analysis/plots_tracheal_epithelium.Rmd" target="_blank">d35cb03</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/368a74a30418728c7ae896e6ec1c567e24ace79f/docs/plots_tracheal_epithelium.html" target="_blank">368a74a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added some text to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/223406b7e0fc6e73efd027e6d1bc00b0b0ad5c6d/analysis/plots_tracheal_epithelium.Rmd" target="_blank">223406b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/aca46cc698358965ad25399bf7b19f74a6b5df23/docs/plots_tracheal_epithelium.html" target="_blank">aca46cc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added manual clustering of droplet samples based on PCs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/38f811b28b8ba3f3996f2551c0e0a4eace930da3/analysis/plots_tracheal_epithelium.Rmd" target="_blank">38f811b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/343747ead094b567a93133bdba3411bff3c25667/analysis/plots_tracheal_epithelium.Rmd" target="_blank">343747e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Small edit to figure dimensions.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/5a35bbd6b181f322bc808840a2df3c20a6b80d94/docs/plots_tracheal_epithelium.html" target="_blank">5a35bbd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added labeled PCA plot; adjusted plot dimensions in
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fb91075ba717e53382f9ae3ea99b065256687108/analysis/plots_tracheal_epithelium.Rmd" target="_blank">fb91075</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/8b9b5289e71be99509c2adf86e0bfead46cb50b1/docs/plots_tracheal_epithelium.html" target="_blank">8b9b528</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added more PCA plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ee7cbf157775bd42c882bb897efac8e07994c207/analysis/plots_tracheal_epithelium.Rmd" target="_blank">ee7cbf1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c517ea207b696aa2815c2c77963d6c305373b7d3/docs/plots_tracheal_epithelium.html" target="_blank">c517ea2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Small fix to one of the PCA plots in plots_tracheal_epithelium.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8f5c210a6a93dd5e5c25d35f100e2f3b88f1c3f3/analysis/plots_tracheal_epithelium.Rmd" target="_blank">8f5c210</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/01afbd2b764fd3d6724afa103aeefa2e4c91ff60/docs/plots_tracheal_epithelium.html" target="_blank">01afbd2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Added some PC plots to the plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f1c7d028ef9fa5c96cf785283b567f98c26b365b/analysis/plots_tracheal_epithelium.Rmd" target="_blank">f1c7d02</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0a04fc1929d7987b10f019c015f116aa46705bcc/docs/plots_tracheal_epithelium.html" target="_blank">0a04fc1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Added abundance plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f914f7e28e63914b21b1b6a00c7c73e81ee1a961/analysis/plots_tracheal_epithelium.Rmd" target="_blank">f914f7e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/61917ad82d0229261c265c7546d2cb48e3ce2a67/analysis/plots_tracheal_epithelium.Rmd" target="_blank">61917ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Working on new analysis, plots_tracheal_epithelium.Rmd.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><em>TO DO: Add introductory text here.</em></p>
<p>Load the packages used in the analysis below, as well as additional functions that we will use to generate some of the plots.</p>
<pre class="r"><code>library(dplyr)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/plots.R&quot;)</code></pre>
<div id="droplet-data" class="section level2">
<h2>Droplet data</h2>
<p>We begin with the droplet data. Note that the count data are no longer needed at this stage.</p>
<pre class="r"><code>load(&quot;../data/droplet.RData&quot;)
samples_droplet &lt;- samples
rm(samples,counts)</code></pre>
<p>Load the <span class="math inline">\(k = 7\)</span> Poisson NMF model fit.</p>
<pre class="r"><code>fit_droplet &lt;- readRDS(&quot;../output/droplet/rds/fit-droplet-scd-ex-k=7.rds&quot;)$fit</code></pre>
<p>The Montoro <em>et al</em> (2018) article mentions that some epithelial cell types are abundant whereas others are rare. The topics inferred from the droplet data reflect this:</p>
<pre class="r"><code>p1 &lt;- create_abundance_plot(fit_droplet)
print(p1)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/abundance-plot-droplet-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-abundance-plot-droplet-1">
Past versions of abundance-plot-droplet-1.png
</button>
</p>
<div id="fig-abundance-plot-droplet-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0a04fc1929d7987b10f019c015f116aa46705bcc/docs/figure/plots_tracheal_epithelium.Rmd/abundance-plot-droplet-1.png" target="_blank">0a04fc1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The first topic—which is not actually visible in this bar chart—is indeed very rare; only 43 out of 7,193 samples have a greater than 10% contribution from this topic.</p>
<pre class="r"><code>sum(poisson2multinom(fit_droplet)$L[,1] &gt; 0.1)
# [1] 43</code></pre>
<p>In this next part of the analysis, we perform PCA on the estimated topic proportions to explore structure in the data as inferred by the topic model. Typically, a nonlinear embedding method such as <em>t</em>-SNE or UMAP is used to visualize the structure. The disadvantage of such methods is that it can often be difficult to get the (many) tuning parameters right, they can be slow when applied to large data sets, and the embeddings are not unique; by contrast, PCA has no tuning parameters, and the principal components (PCs) are unique.</p>
<pre class="r"><code>fit &lt;- poisson2multinom(fit_droplet)
pca &lt;- prcomp(fit$L)$x</code></pre>
<p>In the projection onto four of the PCs—PCs 1, 2, 5 and 6—we can delineate 4 clusters. A fifth subset (“E”) is used as a “background” cluster. (Note that PCs 3 and 4 do not reveal any additional substrcture, so are not shown.)</p>
<pre class="r"><code>n   &lt;- nrow(pca)
x   &lt;- rep(&quot;E&quot;,n)
pc1 &lt;- pca[,&quot;PC1&quot;]
pc2 &lt;- pca[,&quot;PC2&quot;]
pc6 &lt;- pca[,&quot;PC6&quot;]
x[pc2 &gt; -0.1]  &lt;- &quot;A&quot;
x[pc6 &lt; -0.04] &lt;- &quot;B&quot;
x[(pc1 - 0)^2 + (pc2 + 0.75)^2 &lt; 0.09]  &lt;- &quot;C&quot;
x[(pc1 - 0.5)^2 + (pc2 + 0.9)^2 &lt; 0.04] &lt;- &quot;D&quot;
samples_droplet$cluster &lt;- x
p1 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC1&quot;,&quot;PC2&quot;),x) +
      labs(fill = &quot;cluster&quot;)
p2 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC5&quot;,&quot;PC6&quot;),x) +
      labs(fill = &quot;cluster&quot;)
plot_grid(p1,p2)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/droplet-clustering-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-droplet-clustering-2-1">
Past versions of droplet-clustering-2-1.png
</button>
</p>
<div id="fig-droplet-clustering-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8/docs/figure/plots_tracheal_epithelium.Rmd/droplet-clustering-2-1.png" target="_blank">e11855b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/adda33faa241fa7ea1abe8bf5eff706d459f5547/docs/figure/plots_tracheal_epithelium.Rmd/droplet-clustering-2-1.png" target="_blank">adda33f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0a16b60e3247e5a35fa4637e0133965db35eab0b/docs/figure/plots_tracheal_epithelium.Rmd/droplet-clustering-2-1.png" target="_blank">0a16b60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The vast majority of the cells are in cluster A:</p>
<pre class="r"><code>table(x)
# x
#    A    B    C    D    E 
# 6533   50  162  375   73</code></pre>
<p>Cluster A further subdivides into two not-quite-so-distinct subclusters, otherwise there does not appear to be any other interesting substructure to delineate:</p>
<pre class="r"><code>rows &lt;- which(samples_droplet$cluster == &quot;A&quot;)
fit  &lt;- select(poisson2multinom(fit_droplet),loadings = rows)
pca  &lt;- prcomp(fit$L)$x
n    &lt;- nrow(pca)
x    &lt;- rep(&quot;A1&quot;,n)
pc1  &lt;- pca[,1]
x[pc1 &gt; 0.2] &lt;- &quot;A2&quot;
samples_droplet[rows,&quot;cluster&quot;] &lt;- x
p3 &lt;- pca_plot_with_labels(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),x) +
      labs(fill = &quot;cluster&quot;)
print(p3)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/droplet-clustering-4-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>In summary, we have subdivided the data into 6 subsets:</p>
<pre class="r"><code>samples_droplet$cluster &lt;- factor(samples_droplet$cluster)
table(samples_droplet$cluster)
# 
#   A1   A2    B    C    D    E 
# 3968 2565   50  162  375   73</code></pre>
<p>The structure plot summarizes the topic proportions in each of these 6 subsets:</p>
<pre class="r"><code>set.seed(2)
droplet_topic_colors &lt;- c(&quot;gold&quot;,&quot;royalblue&quot;,&quot;turquoise&quot;,&quot;greenyellow&quot;,
                          &quot;forestgreen&quot;,&quot;firebrick&quot;,&quot;olivedrab&quot;)
droplet_topics &lt;- c(1,3,7,4,5,6,2)
rows &lt;- sort(c(sample(which(samples_droplet$cluster == &quot;A1&quot;),800),
               sample(which(samples_droplet$cluster == &quot;A2&quot;),500),
               which(samples_droplet$cluster == &quot;B&quot;),
               which(samples_droplet$cluster == &quot;C&quot;),
               sample(which(samples_droplet$cluster == &quot;D&quot;),200),
               which(samples_droplet$cluster == &quot;E&quot;),200))
p4 &lt;- structure_plot(select(poisson2multinom(fit_droplet),loadings = rows),
                     grouping = samples_droplet[rows,&quot;cluster&quot;],
                     topics = droplet_topics,
                     colors = droplet_topic_colors[droplet_topics],
                     perplexity = c(100,70,12,50,50,20),
                     n = Inf,gap = 40,num_threads = 4,verbose = FALSE)
print(p4)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/droplet-structure-plot-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>The bulk of the samples lie on a continuous gradient between topics 2 and 5. There is a smaller cluster at the bottom of this plot, with high contributions from topic 6.</p>
<p>Along these PCs, we see that topics 3, 4, 5 and 7 exist in many combinations, with no apparent discrete populations.</p>
<p>Topic 1 captures a very small discrete population of cells:</p>
<p>In summary, topics 1 and 6 pick up discrete “cell types”, whereas the other topics characterize more continuous variation in gene expression, perhaps cell types along a continuous trajectory of development. There are some other discrete clusters that seem to be composed of distinct combinations of topics that we will neeed to examine more closely.</p>
<p>These next few scatterplots show the same PCs as before, with the samples labeled according to their assignment to these clusters:</p>
<pre class="r"><code>droplet_cluster_colors &lt;- c(&quot;gray&quot;,&quot;darkblue&quot;,&quot;gold&quot;,&quot;yellowgreen&quot;,&quot;firebrick&quot;)
p9 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC1&quot;,&quot;PC2&quot;),samples_droplet$cluster,
                           droplet_cluster_colors) + labs(fill = &quot;cluster&quot;)
p10 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC4&quot;,&quot;PC5&quot;),samples_droplet$cluster,
                            droplet_cluster_colors) + labs(fill = &quot;cluster&quot;)
p11 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC5&quot;,&quot;PC6&quot;),samples_droplet$cluster,
                            droplet_cluster_colors) + labs(fill = &quot;cluster&quot;)
plot_grid(p9,p10,p11)</code></pre>
<p>In the following, we will treat the more abundant clusters, <span class="math inline">\(c_1\)</span> and <span class="math inline">\(c_4\)</span>, separately from the rest of the samples.</p>
<p><strong>TO DO:</strong> Add text here.</p>
<pre class="r"><code>p12 &lt;- pca_plot(poisson2multinom(fit_pulseseq),pcs = 3:4,k = 7)
p13 &lt;- pca_plot(poisson2multinom(fit_pulseseq),pcs = 5:6,k = 2)
plot_grid(p12,p13)</code></pre>
<p><strong>TO DO:</strong> Add text here.</p>
<pre class="r"><code>droplet_topic_colors &lt;- c(&quot;gold&quot;,&quot;royalblue&quot;,&quot;turquoise&quot;,&quot;greenyellow&quot;,
                          &quot;forestgreen&quot;,&quot;firebrick&quot;,&quot;olivedrab&quot;)
names(droplet_topic_colors) &lt;- paste0(&quot;k&quot;,1:7)                       
topics &lt;- c(&quot;k1&quot;,&quot;k3&quot;,&quot;k4&quot;,&quot;k5&quot;,&quot;k7&quot;,&quot;k2&quot;,&quot;k6&quot;)
set.seed(1)
fit_droplet_rare &lt;- select(poisson2multinom(fit_droplet),
                           loadings = which(samples_droplet$cluster != &quot;c1&quot; &amp;
                                            samples_droplet$cluster != &quot;c4&quot;))
p12 &lt;- structure_plot(fit_droplet_rare,verbose = FALSE,perplexity = 50,
                      topics = topics,
                      colors = droplet_topic_colors[topics])
print(p12)</code></pre>
<p><strong>TO DO:</strong> Add text here.</p>
<pre class="r"><code>topics &lt;- c(&quot;k1&quot;,&quot;k3&quot;,&quot;k4&quot;,&quot;k7&quot;,&quot;k5&quot;,&quot;k2&quot;,&quot;k6&quot;)
set.seed(1)
fit_droplet_abundant &lt;-
  select(poisson2multinom(fit_droplet),
                          loadings = which(samples_droplet$cluster == &quot;c1&quot; |
                                           samples_droplet$cluster == &quot;c4&quot;))
p13 &lt;- structure_plot(fit_droplet_abundant,n = 2000,perplexity = 50,
                      verbose = FALSE,topics = topics,
                      colors = droplet_topic_colors[topics],
                      scaling = c(1,1,1,1,2,1,1))
print(p13)</code></pre>
</div>
<div id="pulse-seq-data" class="section level2">
<h2>Pulse-seq data</h2>
<pre class="r"><code>load(&quot;../data/pulseseq.RData&quot;)
samples_pulseseq &lt;- samples
rm(samples,counts)</code></pre>
<p>Load the <span class="math inline">\(k = 9\)</span> and <span class="math inline">\(k = 11\)</span> Poisson NMF fits for the pulse-seq data.</p>
<pre class="r"><code>fit_pulseseq9 &lt;- readRDS(&quot;../output/pulseseq/rds/fit-pulseseq-scd-ex-k=9.rds&quot;)$fit
fit_pulseseq11 &lt;- readRDS(&quot;../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds&quot;)$fit</code></pre>
<p>Likewise, we also pick up rare and abundant topics in the pulse-seq data:</p>
<pre class="r"><code>p2 &lt;- create_abundance_plot(fit_pulseseq)
print(p2)</code></pre>
<pre class="r"><code>temp &lt;- select(poisson2multinom(fit_droplet),
               loadings = which(samples_droplet$tissue == &quot;Ionocyte&quot;))
structure_plot(temp,rows = order(temp$L[,3]),
                      topics = topic_ordering,
                      colors = droplet_topic_colors[topic_ordering])
temp2 &lt;- select(poisson2multinom(fit_droplet),
                loadings = which(samples_droplet$tissue == &quot;Tuft&quot;))
structure_plot(temp2,perplexity = 30,
               topics = topic_ordering,
               colors = droplet_topic_colors[topic_ordering])
temp3 &lt;- select(poisson2multinom(fit_droplet),
                loadings = which(samples_droplet$tissue == &quot;Neuroendocrine&quot;))
structure_plot(temp3,perplexity = 30,
               topics = topic_ordering,
               colors = droplet_topic_colors[topic_ordering])</code></pre>
<p>It is helpful to compare these results with clustering reported in the Montoro <em>et al</em> (2018) paper. To make this comparison, we layer the 7 clusters on top of these PCs:</p>
<pre class="r"><code>droplet_celltype_colors &lt;-
  c(&quot;royalblue&quot;,      # Basal
    &quot;firebrick&quot;,      # Ciliated
    &quot;forestgreen&quot;,    # Club
    &quot;gold&quot;,           # Goblet
    &quot;darkmagenta&quot;,    # Ionocyte
    &quot;darkorange&quot;,     # Neuroendocrine
    &quot;lightsteelblue&quot;) # Tuft
p9 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC1&quot;,&quot;PC2&quot;),samples_droplet$tissue,
                           droplet_celltype_colors) + labs(fill = &quot;celltype&quot;)
p10 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC4&quot;,&quot;PC5&quot;),samples_droplet$tissue,
                            droplet_celltype_colors) + labs(fill = &quot;celltype&quot;)
p11 &lt;- pca_plot_with_labels(fit_droplet,c(&quot;PC5&quot;,&quot;PC6&quot;),samples_droplet$tissue,
                            droplet_celltype_colors) + labs(fill = &quot;celltype&quot;)
plot_grid(p9,p10,p11)</code></pre>
<p><strong>TO DO:</strong> Add Structure plot9s) to compare Montoro <em>et al</em> (2018) clustering.</p>
<p>And likewise in the pulse-seq data:</p>
<pre class="r"><code>p5 &lt;- basic_pca_plot(fit_pulseseq,c(&quot;PC3&quot;,&quot;PC4&quot;))
p6 &lt;- basic_pca_plot(fit_pulseseq,c(&quot;PC5&quot;,&quot;PC6&quot;))
plot_grid(p5,p6)</code></pre>
<pre class="r"><code>ggplot(cbind(pca$x,data.frame(k2 = fit$L[,2])),
       aes(x = PC1,y = PC2,fill = k2)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = 0.5) +
  theme_cowplot(font_size = 12)
ggplot(cbind(pca$x,data.frame(k5 = fit$L[,5])),
       aes(x = PC1,y = PC2,fill = k5)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = 0.5) +
  theme_cowplot(font_size = 12)
ggplot(cbind(pca$x,data.frame(k6 = fit$L[,6])),
       aes(x = PC1,y = PC2,fill = k6)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = 0.5) +
  theme_cowplot(font_size = 12)
ggplot(cbind(pca$x,data.frame(k1 = fit$L[,1])),
       aes(x = PC5,y = PC6,fill = k1)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = 0.5) +
  theme_cowplot(font_size = 12)</code></pre>
<pre class="r"><code>ggplot(cbind(pca$x,data.frame(k3 = fit$L[,3])),
       aes(x = PC1,y = PC2,fill = k3)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = 0.5) +
  theme_cowplot(font_size = 12)
ggplot(cbind(pca$x,data.frame(k4 = fit$L[,4])),
       aes(x = PC1,y = PC2,fill = k4)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_gradient2(low = &quot;deepskyblue&quot;,mid = &quot;gold&quot;,high = &quot;orangered&quot;,
                       midpoint = 0.5) +
  theme_cowplot(font_size = 12)</code></pre>
<pre class="r"><code>ggplot(cbind(samples_droplet,pca$x),aes(x = PC1,y = PC2,fill = tissue)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_manual(values = clrs) +
  theme_cowplot(font_size = 12)
ggplot(cbind(samples_droplet,pca$x),aes(x = PC5,y = PC6,fill = tissue)) +
  geom_point(shape = 21,color = &quot;white&quot;,,size = 2) +
  scale_fill_manual(values = clrs) +
  theme_cowplot(font_size = 12)</code></pre>
<pre class="r"><code>clrs &lt;- c(&quot;royalblue&quot;,      # basal
          &quot;firebrick&quot;,      # ciliated
          &quot;forestgreen&quot;,    # club
          &quot;gold&quot;,           # goblet
          &quot;darkmagenta&quot;,    # ionocyte
          &quot;darkorange&quot;,     # neuroendocrine
          &quot;tomato&quot;,         # proliferating
          &quot;darkgray&quot;)       # tuft
ggplot(cbind(samples_droplet,pca$x),aes(x = PC1,y = PC2,fill = tissue)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 2) +
  scale_fill_manual(values = clrs) +
  theme_cowplot(font_size = 12)
ggplot(cbind(samples_droplet,pca$x),aes(x = PC5,y = PC6,fill = tissue)) +
  geom_point(shape = 21,color = &quot;white&quot;,,size = 2) +
  scale_fill_manual(values = clrs) +
  theme_cowplot(font_size = 12)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.5
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0      ggplot2_3.3.0      fastTopics_0.3-165 dplyr_0.8.3       
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.0        Rcpp_1.0.5           lattice_0.20-38     
#  [4] tidyr_1.0.0          prettyunits_1.1.1    assertthat_0.2.1    
#  [7] zeallot_0.1.0        rprojroot_1.3-2      digest_0.6.23       
# [10] R6_2.4.1             backports_1.1.5      MatrixModels_0.4-1  
# [13] evaluate_0.14        coda_0.19-3          httr_1.4.1          
# [16] pillar_1.4.3         rlang_0.4.5          progress_1.2.2      
# [19] lazyeval_0.2.2       data.table_1.12.8    irlba_2.3.3         
# [22] SparseM_1.78         whisker_0.4          Matrix_1.2-18       
# [25] rmarkdown_2.3        labeling_0.3         Rtsne_0.15          
# [28] stringr_1.4.0        htmlwidgets_1.5.1    munsell_0.5.0       
# [31] compiler_3.6.2       httpuv_1.5.2         xfun_0.11           
# [34] pkgconfig_2.0.3      mcmc_0.9-6           htmltools_0.4.0     
# [37] tidyselect_0.2.5     tibble_2.1.3         workflowr_1.6.2.9000
# [40] quadprog_1.5-8       viridisLite_0.3.0    crayon_1.3.4        
# [43] withr_2.1.2          later_1.0.0          MASS_7.3-51.4       
# [46] grid_3.6.2           jsonlite_1.6         gtable_0.3.0        
# [49] lifecycle_0.1.0      git2r_0.26.1         magrittr_1.5        
# [52] scales_1.1.0         RcppParallel_5.0.2   stringi_1.4.3       
# [55] farver_2.0.1         fs_1.3.1             promises_1.1.0      
# [58] vctrs_0.2.1          tools_3.6.2          glue_1.3.1          
# [61] purrr_0.3.3          hms_0.5.2            yaml_2.2.0          
# [64] colorspace_1.4-1     plotly_4.9.2         knitr_1.26          
# [67] quantreg_5.54        MCMCpack_1.4-5</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
