<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Visualize and interpret topics in PBMC data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Visualize and interpret topics in PBMC data</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree070f612e95a8ff9ad8f82b4df8587e862ade37fftargetblank070f612a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/070f612e95a8ff9ad8f82b4df8587e862ade37ff" target="_blank">070f612</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree070f612e95a8ff9ad8f82b4df8587e862ade37fftargetblank070f612a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/070f612e95a8ff9ad8f82b4df8587e862ade37ff" target="_blank">070f612</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plots_pbmc.Rmd</code>) and HTML (<code>docs/plots_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/070f612e95a8ff9ad8f82b4df8587e862ade37ff/analysis/plots_pbmc.Rmd" target="_blank">070f612</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5ee1a1d0167cf8492293fbdba38467e129bf264e/analysis/plots_pbmc.Rmd" target="_blank">5ee1a1d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
Made some refinements to the 68k pbmc analysis in plots_pbmc.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/25a24f7d55ff38cf99b1c23f3e3e7bfb84419c67/docs/plots_pbmc.html" target="_blank">25a24f7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/655dfae4b6ca1aca66e19dbaafaa58619f280365/analysis/plots_pbmc.Rmd" target="_blank">655dfae</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/208d263faab0770c43953404595fdb98748755fa/docs/plots_pbmc.html" target="_blank">208d263</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
Added hex plots to plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8bfc9089f5f2eef21ae638c42df50e5daa51fde4/analysis/plots_pbmc.Rmd" target="_blank">8bfc908</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d4f9ec757dfc4121de56185b8a878f1eecd8f178/analysis/plots_pbmc.Rmd" target="_blank">d4f9ec7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
A couple small edits to the text in plots_pbmc.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/30d2ef1fa568ca329fe63d8bc498c3dec56ad55e/docs/plots_pbmc.html" target="_blank">30d2ef1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
Added hex plot to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6507d90c6920dcf58f15c18a639dfbe4a2412266/analysis/plots_pbmc.Rmd" target="_blank">6507d90</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3ab7da1135c1021866310fb12a88c4ad3b237395/analysis/plots_pbmc.Rmd" target="_blank">3ab7da1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
A few minor edits.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2defe6d80efff8e3f48136d45ef1464755c0d740/analysis/plots_pbmc.Rmd" target="_blank">2defe6d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added crosstab plot to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e23f091213b96c5b85f281bcf199e157fca39265/docs/plots_pbmc.html" target="_blank">e23f091</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added a few more notes to plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ca90f127afa06582b290f3cc36fa16296ca446e/analysis/plots_pbmc.Rmd" target="_blank">0ca90f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/94370c23e5d440eedce99a6cd04827eef1d04bc5/docs/plots_pbmc.html" target="_blank">94370c2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added notes adjoining results on 68k data in plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c6e754d5f359e69adf4e4931b415c444d0a1e5b3/analysis/plots_pbmc.Rmd" target="_blank">c6e754d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/399c5976cb5d054fae7e214567e39a71315ce0aa/docs/plots_pbmc.html" target="_blank">399c597</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added another few clusters to 68k fit, revised structure plot, and
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1a0c2d87fc7537d01c7bf1a36cdeb03684dfae90/analysis/plots_pbmc.Rmd" target="_blank">1a0c2d8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/eac2d23a7e084d60d7821643a185ad2e45d8092a/docs/plots_pbmc.html" target="_blank">eac2d23</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Revised the cross-tab plot, and added notes to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/2d156b8fefbcb1c1c65e7040c43eedff5fbf5657/docs/plots_pbmc.html" target="_blank">2d156b8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Revised the cross-tab plot, and added notes to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e8971d86164615e8337d2690a577bb3e6d465acd/analysis/plots_pbmc.Rmd" target="_blank">e8971d8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9d8bb284f6288d09a547d7ee44ffebe47278da1b/analysis/plots_pbmc.Rmd" target="_blank">9d8bb28</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Revised structure plot in plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/27f6f51e6d680d7ddd529dbe6aad301f941637d2/analysis/plots_pbmc.Rmd" target="_blank">27f6f51</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
A couple small adjustments to the plotting settings in plots_pbmc.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/abb846e9e32e6a27cd74c5d5d0dc0ca2799d60f6/docs/plots_pbmc.html" target="_blank">abb846e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added dotplot to compare clusters vs. cell-types.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/45c3c322b79cc885c82b3411ba56c91d41062956/analysis/plots_pbmc.Rmd" target="_blank">45c3c32</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/f53c86ce2e664d45e7e8b7299bfada46ae968f15/docs/plots_pbmc.html" target="_blank">f53c86c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
<td>
Made a few small improvements to plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f7c157a99f03eb371705a3201525f9c099aa0eda/analysis/plots_pbmc.Rmd" target="_blank">f7c157a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e13c17e5af93b7e4decc2bbfac93988a3339f575/analysis/plots_pbmc.Rmd" target="_blank">e13c17e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
<td>
Removed reordering of 68k fit, and fixed downstream PCA analyses.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/01a0c8bbcec855b7ef0d146d10e1d0d708c24684/analysis/plots_pbmc.Rmd" target="_blank">01a0c8b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
<td>
Fixed colours in 68k structure plot.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a0cb7c61cd959b2882f75eb4de7a3eda1eb07b0b/docs/plots_pbmc.html" target="_blank">a0cb7c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
<td>
Added 68k fit structure plot to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5b2fb5c11d350f319b9c39d26c37a0d9a8be66b9/analysis/plots_pbmc.Rmd" target="_blank">5b2fb5c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b6489db5d0c0b21a2856fd9ce0fac3045ad76688/docs/plots_pbmc.html" target="_blank">b6489db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
Re-built plots_pbmc with new PCA plots from 68k fit.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/89d98edb7828de05b62242c529b19e6b98ec8cad/analysis/plots_pbmc.Rmd" target="_blank">89d98ed</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
Removed basic_pca_plot in plots.R; working on PCA plots for 68k fit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/13ee038ed5f2e1757c215510ac94e68870ae4074/docs/plots_pbmc.html" target="_blank">13ee038</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
Revised notes in plots_pbmc.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e766da56adffe7e2e06cdcd795bb7fb7a36b0d7d/analysis/plots_pbmc.Rmd" target="_blank">e766da5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/dbd5882a4e6b5a20c0d1a04e9a3c899f4baff6b8/analysis/plots_pbmc.Rmd" target="_blank">dbd5882</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
Added some explanatory text to plots_pbmc.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/97d7e8692b0a40a53f6afc1ed5ee5067b1665d33/docs/plots_pbmc.html" target="_blank">97d7e86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
Fixed subclustering of cluster A in purified PBMC data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/005b217f97b1e5cf606c5e2fbb1275ba9753116e/analysis/plots_pbmc.Rmd" target="_blank">005b217</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/59777e743c15ee5a5576d476bf68227bfb2c5bf8/docs/plots_pbmc.html" target="_blank">59777e7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
Added table comparing Zheng et al (2017) labeling with clusters in
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9cd4a08b3c3352eaed9f8fdf09d0f1ce0a1276c3/analysis/plots_pbmc.Rmd" target="_blank">9cd4a08</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c87ddf8d946c9c5b08d51f7aa13e49181182062f/docs/plots_pbmc.html" target="_blank">c87ddf8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
Added PBMC purified structure plot with the new clustering (A1, A2, B,
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c6dd5f23d5d791a9b4f4f66fecf9365617d2be69/analysis/plots_pbmc.Rmd" target="_blank">c6dd5f2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/ce421ae6b4930c9ab4094deb192af6933b1924c7/docs/plots_pbmc.html" target="_blank">ce421ae</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
A few small revisions to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ddc6de4d725601617461a520887e5a576ef7cec9/analysis/plots_pbmc.Rmd" target="_blank">ddc6de4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a406a2f59205b74fc2effbb9620cb3d7d445eb74/docs/plots_pbmc.html" target="_blank">a406a2f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
Added back first PCA plot for 68k pbmc data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8daa1311212127a8d82836be381898d738985322/analysis/plots_pbmc.Rmd" target="_blank">8daa131</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/7900d171e513796cb2e56f95abf713355845a5ae/docs/plots_pbmc.html" target="_blank">7900d17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
Revamping the process of defining clusters in plots_pbmc using PCA.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/14dd774ec71c4d5b39b33da03dbe7efcc6c71415/analysis/plots_pbmc.Rmd" target="_blank">14dd774</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/944ce0c669f639989be997b97a5e60216854a1c1/analysis/plots_pbmc.Rmd" target="_blank">944ce0c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
Working on new PCA plots for purified PBMC c3 subclusters.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b05232dfedc5ff70024a5a13682fc03ff0533ade/docs/plots_pbmc.html" target="_blank">b05232d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d0e9f6eaf7136021833d30a302c0fc06f045a9d0/analysis/plots_pbmc.Rmd" target="_blank">d0e9f6e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/fbb0697684addb55df25f7501714a5eac8189e6f/docs/plots_pbmc.html" target="_blank">fbb0697</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/cdca13ec3fb3f0a81510038deac7a7cb1c4910a0/analysis/plots_pbmc.Rmd" target="_blank">cdca13e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/216027a8630c038d35ff5149539a552cc8650fa7/docs/plots_pbmc.html" target="_blank">216027a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
Re-built plots_pbmc webpage with structure plots.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/98888b2fae10880621bb3efd89cc53901190ce81/analysis/plots_pbmc.Rmd" target="_blank">98888b2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
Added structure plots to plots_pbmc.R.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/93109937d813d7c9cafacec2b45388af1d7c5e5a/docs/plots_pbmc.html" target="_blank">9310993</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
Revised the PCA plots in pbmc_plots.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3a746ff564ce29d445936f93a4d70bdd202562f2/analysis/plots_pbmc.Rmd" target="_blank">3a746ff</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1091dd00a2edd2cbad7d9fcd05ea9830da22e527/analysis/plots_pbmc.Rmd" target="_blank">1091dd0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
<td>
Added code for Ternary plots to plots_pbmc.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/6d3d7e4e855a46f2b154a459db16dd9fbaa58ccf/docs/plots_pbmc.html" target="_blank">6d3d7e4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added 68k PCA plots to plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6ec82cafc481b6839d49aa11b1f2c4568fbd0592/analysis/plots_pbmc.Rmd" target="_blank">6ec82ca</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/38f07a2440114be440e2efb027050b0e2b13bf5f/docs/plots_pbmc.html" target="_blank">38f07a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
A few small revisions to the plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fd0316f50098b3f0daabcc56e3861be698737068/analysis/plots_pbmc.Rmd" target="_blank">fd0316f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/606cd97bc0567099ab728910c625f96d0820b62f/docs/plots_pbmc.html" target="_blank">606cd97</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added basic PCA plots to plots_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1b41a6061bee5e770f1bdb75fd1d5bd5d90d9fdc/analysis/plots_pbmc.Rmd" target="_blank">1b41a60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/d6e5d3958d6841c1f5b83f5276f29b2906258bf4/docs/plots_pbmc.html" target="_blank">d6e5d39</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added PCA plot with purified PBMC clustering to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/99301a7a75c1e4c7124853dbcae9b8e4e34f4718/analysis/plots_pbmc.Rmd" target="_blank">99301a7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bf23ca096435ce5829d49f79624deb0e64846fb0/analysis/plots_pbmc.Rmd" target="_blank">bf23ca0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added manual labeling of purified PBMC data to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0c1b570c3d96a2191c3a065785cad4fa073d777a/docs/plots_pbmc.html" target="_blank">0c1b570</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
First build on plots_pbmc page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/eb7f776d4d4ff6cd9f4851ed02d6a620ce1386cf/analysis/plots_pbmc.Rmd" target="_blank">eb7f776</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
workflowr::wflow_publish(“plots_pbmc.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we examine and compare the topic modeling results for the two closely related data sets from <a href="https://doi.org/10.1038/ncomms14049">Zheng <em>et al</em> (2017)</a>, the mixture of FACS-purified PBMC data sets, and the “unsorted” 68k PBMC data. The goal of this analysis is to illustrate how the topic models fitted to these data sets can be used to learn about structure in the data. In particular, we would like to identify clusters, and interpret clusters and topics as “cell types” or “gene expression programs”.</p>
<p>Load the packages used in the analysis below, as well as additional functions that will be used to generate some of the plots.</p>
<pre class="r"><code>library(dplyr)
library(fastTopics)
library(ggplot2)
library(cowplot)
source(&quot;../code/plots.R&quot;)</code></pre>
<div id="mixture-of-facs-purified-pbmc-data" class="section level2">
<h2>Mixture of FACS-purified PBMC data</h2>
<p>We begin with the mixture of FACS-purified PBMC data. Note that the count data are no longer needed at this stage.</p>
<pre class="r"><code>load(&quot;../data/pbmc_purified.RData&quot;)
samples_purified &lt;- samples
rm(samples,genes,counts)</code></pre>
<p>Load the <span class="math inline">\(k = 6\)</span> Poisson NMF model fit.</p>
<pre class="r"><code>fit_purified &lt;-
  readRDS(&quot;../output/pbmc-purified/rds/fit-pbmc-purified-scd-ex-k=6.rds&quot;)$fit</code></pre>
<p>Here, we explore the structure of the single-cell data as inferred by the topic model. Specifically, we use PCA to uncover structure in the estimated topic proportions of the multinomial topic model. Although PCA is simple, we will see that it works well, both for visualization and identifying clusters, and avoids the complications of the popular <em>t</em>-SNE and UMAP nonlinear dimensionality reduction methods. (Note that, since the topic proportions sum to 1, there are only 5 PCs to examine, not 6.)</p>
<pre class="r"><code>fit &lt;- poisson2multinom(fit_purified)
pca &lt;- prcomp(fit$L)$x</code></pre>
<p>Three large clusters are evident from first two PCs. We label the three large clusters as “A”, “B” and “C”.</p>
<p>Since there are so many samples, the scatterplot suffers from “overplotting”. So it also helpful to view this PC projection as a density plot.</p>
<pre class="r"><code>n   &lt;- nrow(pca)
x   &lt;- rep(&quot;C&quot;,n)
pc1 &lt;- pca[,&quot;PC1&quot;]
pc2 &lt;- pca[,&quot;PC2&quot;]
x[pc1 + 0.2 &gt; pc2] &lt;- &quot;A&quot;
x[pc2 &gt; 0.25] &lt;- &quot;B&quot;
x[(pc1 + 0.4)^2 + (pc2 + 0.1)^2 &lt; 0.07] &lt;- &quot;C&quot;
samples_purified$cluster &lt;- x
p1 &lt;- pca_plot_with_labels(fit_purified,c(&quot;PC1&quot;,&quot;PC2&quot;),
                           samples_purified$cluster) +
      labs(fill = &quot;cluster&quot;)
p2 &lt;- pca_hex_plot(fit_purified,c(&quot;PC1&quot;,&quot;PC2&quot;))
plot_grid(p1,p2,rel_widths = c(9,10))</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-purified-clustering-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-purified-clustering-2-1">
Past versions of pbmc-purified-clustering-2-1.png
</button>
</p>
<div id="fig-pbmc-purified-clustering-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/30d2ef1fa568ca329fe63d8bc498c3dec56ad55e/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-2-1.png" target="_blank">30d2ef1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7900d171e513796cb2e56f95abf713355845a5ae/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-2-1.png" target="_blank">7900d17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/38f07a2440114be440e2efb027050b0e2b13bf5f/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-2-1.png" target="_blank">38f07a2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Most of the samples are in cluster A:</p>
<pre class="r"><code>table(x)
# x
#     A     B     C 
# 72614 10439 11602</code></pre>
<p>A small number of outlying data points do not seem to belong to any the three clusters, or they fall in between the clusters. For these data points, we assign them (rather arbitrarily) to one of the three clusters.</p>
<p>From these plots, there also also appears to be finer scale structure. For example, judging by the density plot, cluster A appears to split into two subclusters. We will examine this finer scale structure below.</p>
<p>Also note that other PCs beyond the first two may also sometimes reveal additional clustering, and we will see examples of this in the 68k PBMC data.</p>
<p>Within cluster C, there are two mostly well-defined subclusters (labeled “C1” and “C2”). There appear to be at least a couple other smaller, less well-defined subclusters, but in this analysis we focus on the largest, most obvious clusters.</p>
<pre class="r"><code>rows &lt;- which(samples_purified$cluster == &quot;C&quot;)
fit  &lt;- select(poisson2multinom(fit_purified),loadings = rows)
pca  &lt;- prcomp(fit$L)$x
n    &lt;- nrow(pca)
x    &lt;- rep(&quot;C3&quot;,n)
pc1  &lt;- pca[,1]
pc2  &lt;- pca[,2]
x[pc1 &lt; 0 &amp; pc2 &lt; 0.4] &lt;- &quot;C1&quot;
x[pc1 &gt; 0.5 &amp; pc2 &lt; 0.15] &lt;- &quot;C2&quot;
samples_purified[rows,&quot;cluster&quot;] &lt;- x
p3 &lt;- pca_plot_with_labels(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),x) +
      labs(fill = &quot;cluster&quot;)
p4 &lt;- pca_hex_plot(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),bins = c(0,1,5,10,100,Inf))
plot_grid(p3,p4,rel_widths = c(9,10))</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-purified-clustering-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-purified-clustering-4-1">
Past versions of pbmc-purified-clustering-4-1.png
</button>
</p>
<div id="fig-pbmc-purified-clustering-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/208d263faab0770c43953404595fdb98748755fa/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-4-1.png" target="_blank">208d263</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7900d171e513796cb2e56f95abf713355845a5ae/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-4-1.png" target="_blank">7900d17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The two subclusters, C1 and C2, account for most of the samples in cluster C. We also define a third subset, C3—a “background cluster”—containing all the samples that were not assigned to C1 or C2.</p>
<pre class="r"><code>table(x)
# x
#   C1   C2   C3 
# 7822 2909  871</code></pre>
<p>Now we turn to cluster A. Within this cluster, there is a large subcluster, which we label as “A1”; the subset of samples that are not assigned to this cluster are labeled “A2”. (The A1 subcluster is much less distinct than the other clusters we have seen so far, and may not show up clearly in this scatterplot—it is more apparent from the density plot.) Otherwise, there is no obvious additional clustering of the samples within cluster A.</p>
<pre class="r"><code>rows &lt;- which(samples_purified$cluster == &quot;A&quot;)
fit  &lt;- select(poisson2multinom(fit_purified),loadings = rows)
pca  &lt;- prcomp(fit$L)$x
n    &lt;- nrow(fit$L)
x    &lt;- rep(&quot;A2&quot;,n)
pc1  &lt;- pca[,1]
pc2  &lt;- pca[,2]
x[pc1 &gt; 0.55 - pc2 | pc1 &gt; 0.65] &lt;- &quot;A1&quot;
samples_purified[rows,&quot;cluster&quot;] &lt;- x
p5 &lt;- pca_plot_with_labels(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),x) +
      labs(fill = &quot;cluster&quot;)
p6 &lt;- pca_hex_plot(fit,c(&quot;PC1&quot;,&quot;PC2&quot;))
plot_grid(p5,p6,rel_widths = c(9,10))</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-purified-clustering-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-purified-clustering-6-1">
Past versions of pbmc-purified-clustering-6-1.png
</button>
</p>
<div id="fig-pbmc-purified-clustering-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/208d263faab0770c43953404595fdb98748755fa/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-6-1.png" target="_blank">208d263</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/97d7e8692b0a40a53f6afc1ed5ee5067b1665d33/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-6-1.png" target="_blank">97d7e86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7900d171e513796cb2e56f95abf713355845a5ae/docs/figure/plots_pbmc.Rmd/pbmc-purified-clustering-6-1.png" target="_blank">7900d17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In summary, we have subdivided the data into 6 subsets:</p>
<pre class="r"><code>samples_purified$cluster &lt;- factor(samples_purified$cluster)
table(samples_purified$cluster)
# 
#    A1    A2     B    C1    C2    C3 
#  8352 64262 10439  7822  2909   871</code></pre>
<p>We also inspected principal components individually in each of these 6 clusters and we did not find any of clear-cut examples of subclustering withing these clusters.</p>
<p>The structure plot summarizes the topic proportions in each of these 6 subsets:</p>
<pre class="r"><code>set.seed(1)
pbmc_purified_topic_colors &lt;- c(&quot;gold&quot;,&quot;forestgreen&quot;,&quot;dodgerblue&quot;,
                                &quot;gray&quot;,&quot;greenyellow&quot;,&quot;magenta&quot;)
pbmc_purified_topics &lt;- c(2,5,3,1,4,6)
rows &lt;- sort(c(sample(which(samples_purified$cluster == &quot;A1&quot;),250),
               sample(which(samples_purified$cluster == &quot;A2&quot;),1200),
               sample(which(samples_purified$cluster == &quot;B&quot;),250),
               sample(which(samples_purified$cluster == &quot;C1&quot;),250),
               sample(which(samples_purified$cluster == &quot;C2&quot;),200),
               sample(which(samples_purified$cluster == &quot;C3&quot;),200)))
p7 &lt;- structure_plot(select(poisson2multinom(fit_purified),loadings = rows),
                     grouping = samples_purified[rows,&quot;cluster&quot;],
                     topics = pbmc_purified_topics,
                     colors = pbmc_purified_topic_colors[pbmc_purified_topics],
                     n = Inf,perplexity = c(70,100,70,70,50,50),
                     gap = 40,num_threads = 4,verbose = FALSE)
print(p7)</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-purified-structure-plot-1">
Past versions of pbmc-purified-structure-plot-1.png
</button>
</p>
<div id="fig-pbmc-purified-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/208d263faab0770c43953404595fdb98748755fa/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">208d263</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/eac2d23a7e084d60d7821643a185ad2e45d8092a/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">eac2d23</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2d156b8fefbcb1c1c65e7040c43eedff5fbf5657/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">2d156b8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/abb846e9e32e6a27cd74c5d5d0dc0ca2799d60f6/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">abb846e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f53c86ce2e664d45e7e8b7299bfada46ae968f15/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">f53c86c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/13ee038ed5f2e1757c215510ac94e68870ae4074/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">13ee038</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/97d7e8692b0a40a53f6afc1ed5ee5067b1665d33/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">97d7e86</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/59777e743c15ee5a5576d476bf68227bfb2c5bf8/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">59777e7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c87ddf8d946c9c5b08d51f7aa13e49181182062f/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">c87ddf8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7900d171e513796cb2e56f95abf713355845a5ae/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">7900d17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fbb0697684addb55df25f7501714a5eac8189e6f/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">fbb0697</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/216027a8630c038d35ff5149539a552cc8650fa7/docs/figure/plots_pbmc.Rmd/pbmc-purified-structure-plot-1.png" target="_blank">216027a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Out of the 6 topics, 4 of them (<span class="math inline">\(k = 2, 3, 4, 5\)</span>) align closely with clusters (labeled A1, B, C1, C2). And, indeed, they align closely with their inclusion in the individual FACS-purified data sets:</p>
<pre class="r"><code>with(samples_purified,table(celltype,cluster))
#                               cluster
# celltype                          A1    A2     B    C1    C2    C3
#   CD19+ B                          0     3 10073     0     0     9
#   CD14+ Monocyte                   0    30     8     1  2369   204
#   CD34+                            4    43   352  7740   539   554
#   CD4+ T Helper2                   1 11183     0    16     0    13
#   CD56+ NK                      8285    78     0    17     1     4
#   CD8+ Cytotoxic T                60 10096     0     0     0    53
#   CD4+/CD45RO+ Memory              0 10201     0    19     0     4
#   CD8+/CD45RA+ Naive Cytotoxic     1 11945     3     0     0     4
#   CD4+/CD45RA+/CD25- Naive T       1 10440     1    25     0    12
#   CD4+/CD25 T Reg                  0 10243     2     4     0    14</code></pre>
<p>Based on the above results, we make a few observations:</p>
<ul>
<li><p>Because of their close correspondence, subsequent analysis of topics 2, 3, 4 and 5 should yield similar results to analyzing the clusters A1, B, C1, C2. For example, cluster B corresponds almost exactly to the B-cell data set. The largest cluster, cluster A2, is mostly comprised of T-cells.</p></li>
<li><p>Cluster A2—see also the PCA plot above—is an example where analyzing the most prevalent topics (<span class="math inline">\(k = 1, 6\)</span>) will yield different results than a cluster-based analysis.</p></li>
<li><p>Many samples labeled as “CD34+” are not assigned to the CD34+ cluster (C1). This reflects the fact that this population was much less pure (45%) than the others, so we would expect some cells labeled as “CD34+” to not necessarily be CD34+ cells.</p></li>
<li><p>Cluster C3 is a heterogeneous cluster with a relatively small number of samples (790) that could potentially contain additional clusters of biological relevance, but will likely be more challenging to analyze and interpret than the other clusters, so we do not investigate this cluster further.</p></li>
</ul>
<p>In summary, a cluster-based analysis and topic-based analysis should yield mostly similar results, except for the analysis of cluster A2, which should benefit from a topic-based analysis.</p>
</div>
<div id="unsorted-68k-pbmc-data" class="section level2">
<h2>Unsorted 68k PBMC data</h2>
<p>Next, we turn to the 68k data set.</p>
<pre class="r"><code>load(&quot;../data/pbmc_68k.RData&quot;)
samples_68k &lt;- samples
rm(samples,genes,counts)</code></pre>
<p>Load the <span class="math inline">\(k = 6\)</span> Poisson NMF model fit, and compute PCs from the topic proportions.</p>
<pre class="r"><code>fit_68k &lt;- readRDS(&quot;../output/pbmc-68k/rds/fit-pbmc-68k-scd-ex-k=6.rds&quot;)$fit
fit &lt;- poisson2multinom(fit_68k)
pca &lt;- prcomp(fit$L)$x</code></pre>
<p>From the <span class="math inline">\(k = 6\)</span> fit, we find least three distinct clusters in the projection onto PCs 3 and 4. We label these clusters “A”, “B” and “C”, as above, while cautioning that this labeling does not imply a connection with the purified PBMC clusters above.</p>
<pre class="r"><code>n   &lt;- nrow(pca)
x   &lt;- rep(&quot;A&quot;,n)
pc3 &lt;- pca[,&quot;PC3&quot;]
pc4 &lt;- pca[,&quot;PC4&quot;]
x[pc4 &lt; -0.12 | pc3/1.9 - 0.17 &gt; pc4] &lt;- &quot;B&quot;
x[pc4 &lt; -0.75] &lt;- &quot;C&quot;
samples_68k$cluster &lt;- x
p8 &lt;- pca_plot_with_labels(fit_68k,c(&quot;PC3&quot;,&quot;PC4&quot;),x) +
      labs(fill = &quot;cluster&quot;)
p9 &lt;- pca_hex_plot(fit_68k,c(&quot;PC3&quot;,&quot;PC4&quot;))
plot_grid(p8,p9,rel_widths = c(9,10))</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-68k-clustering-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The vast majority of the cells are in cluster A.</p>
<pre class="r"><code>table(samples_68k$cluster)
# 
#     A     B     C 
# 63405  5009   165</code></pre>
<p>The wide range in the sizes of these clusters is striking; the smallest cluster (C) is less than 1% the size of the largest (A1). By contrast, community detection methods such as the Louvain algorithm are preferentially biased toward more uniformly sized clusters (this is a known limitation of community detection methods).</p>
<p>Looking more closely at the top two PCs in cluster B, we identify two large clusters, with the remaining samples assigned to a “background” cluster, B3.</p>
<pre class="r"><code>rows &lt;- which(samples_68k$cluster == &quot;B&quot;)
fit  &lt;- select(poisson2multinom(fit_68k),loadings = rows)
pca  &lt;- prcomp(fit$L)$x
n    &lt;- nrow(pca)
x    &lt;- rep(&quot;B3&quot;,n)
pc1  &lt;- pca[,&quot;PC1&quot;]
x[pc1 &gt; -0.12] &lt;- &quot;B1&quot;
x[pc1 &lt; -0.3]  &lt;- &quot;B2&quot;
samples_68k[rows,&quot;cluster&quot;] &lt;- x
p10 &lt;- pca_plot_with_labels(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),x) +
       labs(fill = &quot;cluster&quot;)
p11 &lt;- pca_hex_plot(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),bins = c(0,1,5,10,20,Inf))
plot_grid(p10,p11,rel_widths = c(9,10))</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-68k-clustering-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Cluster A subdivides fairly into two large clusters, labeled A1 and A2. The remaining in-between samples are assigned to subset A3.</p>
<pre class="r"><code>rows &lt;- which(samples_68k$cluster == &quot;A&quot;)
fit  &lt;- select(poisson2multinom(fit_68k),loadings = rows)
pca  &lt;- prcomp(fit$L)$x
n    &lt;- nrow(pca)
x    &lt;- rep(&quot;A3&quot;,n)
pc2  &lt;- pca[,&quot;PC2&quot;]
pc3  &lt;- pca[,&quot;PC3&quot;]
x[2.5*pc3 &lt; 0.4 - pc2] &lt;- &quot;A1&quot;
x[pc3 &gt; 0.75 - pc2] &lt;- &quot;A2&quot;
samples_68k[rows,&quot;cluster&quot;] &lt;- x
p12 &lt;- pca_plot_with_labels(fit,c(&quot;PC2&quot;,&quot;PC3&quot;),x) +
       labs(fill = &quot;cluster&quot;)
p13 &lt;- pca_hex_plot(fit,c(&quot;PC2&quot;,&quot;PC3&quot;),bins = c(0,1,5,10,100,Inf))
plot_grid(p12,p13,rel_widths = c(9,10))</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-68k-clustering-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-68k-clustering-4-1">
Past versions of pbmc-68k-clustering-4-1.png
</button>
</p>
<div id="fig-pbmc-68k-clustering-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f53c86ce2e664d45e7e8b7299bfada46ae968f15/docs/figure/plots_pbmc.Rmd/pbmc-68k-clustering-4-1.png" target="_blank">f53c86c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b6489db5d0c0b21a2856fd9ce0fac3045ad76688/docs/figure/plots_pbmc.Rmd/pbmc-68k-clustering-4-1.png" target="_blank">b6489db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Within cluster A, the vast majority of the samples are assigned to the A1 subcluster:</p>
<pre class="r"><code>table(x)
# x
#    A1    A2    A3 
# 59294  3589   522</code></pre>
<pre class="r"><code>rows &lt;- which(samples_68k$cluster == &quot;A1&quot;)
fit  &lt;- select(poisson2multinom(fit_68k),loadings = rows)
p14 &lt;- pca_plot(fit,k = 3:4)
print(p14)</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-68k-clustering-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-68k-clustering-6-1">
Past versions of pbmc-68k-clustering-6-1.png
</button>
</p>
<div id="fig-pbmc-68k-clustering-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/399c5976cb5d054fae7e214567e39a71315ce0aa/docs/figure/plots_pbmc.Rmd/pbmc-68k-clustering-6-1.png" target="_blank">399c597</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p15 &lt;- pca_hex_plot(fit,c(&quot;PC1&quot;,&quot;PC2&quot;),bins = c(0,1,10,20,100,Inf))
print(p15)</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-68k-clustering-7-1.png" width="420" style="display: block; margin: auto;" /></p>
<p>In summary, we have subdivided these data into 7 subsets:</p>
<pre class="r"><code>samples_68k$cluster &lt;- factor(samples_68k$cluster)
table(samples_68k$cluster)
# 
#    A1    A2    A3    B1    B2    B3     C 
# 59294  3589   522  4011   819   179   165</code></pre>
<p>The structure plot summarizes the topic proportions in each of these 7 subsets:</p>
<pre class="r"><code>set.seed(1)
pbmc_68k_topic_colors &lt;- c(&quot;yellow&quot;,&quot;lightskyblue&quot;,&quot;salmon&quot;,
                           &quot;firebrick&quot;,&quot;royalblue&quot;,&quot;olivedrab&quot;)
pbmc_68k_topics &lt;- c(2,5,1,3,4,6)
rows &lt;- sort(c(sample(which(samples_68k$cluster == &quot;A1&quot;),1200),
               sample(which(samples_68k$cluster == &quot;A2&quot;),500),
               sample(which(samples_68k$cluster == &quot;A3&quot;),300),
               sample(which(samples_68k$cluster == &quot;B1&quot;),500),
               sample(which(samples_68k$cluster == &quot;B2&quot;),300),
               which(samples_68k$cluster == &quot;B3&quot;),
               which(samples_68k$cluster == &quot;C&quot;)))
p16 &lt;- structure_plot(select(poisson2multinom(fit_68k),loadings = rows),
                      grouping = samples_68k[rows,&quot;cluster&quot;],
                      topics = pbmc_68k_topics,
                      colors = pbmc_68k_topic_colors[pbmc_68k_topics],
                      perplexity = c(100,100,50,100,50,50,50),
                      n = Inf,gap = 40,num_threads = 4,verbose = FALSE)
print(p16)</code></pre>
<p><img src="figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" width="900" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pbmc-68k-structure-plot-1">
Past versions of pbmc-68k-structure-plot-1.png
</button>
</p>
<div id="fig-pbmc-68k-structure-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/399c5976cb5d054fae7e214567e39a71315ce0aa/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">399c597</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/abb846e9e32e6a27cd74c5d5d0dc0ca2799d60f6/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">abb846e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f53c86ce2e664d45e7e8b7299bfada46ae968f15/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">f53c86c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a0cb7c61cd959b2882f75eb4de7a3eda1eb07b0b/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">a0cb7c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7900d171e513796cb2e56f95abf713355845a5ae/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">7900d17</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fbb0697684addb55df25f7501714a5eac8189e6f/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">fbb0697</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/216027a8630c038d35ff5149539a552cc8650fa7/docs/figure/plots_pbmc.Rmd/pbmc-68k-structure-plot-1.png" target="_blank">216027a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>These subsets do not align as closely with the cell-type labeling inferred by Zheng et al (2017), which is not surprising considering that this labeling is based on the FACS-purified data set.</p>
<pre class="r"><code>with(samples_68k,table(celltype,cluster))
#                               cluster
# celltype                          A1    A2    A3    B1    B2    B3     C
#   CD14+ Monocyte                   6     1     1  2840     1    11     2
#   CD19+ B                       1983  3577   310     0    37     1     0
#   CD34+                           10     5    47    30    21     1   163
#   CD4+ T Helper2                  67     0    16     6     8     0     0
#   CD4+/CD25 T Reg               6174     0    11     2     0     0     0
#   CD4+/CD45RA+/CD25- Naive T    1863     1     4     1     3     1     0
#   CD4+/CD45RO+ Memory           3059     0     0     2     0     0     0
#   CD56+ NK                      8734     0    24    12     1     5     0
#   CD8+ Cytotoxic T             20672     1    74    24     0     2     0
#   CD8+/CD45RA+ Naive Cytotoxic 16649     0     9     0     5     3     0
#   Dendritic                       77     4    26  1094   743   155     0</code></pre>
<p>A few notes about these results:</p>
<ul>
<li><p>As in the purified fit, here we identify a B-cells cluster (B) and topic (5) that closely match.</p></li>
<li><p>We also identify what is most likely a cluster of CD34+ cells (C), although the corresponding topic (6) is not distinctive to this cluster, so it remains to be seen if topic 6 also characterises CD34+ cells.</p></li>
<li><p>Unlike the FACS-purified data, don’t identify a clear-cut cluster for NK cells; the NK cells are mixed in with the T-cells (subset A1). NK cells will emerge only after subsequent analysis of topic 3.</p></li>
<li><p>Some of the distinct clusters (e.g., B1a, B1b, B2) are characterized by mixtures of topics, so if these clusters do indeed correspond to interesting cell types, analyzing the topics alone may not shed light onto these cell types.</p></li>
</ul>
<p>In summary, the topics and clusters seem to offer very much complementary biological insights, although subsequent analysis is needed to determine what these insights are.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.5
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0      ggplot2_3.3.0      fastTopics_0.3-165 dplyr_0.8.3       
# 
# loaded via a namespace (and not attached):
#  [1] ggrepel_0.9.0        Rcpp_1.0.5           lattice_0.20-38     
#  [4] tidyr_1.0.0          prettyunits_1.1.1    assertthat_0.2.1    
#  [7] zeallot_0.1.0        rprojroot_1.3-2      digest_0.6.23       
# [10] R6_2.4.1             backports_1.1.5      MatrixModels_0.4-1  
# [13] evaluate_0.14        coda_0.19-3          httr_1.4.1          
# [16] pillar_1.4.3         rlang_0.4.5          progress_1.2.2      
# [19] lazyeval_0.2.2       data.table_1.12.8    irlba_2.3.3         
# [22] SparseM_1.78         hexbin_1.28.0        whisker_0.4         
# [25] Matrix_1.2-18        rmarkdown_2.3        labeling_0.3        
# [28] Rtsne_0.15           stringr_1.4.0        htmlwidgets_1.5.1   
# [31] munsell_0.5.0        compiler_3.6.2       httpuv_1.5.2        
# [34] xfun_0.11            pkgconfig_2.0.3      mcmc_0.9-6          
# [37] htmltools_0.4.0      tidyselect_0.2.5     tibble_2.1.3        
# [40] workflowr_1.6.2.9000 quadprog_1.5-8       viridisLite_0.3.0   
# [43] crayon_1.3.4         withr_2.1.2          later_1.0.0         
# [46] MASS_7.3-51.4        grid_3.6.2           jsonlite_1.6        
# [49] gtable_0.3.0         lifecycle_0.1.0      git2r_0.26.1        
# [52] magrittr_1.5         scales_1.1.0         RcppParallel_5.0.2  
# [55] stringi_1.4.3        farver_2.0.1         fs_1.3.1            
# [58] promises_1.1.0       vctrs_0.2.1          tools_3.6.2         
# [61] glue_1.3.1           purrr_0.3.3          hms_0.5.2           
# [64] yaml_2.2.0           colorspace_1.4-1     plotly_4.9.2        
# [67] knitr_1.26           quantreg_5.54        MCMCpack_1.4-5</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
