<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Visualize and interpret droplet and pulse-seq topics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Visualize and interpret droplet and pulse-seq topics</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree2db338f90b7dfb0bfafc363b9cfa73dbd40e6df5targetblank2db338fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/2db338f90b7dfb0bfafc363b9cfa73dbd40e6df5" target="_blank">2db338f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree2db338f90b7dfb0bfafc363b9cfa73dbd40e6df5targetblank2db338fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/2db338f90b7dfb0bfafc363b9cfa73dbd40e6df5" target="_blank">2db338f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plots_tracheal_epithelium.Rmd</code>) and HTML (<code>docs/plots_tracheal_epithelium.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2db338f90b7dfb0bfafc363b9cfa73dbd40e6df5/analysis/plots_tracheal_epithelium.Rmd" target="_blank">2db338f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-29
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/4dbb46bce0584520f0582b7e6035eecbb30ae8f1/analysis/plots_tracheal_epithelium.Rmd" target="_blank">4dbb46b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-28
</td>
<td>
A couple small edits to the test in the plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/28f8ec5088481200eeb4048d4f76c9a9813426ab/analysis/plots_tracheal_epithelium.Rmd" target="_blank">28f8ec5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-28
</td>
<td>
z-scores in zscore_scatterplot are now shown on sqrt-scale.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bd72ce4eaf639a0f5b0fce5e0dcc53d679f8ddcb/analysis/plots_tracheal_epithelium.Rmd" target="_blank">bd72ce4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-28
</td>
<td>
Improved plots for ionocytes and ciliated topics/clusters in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ff9d6ef4c14b8e031f1f3fb29a26d442012e6720/analysis/plots_tracheal_epithelium.Rmd" target="_blank">ff9d6ef</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-28
</td>
<td>
Added scatterplots for ciliated cells to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1fab15a9e98f3221a3242f76ad71bd88b6c1ae55/analysis/plots_tracheal_epithelium.Rmd" target="_blank">1fab15a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-28
</td>
<td>
Added plots for proliferating cells cluster to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/920c62f47ccdafccb4559f919e52d816373197c0/analysis/plots_tracheal_epithelium.Rmd" target="_blank">920c62f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-27
</td>
<td>
Made a couple small edits to plots_tracheal_epithelium.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/50691bf63953cac2d4a96d41b66e6befb733e989/analysis/plots_tracheal_epithelium.Rmd" target="_blank">50691bf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
<td>
Working on volcano plots and scatterplots for club cells in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0e10adfe511026784a56133da08883eaaf486e24/analysis/plots_tracheal_epithelium.Rmd" target="_blank">0e10adf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
<td>
Working on volcano plots and scatterplots for basal and hillock cells in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/6272c693d94ade551405c8ec21330bda16e1a838/docs/plots_tracheal_epithelium.html" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f9fe3eb4d532228fe192425e0191e5f44952073a/analysis/plots_tracheal_epithelium.Rmd" target="_blank">f9fe3eb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/95f91d46a8ac16ffc8c8b9e84e77f4b7753dd83f/analysis/plots_tracheal_epithelium.Rmd" target="_blank">95f91d4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
<td>
Improved functions logfoldchange_scatterplot and zscores_scatterplot in plots.R.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/73ef43975b21edb847356104b0f20f3bb0b55fb5/analysis/plots_tracheal_epithelium.Rmd" target="_blank">73ef439</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
<td>
Made some improvements to zscore_scatterplot in plots.R.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/56bb5fdc9321a6ce288f6afc63c47855866446b5/analysis/plots_tracheal_epithelium.Rmd" target="_blank">56bb5fd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
Adding some scatterplots and volcano plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/plots_tracheal_epithelium.html" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
Added volcano plots for T+N celels to plots_tracheal_epithelium
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7262a9632efd20559839c3a8480faaf0c3b583cd/analysis/plots_tracheal_epithelium.Rmd" target="_blank">7262a96</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/30194ad9cc98861d418fed48d8d4565a3742c31b/docs/plots_tracheal_epithelium.html" target="_blank">30194ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
Added volcano plots and scatterplot for goblet cells.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9e590031f4c16891143ce55a7b870ea04b59fe6c/analysis/plots_tracheal_epithelium.Rmd" target="_blank">9e59003</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c0fa2db2320a5627d877475e7e86d56710b76603/docs/plots_tracheal_epithelium.html" target="_blank">c0fa2db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
Revised plots for ionocytes cluster in plots_tracheal_epithelium
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/81506fb324f25668a3f51cf79edc380a00ce4871/analysis/plots_tracheal_epithelium.Rmd" target="_blank">81506fb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/78d64e9067ba91b6010a14e255b5b98331fb1010/docs/plots_tracheal_epithelium.html" target="_blank">78d64e9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
<td>
Added ionocytes volcano plot to plots_tracheal_epithelium.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f0bdbda1a52710b579bc49aef78bbf183c170a11/analysis/plots_tracheal_epithelium.Rmd" target="_blank">f0bdbda</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/361eded8d39fef83e29ceba7c42a2f3aa0ae4e9e/docs/plots_tracheal_epithelium.html" target="_blank">361eded</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
<td>
Added text to accompany ciliated cells volcano plots.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8198c07fd1f92ea5ae43cc52b33f2a1b4288f340/analysis/plots_tracheal_epithelium.Rmd" target="_blank">8198c07</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/23f87bb202ad31fcb5ebc7aaaba8bf0344e09343/docs/plots_tracheal_epithelium.html" target="_blank">23f87bb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
<td>
Improved volcano plots for ciliated cell type in
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/516a32ee85f8000ea1fd0998fc5030b2db66f660/analysis/plots_tracheal_epithelium.Rmd" target="_blank">516a32e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0ed5061eff987ada9a6387e5a2dd43b314014be5/docs/plots_tracheal_epithelium.html" target="_blank">0ed5061</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
Added some rough, first-draft volcano plots to
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6ccf99810c29df43cfb1efe7c7a636c8058ed8fb/analysis/plots_tracheal_epithelium.Rmd" target="_blank">6ccf998</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/de36d3106fd4290326ae4c105179949a80fef267/analysis/plots_tracheal_epithelium.Rmd" target="_blank">de36d31</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
Added volcano plots for ciliated cells to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/082352a69f0c7749ec5296a4555e6b230ec58bcb/analysis/plots_tracheal_epithelium.Rmd" target="_blank">082352a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
Added steps to plots_tracheal_epithelium analysis to compute differential expression statistics.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/06d0b30f47d955a1f399fc678ba6bfbe1b0c07a2/docs/plots_tracheal_epithelium.html" target="_blank">06d0b30</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
I’m starting to revamp the plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0877a1fbd66259386965572ab33f1c4697937348/analysis/plots_tracheal_epithelium.Rmd" target="_blank">0877a1f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/5af46f1322d6b97d127e70e4cf5e9a9060c77126/analysis/plots_tracheal_epithelium.Rmd" target="_blank">5af46f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-20
</td>
<td>
Working on Structure plot for droplet data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c0725777d2e931d27accb424694ebad549c64da6/analysis/plots_tracheal_epithelium.Rmd" target="_blank">c072577</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-19
</td>
<td>
Re-created structure plot for pulseseq data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/942486b7eb0536b69a20ae4e4502c20ba58d235e/analysis/plots_tracheal_epithelium.Rmd" target="_blank">942486b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-18
</td>
<td>
Fixing merge issue.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8cf758e751db6882c4ab867258140a908d230f42/analysis/plots_tracheal_epithelium.Rmd" target="_blank">8cf758e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-12
</td>
<td>
Working on improvements to clustering of pulse-seq data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a128de5ef56450d69bfd584935bb294d71120d9d/analysis/plots_tracheal_epithelium.Rmd" target="_blank">a128de5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-12
</td>
<td>
Revamping the analysis of the pulseseq data in plots_tracheal_epithelium.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3ab7da1135c1021866310fb12a88c4ad3b237395/analysis/plots_tracheal_epithelium.Rmd" target="_blank">3ab7da1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
A few minor edits.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/f2e0b23bed4cda52183098bf6a910703435b35df/docs/plots_tracheal_epithelium.html" target="_blank">f2e0b23</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Fixed dimensions of PCA plots in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/7b59815b30e32e2d5f588bf89411bee7a3311963/analysis/plots_tracheal_epithelium.Rmd" target="_blank">7b59815</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/55896110708bbeb391769a090397a28a1543e9d2/docs/plots_tracheal_epithelium.html" target="_blank">5589611</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added PCA plots and structure plots from pulseseq data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b731a4a8be11b3d714d52d28a6c7fd9196630e43/analysis/plots_tracheal_epithelium.Rmd" target="_blank">b731a4a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
workflowr::wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c3c1b123eda35d894e3f2ffc836aaec5650530e6/docs/plots_tracheal_epithelium.html" target="_blank">c3c1b12</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/2defe6d80efff8e3f48136d45ef1464755c0d740/analysis/plots_tracheal_epithelium.Rmd" target="_blank">2defe6d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Added crosstab plot to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/97c13c2b59f769918858e89c6374d5e7049c6e37/docs/plots_tracheal_epithelium.html" target="_blank">97c13c2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8/analysis/plots_tracheal_epithelium.Rmd" target="_blank">e11855b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Working on revised analysis of droplet and pulse-seq data sets.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/e11855bf04b544e1fbfd4ef90b0162a8aa8d6df8/docs/plots_tracheal_epithelium.html" target="_blank">e11855b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-25
</td>
<td>
Working on revised analysis of droplet and pulse-seq data sets.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bf23ca096435ce5829d49f79624deb0e64846fb0/analysis/plots_tracheal_epithelium.Rmd" target="_blank">bf23ca0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added manual labeling of purified PBMC data to plots_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/077d3d55ca6da2879de67f6d1e866f0b5ef5ac99/analysis/plots_tracheal_epithelium.Rmd" target="_blank">077d3d5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Added k=9 and k=11 pulseseq fits to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0ce960489da333344f062deaac7bffdbaf8b5838/docs/plots_tracheal_epithelium.html" target="_blank">0ce9604</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
Re-built plots_tracheal_epithelium with fastTopics 0.3-162.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/961570e0d6a33b7b3908c6feb3beca085c401268/analysis/plots_tracheal_epithelium.Rmd" target="_blank">961570e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-20
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/b17bfa4f7de6a68e6b6cee8ed7620188a8411230/docs/plots_tracheal_epithelium.html" target="_blank">b17bfa4</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added pulseseq PCA plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/76dc0c6859e954265dc836ca481a7c06cc174866/analysis/plots_tracheal_epithelium.Rmd" target="_blank">76dc0c6</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c70612f1d996cac0442f6b5346aa812e5e51cfa4/analysis/plots_tracheal_epithelium.Rmd" target="_blank">c70612f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Revised structure plot settings for abundant droplet samples in plots_tracheal_epithelium.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/adda33faa241fa7ea1abe8bf5eff706d459f5547/docs/plots_tracheal_epithelium.html" target="_blank">adda33f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Fixed another structure plot in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/29a9258d418e4df1553836d2e1a3de7073ce2e3e/analysis/plots_tracheal_epithelium.Rmd" target="_blank">29a9258</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0a16b60e3247e5a35fa4637e0133965db35eab0b/docs/plots_tracheal_epithelium.html" target="_blank">0a16b60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Fixed structure plot in plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/3a7bd7465276742c305377eb2737c52e0e031abe/analysis/plots_tracheal_epithelium.Rmd" target="_blank">3a7bd74</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/f4bdf191a428ef8917fdd6f9e386653350125a0d/docs/plots_tracheal_epithelium.html" target="_blank">f4bdf19</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added explanatory text and improved PC-based manual clustering of
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c7b77ee2b8b9d0ac075a4cfcf78a1a755ca868d6/analysis/plots_tracheal_epithelium.Rmd" target="_blank">c7b77ee</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/70a4a60fe8655bd8d5609ed02a7f3718c41ae6fa/analysis/plots_tracheal_epithelium.Rmd" target="_blank">70a4a60</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added note to plots_tracheal_epithelium.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/fb21b3b781ba22db8b89d1fb531cf91e63e4e1d5/docs/plots_tracheal_epithelium.html" target="_blank">fb21b3b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added very initial Structure plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/d35cb03fade4ee2fc5da4880b31c21d1c2bb9dda/analysis/plots_tracheal_epithelium.Rmd" target="_blank">d35cb03</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/368a74a30418728c7ae896e6ec1c567e24ace79f/docs/plots_tracheal_epithelium.html" target="_blank">368a74a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added some text to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/223406b7e0fc6e73efd027e6d1bc00b0b0ad5c6d/analysis/plots_tracheal_epithelium.Rmd" target="_blank">223406b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/aca46cc698358965ad25399bf7b19f74a6b5df23/docs/plots_tracheal_epithelium.html" target="_blank">aca46cc</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added manual clustering of droplet samples based on PCs.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/38f811b28b8ba3f3996f2551c0e0a4eace930da3/analysis/plots_tracheal_epithelium.Rmd" target="_blank">38f811b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/343747ead094b567a93133bdba3411bff3c25667/analysis/plots_tracheal_epithelium.Rmd" target="_blank">343747e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Small edit to figure dimensions.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/5a35bbd6b181f322bc808840a2df3c20a6b80d94/docs/plots_tracheal_epithelium.html" target="_blank">5a35bbd</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added labeled PCA plot; adjusted plot dimensions in
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fb91075ba717e53382f9ae3ea99b065256687108/analysis/plots_tracheal_epithelium.Rmd" target="_blank">fb91075</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/8b9b5289e71be99509c2adf86e0bfead46cb50b1/docs/plots_tracheal_epithelium.html" target="_blank">8b9b528</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
Added more PCA plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ee7cbf157775bd42c882bb897efac8e07994c207/analysis/plots_tracheal_epithelium.Rmd" target="_blank">ee7cbf1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-19
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/c517ea207b696aa2815c2c77963d6c305373b7d3/docs/plots_tracheal_epithelium.html" target="_blank">c517ea2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Small fix to one of the PCA plots in plots_tracheal_epithelium.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/8f5c210a6a93dd5e5c25d35f100e2f3b88f1c3f3/analysis/plots_tracheal_epithelium.Rmd" target="_blank">8f5c210</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/01afbd2b764fd3d6724afa103aeefa2e4c91ff60/docs/plots_tracheal_epithelium.html" target="_blank">01afbd2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Added some PC plots to the plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f1c7d028ef9fa5c96cf785283b567f98c26b365b/analysis/plots_tracheal_epithelium.Rmd" target="_blank">f1c7d02</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/0a04fc1929d7987b10f019c015f116aa46705bcc/docs/plots_tracheal_epithelium.html" target="_blank">0a04fc1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Added abundance plots to plots_tracheal_epithelium analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/f914f7e28e63914b21b1b6a00c7c73e81ee1a961/analysis/plots_tracheal_epithelium.Rmd" target="_blank">f914f7e</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
wflow_publish(“plots_tracheal_epithelium.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/61917ad82d0229261c265c7546d2cb48e3ce2a67/analysis/plots_tracheal_epithelium.Rmd" target="_blank">61917ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-18
</td>
<td>
Working on new analysis, plots_tracheal_epithelium.Rmd.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we closely examine the topic modeling results for the two epithelial airway data sets (droplet and pulse-seq), and investigate the benefits of modeling single cells as <em>mixtures</em> of gene expression programs in these data sets. In particular, we will compare and contrast differential expression in clusters vs. topics.</p>
<p>Load the packages used in the analysis below, as well as additional functions that we will use to generate some of the plots.</p>
<pre class="r"><code>library(Matrix)
library(dplyr)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source(&quot;../code/plots.R&quot;)</code></pre>
<div id="load-data-and-results" class="section level2">
<h2>Load data and results</h2>
<p>Load the smaller “droplet” data set, the <span class="math inline">\(k = 7\)</span> Poisson NMF model fit for these data, the 8 clusters identified in the <a href="clusters_droplet.html">clustering analysis</a>, and the results of the differential expression analysis.</p>
<pre class="r"><code>load(&quot;../data/droplet.RData&quot;)
load(&quot;../output/droplet/diff-count-droplet.RData&quot;)
counts_droplet &lt;- counts
samples_droplet &lt;- readRDS(&quot;../output/droplet/clustering-droplet.rds&quot;)
fit_droplet &lt;- readRDS(&quot;../output/droplet/rds/fit-droplet-scd-ex-k=7.rds&quot;)$fit
diff_count_droplet &lt;- diff_count_topics
diff_count_clusters_droplet &lt;- diff_count_clusters
diff_count_merge_droplet &lt;- diff_count_merge_club
rm(samples,counts)
rm(diff_count_topics,diff_count_clusters,diff_count_merge_club)</code></pre>
<p>For reference, we show here the Structure plot from the <a href="clusters_droplet.html">clustering analysis of the droplet data</a>. This Structure plot summarizes the topic proportions in each of the 8 subsets (including the background cluster).</p>
<p><img src="figure/clusters_droplet.Rmd/structure-plot-1.png" /></p>
<p>Next, we load the larger “pulse-seq” data set, the <span class="math inline">\(k = 11\)</span> Poisson NMF model fit for these data, and the 7 clusters identified in the <a href="clusters_pulseseq.html">clustering analysis</a>.</p>
<pre class="r"><code>load(&quot;../data/pulseseq.RData&quot;)
load(&quot;../output/pulseseq/diff-count-pulseseq.RData&quot;)
counts_pulseseq  &lt;- counts
samples_pulseseq &lt;- readRDS(&quot;../output/pulseseq/clustering-pulseseq.rds&quot;)
fit_pulseseq &lt;- readRDS(&quot;../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds&quot;)$fit
diff_count_pulseseq &lt;- diff_count_topics
diff_count_clusters_pulseseq &lt;- diff_count_clusters
diff_count_merge_pulseseq &lt;- diff_count_merge_bc
rm(samples,counts)
rm(diff_count_topics,diff_count_clusters,diff_count_merge_bc)</code></pre>
<p>For reference, we show here the Structure plot from the <a href="clusters_pulseseq.html">clustering analysis of the pulse-seq data</a>. This Structure plot summarizes the topic proportions in each of the 7 subsets (including the background cluster):</p>
<p><img src="figure/clusters_pulseseq.Rmd/structure-plot-1.png" /></p>
</div>
<div id="ciliated-cells" class="section level2">
<h2>Ciliated cells</h2>
<p>We begin with the cluster that captures ciliated cells. This cluster is one of the most distinctive in both the droplet and pulse-seq data sets. Ciliated cells are abundant, though not as much as basal and club cells.</p>
<pre class="r"><code>ciliated_genes &lt;- c(&quot;Ccdc113&quot;,&quot;Ccdc153&quot;,&quot;Cdhr3&quot;,&quot;Foxj1&quot;,&quot;Lztfl1&quot;,&quot;Mlf1&quot;)
p1 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,&quot;Cil&quot;,
                                          ciliated_genes,
                                          label_above_quantile = 0.998)
print(p1)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plot-droplet-cil-1">
Past versions of volcano-plot-droplet-cil-1.png
</button>
</p>
<div id="fig-volcano-plot-droplet-cil-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6272c693d94ade551405c8ec21330bda16e1a838/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/30194ad9cc98861d418fed48d8d4565a3742c31b/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">30194ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c0fa2db2320a5627d877475e7e86d56710b76603/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">c0fa2db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/78d64e9067ba91b6010a14e255b5b98331fb1010/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">78d64e9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/361eded8d39fef83e29ceba7c42a2f3aa0ae4e9e/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">361eded</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/23f87bb202ad31fcb5ebc7aaaba8bf0344e09343/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">23f87bb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed5061eff987ada9a6387e5a2dd43b314014be5/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-cil-1.png" target="_blank">0ed5061</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Marker genes and transcription factors identified in Montoro <em>et al</em> (2018) are highlighted in black, and other top differentially expressed genes are shown with gray labels.</p>
<p>We obtain similar top differentially expressed genes in the pulse-seq data:</p>
<pre class="r"><code>p2 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,&quot;Cil&quot;,
                                          ciliated_genes,
                                          label_above_quantile = 0.998)
print(p2)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plot-pulseseq-cil-1">
Past versions of volcano-plot-pulseseq-cil-1.png
</button>
</p>
<div id="fig-volcano-plot-pulseseq-cil-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6272c693d94ade551405c8ec21330bda16e1a838/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c0fa2db2320a5627d877475e7e86d56710b76603/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">c0fa2db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/78d64e9067ba91b6010a14e255b5b98331fb1010/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">78d64e9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/361eded8d39fef83e29ceba7c42a2f3aa0ae4e9e/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">361eded</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/23f87bb202ad31fcb5ebc7aaaba8bf0344e09343/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">23f87bb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed5061eff987ada9a6387e5a2dd43b314014be5/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-cil-1.png" target="_blank">0ed5061</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The topic for the ciliated cell-type (<span class="math inline">\(k = 7\)</span>) corresponds very closely to the “Cil” cluster in the pulse-seq data. In addition, gene enrichments are considerably stronger in the topic for ciliated genes, particularly so for some genes with lower expression levels.</p>
<pre class="r"><code>p3 &lt;- zscores_scatterplot(diff_count_clusters_pulseseq,
                          diff_count_pulseseq,&quot;Cil&quot;,&quot;k7&quot;,ciliated_genes,
                          xlab = &quot;cluster Cil&quot;,ylab = &quot;topic 7&quot;)
p4 &lt;- beta_scatterplot(diff_count_clusters_pulseseq,diff_count_pulseseq,
                       &quot;Cil&quot;,&quot;k7&quot;,ciliated_genes,
                       xlab = &quot;cluster Cil&quot;,ylab = &quot;topic 7&quot;)
plot_grid(p3,p4)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/scatterplots-pulseseq-Cil-vs-k7-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="ionocytes" class="section level2">
<h2>Ionocytes</h2>
<p>In the pulse-seq data, we identify a distinctive cluster for the newly discovered, rare “ionocyte” cell type. Gene expression in these cells is not fully captured by any single topic, yet the mixture of topics forms a distinctive cluster.</p>
<pre class="r"><code>ionocyte_genes &lt;- c(&quot;Ascl3&quot;,&quot;Asgr1&quot;,&quot;Atp6v0d2&quot;,&quot;Atp6v1c2&quot;,&quot;Cftr&quot;,&quot;Foxi1&quot;,
                    &quot;Moxd1&quot;,&quot;P2ry14&quot;,&quot;Stap1&quot;)
p5 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,&quot;I&quot;,
                                          ionocyte_genes,
                                          label_above_quantile = 0.998)
print(p5)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" width="528" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plot-pulseseq-I-1">
Past versions of volcano-plot-pulseseq-I-1.png
</button>
</p>
<div id="fig-volcano-plot-pulseseq-I-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6272c693d94ade551405c8ec21330bda16e1a838/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/30194ad9cc98861d418fed48d8d4565a3742c31b/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">30194ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c0fa2db2320a5627d877475e7e86d56710b76603/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">c0fa2db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/78d64e9067ba91b6010a14e255b5b98331fb1010/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">78d64e9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/23f87bb202ad31fcb5ebc7aaaba8bf0344e09343/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">23f87bb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed5061eff987ada9a6387e5a2dd43b314014be5/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-I-1.png" target="_blank">0ed5061</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We do not identify a cluster for ionocytes in the droplet data. Judging by expression of the <em>Foxi1</em> ionocyte marker gene, only a handful of cells in the droplet data are ionocytes:</p>
<pre class="r"><code>p6 &lt;- pca_plot_with_counts(fit_droplet,counts_droplet[,&quot;Foxi1&quot;],1:2)
print(p6)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" width="450" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-plot-droplet-Foxi1-1">
Past versions of pca-plot-droplet-Foxi1-1.png
</button>
</p>
<div id="fig-pca-plot-droplet-Foxi1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/30194ad9cc98861d418fed48d8d4565a3742c31b/docs/figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" target="_blank">30194ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/c0fa2db2320a5627d877475e7e86d56710b76603/docs/figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" target="_blank">c0fa2db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/78d64e9067ba91b6010a14e255b5b98331fb1010/docs/figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" target="_blank">78d64e9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/23f87bb202ad31fcb5ebc7aaaba8bf0344e09343/docs/figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" target="_blank">23f87bb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed5061eff987ada9a6387e5a2dd43b314014be5/docs/figure/plots_tracheal_epithelium.Rmd/pca-plot-droplet-Foxi1-1.png" target="_blank">0ed5061</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="goblet-cells" class="section level2">
<h2>Goblet cells</h2>
<p>Consistent with Montoro <em>et al</em> (2018), we identify a cluster of Goblet cells in the droplet data.</p>
<pre class="r"><code>goblet_genes &lt;- &quot;Gp2&quot;
p7 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,&quot;G&quot;,
                                          goblet_genes,
                                          label_above_quantile = 0.998)
print(p7)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-G-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plot-droplet-G-1">
Past versions of volcano-plot-droplet-G-1.png
</button>
</p>
<div id="fig-volcano-plot-droplet-G-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6272c693d94ade551405c8ec21330bda16e1a838/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-G-1.png" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-G-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/30194ad9cc98861d418fed48d8d4565a3742c31b/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-G-1.png" target="_blank">30194ad</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/23f87bb202ad31fcb5ebc7aaaba8bf0344e09343/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-G-1.png" target="_blank">23f87bb</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0ed5061eff987ada9a6387e5a2dd43b314014be5/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-G-1.png" target="_blank">0ed5061</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Topic <span class="math inline">\(k = 1\)</span> is unique to this cluster suggesting that the this topic characterizes the goblet cell type. Indeed, there is a very close correspondence between the topic and cluster, with several characteristic genes (e.g. <em>Gp2</em>) showing somewhat stronger enrichment in the topic.</p>
<pre class="r"><code>p8 &lt;- zscores_scatterplot(diff_count_clusters_droplet,
                          diff_count_droplet,&quot;G&quot;,&quot;k1&quot;,goblet_genes,
                          label_above_score = 200,
                          xlab = &quot;cluster G&quot;,ylab = &quot;topic 1&quot;)
p9 &lt;- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                       &quot;G&quot;,&quot;k1&quot;,goblet_genes,xlab = &quot;cluster G&quot;,
                       ylab = &quot;topic 1&quot;)
plot_grid(p8,p9)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/scatterplots-droplet-G-vs-k1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="tuft-and-pulmonary-neuroendocrine-cells" class="section level2">
<h2>Tuft and pulmonary neuroendocrine cells</h2>
<p>In both data sets, we identify clusters for tuft and pulmonary neuroendocrine cells. The fitted topic models do not distinguish between these two rare cell types; we identify these cell types as a single cluster.</p>
<pre class="r"><code>tuft_neuroendocrine_genes &lt;- c(&quot;Ascl1&quot;,&quot;Ascl2&quot;,&quot;Ascl3&quot;,&quot;Chga&quot;,&quot;Dclk1&quot;,&quot;Rgs13&quot;)
p8 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,&quot;T+N&quot;,
                                          tuft_neuroendocrine_genes,
                                          label_above_quantile = 0.998)
print(p8)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-T+N-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plot-droplet-T+N-1">
Past versions of volcano-plot-droplet-T+N-1.png
</button>
</p>
<div id="fig-volcano-plot-droplet-T+N-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6272c693d94ade551405c8ec21330bda16e1a838/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-T+N-1.png" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-droplet-T+N-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here is the volcano plot from the pulse-seq data:</p>
<pre class="r"><code>p9 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,&quot;T+N&quot;,
                                          tuft_neuroendocrine_genes,
                                          label_above_quantile = 0.998)
print(p9)</code></pre>
<p><img src="figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-T+N-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plot-pulseseq-T+N-1">
Past versions of volcano-plot-pulseseq-T+N-1.png
</button>
</p>
<div id="fig-volcano-plot-pulseseq-T+N-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6272c693d94ade551405c8ec21330bda16e1a838/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-T+N-1.png" target="_blank">6272c69</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/6a9691b6dc7c3b314755caf9ad39e71dedf27291/docs/figure/plots_tracheal_epithelium.Rmd/volcano-plot-pulseseq-T+N-1.png" target="_blank">6a9691b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-09-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="basal-cells" class="section level2">
<h2>Basal cells</h2>
<p>We now move on to the large majority of cells in each of the epithelial airway data sets that do not break down into distinct clusters.</p>
<p>Abundant basal cells in droplet data:</p>
<pre class="r"><code>basal_genes &lt;- c(&quot;Aqp3&quot;,&quot;Krt5&quot;,&quot;Dapl1&quot;,&quot;Hspa1a&quot;,&quot;Trp63&quot;)
p10 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,&quot;B&quot;,
                                           basal_genes,
                                           label_above_quantile = 0.995)
print(p10)</code></pre>
<p>Compare basal topic (<span class="math inline">\(k = 2\)</span>) against basal cluster in droplet data:</p>
<pre class="r"><code>p11 &lt;- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                        &quot;B&quot;,&quot;k2&quot;,basal_genes,labs = &quot;cluster B&quot;) +
  labs(x = &quot;cluster B&quot;,y = &quot;topic 2&quot;,title = &quot;log-fold change (\u03b2)&quot;)
print(p11)</code></pre>
<p>Basal cells in pulse-seq data:</p>
<pre class="r"><code>p12 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,&quot;B&quot;,
                                           basal_genes,
                                           label_above_quantile = 0.995)
print(p12)</code></pre>
<p>Compare combined basal topics (<span class="math inline">\(k = 1, 3, 9\)</span>) against cluster in pulse-seq data:</p>
<pre class="r"><code>p13 &lt;- beta_scatterplot(diff_count_clusters_pulseseq,
                        diff_count_merge_pulseseq,
                        &quot;B&quot;,&quot;k1+k3+k9&quot;,basal_genes,label_above_score = Inf) +
  labs(x = &quot;cluster B&quot;,y = &quot;topics 1, 3 and 9&quot;,
       title = &quot;log-fold change (\u03b2)&quot;)
print(p13)</code></pre>
</div>
<div id="club-cells" class="section level2">
<h2>Club cells</h2>
<p>Club cells in droplet data:</p>
<pre class="r"><code>club_genes &lt;- c(&quot;Nfia&quot;,&quot;Cbr2&quot;,&quot;Krt15&quot;,&quot;Cyp2f2&quot;,&quot;Lypd2&quot;,&quot;Scgb1a1&quot;)
p16 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,&quot;C&quot;,
                                           club_genes,
                                           label_above_quantile = 0.995)
print(p16)</code></pre>
<p>Club cells in pulse-seq data:</p>
<pre class="r"><code>p17 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,&quot;C&quot;,
                                           club_genes,
                                           label_above_quantile = 0.995)
print(p17)</code></pre>
<p>Club topic in droplet data:</p>
<pre class="r"><code>p18 &lt;- beta_scatterplot(diff_count_clusters_droplet,diff_count_merge_droplet,
                        &quot;C&quot;,&quot;k5+k7&quot;,club_genes,label_above_score = Inf)
print(p18)</code></pre>
<p>Club topic in pulse-seq data:</p>
<pre class="r"><code>p19 &lt;- beta_scatterplot(diff_count_clusters_pulseseq,diff_count_merge_pulseseq,
                        &quot;C&quot;,&quot;k4+k5+k6+k8+k10&quot;,club_genes,
                        label_above_score = Inf)
print(p19)</code></pre>
</div>
<div id="proliferating-cells" class="section level2">
<h2>Proliferating cells</h2>
<p>Proliferating cells in pulse-seq data:</p>
<pre class="r"><code>cell_cycle_genes &lt;- c(&quot;Cdk1&quot;,&quot;Ube2c&quot;,&quot;Top2a&quot;)
p14 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,&quot;P&quot;,
                                           cell_cycle_genes,
                                           label_above_quantile = 0.998)
print(p14)</code></pre>
<pre class="r"><code>qt_random_tie &lt;- function (x) {
  y = rank(x,ties.method = &quot;random&quot;)
  return(qqnorm(y,plot.it = FALSE)$x)
}
rows &lt;- with(samples_pulseseq,which(cluster == &quot;B&quot; | cluster == &quot;P&quot;))
fit2 &lt;- select(poisson2multinom(fit_pulseseq),loadings = rows)
X &lt;- counts_pulseseq[rows,cell_cycle_genes]
Y &lt;- apply(X,2,qt_random_tie)
rownames(Y) &lt;- rownames(X)
score &lt;- rowSums(Y)
pdat &lt;- as.data.frame(prcomp(fit2$L)$x)
pdat &lt;- cbind(pdat,data.frame(score = score))
p15 &lt;- ggplot(pdat,aes(x = PC5,y = PC6,fill = score)) +
  geom_point(shape = 21,color = &quot;white&quot;,size = 1.25) +
  scale_fill_gradientn(colors = c(&quot;darkblue&quot;,&quot;royalblue&quot;,
                       &quot;lightskyblue&quot;,&quot;darkorange&quot;,&quot;firebrick&quot;)) +
  theme_cowplot(font_size = 10)</code></pre>
</div>
<div id="hillock-cells" class="section level2">
<h2>Hillock cells</h2>
<p>Hillock cells in droplet data:</p>
<pre class="r"><code>hillock_genes &lt;- c(&quot;Anxa1&quot;,&quot;Cldn3&quot;,&quot;Ecm1&quot;,&quot;Krt13&quot;,&quot;Krt4&quot;,&quot;Lgals3&quot;,&quot;S100a11&quot;)
p14 &lt;- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,&quot;H&quot;,
                                           hillock_genes,
                                           label_above_quantile = 0.998)
print(p14)</code></pre>
<p>Compare Hillock topic (<span class="math inline">\(k = 4\)</span>) against cluster in droplet data:</p>
<pre class="r"><code>p15 &lt;- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                        &quot;H&quot;,&quot;k4&quot;,hillock_genes,label_above_score = Inf) +
  labs(x = &quot;cluster H&quot;,y = &quot;topic 4&quot;,title = &quot;log-fold change (\u03b2)&quot;)
print(p15)
p15 &lt;- zscores_scatterplot(diff_count_clusters_droplet$Z[,&quot;H&quot;],
                           diff_count_droplet$Z[,4],
                           diff_count_droplet$colmeans,
                           colnames(counts_droplet),
                           label_above_score = 100,
                           zmax = 400) +
  labs(x = &quot;cluster H&quot;,y = &quot;topic 4&quot;,title = &quot;z-scores&quot;)</code></pre>
<p>TO DO: Hillock topic (<span class="math inline">\(k = 1\)</span>) in pulse-seq data:</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.6
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0      ggrepel_0.9.0      ggplot2_3.3.0      fastTopics_0.3-177
# [5] dplyr_0.8.3        Matrix_1.2-18     
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.5           lattice_0.20-38      tidyr_1.0.0         
#  [4] prettyunits_1.1.1    assertthat_0.2.1     zeallot_0.1.0       
#  [7] rprojroot_1.3-2      digest_0.6.23        R6_2.4.1            
# [10] backports_1.1.5      MatrixModels_0.4-1   evaluate_0.14       
# [13] coda_0.19-3          httr_1.4.1           pillar_1.4.3        
# [16] rlang_0.4.5          progress_1.2.2       lazyeval_0.2.2      
# [19] data.table_1.12.8    irlba_2.3.3          SparseM_1.78        
# [22] whisker_0.4          rmarkdown_2.3        labeling_0.3        
# [25] Rtsne_0.15           stringr_1.4.0        htmlwidgets_1.5.1   
# [28] munsell_0.5.0        compiler_3.6.2       httpuv_1.5.2        
# [31] xfun_0.11            pkgconfig_2.0.3      mcmc_0.9-6          
# [34] htmltools_0.4.0      tidyselect_0.2.5     tibble_2.1.3        
# [37] workflowr_1.6.2.9000 quadprog_1.5-8       viridisLite_0.3.0   
# [40] crayon_1.3.4         withr_2.1.2          later_1.0.0         
# [43] MASS_7.3-51.4        grid_3.6.2           jsonlite_1.6        
# [46] gtable_0.3.0         lifecycle_0.1.0      git2r_0.26.1        
# [49] magrittr_1.5         scales_1.1.0         RcppParallel_4.4.2  
# [52] stringi_1.4.3        farver_2.0.1         fs_1.3.1            
# [55] promises_1.1.0       vctrs_0.2.1          tools_3.6.2         
# [58] glue_1.3.1           purrr_0.3.3          hms_0.5.2           
# [61] yaml_2.2.0           colorspace_1.4-1     plotly_4.9.2        
# [64] knitr_1.26           quantreg_5.54        MCMCpack_1.4-5</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
