<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>Perform differential expression analysis on topics and clusters identified in mixture of FACS-purified PBMC data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/single-cell-topics">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Perform differential expression analysis on topics and clusters identified in mixture of FACS-purified PBMC data</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2.9000). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
diff-count-analysis-clusters
</li>
<li>
diff-count-analysis-subpop
</li>
<li>
diff-count-analysis-t
</li>
<li>
diff-count-analysis-topics
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>plots_purified_pbmc_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreeaecd70036aae5c383667631ec6234a278bea92b3targetblankaecd700a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/aecd70036aae5c383667631ec6234a278bea92b3" target="_blank">aecd700</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstreeaecd70036aae5c383667631ec6234a278bea92b3targetblankaecd700a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/aecd70036aae5c383667631ec6234a278bea92b3" target="_blank">aecd700</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/
    Ignored:    data/droplet.RData
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    output/droplet/diff-count-droplet.RData
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/diff-count-pulseseq.RData
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/plots_purified_pbmc_cache/
    Untracked:  plots/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/plots_purified_pbmc.Rmd</code>) and HTML (<code>docs/plots_purified_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/aecd70036aae5c383667631ec6234a278bea92b3/analysis/plots_purified_pbmc.Rmd" target="_blank">aecd700</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-02-09
</td>
<td>
workflowr::wflow_publish(“plots_purified_pbmc.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/1f0eb8cf964185858bb1a1ad82035351424b0550/analysis/plots_purified_pbmc.Rmd" target="_blank">1f0eb8c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-02-08
</td>
<td>
More improvements to volcano plots in plots_purified_pbmc.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/be277389ea8d2bb0c390c95e4be08713ab595dfb/analysis/plots_purified_pbmc.Rmd" target="_blank">be27738</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-02-08
</td>
<td>
Improved volcano plots for FACS cell populations in plots_purified_pbmc.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/ab3eeb82557ac15852443a50d3aad409242690c7/docs/plots_purified_pbmc.html" target="_blank">ab3eeb8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0b7339714de3e3fc87c26012b646aa1af42b06e2/analysis/plots_purified_pbmc.Rmd" target="_blank">0b73397</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
<td>
Added more volcano plots to plots_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/42ebc62099a4d7af6134fbfea745d3d7b6cdfbf3/docs/plots_purified_pbmc.html" target="_blank">42ebc62</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
<td>
Added volcano plots for FACS populations in plots_purified_pbmc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/4edefd08c545d4440ecfc1fce9c3e70dda27784b/analysis/plots_purified_pbmc.Rmd" target="_blank">4edefd0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
<td>
workflowr::wflow_publish(“plots_purified_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/01c11399743c5e2741dd6f6c45cd359b4dc43fdc/analysis/plots_purified_pbmc.Rmd" target="_blank">01c1139</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
<td>
Adding new code for volcano plots in plots_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a4bc59be3a7313cdcf5df4daaab580568ddcfa20/docs/plots_purified_pbmc.html" target="_blank">a4bc59b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
Re-built plots_purified_pbmc page after removing cache.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/b92d4db07c0205516d832edb1a7adcbd820102c4/analysis/plots_purified_pbmc.Rmd" target="_blank">b92d4db</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
workflowr::wflow_publish(“plots_purified_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/a0b6c2b221c0b5c52edef847b0c1d655e8891a16/docs/plots_purified_pbmc.html" target="_blank">a0b6c2b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ed8a595759596561c82916b2e9dbab5fea522b1d/analysis/plots_purified_pbmc.Rmd" target="_blank">ed8a595</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
workflowr::wflow_publish(“plots_purified_pbmc.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/04113408c52e484a058dfa1525a28dfb9f86d646/docs/plots_purified_pbmc.html" target="_blank">0411340</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
Added scatterplots to plots_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e16bf807a27c4d858072cea35cd6104b204063d7/analysis/plots_purified_pbmc.Rmd" target="_blank">e16bf80</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
<td>
workflowr::wflow_publish(“plots_purified_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a12c42a653f1e24b627dbb48b168d397e2260360/analysis/plots_purified_pbmc.Rmd" target="_blank">a12c42a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-05
</td>
<td>
Implemented function lfc_scatterplot in functions_for_plots_purified_pbmc.R.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/9fd0455eaeaa690255d71b8e5e2696c52baf90f8/analysis/plots_purified_pbmc.Rmd" target="_blank">9fd0455</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-05
</td>
<td>
Added steps to save volcano plots in plots_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/fad8e3d9888206e882551f96333364e6203b6599/docs/plots_purified_pbmc.html" target="_blank">fad8e3d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-05
</td>
<td>
First build of plots_purified_pbmc page.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/bf079301ee8ba4d6de8c69e41c9c9ea0eb5fb6a2/analysis/plots_purified_pbmc.Rmd" target="_blank">bf07930</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-05
</td>
<td>
workflowr::wflow_publish(“plots_purified_pbmc.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/40c2d84dc397204d1a0537e906f102d87bde834a/analysis/plots_purified_pbmc.Rmd" target="_blank">40c2d84</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-04
</td>
<td>
Working on log-fold change scatterplots in plots_purified_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e437ddf609cb340d500ccf63e7fc6735270f7b4a/analysis/plots_purified_pbmc.Rmd" target="_blank">e437ddf</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-04
</td>
<td>
Working on volcano plots in plots_purified_pbmc analysis.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here we perform a differential expression analysis using the topic model fit to the mixture of FACS-purified data, as well as the clusters identified from this topic model.</p>
<p>Load the packages used in the analysis below, as well as additional functions that we will use to generate some of the plots.</p>
<pre class="r"><code>library(Matrix)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source(&quot;../code/functions_for_plots_purified_pbmc.R&quot;)</code></pre>
<p>Load the count data, the <span class="math inline">\(K = 6\)</span> topic model fit, and the 7 clusters identified in the <a href="clusters_purified_pbmc.html">clustering analysis</a>.</p>
<pre class="r"><code>load(&quot;../data/pbmc_purified.RData&quot;)
fit &lt;- readRDS(file.path(&quot;../output/pbmc-purified/rds&quot;,
                         &quot;fit-pbmc-purified-scd-ex-k=6.rds&quot;))$fit
fit &lt;- poisson2multinom(fit)
samples &lt;- readRDS(&quot;../output/pbmc-purified/clustering-pbmc-purified.rds&quot;)</code></pre>
<p>Perform differential expression analysis using the FACS labeling:</p>
<pre class="r"><code>celltype &lt;- as.character(samples$celltype)
celltype[celltype == &quot;CD4+/CD45RA+/CD25- Naive T&quot; |
         celltype == &quot;CD4+/CD45RO+ Memory&quot; |
         celltype == &quot;CD8+/CD45RA+ Naive Cytotoxic&quot; |
         celltype == &quot;CD4+ T Helper2&quot; |
         celltype == &quot;CD4+/CD25 T Reg&quot;] &lt;- &quot;T cell&quot;
celltype &lt;- factor(celltype)
table(celltype)
diff_count_facs &lt;- diff_count_clusters(celltype,counts)
# celltype
#   CD14+ Monocyte          CD19+ B            CD34+         CD56+ NK 
#             2612            10085             9232             8385 
# CD8+ Cytotoxic T           T cell 
#            10209            54132 
# Fitting 21952 x 6 = 131712 univariate Poisson models.
# Computing log-fold change statistics.
# Stabilizing log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>These volcano plots summarize the results of the differential expression analysis using the FACS labeling:</p>
<pre class="r"><code>p1 &lt;- volcano_plot(diff_count_facs,&quot;CD19+ B&quot;,genes$symbol,
                   label_above_quantile = 0.9995,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;B cells&quot;)
p2 &lt;- volcano_plot(diff_count_facs,&quot;CD14+ Monocyte&quot;,genes$symbol,
                   label_above_quantile = 0.9995,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;CD14+ cells&quot;)
p3 &lt;- volcano_plot(diff_count_facs,&quot;CD34+&quot;,genes$symbol,
                   label_above_quantile = 0.999,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;CD34+ cells&quot;)
p4 &lt;- volcano_plot(diff_count_facs,&quot;CD56+ NK&quot;,genes$symbol,
                   label_above_quantile = 0.9995,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;NK cells&quot;)
p5 &lt;- volcano_plot(diff_count_facs,&quot;CD8+ Cytotoxic T&quot;,genes$symbol,
                   label_above_quantile = 0.9995,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;CD8+ T cells&quot;)
p6 &lt;- volcano_plot(diff_count_facs,&quot;T cell&quot;,genes$symbol,
                   label_above_quantile = 0.999,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;T cells&quot;)
plot_grid(p1,p2,p3,p4,p5,p6,nrow = 2,ncol = 3)</code></pre>
<p><img src="figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-1-1">
Past versions of volcano-plots-1-1.png
</button>
</p>
<div id="fig-volcano-plots-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ab3eeb82557ac15852443a50d3aad409242690c7/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" target="_blank">ab3eeb8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/42ebc62099a4d7af6134fbfea745d3d7b6cdfbf3/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" target="_blank">42ebc62</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a4bc59be3a7313cdcf5df4daaab580568ddcfa20/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" target="_blank">a4bc59b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a0b6c2b221c0b5c52edef847b0c1d655e8891a16/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" target="_blank">a0b6c2b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/04113408c52e484a058dfa1525a28dfb9f86d646/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" target="_blank">0411340</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fad8e3d9888206e882551f96333364e6203b6599/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-1-1.png" target="_blank">fad8e3d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Perform differential expression analysis using the clusters:</p>
<pre class="r"><code>table(samples$cluster)
diff_count_clusters &lt;- diff_count_clusters(samples$cluster,counts)
# 
#         B     CD14+     CD34+      CD8+ dendritic        NK         T 
#     10439      2956      8237      3757       308      8380     60578 
# Fitting 21952 x 7 = 153664 univariate Poisson models.
# Computing log-fold change statistics.
# Stabilizing log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>These volcano plots summarize the results of the differential expression analysis using the clusters:</p>
<pre class="r"><code>p7 &lt;- volcano_plot(diff_count_clusters,&quot;B&quot;,genes$symbol,
                   label_above_quantile = 0.999,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;B cells&quot;)
# 10134 out of 18417 data points will be included in plot
p8 &lt;- volcano_plot(diff_count_clusters,&quot;CD14+&quot;,genes$symbol,
                   label_above_quantile = 0.999,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;CD14+ cells&quot;)
# 10130 out of 18417 data points will be included in plot
p9 &lt;- volcano_plot(diff_count_clusters,&quot;CD34+&quot;,genes$symbol,
                   label_above_quantile = 0.999,
                   subsample_below_quantile = 0.5,
                   filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;CD34+ cells&quot;)
# 10130 out of 18417 data points will be included in plot
p10 &lt;- volcano_plot(diff_count_clusters,&quot;dendritic&quot;,genes$symbol,
                    label_above_quantile = 0.999,
                    subsample_below_quantile = 0.5,
                    filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;dendritic cells&quot;)
# 10130 out of 18417 data points will be included in plot
p11 &lt;- volcano_plot(diff_count_clusters,&quot;NK&quot;,genes$symbol,
                    label_above_quantile = 0.999,
                    subsample_below_quantile = 0.5,
                    filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;NK cells&quot;)
# 10130 out of 18417 data points will be included in plot
p12 &lt;- volcano_plot(diff_count_clusters,&quot;CD8+&quot;,genes$symbol,
                    label_above_quantile = 0.999,
                    subsample_below_quantile = 0.5,
                    filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;CD8+ T cells&quot;)
# 10130 out of 18417 data points will be included in plot
p13 &lt;- volcano_plot(diff_count_clusters,&quot;T&quot;,genes$symbol,
                    label_above_quantile = 0.999,
                    subsample_below_quantile = 0.5,
                    filter_low_counts = 5e-5) +
  guides(fill = &quot;none&quot;) +
  ggtitle(&quot;T cells&quot;)
# 10131 out of 18417 data points will be included in plot
plot_grid(p7,p8,p9,p10,p11,p12,p13,nrow = 3,ncol = 3)</code></pre>
<p><img src="figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-volcano-plots-2-1">
Past versions of volcano-plots-2-1.png
</button>
</p>
<div id="fig-volcano-plots-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/ab3eeb82557ac15852443a50d3aad409242690c7/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" target="_blank">ab3eeb8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/42ebc62099a4d7af6134fbfea745d3d7b6cdfbf3/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" target="_blank">42ebc62</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a4bc59be3a7313cdcf5df4daaab580568ddcfa20/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" target="_blank">a4bc59b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a0b6c2b221c0b5c52edef847b0c1d655e8891a16/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" target="_blank">a0b6c2b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/04113408c52e484a058dfa1525a28dfb9f86d646/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" target="_blank">0411340</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/fad8e3d9888206e882551f96333364e6203b6599/docs/figure/plots_purified_pbmc.Rmd/volcano-plots-2-1.png" target="_blank">fad8e3d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2021-01-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Perform differential expression analysis using the multinomial topic model, after removing the dendritic cells cluster, and after merging the two T cell topics:</p>
<pre class="r"><code>rows &lt;- which(samples$cluster != &quot;dendritic&quot;)
fit_merged &lt;- select_loadings(fit,loadings = rows)
fit_merged &lt;- merge_topics(fit,c(1,6))
diff_count_topics &lt;- diff_count_analysis(fit_merged,counts)
# Fitting 21952 x 5 = 109760 univariate Poisson models.
# Computing log-fold change statistics.
# Stabilizing log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Perform differential expression analysis on the T cells only:</p>
<pre class="r"><code>rows &lt;- which(samples$cluster == &quot;T&quot;)
fit_t &lt;- select_loadings(fit,loadings = rows)
diff_count_t &lt;- diff_count_analysis(fit_t,counts[rows,])
# Fitting 21952 x 6 = 131712 univariate Poisson models.
# Computing log-fold change statistics.
# Stabilizing log-fold change estimates using adaptive shrinkage.</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>These volcano plots summarize the results of the differential expression analysis using the topic model:</p>
<pre class="r"><code>volcano_plot(diff_count_topics,&quot;k3&quot;,genes$symbol,
             label_above_quantile = 0.998) +
  ggtitle(&quot;topic 3 (B cells)&quot;)</code></pre>
<p><img src="figure/plots_purified_pbmc.Rmd/volcano-plots-3-1.png" width="840" style="display: block; margin: auto;" /></p>
<p>Here are more volcano plots:</p>
<pre class="r"><code>p2 &lt;- volcano_plot_better(diff_count_topics,&quot;k3&quot;,genes$symbol,0.998,
                          &quot;topic 3 (B cells)&quot;)
p3 &lt;- volcano_plot_better(diff_count_topics,&quot;k4&quot;,genes$symbol,0.998,
                          &quot;topic 4 (NK cells)&quot;)
p4 &lt;- volcano_plot_better(diff_count_topics,&quot;k2&quot;,genes$symbol,0.998,
                          &quot;topic 2 (CD14+)&quot;)
p5 &lt;- volcano_plot_better(diff_count_topics,&quot;k5&quot;,genes$symbol,0.998,
                          &quot;topic 5 (CD34+)&quot;)
p6 &lt;- volcano_plot_better(diff_count_topics,&quot;k1+k6&quot;,genes$symbol,0.998,
                          &quot;topics 1 + 6 (T cells)&quot;)
p7 &lt;- volcano_plot_better(diff_count_t,&quot;k1&quot;,genes$symbol,0.997,
                          &quot;topic 1 (CD4+/CD8+)&quot;)
plot_grid(p2,p3,p4,p5,p6,p7,p8,p9,nrow = 3,ncol = 3)</code></pre>
<p>The results of the differential expression analyses can also be browsed in interactive volcano plots:</p>
<pre class="r"><code>volcano_plotly(diff_count_topics,&quot;k3&quot;,&quot;volcano_plot_purified_pbmc_bcells.html&quot;,
               genes$symbol,title = &quot;topic 3 (B cells)&quot;)
volcano_plotly(diff_count_topics,&quot;k4&quot;,&quot;volcano_plot_purified_pbmc_nk.html&quot;,
               genes$symbol,title = &quot;topic 4 (NK cells)&quot;)
volcano_plotly(diff_count_topics,&quot;k2&quot;,&quot;volcano_plot_purified_pbmc_cd14.html&quot;,
               genes$symbol,title = &quot;topic 2 (CD14+)&quot;)
volcano_plotly(diff_count_topics,&quot;k5&quot;,&quot;volcano_plot_purified_pbmc_cd34.html&quot;,
               genes$symbol,title = &quot;topic 5 (CD34+)&quot;)
volcano_plotly(diff_count_topics,&quot;k1+k6&quot;,
               &quot;volcano_plot_purified_pbmc_tcells.html&quot;,genes$symbol,
               title = &quot;topics 1 + 6 (T cells)&quot;)
volcano_plotly(diff_count_t,&quot;k1&quot;,&quot;volcano_plot_purified_pbmc_cd4cd8.html&quot;,
               genes$symbol,title = &quot;topic 1 (CD4+/CD8+)&quot;)
volcano_plotly(diff_count_clusters,&quot;CD8+&quot;,
               &quot;volcano_plot_purified_pbmc_cd8.html&quot;,genes$symbol,
               title = &quot;CD8+ cluster&quot;)
volcano_plotly(diff_count_clusters,&quot;dendritic&quot;,
               &quot;volcano_plot_purified_pbmc_dendritic.html&quot;,genes$symbol,
               title = &quot;dendritic cells cluster&quot;)</code></pre>
<p>The interactive volcano plots can also be viewed by clicking on these links:</p>
<ul>
<li><p><a href="volcano_plot_purified_pbmc_bcells.html">topic 3 (B cells)</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_nk.html">topic 4 (NK cells)</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_cd14.html">topic 2 (CD14+)</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_cd34.html">topic 5 (CD34+)</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_tcells.html">topics 1 and 6 (T cells)</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_cd4cd8.html">topic 1 (CD4+/CD8+ lineage differentiation)</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_cd8.html">CD8+ cluster</a></p></li>
<li><p><a href="volcano_plot_purified_pbmc_dendritic.html">dendritic cells cluster</a></p></li>
</ul>
<pre class="r"><code>p10 &lt;- lfc_scatterplot(diff_count_facs,diff_count_topics,&quot;CD19+ B&quot;,&quot;k3&quot;,
                       genes$symbol,label_above_quantile = 0.998,
                       xlab = &quot;B cells FACS subpopulation&quot;,ylab = &quot;topic 3&quot;)
p11 &lt;- lfc_scatterplot(diff_count_facs,diff_count_topics,&quot;CD56+ NK&quot;,&quot;k4&quot;,
                       genes$symbol,label_above_quantile = 0.998,
                       xlab = &quot;NK cells FACS subpopulation&quot;,ylab = &quot;topic 4&quot;)
p12 &lt;- lfc_scatterplot(diff_count_facs,diff_count_topics,&quot;CD14+ Monocyte&quot;,&quot;k2&quot;,
                       genes$symbol,label_above_quantile = 0.998,
                       xlab = &quot;CD14+ FACS subpopulation&quot;,ylab = &quot;topic 2&quot;)
p13 &lt;- lfc_scatterplot(diff_count_facs,diff_count_topics,&quot;CD34+&quot;,&quot;k5&quot;,
                       genes$symbol,label_above_quantile = 0.998,
                       xlab = &quot;CD34+ FACS subpopulation&quot;,ylab = &quot;topic 4&quot;)
p14 &lt;- lfc_scatterplot(diff_count_facs,diff_count_topics,&quot;T cell&quot;,&quot;k1+k6&quot;,
                       genes$symbol,label_above_quantile = 0.998,
                       xlab = &quot;T cells FACS subpopulation&quot;,
                       ylab = &quot;topics 1 + 6&quot;)
plot_grid(p10,p11,p12,p13,p14,nrow = 3,ncol = 2)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.2 (2019-12-12)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.7
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] cowplot_1.0.0     ggrepel_0.9.0     ggplot2_3.3.0     fastTopics_0.4-33
# [5] Matrix_1.2-18    
# 
# loaded via a namespace (and not attached):
#  [1] httr_1.4.2           tidyr_1.0.0          jsonlite_1.6        
#  [4] viridisLite_0.3.0    RcppParallel_4.4.2   assertthat_0.2.1    
#  [7] mixsqp_0.3-44        yaml_2.2.0           progress_1.2.2      
# [10] pillar_1.4.3         backports_1.1.5      lattice_0.20-38     
# [13] quantreg_5.54        glue_1.3.1           quadprog_1.5-8      
# [16] digest_0.6.23        promises_1.1.0       colorspace_1.4-1    
# [19] htmltools_0.4.0      httpuv_1.5.2         pkgconfig_2.0.3     
# [22] invgamma_1.1         SparseM_1.78         purrr_0.3.3         
# [25] scales_1.1.0         whisker_0.4          later_1.0.0         
# [28] Rtsne_0.15           MatrixModels_0.4-1   git2r_0.26.1        
# [31] tibble_2.1.3         farver_2.0.1         withr_2.1.2         
# [34] ashr_2.2-51          lazyeval_0.2.2       magrittr_1.5        
# [37] crayon_1.3.4         mcmc_0.9-6           evaluate_0.14       
# [40] fs_1.3.1             MASS_7.3-51.4        truncnorm_1.0-8     
# [43] tools_3.6.2          data.table_1.12.8    prettyunits_1.1.1   
# [46] hms_0.5.2            lifecycle_0.1.0      stringr_1.4.0       
# [49] MCMCpack_1.4-5       plotly_4.9.2         munsell_0.5.0       
# [52] irlba_2.3.3          compiler_3.6.2       rlang_0.4.5         
# [55] grid_3.6.2           htmlwidgets_1.5.1    labeling_0.3        
# [58] rmarkdown_2.3        gtable_0.3.0         codetools_0.2-16    
# [61] R6_2.4.1             knitr_1.26           dplyr_0.8.3         
# [64] zeallot_0.1.0        workflowr_1.6.2.9000 rprojroot_1.3-2     
# [67] stringi_1.4.3        SQUAREM_2017.10-1    Rcpp_1.0.5          
# [70] vctrs_0.2.1          tidyselect_0.2.5     xfun_0.11           
# [73] coda_0.19-3</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
