<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Peter Carbonetto" />


<title>Prepare 68k PBMC data for topic modeling analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">single-cell topics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Prepare 68k PBMC data for topic modeling analysis</h1>
<h4 class="author"><em>Peter Carbonetto</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>single-cell-topics/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree16dcd2ae2149d3c244c1cd2f5b9b6627bd47a58ctargetblank16dcd2aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/single-cell-topics/tree/16dcd2ae2149d3c244c1cd2f5b9b6627bd47a58c" target="_blank">16dcd2a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabsinglecelltopicstree16dcd2ae2149d3c244c1cd2f5b9b6627bd47a58ctargetblank16dcd2aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/single-cell-topics/tree/16dcd2ae2149d3c244c1cd2f5b9b6627bd47a58c" target="_blank">16dcd2a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/GSE103354_PulseSeq_UMI_counts.rds
    Ignored:    data/GSE103354_Trachea_droplet_UMIcounts.txt.gz
    Ignored:    data/b_cells/
    Ignored:    data/cd14_monocytes/
    Ignored:    data/cd34_filtered/
    Ignored:    data/cd4_t_helper/
    Ignored:    data/cd56_nk/
    Ignored:    data/cytotoxic_t/
    Ignored:    data/droplet.RData
    Ignored:    data/memory_t/
    Ignored:    data/naive_cytotoxic/
    Ignored:    data/naive_t/
    Ignored:    data/pbmc_68k.RData
    Ignored:    data/pbmc_68k/
    Ignored:    data/pbmc_purified.RData
    Ignored:    data/pulseseq.RData
    Ignored:    data/regulatory_t/
    Ignored:    output/droplet/fits-droplet.RData
    Ignored:    output/droplet/rds/
    Ignored:    output/pbmc-68k/fits-pbmc-68k.RData
    Ignored:    output/pbmc-68k/rds/
    Ignored:    output/pbmc-purified/fits-pbmc-purified.RData
    Ignored:    output/pbmc-purified/rds/
    Ignored:    output/pulseseq/fits-pulseseq.RData
    Ignored:    output/pulseseq/rds/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/prepare_68k_pbmc.Rmd</code>) and HTML (<code>docs/prepare_68k_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/16dcd2ae2149d3c244c1cd2f5b9b6627bd47a58c/analysis/prepare_68k_pbmc.Rmd" target="_blank">16dcd2a</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-17
</td>
<td>
Merge branch ‘master’ of github.com:stephenslab/single-cell-topics.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/a3972f19b898355af3871c7959d790f35ccd1996/analysis/prepare_68k_pbmc.Rmd" target="_blank">a3972f1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-17
</td>
<td>
Improved code in prepare_68k_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/0b0b5a5f7e6f4617a28d6c9e99f06bb1d992fc5d/analysis/prepare_68k_pbmc.Rmd" target="_blank">0b0b5a5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-08-11
</td>
<td>
Working on plots to visualize droplet and 68k pbmc topics.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/dab7448538f16ce2a1e74b4875ef0a1bedd528eb/docs/prepare_68k_pbmc.html" target="_blank">dab7448</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
Made a couple minor revisions to prepare_68k_pbmc.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/4a79355af4e4a57eb4010e87cc0d8041bbc52ca8/analysis/prepare_68k_pbmc.Rmd" target="_blank">4a79355</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
wflow_publish(“prepare_68k_pbmc.Rmd”, view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/single-cell-topics/9622db8e84a502905e6bebdc97a83a1a6b6b2f88/docs/prepare_68k_pbmc.html" target="_blank">9622db8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
Built first draft of prepare_68k_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/e684eab5307b574219b366c55e00f4e75d3b7285/analysis/prepare_68k_pbmc.Rmd" target="_blank">e684eab</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
More improvements to prepare_68k_pbmc analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/single-cell-topics/blob/29e6a6d4ab573eb061754b4077381276500acba4/analysis/prepare_68k_pbmc.Rmd" target="_blank">29e6a6d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2020-07-28
</td>
<td>
Working on R Markdown for preparation of 68k pbmc data set.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>These are the steps taken to prepare the unsorted 68k PBMC data from <a href="https://doi.org/10.1038/ncomms14049">Zheng <em>et al</em> (2017)</a> for topic modeling analysis.</p>
<p>Load the packages used in the analysis below, and some functions written specifically for loading and preparing these data.</p>
<pre class="r"><code>library(tools)
library(Matrix)
library(readr)
source(&quot;../code/pbmc.R&quot;)</code></pre>
<p>Before loading the data, download the “Gene/cell matrix (filtered)” tar.gz file for the “Fresh 68k PBMCs (Donor A) data set from <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets">10x Genomics website</a>, download <code>68k_pbmc_barcodes_annotation.tsv</code> from the <a href="https://github.com/10XGenomics/single-cell-3prime-paper">companion repository on GitHub</a>, and then save and compress these files to the <code>data/pbmc_68k</code> directory.</p>
<pre class="r"><code>samples &lt;- read_barcodes_with_annotations(file.path(&quot;../data/pbmc_68k&quot;,
             &quot;68k_pbmc_barcodes_annotation.tsv.gz&quot;))
# Parsed with column specification:
# cols(
#   TSNE.1 = col_double(),
#   TSNE.2 = col_double(),
#   barcodes = col_character(),
#   celltype = col_character()
# )
genes   &lt;- read_genes(&quot;../data/pbmc_68k/genes.tsv.gz&quot;)
counts  &lt;- create_counts_matrix(&quot;../data/pbmc_68k/matrix.mtx.gz&quot;,
                                sample.names = samples$barcode,
                                gene.names = genes$ensembl)</code></pre>
<p>Remove genes that are not expressed in any of the cells.</p>
<pre class="r"><code>j      &lt;- which(colSums(counts &gt; 0) &gt;= 1)
genes  &lt;- genes[j,]
counts &lt;- counts[,j]</code></pre>
<p>After filtering out genes with no expression, the count data are stored in a 68,579 x 20,387 sparse matrix, with rows corresponding to samples, and columns corresponding to genes. This matrix is very sparse—less than 3% of the counts are greater than zero:</p>
<pre class="r"><code>cat(sprintf(&quot;Number of samples: %d\n&quot;,nrow(counts)))
cat(sprintf(&quot;Number of genes: %d\n&quot;,ncol(counts)))
cat(sprintf(&quot;Proportion of counts that are non-zero: %0.1f%%.\n&quot;,
            100*mean(counts &gt; 0)))
# Number of samples: 68579
# Number of genes: 20387
# Proportion of counts that are non-zero: 2.7%.</code></pre>
<p>Save the processed data.</p>
<pre class="r"><code>save(list = c(&quot;samples&quot;,&quot;genes&quot;,&quot;counts&quot;),
     file = &quot;pbmc_68k.RData&quot;)
# resaveRdaFiles(&quot;pbmc_68k.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.5.1 (2018-07-02)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] tools     stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
# [1] readr_1.3.1   Matrix_1.2-15
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.4.6    knitr_1.20      whisker_0.3-2   magrittr_1.5   
#  [5] workflowr_1.6.2 hms_0.4.2       lattice_0.20-38 R6_2.3.0       
#  [9] rlang_0.4.0     stringr_1.3.1   grid_3.5.1      git2r_0.26.1   
# [13] htmltools_0.3.6 yaml_2.2.0      rprojroot_1.3-2 digest_0.6.18  
# [17] tibble_2.1.1    crayon_1.3.4    later_0.7.5     promises_1.0.1 
# [21] fs_1.3.1        glue_1.3.0      evaluate_0.12   rmarkdown_1.10 
# [25] stringi_1.2.4   pillar_1.3.1    compiler_3.5.1  backports_1.1.2
# [29] httpuv_1.4.5    pkgconfig_2.0.2</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
