---
title: "Identify clusters in droplet data using topic model"
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we perform PCA on the topic proportions to identify clusters in
the droplet data. 

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, as well as additional
functions that we will use to generate some of the plots.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(dplyr)
library(fastTopics)
library(ggplot2)
library(cowplot)
source("../code/plots.R")
```

Load data and results
---------------------

Load the droplet data.

```{r load-data}
load("../data/droplet.RData")
```

Load the $k = 7$ Poisson NMF model fit.

```{r load-fit}
fit <- readRDS("../output/droplet/rds/fit-droplet-scd-ex-k=7.rds")$fit
```

Identify clusters from principal components
-------------------------------------------

To identify clusters, we begin by plotting PCs computed from the topic
proportions. (Note that only 6 PCs are needed for 7 topics.)

```{r clustering-1, fig.width=8, fig.height=2.5}
p1 <- pca_plot(poisson2multinom(fit),pcs = 1:2,fill = "none")
p2 <- pca_plot(poisson2multinom(fit),pcs = 3:4,fill = "none")
p3 <- pca_plot(poisson2multinom(fit),pcs = 5:6,fill = "none")
plot_grid(p1,p2,p3,nrow = 1,ncol = 3)
```

Some of the structure is more evident from "hexbin" plots showing the
density of the points.

```{r clustering-2, fig.width=8, fig.height=2.5}
breaks <- c(0,1,5,10,100,Inf)
p4 <- pca_hexbin_plot(poisson2multinom(fit),pcs = 1:2,breaks = breaks)
p5 <- pca_hexbin_plot(poisson2multinom(fit),pcs = 3:4,breaks = breaks)
p6 <- pca_hexbin_plot(poisson2multinom(fit),pcs = 5:6,breaks = breaks)
p4 <- p4 + guides(fill = "none")
p5 <- p5 + guides(fill = "none")
p6 <- p6 + guides(fill = "none")
plot_grid(p4,p5,p6,nrow = 1,ncol = 3)
```

From these PCA plots, we define 4 clusters, labeled A, Cil, G and
T+N. (The reasoning behind these labels will become clear later.)
Points that do not fit in any of these clusters are assigned to a
"background cluster", labeled U.

```{r clustering-3}
pca <- prcomp(poisson2multinom(fit)$L)$x
x   <- rep("U",nrow(pca))
pc1 <- pca[,1]
pc2 <- pca[,2]
pc6 <- pca[,6]
x[pc2 > -0.15] <- "A"
x[pc1 > 0.3 & pc2 < -0.75] <- "Cil"
x[pc1 <= 0.3 & pc2 >= -0.75 & pc2 < -0.4] <- "T+N"
x[pc6 < -0.05] <- "G"
```

There is additional substructure in cluster A, which is more apparent
in the projection onto the top 2 PCs computed from cluster A only.

```{r clustering-4, fig.width=6, fig.height=3}
rows <- which(x == "A")
fit2 <- select(poisson2multinom(fit),loadings = rows)
p7   <- pca_plot(fit2,fill = "none")
p8   <- pca_hexbin_plot(fit2,breaks = breaks) + guides(fill = "none")
plot_grid(p7,p8)
```

The variation in PCs 1 and 2 is mostly produced by topics 2, 4 and 5.

```{r clustering-5, fig.width=5.5, fig.height=4.5}
p9 <- pca_plot(fit2,k = c(2,4,5))
print(p9)
```

Topic 4 in particular corresponds closely to expression of *Krt13*
which was identified as being uniquely expressed by transitional
"hillock" cells.

```{r clustering-6, fig.width=3.25, fig.height=2.5}
p10 <- pca_plot(fit2,fill = log10(counts[rows,"Krt13"])) +
       labs(fill = "log10(count)",title = "Krt13")
print(p10)
```

We label the three more-or-less distinct subclusters as B, C and H,
and assign the remaining "in between" data points to a new "background
cluster", B+C.

```{r clustering-7}
pca <- prcomp(fit2$L)$x
pc1 <- pca[,1]
pc2 <- pca[,2]
y   <- rep("B+C",nrow(pca))
y[pc1 < 0.1] <- "B"
y[pc1 > 0.4 & pc2 < 0.45] <- "C"
y[pc2 > 0.55] <- "H"
x[rows] <- y
```

In summary, we have subdivided the droplet data into 8 subsets. The
substructure is more clear when we plot PCs separately for the
abundant and rare cell-types.

```{r clustering-8, fig.width=6, fig.height=2.5, message=FALSE}
samples$cluster <- factor(x,c("B","C","B+C","H","Cil","T+N","G","U"))
abundant        <- c("B","C","B+C","H")
rare            <- c("Cil","T+N","G","U")
cluster_colors <- c("royalblue",   # B
                    "forestgreen", # C
                    "slategray",   # B+C
                    "turquoise",   # H
					"firebrick",   # Cil
					"darkorange",  # T+N
					"gold",        # G
					"gainsboro")   # U
rows1 <- which(is.element(samples$cluster,abundant))
rows2 <- which(is.element(samples$cluster,rare))
fit1  <- select(poisson2multinom(fit),loadings = rows1)
fit2  <- select(poisson2multinom(fit),loadings = rows2)
p11a <- pca_plot(fit1,fill = samples[rows1,"cluster"]) +
        scale_fill_manual(values = cluster_colors,drop = FALSE) +
	    labs(fill = "cluster")
p11b <- pca_plot(fit2,fill = samples[rows2,"cluster"]) +
        scale_fill_manual(values = cluster_colors,drop = FALSE) +
	    labs(fill = "cluster")
plot_grid(p11a,p11b,nrow = 1,ncol = 2)
```

The clusters identified here correspond well to the Montoro *et al*
(2018) clustering, with some exceptions (e.g., we do not identify an
ionocytes cluster, and the neuroendocrine and tuft cells are included
in the same cluster).

```{r clustering-9}
with(samples,table(tissue,cluster))
```

This close correspondence is also clear from the PCA plots:

```{r clustering-10, fig.width=7.5, fig.height=2.5, message=FALSE}
tissue_colors <- c("royalblue",   # basal
				   "firebrick",   # ciliated
                   "forestgreen", # club
     			   "gold",        # goblet
				   "darkmagenta", # ionocyte
				   "darkorange",  # neuroendocrine
                   "skyblue")     # tuft
p12a <- pca_plot(fit1,fill = samples[rows1,"tissue"]) +
        scale_fill_manual(values = tissue_colors,drop = FALSE) +
	    labs(fill = "cluster")
p12b <- pca_plot(fit2,fill = samples[rows2,"tissue"]) +
        scale_fill_manual(values = tissue_colors,drop = FALSE) +
	    labs(fill = "cluster")
plot_grid(p12a,p12b,nrow = 1,ncol = 2)
```

By computing inter-cluster and inter-topic *total variation distances*
in relative expression levels, we see that the clusters identified
above show greater differentiation in gene expression, and the topics
show more differentiation than the clusters.

```{r clustering-11}
fit_montoro <- init_poisson_nmf_from_clustering(counts,samples$tissue)
fit_cluster <- init_poisson_nmf_from_clustering(counts,samples$cluster)
fit_merge   <- merge_topics(poisson2multinom(fit),c("k5","k7"))
d_montoro   <- totalvardist(poisson2multinom(fit_montoro)$F)
d_cluster   <- totalvardist(poisson2multinom(fit_cluster)$F[,-c(3,8)])
d_topics    <- totalvardist(fit_merge$F[,-3])
cat("Montoro et al (2018) clustering:\n")
print(d_montoro,digits = 3)
cat("Our clustering:\n")
print(d_cluster,digits = 3)
cat("Topics:\n")
print(d_topics,digits = 3)
```

Here is a plot summarizing these differences:

```{r clustering-12, fig.height=2, fig.width=2.5}
pdat <-
  rbind(data.frame(method="montoro.et.al",d=d_montoro[upper.tri(d_montoro)]),
        data.frame(method="clusters",     d=d_cluster[upper.tri(d_cluster)]),
        data.frame(method="topics",       d=d_topics[upper.tri(d_topics)]))
p13 <- ggplot(pdat,aes(x = method,y = d)) +
  geom_boxplot(width = 0.25) +
  labs(x = "",y = "total variation dist") +
  theme_cowplot(font_size = 9)
print(p13)
```

This next plot provides a more direct comparison of the total
variation distances among the 5 clusters that are comparable to the
Montoro *et al* clusters:

```{r clustering-13, fig.height=2.25, fig.width=2.5}
fit_montoro <- merge_topics(poisson2multinom(fit_montoro),
                            c("Tuft","Neuroendocrine"))
d_montoro <- totalvardist(fit_montoro$F[,c("Basal","Ciliated","Club","Goblet",
                                           "Tuft+Neuroendocrine")])
d_cluster <-
  totalvardist(poisson2multinom(fit_cluster)$F[,c("B","Cil","C","G","T+N")])
pdat <- data.frame(montoro  = d_montoro[upper.tri(d_montoro)],
                   clusters = d_cluster[upper.tri(d_cluster)])
p14 <- ggplot(pdat,aes(x = montoro,y = clusters)) +
  geom_point(shape = 21,size = 2,color = "white",fill = "dodgerblue") +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  xlim(0.2,0.7) + 
  ylim(0.2,0.7) + 
  labs(x = "Montoro et al clusters",y = "our clusters") +
  theme_cowplot(font_size = 9)
print(p14)
```

Structure plot
--------------

The structure plot summarizes the topic proportions in each of these 8
subsets:

```{r structure-plot, fig.width=8, fig.height=1.5}
set.seed(1)
topic_colors <- c("gold","royalblue","salmon","turquoise","olivedrab",
                  "firebrick","forestgreen")
topics <- c(3,4,5,1,7,2,6)
rows <- sort(c(sample(which(samples$cluster == "B"),800),
               sample(which(samples$cluster == "C"),800),
			   which(samples$cluster == "B+C"),
               which(samples$cluster == "H"),
               which(samples$cluster == "Cil"),
               which(samples$cluster == "T+N"),
               which(samples$cluster == "G"),
			   which(samples$cluster == "U")))
p15 <- structure_plot(select(poisson2multinom(fit),loadings = rows),
                      grouping = samples[rows,"cluster"],
                      topics = topics,colors = topic_colors[topics],
		  			  perplexity = c(70,70,30,30,50,30,12,18),
                      n = Inf,gap = 30,num_threads = 4,verbose = FALSE)
print(p15)
```

Based on this structure plot, and the above results, we roughly
subdivide the droplet data into two: (1) the Cil, T+N and G clusters
that give rise to well-separated clusters, and (2) the B, C, B+C and H
subsets that contain interesting substructure but much less distinct
clustering. Therefore,the cluster labels B, C, B+C and H are useful as
a guide but should be taken with a grain of salt as the boundaries
between these clusters are somewhat arbitrary.

Note the distribution of the topics in cluster C suggests that there
is further substantial heterogeneity in these cells beyond what can be
captured by identifying "hard" clusters. In particular, there is
additional continuous variation in gene expression primarily captured
by topics 5 and 7:

```{r pca-plot-club, fig.width=6, fig.height=4.25}
p16 <- pca_plot(fit1,pcs = 2:3,k = c(4,5,7))
print(p16)
```

Although subtle, there is variation in topics 2 and 6 within the T+N
cluster that tracks closely with the tuft (here signaled by gene
*Gnat3*) and pulmonary neuroendocrine (*Chga*) cell-types:

```{r pca-plot-tuft-neuroendocrine, fig.width=6, fig.height=4.5}
p17 <- pca_plot(poisson2multinom(fit),k = 2)
p18 <- pca_plot(poisson2multinom(fit),k = 6)
p19 <- pca_plot(poisson2multinom(fit),fill = log10(counts[,"Chga"])) +
       labs(fill = "log10(count)",title = "Chga")
p20 <- pca_plot(poisson2multinom(fit),fill = log10(counts[,"Gnat3"])) +
       labs(fill = "log10(count)",title = "Gnat3")
plot_grid(p17,p18,p19,p20)
```

Analysis of single-cell likelihoods
-----------------------------------

*Add text.*

```{r likelihood-1}
fit_merge      <- merge_topics(poisson2multinom(fit),c("k5","k7"))
fit_cluster    <- init_poisson_nmf_from_clustering(counts,samples$cluster)
fit_montoro    <- init_poisson_nmf_from_clustering(counts,samples$tissue)
fit_cluster    <- poisson2multinom(fit_cluster)
fit_montoro    <- poisson2multinom(fit_montoro)
loglik_topics  <- loglik_multinom_topic_model(counts,fit_merge)
loglik_cluster <- loglik_multinom_topic_model(counts,fit_cluster)
loglik_montoro <- loglik_multinom_topic_model(counts,fit_montoro)
```

*Add text here.*

```{r likelihood-2, fig.width=7.5, fig.height=4, message=FALSE}
minloglik <- -20000
p1 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"Basal",
                         "royalblue",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
p2 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"Club",
                         "forestgreen",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
p3 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,
                         "Ciliated","firebrick",minloglik,
                         "Montoro et al (2018) cluster","our clusters")
p4 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,
						 "Neuroendocrine","darkorange",minloglik,
                         "Montoro et al (2018) cluster","our clusters")
p5 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"Tuft",
                         "dodgerblue",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
p6 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"Goblet",
						 "gold",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
plot_grid(p1,p2,p3,p4,p5,p6,nrow = 2,ncol = 3)
```

*Add text here.*

```{r likelihood-3}
```

*Add text here.*

```{r likelihood-4}
```

Save results
------------

Save the clustering of the droplet data to an RDS file.

```{r save-results}
saveRDS(samples,"clustering-droplet.rds")
```
