---
title: "Identify clusters in pulse-seq data using topic model"
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we perform PCA on the topic proportions to identify clusters in
the pulse-seq data. 

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, as well as additional
functions that we will use to generate some of the plots.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(fastTopics)
library(ggplot2)
library(cowplot)
source("../code/plots.R")
```

Load the count data.

```{r load-data}
load("../data/pulseseq.RData")
x <- as.character(samples$tissue)
x[x == "club (hillock-associated)"] <- "club"
x[x == "goblet.1" | x == "goblet.2" | x == "goblet.progenitor"] <- "goblet"
x[x == "tuft.1" | x == "tuft.2" | x == "tuft.progenitor"] <- "tuft"
samples$tissue <- factor(x)
```

Load the $K = 11$ Poisson NMF model fit.

```{r load-fit}
fit <- readRDS("../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds")$fit
fit <- poisson2multinom(fit)
```

From the PCs of the topic proportions PCA plots, we define 4 clusters,
labeled A, D, Cil and T+N. (The reasoning behind these cluster labels
will become clear later.) Points that do not fit in any of these
clusters are assigned to an "unknown" cluster, labeled U for
"unknown".

```{r clustering-1}
pca <- prcomp(fit$L)$x
x   <- rep("U",nrow(pca))
pc3 <- pca[,3]
pc4 <- pca[,4]
pc5 <- pca[,5]
pc6 <- pca[,6]
x[pc4 < 5.5*pc3 + 0.5] <- "A"
x[(pc3 + 0.7)^2 + (pc4 - 0.1)^2 < 0.18^2] <- "Cil"
x[x == "A" & pc6 > 1.3*pc5 + 0.87] <- "T+N"
x[x == "A" & pc5 > -0.4 & pc6 < 1.3*pc5 + 0.49] <- "D"
```

We can refine the small cluster A somewhat using the PCs computed from
cluster A only. We label this refined cluster as "I", and the rest are
added back to the background cluster (U).

```{r clustering-2}
rows <- which(x == "A")
fit2 <- select_loadings(fit,loadings = rows)
pca  <- prcomp(fit2$L)$x
y    <- rep("U",nrow(pca))
pc1  <- pca[,1]
y[pc1 > -0.1] <- "I"
x[rows] <- y
```

```{r clustering-6, fig.width=6, fig.height=2.75}
```

We subdivide the data points in cluster D, somewhat arbitrarily, into
two subsets, which we label as B and C.

```{r clustering-3}
rows <- which(x == "D")
fit2 <- select_loadings(fit,loadings = rows)
pca  <- prcomp(fit2$L)$x
y    <- rep("C",nrow(pca))
pc1  <- pca[,1]
pc2  <- pca[,2]
y[pc2 > -pc1 - 0.15] <- "B"
x[rows] <- y
```

Within cluster B, from PCs 5 and 6 we define a new cluster, "P",
recognizing that this cluster is not particularly distinct:

```{r clustering-4}
rows <- which(x == "B")
fit2 <- select_loadings(fit,loadings = rows)
pca  <- prcomp(fit2$L)$x
y    <- rep("B",nrow(pca))
pc5  <- pca[,5]
pc6  <- pca[,6]
y[pc5 > 0.1 & pc6 < 0] <- "P"
x[rows] <- y
samples$cluster <- factor(x,c("B","C","P","Cil","T+N","I","U"))
```

In summary, we have subdivided the pulse-seq data into 7 subsets.
Comparing this to the Montoro *et al* (2018) clustering, we observe
some close correspondence between these subsets and the Montoro *et
al* clustering:

```{r clustering-5}
with(samples,table(tissue,cluster))
```

This close correspondence is also clear from PCA plots of the topic
proportions:

```{r clustering-12, fig.width=8, fig.height=1.5, message=FALSE}
tissue_colors <- c("royalblue",   # basal
				   "firebrick",   # ciliated
                   "forestgreen", # club
     			   "gold",        # goblet
				   "darkmagenta", # ionocyte
				   "darkorange",  # neuroendocrine
				   "peru",        # proliferating
                   "skyblue")     # tuft
abundant <- c("B","C","Cil","U")
rare     <- c("T+N","I","P")
rows1 <- which(is.element(samples$cluster,abundant))
rows2 <- which(is.element(samples$cluster,rare))
fit1  <- select_loadings(fit,loadings = rows1)
fit2  <- select_loadings(fit,loadings = rows2)
p1 <- pca_plot(fit1,pcs = 1:2,fill = samples[rows1,"tissue"]) +
  scale_fill_manual(values = tissue_colors,drop = FALSE) +
  labs(fill = "cluster")
p2 <- pca_plot(fit1,pcs = 3:4,fill = samples[rows1,"tissue"]) +
  scale_fill_manual(values = tissue_colors,drop = FALSE) +
  labs(fill = "cluster")
p3 <- pca_plot(fit2,pcs = 1:2,fill = samples[rows2,"tissue"]) +
  scale_fill_manual(values = tissue_colors,drop = FALSE) +
  labs(fill = "cluster")
plot_grid(p1,p2,p3,nrow = 1)
```

This Structure plot summarizes the topic proportions in each of the 7
subsets:

```{r structure-plot, fig.width=8, fig.height=1.5}
set.seed(1)
topic_colors <- c("turquoise","darkorange","dodgerblue","gold","peru",
                  "greenyellow","firebrick","olivedrab","royalblue",
                  "forestgreen","gray")
topics <- c(11,1,3,4,5,6,8,10,9,2,7)
rows <- sort(c(sample(which(samples$cluster == "B" & fit$L[,"k1"] < 0.5),800),
               sample(which(samples$cluster == "B" & fit$L[,"k1"] > 0.5),200),
               sample(which(samples$cluster == "C"),1000),
               sample(which(samples$cluster == "P"),200),
               sample(which(samples$cluster == "Cil"),200),
               sample(which(samples$cluster == "T+N"),200),
               sample(which(samples$cluster == "I"),150),
               sample(which(samples$cluster == "U"),100)))
p <- structure_plot(select_loadings(fit,loadings = rows),
                    grouping = samples[rows,"cluster"],
                    topics = topics,colors = topic_colors,
		 			perplexity = 70,n = Inf,gap = 30,
					num_threads = 4,verbose = FALSE)
print(p1)
```

Second structure plot after merging some topics:

```{r structure-plot-2, fig.width=8, fig.height=1.5}
set.seed(1)
fit2 <- merge_topics(fit,c("k4","k5","k6","k8","k10"))
fit2 <- merge_topics(fit2,c("k3","k9"))
topic_colors <- c("turquoise","darkorange","firebrick","gray","olivedrab",
                  "royalblue")
topics <- c(4,1,5,6,2,3)
p2 <- structure_plot(select_loadings(fit2,loadings = rows),
                    grouping = samples[rows,"cluster"],
                    topics = topics,colors = topic_colors,
		 			perplexity = 70,n = Inf,gap = 30,
					num_threads = 4,verbose = FALSE)
print(p2)
```

Based on this structure plot, and from the other results above, we
roughly subdivide the pulse-seq data into two subsets: (1) the Cil,
T+N and I clusters that give rise to fairly well-separated clusters,
and (2) the B, C and P subsets that contain interesting substructure
but much less distinct clusters. Therefore, the cluster labels B, C and
P are useful as a guide but should be taken with a grain of salt as
the boundaries between these clusters are somewhat arbitrary.

The structure plot suggests that there is substantial heterogeneity in
the C cluster beyond what can be captured by "hard" clusters. In
particular, three topics ($k = 5, 6, 10$) are largely unique to the
cells in this cluster, and two other topics ($k = 4, 8$) primarily
contribute to the cells in this cluster, although can be found in
small proportions elsewhere.

Save the clustering of the pulse-seq data to an RDS file.

```{r save-results}
saveRDS(samples,"clustering-pulseseq.rds")
```

Analysis of single-cell likelihoods
-----------------------------------

Here we calculate single-cell likelihoods to assess how well the
multinomial topic model captures expression in different cell-types.

```{r likelihood-1, eval=FALSE}
fit <- merge_topics(poisson2multinom(fit),c("k4","k5","k6","k8","k10"))
fit <- merge_topics(fit,c("k1","k3","k9"))
fit_cluster    <- init_poisson_nmf_from_clustering(counts,samples$cluster)
fit_montoro    <- init_poisson_nmf_from_clustering(counts,samples$tissue)
fit_cluster    <- poisson2multinom(fit_cluster)
fit_montoro    <- poisson2multinom(fit_montoro)
loglik_topics  <- loglik_multinom_topic_model(counts,fit)
loglik_cluster <- loglik_multinom_topic_model(counts,fit_cluster)
loglik_montoro <- loglik_multinom_topic_model(counts,fit_montoro)
```

Here we compare the likelihood under gene expression levels estimated
by a hard clustering; specifically, we compare the Montoro *et al*
(2018) clustering against our clustering. Overall, our clustering
appears to better fit the observed expression, particularly in the
most abundant cell types (basal, club).

```{r likelihood-2, fig.width=6, fig.height=6, eval=FALSE}
minloglik <- -30000
p1 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"basal",
                         "royalblue",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
p2 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"club",
						 "forestgreen",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
p3 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,
                         "ciliated","firebrick",minloglik,
                         "Montoro et al (2018) cluster","our clusters")
p4 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,
                         "neuroendocrine","darkorange",minloglik,
                         "Montoro et al (2018) cluster","our clusters")
p5 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"tuft",
                         "dodgerblue",minloglik,"Montoro et al (2018) cluster",
                         "our clusters")
p6 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,
                         "ionocyte","darkmagenta",minloglik,
                         "Montoro et al (2018) cluster","our clusters")
p7 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,
                         "proliferating","peru",minloglik,
                         "Montoro et al (2018) cluster","our clusters")
p8 <- loglik_scatterplot(loglik_montoro,loglik_cluster,samples$tissue,"goblet",
						 "gold",-Inf,"Montoro et al (2018) cluster",
                         "our clusters")
plot_grid(p1,p2,p3,p4,p5,p6,p7,p8,nrow = 3,ncol = 3)
```

Next, we compare the topic-model likelihoods to the clustering-based
likelihoods. It is interesting that although there is no topic
specifically representing ionocytes, the mixture of topics capturing
these cells fits the observed expression levels nearly as well as
estimates from an ionocytes cluster.

```{r likelihood-3,, fig.width=6, fig.height=4, eval=FALSE}
minloglik <- -25000
p9  <- loglik_scatterplot(loglik_cluster,loglik_topics,samples$cluster,"B",
				  "royalblue",minloglik,"cluster","topics")
p10 <- loglik_scatterplot(loglik_cluster,loglik_topics,samples$cluster,"C",
				  "forestgreen",minloglik,"cluster","topics")
p11 <- loglik_scatterplot(loglik_cluster,loglik_topics,samples$cluster,"Cil",
				  "firebrick",minloglik,"cluster","topics")
p12 <- loglik_scatterplot(loglik_cluster,loglik_topics,samples$cluster,"T+N",
				  "darkorange",minloglik,"cluster","topics")
p13 <- loglik_scatterplot(loglik_cluster,loglik_topics,samples$cluster,"I",
				  "darkmagenta",minloglik,"cluster","topics")
p14 <- loglik_scatterplot(loglik_cluster,loglik_topics,samples$cluster,"P",
				  "peru",minloglik,"cluster","topics")
plot_grid(p9,p10,p11,p12,p13,p14,nrow = 2,ncol = 3)
```

We observe similar trends when comparing the topic model likelihoods
against likelihoods obtained by the Montoro *et al* (2018) clustering.
One difference here is that tuft and neuroendocrine cells are split
into two clusters. But, judging by the likelihoods, this split provides
little improvement in fit.

```{r likelihood-4, fig.width=6, fig.height=6, eval=FALSE}
minloglik <- -30000
p15 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,"basal",
                          "royalblue",minloglik,"Montoro et al (2018) cluster",
                          "topics")
p16 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,
                          "club","forestgreen",minloglik,
                          "Montoro et al (2018) cluster","topics")
p17 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,
                          "ciliated","firebrick",minloglik,
                          "Montoro et al (2018) cluster","topics")
p18 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,
                          "neuroendocrine","darkorange",minloglik,
                          "Montoro et al (2018) cluster","topics")
p19 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,"tuft",
                          "dodgerblue",minloglik,
                          "Montoro et al (2018) cluster","topics")
p20 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,
                          "ionocyte","darkmagenta",minloglik,
                          "Montoro et al (2018) cluster","topics")
p21 <- loglik_scatterplot(loglik_montoro,loglik_topics,samples$tissue,
                          "proliferating","peru",minloglik,
                          "Montoro et al (2018) cluster","topics")
plot_grid(p15,p16,p17,p18,p19,p20,p21,nrow = 3,ncol = 3)
```

Save results
------------
