---
title: "Identify clusters in mixture of FACS-purified PBMC data using topic model"
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we perform PCA on the topic proportions to identify clusters in
the mixture of FACS-purified PBMC data sets.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, as well as additional
functions that we will use to generate some of the plots.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(dplyr)
library(fastTopics)
library(Rtsne)
library(uwot)
library(ggplot2)
library(cowplot)
source("../code/plots.R")
```

Load the count data.

```{r load-data}
load("../data/pbmc_purified.RData")
```

Load the $K = 6$ Poisson NMF model fit.

```{r load-fit}
fit <- readRDS(file.path("../output/pbmc-purified/rds",
                         "fit-pbmc-purified-scd-ex-k=6.rds"))$fit
```

Identify clusters from principal components
-------------------------------------------

From the PCs of the topic proportions, we define clusters for B-cells,
CD14+ cells and CD34+ cells. The remaining cells are assigned to the U
cluster ("U" for "unknown").

```{r clustering-1}
pca <- prcomp(poisson2multinom(fit)$L)$x
n   <- nrow(pca)
x   <- rep("U",n)
pc1 <- pca[,1]
pc2 <- pca[,2]
pc3 <- pca[,3]
pc4 <- pca[,4]
pc5 <- pca[,5]
x[pc2 > 0.25] <- "B"
x[pc3 < -0.2 & pc4 < 0.2] <- "CD34+"
x[(pc4 + 0.1)^2 + (pc5 - 0.8)^2 < 0.07] <- "CD14+"
```

Next, we define an NK cells cluster from the top 2 PCs of the topic
proportions in the U ("unknown") cells.

```{r clustering-2}
rows <- which(x == "U")
n    <- length(rows)
fit2 <- select(poisson2multinom(fit),loadings = rows)
pca  <- prcomp(fit2$L)$x
y    <- rep("U",n)
pc1  <- pca[,1]
pc2  <- pca[,2]
y[pc1 < -0.3 & 1.1*pc1 < -pc2 - 0.57] <- "NK"
x[rows] <- y
```

Among the remaining cells, we define a much less distinct cluster of
CD8+ cells. The rest are labeled as T-cells.

```{r clustering-3}
rows <- which(x == "U")
n    <- length(rows)
fit2 <- select(poisson2multinom(fit),loadings = rows)
pca  <- prcomp(fit2$L)$x
y    <- rep("T",n)
pc1  <- pca[,1]
pc2  <- pca[,2]
y[pc1 < 0.25 & pc2 < -0.15] <- "CD8+"
x[rows] <- y
```

In summary, we have subdivided the cells into 6 subsets. The
substructure becomes more clear when we plot T cells separately from
the other cells.

```{r clustering-4, fig.width=6, fig.height=4, message=FALSE}
cluster_colors <- c("dodgerblue","forestgreen","darkmagenta",
                    "red","gray","darkorange")
samples$cluster <- factor(x)
x     <- with(samples,cluster == "T" | cluster == "CD8+")
rows1 <- which(!x)
rows2 <- which(x)
p1 <- pca_plot(select(poisson2multinom(fit),loadings = rows1),
               fill = samples$cluster[rows1]) +
  scale_fill_manual(values = cluster_colors,drop = FALSE) +
  labs(fill = "cluster")
p2 <- pca_plot(select(poisson2multinom(fit),loadings = rows2),
               fill = samples$cluster[rows2,drop = FALSE]) +
  scale_fill_manual(values = cluster_colors,drop = FALSE) +
  labs(fill = "cluster")
p3 <- pca_hexbin_plot(select(poisson2multinom(fit),loadings = rows1),bins = 28)
p4 <- pca_hexbin_plot(select(poisson2multinom(fit),loadings = rows2),bins = 28)
plot_grid(p1,p3,p2,p4,nrow = 2,ncol = 2,rel_widths = c(10,11))
```

This clustering corresponds closely to the Zheng *et al* (2017) FACS
cell populations, although there are some differences---for example,
the CD14+ cluster contains a large number of cells identified as CD34+
by FACS.

```{r clustering-5}
with(samples,table(celltype,cluster))
```

This close correspondence is also apparent when we layer the FACS cell
population labels on the PCA plots:

```{r clustering-6, fig.width=8, fig.height=2, message=FALSE}
facs_colors <- c("dodgerblue",  # B-cells
                 "forestgreen", # CD14+
                 "darkmagenta", # CD34+
                 "firebrick",   # T helper cells
                 "gray",        # NK cells
                 "tomato",      # cytotoxic T-cells
                 "yellow",      # memory T-cells
                 "magenta",     # naive cytotoxic 
                 "darkorange",  # naive T-cells
                 "gold")        # regulatory T-cells
p1 <- pca_plot(select(poisson2multinom(fit),loadings = rows1),
               fill = samples[rows1,"celltype"]) +
  scale_fill_manual(values = facs_colors) +
  labs(fill = "FACS subtype")
p2 <- pca_plot(select(poisson2multinom(fit),loadings = rows2),
               fill = samples[rows2,"celltype"]) +
  scale_fill_manual(values = facs_colors) +
  labs(fill = "FACS subtype")
plot_grid(p1,p2)
```

By computing inter-cluster and inter-topic *total variation distances*
in relative expression levels, we see that the clusters identified
above show greater differentiation in gene expression, and the topics
show more differentiation than the clusters.

```{r clustering-7}
fit_zheng <- init_poisson_nmf_from_clustering(counts,samples$celltype)
d_zheng   <- totalvardist(poisson2multinom(fit_zheng)$F)
d_topics  <- totalvardist(poisson2multinom(fit)$F)
```

Here is a plot summarizing these differences:

```{r clustering-8, fig.height=2, fig.width=2}
pdat <-
  rbind(data.frame(method = "zheng.et.al",d = d_zheng[upper.tri(d_zheng)]),
        data.frame(method = "topics",     d = d_topics[upper.tri(d_topics)]))
p3 <- ggplot(pdat,aes(x = method,y = d)) +
  geom_boxplot(width = 0.25) +
  labs(x = "",y = "total variation dist") +
  theme_cowplot(font_size = 9)
print(p3)
```

Structure plot
--------------

The structure plot summarizes the topic proportions in each of these 6
subsets:

```{r structure-plot, fig.width=8, fig.height=1.5}
set.seed(1)
topic_colors <- c("gold","forestgreen","dodgerblue","gray","darkmagenta",
                  "violet")
topics <- c(5,3,2,4,1,6)
rows <- sort(c(sample(which(samples$cluster == "B"),1000),
               sample(which(samples$cluster == "CD14+"),300),
               sample(which(samples$cluster == "CD34+"),500),
               sample(which(samples$cluster == "CD8+"),400),
               sample(which(samples$cluster == "NK"),500),
               sample(which(samples$cluster == "T"),1000)))
p3 <- structure_plot(select(poisson2multinom(fit),loadings = rows),
                     grouping = samples[rows,"cluster"],
                     topics = topics,colors = topic_colors[topics],
   	  			     perplexity = c(70,30,30,30,30,70),
                     n = Inf,gap = 50,num_threads = 4,verbose = FALSE)
print(p3)
```

Note that CD8+ cells are identified as a distinctive mixture of topics
capturing NK cells and T-cells.

Analysis of single-cell likelihoods
-----------------------------------

*Create scatterplots comparing single-cell likelihoods using a "hard"
clustering (based on the FACS subpopulations).*

PCA vs. *t*-SNE and UMAP: an illustration
-----------------------------------------

Here we illustrate one of the benefits of using a simple linear
dimensionality reduction technique, PCA, over more widely used
nonlinear dimensionality reduction methods *t*-SNE and UMAP.  In
particular, we compare a 2-d linear embedding produced by PCA against
nonlinear embeddings generated by *t*-SNE and UMAP.

To begin, draw a random subset of 2,000 cells from the B, CD14+ and
CD34+ clusters identified above. (The main reason for taking a random
subset is that we don't want to wait a long time for *t*-SNE and UMAP
to complete.) Then run PCA on the topic proportions for this random
subset of 2,000 samples.

```{r pca}
set.seed(5)
rows <- which(with(samples,
                   cluster == "B" | cluster == "CD14+" | cluster == "CD34+"))
rows <- sort(sample(rows,2000))
fit2 <- select(poisson2multinom(fit),loadings = rows)
x    <- samples$cluster[rows,drop = TRUE]
p4   <- pca_plot(fit2,fill = x) + labs(fill = "cluster")
```

Next, run *t*-SNE on the topic proportions.

```{r tsne}
tsne <- Rtsne(fit2$L,dims = 2,pca = FALSE,normalize = FALSE,perplexity = 100,
              theta = 0.1,max_iter = 1000,eta = 200,verbose = FALSE)
tsne$x <- tsne$Y
colnames(tsne$x) <- c("tsne1","tsne2")
p5 <- pca_plot(fit2,out.pca = tsne,fill = x) + labs(fill = "cluster")
```

Then run UMAP on the topic proportions.

```{r umap}
out.umap <- umap(fit2$L,n_neighbors = 30,metric = "euclidean",n_epochs = 1000,
                 min_dist = 0.1,scale = "none",learning_rate = 1,
                 verbose = FALSE)
out.umap <- list(x = out.umap)
colnames(out.umap$x) <- c("umap1","umap2")
p6 <- pca_plot(fit2,out.pca = out.umap,fill = x) + labs(fill = "cluster")
```

```{r pca-vs-tsne, fig.width=8, fig.height=2}
plot_grid(p4,p5,p6,nrow = 1)
```

By the projection of the samples onto the first two PCs, we see that
the B-cells cluster is distinct from the others, whereas the CD14+ and
CD34+ cells do not separate quite so neatly.

However, this this finer scale detail is not captured in the *t*-SNE
and UMAP embeddings; this illustrates the tendency of *t*-SNE and UMAP
to accentuate clusters in the data at the risk of distorting or
obscure finer scale substructure.

Note that the first 2 PCs should be sufficient for capturing the full
structure in the topic proportions as they explain >96% of the
variance:

```{r prcomp-summary}
summary(prcomp(fit2$L))
```

Save results
------------

Save the clustering of the PBMC data to an RDS file.

```{r save-results}
saveRDS(samples,"clustering-pbmc-purified.rds")
```

```{r structure-plot-for-paper, echo=FALSE, eval=FALSE}
set.seed(1)
rows1   <- sample(which(samples$cluster == "B"),1000)
rows2   <- sample(which(samples$cluster == "CD14+"),300)
fit1    <- select(poisson2multinom(fit),loadings = rows1)
fit2    <- select(poisson2multinom(fit),loadings = rows2)
topics1 <- order(colMeans(fit1$L))
topics2 <- order(colMeans(fit2$L))
p1 <- structure_plot(fit1,topics = topics1,colors = topic_colors[topics1],
                     perplexity = 70,num_threads = 4) + ggtitle("B-cells")
p2 <- structure_plot(fit2,topics = topics2,colors = topic_colors[topics2],
					 perplexity = 30,num_threads = 4) + ggtitle("CD14+")
plot_grid(p1,p2)
```
