---
title: "Evaluation of DE analysis with more than two topics"
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we expand the evaluation of the new topic-model-based DE analysis
methods to consider the case of more than two topics. When comparing
multiple groups, clusters or topics it is less clear what it means for
a gene to be "differentially expressed". So to simplify the evaluation
we simulate the data so that each gene is differentially expressed in
at most one topic.

In this evaluation can no longer compare to DESeq2 because DESeq2 is
designed to compare gene expression across two groups. Still, we can
closely examine the benefit of accounting for uncertainty, and in so
doing we will see that the benefits become more pronounced because our
analysis factors in correlations in the estimates across the different
topics.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.show = "hold",fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, and some additional
functions for simulating the data.

```{r load-pkgs, message=FALSE, warning=FALSE}
library(Matrix)
library(fastTopics)
library(ggplot2)
library(cowplot)
source("../code/de_analysis_functions.R")
```

Simulate UMI counts
-------------------

We begin by simulating UMI count data from a rank-6 Poisson NMF model
such that roughly half of the genes have the same expression rate
across all topics, and the other half have a different rate of
expression in exactly one out of the six topics. The details of the
simulation are similar to the two-topic case.

```{r simulate-counts}
set.seed(1)
m <- 10000
k <- 6
dat <- simulate_manytopic_umi_data(m = m,k = k)
X <- dat$X
F <- dat$F
L <- dat$L
```

Fit multinomial topic model to UMI counts
-----------------------------------------

Now we fit a multinomial topic model to the simulated UMI count data.
To simplify evaluation, we assume that the topic proportions are
known, and fix them to their ground-truth values. In this way, the
only error that can arise is in the estimates of the expression rates
$f_{ij}$.

```{r fit-poisson-nmf, warning=FALSE, message=FALSE, cache=TRUE}
fit0 <- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit <- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = "scd",
                       update.loadings = NULL,verbose = "none")
fit <- poisson2multinom(fit)
summary(fit)
```

DE analysis with and without shrinkage
--------------------------------------

First we perform a DE analysis *without* shrinking the LFC estimates:

```{r de-noshrink, message=FALSE, cache=TRUE}
set.seed(1)
de.noshrink <- de_analysis(fit,X,shrink.method = "none",
                           control = list(ns = 10000,nc = 2))
```

Next we perform a second DE analysis using adaptive shrinkage to
shrink the LFC estimates:

```{r de-with-ash, message=FALSE, cache=TRUE}
set.seed(1)
de <- de_analysis(fit,X,shrink.method = "ash",control = list(ns = 1e4,nc = 2))
```

To measure differential expression across multiple topics, we use the
"least extreme" log-fold change, which, for topic $k$, is defined as
the LFC between topic $k$ and each other topic $j \neq k$ with a
expression ratio closest to 1. Then we compare estimates of the "least
extreme" LFCs with and without shrinkage.

As expected, smaller estimates, and estimates corresponding to genes
with lower expression, tend to be more strongly shrunk toward zero.

```{r shrink-vs-noshrink-scatterplot, fig.height=2.75, fig.width=4.5}
pdat <- data.frame(noshrink  = c(de.noshrink$postmean),
                   shrink    = c(de$postmean),
                   log10mean = rep(log10(de$f0),k))
p1 <- ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -4) +
  labs(x = "original LFC estimates",
       y = "shrunken LFC estimates") +
  theme_cowplot()
print(p1)
```

The next two bar charts show the overall impact of the shrinkage on
the distribution of the $z$-scores; dark blue bars are for
diifferentially expressed genes, and orange bars are for
non-differentially expressed genes.

```{r shrink-vs-noshrink-zscores, fig.height=5, fig.width=8, warning=FALSE}
nonzero_lfc <- matrix(FALSE,m,k)
for (i in 1:m) {
  y <- dat$F[i,]
  if (max(y) - min(y) > 1e-8) {
    j <- which.max((y - mean(y))^2)
    nonzero_lfc[i,j] <- TRUE
  }
}
pdat <- data.frame(noshrink = clamp(c(de.noshrink$z),-4,+4),
                   shrink   = clamp(c(de$z),-4,+4),
                   de       = c(nonzero_lfc))
p2 <- ggplot(subset(pdat,de),aes(x = noshrink,color = de)) +
  geom_histogram(color = "darkblue",fill = "darkblue",bins = 64) +
  coord_cartesian(xlim = c(-4,+4),,ylim = c(0,400)) +
  labs(x = "z-score",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p3 <- ggplot(subset(pdat,de),aes(x = shrink,color = de)) +
  geom_histogram(color = "darkblue",fill = "darkblue",bins = 64) +
  coord_cartesian(xlim = c(-4,+4),ylim = c(0,400)) +
  labs(x = "z-score",y = "genes",title = "with shrinkage") +
  theme_cowplot()
p4 <- ggplot(subset(pdat,!de),aes(x = noshrink,color = de)) +
  geom_histogram(color = "darkorange",fill = "darkorange",bins = 64) +
  coord_cartesian(xlim = c(-4,+4)) +
  labs(x = "z-score",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p5 <- ggplot(subset(pdat,!de),aes(x = shrink,color = de)) +
  geom_histogram(color = "darkorange",fill = "darkorange",bins = 64) +
  coord_cartesian(xlim = c(-4,+4)) +
  labs(x = "z-score",y = "genes",title = "with shrinkage") +
  theme_cowplot()
print(plot_grid(p2,p3,p4,p5,nrow = 2,ncol = 2))
```

And the next two bar charts show the overall impact of the shrinkage
on the $p$-values. (The Y axis in the "with shrinkage" plot has been
cut off at 2,500 genes to show in more detail the smaller $p$-values
identifying true positives.)

```{r shrink-vs-noshrink-pvalues, fig.height=2.75, fig.width=8, warning=FALSE}
pdat <- data.frame(noshrink = 10^(-c(de.noshrink$lpval)),
                   shrink   = 10^(-c(de$lpval)),
                   de       = factor(c(nonzero_lfc)))
p6 <- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  coord_cartesian(ylim = c(0,2500)) +
  labs(x = "p-value",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p7 <- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  coord_cartesian(ylim = c(0,2500)) +
  labs(x = "p-value",y = "genes",title = "with shrinkage") +
  theme_cowplot()
print(plot_grid(p6,p7))
```

Both the $z$-score and $p$-value plots show that the adaptive
shrinkage step more strongly encourages the LFC estimates for the
non-DE genes toward zero, thereby improving overall accuracy.

Quantifying uncertainty in the LFC estimates
--------------------------------------------

*Add text here.*

```{r point-vs-shrunken-scatterplot, fig.width=8, fig.height=2.5}
pdat <- data.frame(est       = c(de$est),
                   shrink    = c(de$postmean),
				   log10mean = rep(log10(de$f0),k))
p6 <- ggplot(pdat,aes(x = est,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -4) +
  labs(x = "point estimates",
       y = "shrunken estimates") +
  theme_cowplot()
print(p6)
```

*Add text here.*

```{r kl-vs-fasttopics-1, fig.width=8, fig.height=2.5}
D     <- fastTopics:::min_kl_poisson(fit$F)
i     <- which(de$z > 0)
pdat1 <- data.frame(kl = pmin(D[i],5e-6),de = factor(nonzero_lfc[i]))
pdat2 <- data.frame(z = clamp(de$z[i],-2,+2),de = factor(nonzero_lfc[i]))
p7 <- ggplot(pdat1,aes(x = kl,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "K-L divergence",y = "genes") +
  theme_cowplot()
p8 <- ggplot(pdat2,aes(x = z,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "z-score",y = "genes") +
  theme_cowplot()
print(plot_grid(p7,p8))
```

*Add text here.*

```{r kl-vs-fasttopics-2, fig.width=8, fig.height=2.5}
p9  <- p7 + scale_x_continuous(limits = 1e-6 * c(0.5,5.1))
p10 <- p8 + scale_x_continuous(limits = c(0.05,2.1))
print(plot_grid(p9,p10))
```
