---
title: "DE analysis using a topic model: evaluation using simulated data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

*Add summary of the analysis here.*

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.show = "hold",fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, and some additional
functions for simulating the data.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(fastTopics)
library(MCMCpack)
library(ggplot2)
library(cowplot)
source("../code/de_eval_functions.R")
```

Simulate UMI counts
-------------------

We begin by simulating counts intended to "mimic" UMI count data from
a single-cell RNA sequencing experiment. In particular, we simulate
counts $x_{ij}$ from a Poisson NMF model $x_{ij} \sim
\mathrm{Poisson}(\lambda_{ij})$, such that $\lambda_{ij} =
\sum_{k=1}^K l_{ik} f_{jk}$. We focus our attention on the $K = 2$
case to simplify the analysis; comparing gene expression between three
or topics brings some additional complications which aren't necessary
for assessing basic properties of the new DE methods. The simulation
of the topic proportions $l_{ik}$ and the gene expression rates
$f_{jk}$ is described in the comments accompanying function
`simulate_twotopic_umi_data`.

```{r simulate-counts}
set.seed(1)
dat <- simulate_twotopic_umi_data()
X <- dat$X
```

The sample sizes (total counts for each cell) in the simulated data
are roughly normally distributed on the (base-10) log scale with
mean 3.2 and standard deviation 0.2:

```{r sizes, fig.height=2.5, fig.width=3}
s <- rowSums(X)
p1 <- ggplot(data.frame(s = log10(s)),aes(x = s)) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "log10 size",y = "cells") +
  theme_cowplot()
print(p1)
mean(log10(s))
sd(log10(s))
```

We have generated the expression rates so that they are roughly
uniformly distributed on the log scale:

```{r gene-expression-rates, fig.height=2.5, fig.width=3}
p2 <- ggplot(data.frame(f = as.vector(dat$F)),aes(x = log10(f))) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "log10 expression rate",y = "genes") +
  theme_cowplot()
print(p2)
```

About half of the genes have differences in expression between the two
topics, and among the nonzero gene expression differences, the LFCs
are roughly *t*-distibuted with 3 degrees of freedom:

```{r lfc-density, fig.height=2.5, fig.width=3}
lfc <- log2(dat$F[,1]/dat$F[,2])
p3 <- ggplot(data.frame(lfc = lfc[lfc != 0]),aes(x = lfc)) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "LFC",y = "genes") +
  theme_cowplot()
print(p3)
mean(lfc == 0)
```

Fit multinomial topic model to UMI counts
-----------------------------------------

Fit a multinomial topic model to. To simplify this evaluation, we
assume that the topic proportions are known, and set to the ground
truth (that is, the values used to simulate the data), so the only
error that can arise is in the estimates of the factors, or the
relative expression levels.

```{r fit-poisson-nmf, warning=FALSE}
fit0 <- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit <- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = "scd",
                       update.loadings = NULL,verbose = "none")
fit <- poisson2multinom(fit)
summary(fit)
```

DE analysis with and without shrinkage
--------------------------------------

First we perform a DE analysis without applying adaptive shrinkage to
the LFC estimates:

```{r de-noshrink}
set.seed(1)
de.noshrink <- de_analysis(fit,X,shrink.method = "none",
                           control = list(nc = 4))
```

Next we perform a second DE analysis using adaptive shrinkage:

```{r de-with-ash}
set.seed(1)
de <- de_analysis(fit,X,shrink.method = "ash",control = list(nc = 4))
```

*Add text here.*

```{r shrink-vs-noshrink-scatterplot, fig.height=4, fig.width=5}
pdat <- data.frame(noshrink  = de.noshrink$postmean[,1],
                   shrink    = de$postmean[,1],
                   log10mean = log10(de$f0))
p4 <- ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white") +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -3.5) +
  xlim(c(-13,11)) +
  ylim(c(-13,11)) +
  labs(x = "LFC without shrinkage",
       y = "LFC with shrinkage") +
  theme_cowplot()
print(p4)
```

*Add text here.*

```{r shrink-vs-noshrink-zscores, fig.height=3, fig.width=6}
pdat <- data.frame(noshrink = clamp(de.noshrink$z[,1],-8,+8),
                   shrink   = clamp(de$z[,1],-8,+8),
                   de       = factor(lfc == 0))
p5 <- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  ylim(c(0,100))  +
  theme_cowplot()
p6 <- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  ylim(c(0,100))  +
  theme_cowplot()
print(plot_grid(p5,p6))
```
