---
title: "Differential expression analysis using a topic model: illustration in mixture of FACS-purified PBMC data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

The aim of this analysis is to understand by way of illustration the
differences between a "classical" differential expresion analysis
comparing expression among cell types (here we implement this
"classical" DE analysis using [DESEq2][deseq2]), and a differential expression
analysis using the topic model, which allows for grades of membership
to cell types (or more generally cellular expression factors).

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Begin by loading the packages and some function definitions used in
the analysis.

```{r load-pkgs, message=FALSE, warning=FALSE}
library(Matrix)
library(DESeq2)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source("../code/de_analysis_functions.R")
```

Load count data and topic model fit
-----------------------------------

Load the UMI count data for 94,655 cells and 21,952 genes.

```{r load-data}
load("../data/pbmc_purified.RData")
dim(counts)
```

Load the $K = 6$ Poisson NMF model fit to these data, and convert the
Poisson NMF model fit to a multinomial topic model fit.

```{r load-fit}
fit <- readRDS(file.path("../output/pbmc-purified/rds",
                         "fit-pbmc-purified-scd-ex-k=6.rds"))$fit
fit <- poisson2multinom(fit)
```

The cells are subdivided, based on FACS sorting, into 10 "cell
types". Several of the cell types are virtually indistinguishable
based on their gene expression profiles alone, so we combine these
indistinguishable cell types into a single "T cell" cell type. This
results in 5 predefined cell types, the majority of which are T cells:

```{r cell-types}
set.seed(1)
celltype <- as.character(samples$celltype)
celltype[celltype == "CD4+ T Helper2" |
         celltype == "CD4+/CD45RO+ Memory" |
         celltype == "CD8+/CD45RA+ Naive Cytotoxic" |
         celltype == "CD4+/CD45RA+/CD25- Naive T" |
         celltype == "CD8+ Cytotoxic T" |
         celltype == "CD4+/CD25 T Reg"] <- "T cell"
celltype <- factor(celltype,
                   c("CD19+ B","CD14+ Monocyte","CD34+","CD56+ NK","T cell"))
table(celltype)
```

Structure plot
--------------

Next we visualize the structure inferred by the $K = $ topic model
using a "structure plot". The cells in this plot are arranged
horizontally according to their predefined cell type to relate the
topics to these predefinend cell types:

```{r structure-plot, fig.width=8, fig.height=1.8}
topic_colors <- c("gold","forestgreen","dodgerblue","gray",
                  "darkmagenta","violet")
topics <- c(5,3,2,4,1,6)
rows <- sort(c(sample(which(celltype == "CD19+ B"),500),
               sample(which(celltype == "CD14+ Monocyte"),250),
               sample(which(celltype == "CD34+"),500),
               sample(which(celltype == "CD56+ NK"),400),
               sample(which(celltype == "T cell"),1000)))
p1 <- structure_plot(select_loadings(fit,loadings = rows),
                     grouping = celltype[rows],topics = topics,
		             colors = topic_colors,perplexity = 70,n = Inf,gap = 30,
		             num_threads = 4,verbose = FALSE)
print(p1)
```

Some of the topics correspond very closely to the predefined cell
types.  In particular, topics 1 through 4 closely correspond,
respectively, to T cells, CD14+ monocytes (myeloid cells), B cells and
natural killer (NK) cells.

Topic 5 (violet) closely corresponds to the "CD34+" FACS cell type,
but from the structure plot we observe many cells labeled as "CD34+"
with little to no contribution from topic 5, which suggests
mislabeling of the CD34+ cells.

Topic 6 (magenta) does not correspond to any FACS cell type and as we
will see it captures a different characteristic of the
cells---specifically, abundance of ribosomal protein genes.
Therefore, the DE results for topics 1--4 are most comparable to a
classical DE analysis, and we begin with these comparisons. But before
doing this we first assess accuracy of the MCMC computations used in
the topic-model-based DE analysis.

Assessing accuracy of the Monte Carlo estimates
-----------------------------------------------

The topic-model-based DE analysis was performed previously by
simulating the posterior distribution of the LFC statistics via
MCMC. Here we assess accuracy of the MCMC calculations. We performed
the DE analysis twice (using different seeds, each with 100,000 Monte
Carlo samples), so we can compare the posterior mean estimates and
*z*-scores returned by the two `de_analysis` runs.

```{r load-de-analysis}
load("../output/pbmc-purified/de-pbmc-purified-seed=1.RData")
de1 <- de
load("../output/pbmc-purified/de-pbmc-purified-seed=2.RData")
de2 <- de
rm(de)
```

The MCMC estimates of the posterior mean log-fold change (LFC)
estimates are largely consistent:

```{r assess-monte-carlo-1, fig.width=3.25, fig.height=3}
pdat <- data.frame(postmean1 = as.vector(de1$postmean),
                   postmean2 = as.vector(de2$postmean))
pdat1 <- subset(pdat,abs(postmean1) > 0.1 | abs(postmean2) > 0.1)
pdat2 <- subset(pdat,abs(postmean1) <= 0.1 & abs(postmean2) <= 0.1)
pdat2 <- pdat[sample(nrow(pdat2),100),]
pdat  <- rbind(pdat1,pdat2)
p1 <- ggplot(pdat,aes(x = postmean1,y = postmean2)) +
  geom_point(shape = 21,color = "white",fill = "black",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dashed") +
  labs(x = "first posterior mean",y = "second posterior mean") +
  theme_cowplot()
print(p1)
```

The *z*-scores on the other hand are estimated less consistently,
presumably because accurately estimating uncertainty is harder.
Still, the *z*-scores are still are consistent enough in that it is
rare for an LFC to have an lfsr less than 0.05 in one MCMC simulation
and not the other (these are the red points in the scatterplot).  Note
for better visualization *z*-scores larger than 100 (or smaller than
-100) are shown as 100 (or -100) in this plot.

```{r assess-monte-carlo-3, fig.width=4.25, fig.height=3, warning=FALSE}
pdat <- data.frame(z1 = clamp(as.vector(de1$z),-100,+100),
                   z2 = clamp(as.vector(de2$z),-100,+100),
				   lfsr = factor((de1$lfsr < 0.05) + (de2$lfsr < 0.05)))
pdat1 <- subset(pdat,abs(z1) > 0.5 | abs(z2) > 0.5)
pdat2 <- subset(pdat,abs(z1) <= 0.5 & abs(z2) <= 0.5)
pdat2 <- pdat[sample(nrow(pdat2),100),]
pdat  <- rbind(pdat1,pdat2)
p2 <- ggplot(pdat,aes(x = z1,y = z2,fill = lfsr)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  scale_fill_manual(values = c("darkblue","tomato","dodgerblue"),
                    na.value = "white") +
  labs(x = "first z-score",y = "second z-score",fill = "lfsr < 0.05") +
  theme_cowplot()
print(p2)
```

Moving forward, when the two *z*-scores disagree, we use the one that
is nearer to zero.

```{r fix-z-scores}
de             <- de1[c("f0","lower","postmean","upper","z","lfsr")]
class(de)      <- c("topic_model_de_analysis","list")
i              <- which(abs(de2$z) < abs(de1$z))
de$lower[i]    <- de2$lower[i]
de$postmean[i] <- de2$postmean[i]
de$upper[i]    <- de2$upper[i]
de$z[i]        <- de2$z[i]
de$lfsr[i]     <- de$lfsr[i]
```

```{r mcmc-accuracy-plots-for-paper, echo=FALSE, warning=FALSE}
ggsave("../plots/postmean_scatterplot_purified_pbmc.eps",p1,
       height = 4,width = 4)
ggsave("../plots/zscore_scatterplot_purified_pbmc.eps",p2,
	   height = 4,width = 5.25)
```

Load DESeq2 results for all cell types
--------------------------------------

We load the results of the DESeq2 analyes, and combine them into two
data frames: a data frame for the posterior mean LFC estimates, and a
data frame for the *z*-scores.

```{r load-deseq2}
load("../output/pbmc-purified/deseq2-pbmc-purified.RData")
celltypes <- names(deseq)
n <- length(celltypes)
p <- nrow(genes)
deseq2 <- list(postmean = matrix(0,p,n),
               z        = matrix(0,p,n))
rownames(deseq2$postmean) <- genes$ensembl
rownames(deseq2$z)        <- genes$ensembl
colnames(deseq2$postmean) <- celltypes
colnames(deseq2$z)        <- celltypes
for (i in 1:n) {
  deseq2$postmean[,i] <- deseq[[i]]$log2FoldChange
  deseq2$z[,i]        <- with(deseq[[i]],log2FoldChange/lfcSE)
}
deseq <- deseq2
rm(deseq2)
```

Since we filtered out a few lowly expressed genes before running
DESeq2, we subset the fastTopics results to match up with DESeq2.

```{r align-with-deseq2}
rows        <- match(rownames(deseq$z),rownames(de$z))
de$f0       <- de$f0[rows]
de$lower    <- de$lower[rows,]
de$postmean <- de$postmean[rows,]
de$upper    <- de$upper[rows,]
de$z        <- de$z[rows,]
de$lfsr     <- de$lfsr[rows,]
```

DESeq2 vs. fastTopics
---------------------

Comparing the distributions of all *z*-scores, we see that the DE
analysis allowing for grades of membership has many more *z*-scores
near zero, yet still allows for very large $z$-scores, illustrating
the flexibility of the adaptive shrinkage. (Because there are a few
extremely large and extremely small *z*-scores, for better
visualization z-scores larger than 100 in magnitude are shown as 100
or -100.)

```{r z-score-scatterplot, fig.height=3.25, fig.width=3.25}
pdat <-
  data.frame(deseq     =quantile(deseq$z,seq(0,1,length.out=1000),na.rm=TRUE),
             fasttopics=quantile(de$z,seq(0,1,length.out=1000),na.rm=TRUE))
pdat <- transform(pdat,
                  deseq      = clamp(deseq,-100,+100),
                  fasttopics = clamp(fasttopics,-100,+100))
p1 <- ggplot(pdat,aes(x = deseq,y = fasttopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  labs(x = "DESeq2",y = "fastTopics",title = "z-score quantiles") +
  theme_cowplot()
print(p1)
```

```{r zscore-qq-plots-for-paper, echo=FALSE}
ggsave("../plots/zscore_qq_purified_pbmc.eps",p1,height = 4,width = 4)
```

Now let's look closely at the DESeq2 and fastTopics results for B
cells and CD14+ cells. These two cell types are chosen because they
each closely correspond to a topic (topics 2 and 3). Here we focus on
genes for which the *z*-score is greater than 2 in at least one of the
analyses. Genes are colored according to the LFSR estimated in the
fastTopics analysis.

```{r bcells-z-scores, fig.height=3.1, fig.width=8, warning=FALSE}
i <- "CD19+ B"
k <- "k3"
pdat1 <- data.frame(gene                = genes$symbol,
                    postmean.deseq      = deseq$postmean[,i],
                    postmean.fasttopics = de$postmean[,k],
	  			    z.deseq             = deseq$z[,i],
	  			    z.fasttopics        = de$z[,k],
  				    lfsr = cut(de$lfsr[,k],c(-1,0.001,0.01,0.05,Inf)),
  				    stringsAsFactors = FALSE)
j <- which(pdat1$postmean.fasttopics < 8)
pdat1[j,"gene"] <- ""
pdat1 <- subset(pdat1,abs(z.deseq) > 2 | abs(z.fasttopics) > 2)
p1 <- ggplot(pdat1,aes(x = postmean.deseq,y = postmean.fasttopics,
                fill = lfsr,label = gene)) +
  geom_point(shape = 21,color = "white") +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  geom_text_repel(color = "darkgray",size = 2.25,fontface = "italic",
                  segment.color = "darkgray",segment.size = 0.25,
                  min.segment.length = 0,max.overlaps = Inf,na.rm = TRUE) +
  scale_fill_manual(values = c("deepskyblue","gold","orange","coral"),
                    na.value = "gainsboro") +
  xlim(-12.3,13.1) +
  ylim(-10,13.1) +
  labs(x = "DESeq2",y = "fastTopics",
       title = "LFC in B cells") +
  theme_cowplot(font_size = 10)
i <- "CD14+ Monocyte"
k <- "k2"
pdat2 <- data.frame(gene                = genes$symbol,
                    postmean.deseq      = deseq$postmean[,i],
                    postmean.fasttopics = de$postmean[,k],
	  			    z.deseq             = deseq$z[,i],
	  			    z.fasttopics        = de$z[,k],
  				    lfsr = cut(de$lfsr[,k],c(-1,0.001,0.01,0.05,Inf)),
  				    stringsAsFactors = FALSE)
j <- which(pdat2$postmean.fasttopics < 10)
pdat2[j,"gene"] <- ""
pdat2 <- subset(pdat2,abs(z.deseq) > 2 | abs(z.fasttopics) > 2)
p2 <- ggplot(pdat2,aes(x = postmean.deseq,y = postmean.fasttopics,
                fill = lfsr,label = gene)) +
  geom_point(shape = 21,color = "white") +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  geom_text_repel(color = "darkgray",size = 2.25,fontface = "italic",
                  segment.color = "darkgray",segment.size = 0.25,
                  min.segment.length = 0,max.overlaps = Inf,na.rm = TRUE) +
  scale_fill_manual(values = c("deepskyblue","gold","orange","coral"),
                    na.value = "gainsboro") +
  xlim(-8.2,15) +
  ylim(-8.2,15) +
  labs(x = "DESeq2",y = "fastTopics",
       title = "LFC in CD14+ monocytes") +
  theme_cowplot(font_size = 10)
plot_grid(p1,p2)
```

A couple interesting themes emerge from this plot:

+ Many genes with large LFC in the DESeq2 analysis, particularly genes
  with very negative LFCs, have zero LFC in the fastTopics analysis.

+ DESeq2 and fastTopics largely identify the same genes with the
  largest expression increases (largest LFC), but fastTopics estimates
  larger LFCs for these genes.

```{r scatterplots-plots-for-paper, echo=FALSE, warning=FALSE}
ggsave("../plots/scatterplots_purified_pbmc.eps",plot_grid(p1,p2),
	   height = 3,width = 8)
```

The fastTopics DE analysis for all 6 topics is summarized in
the following volcano plots:

```{r volcano-plots, fig.height=10, fig.width=8, warning=FALSE}
my_ggplot_call <- function (dat, plot.title, max.overlaps) {
  i <- which(with(dat,zabs > 0.5 | abs(postmean) > 0.5))
  j <- sample(which(with(dat,zabs <= 1 & abs(postmean) <= 0.5)),100)
  dat <- dat[c(i,j),]
  return(volcano_plot_ggplot_call(dat,plot.title,max.overlaps))
}
p1 <- volcano_plot(de,k = "k1",ymax = 150,labels = genes$symbol,ggplot_call = my_ggplot_call)
p2 <- volcano_plot(de,k = "k2",ymax = 175,labels = genes$symbol,ggplot_call = my_ggplot_call)
p3 <- volcano_plot(de,k = "k3",ymax = 150,labels = genes$symbol,ggplot_call = my_ggplot_call)
p4 <- volcano_plot(de,k = "k4",ymax = 175,labels = genes$symbol,ggplot_call = my_ggplot_call)
p5 <- volcano_plot(de,k = "k5",ymax = 175,labels = genes$symbol,ggplot_call = my_ggplot_call)
p6 <- volcano_plot(de,k = "k6",ymax = 200,labels = genes$symbol,ggplot_call = my_ggplot_call)
plot_grid(p1,p2,p3,p4,p5,p6,nrow = 3,ncol = 2)
```

These same results can also be explored in interactive volcano plots:

```{r bcells-volcano-plotly, results="hide", warning=FALSE}
volcano_plotly(de,k = "k1",ymax = 150,labels = genes$symbol,
               file = "volcano_plot_purified_pbmc_tcells.html",
               width = 600,height = 600)
volcano_plotly(de,k = "k2",ymax = 175,labels = genes$symbol,
               file = "volcano_plot_purified_pbmc_cd14.html",
               width = 600,height = 600)
volcano_plotly(de,k = "k3",ymax = 150,labels = genes$symbol,
               file = "volcano_plot_purified_pbmc_bcells.html",
               width = 600,height = 600)
volcano_plotly(de,k = "k4",ymax = 175,labels = genes$symbol,
               file = "volcano_plot_purified_pbmc_nkcells.html",
               width = 600,height = 600)
volcano_plotly(de,k = "k5",ymax = 175,labels = genes$symbol,
               file = "volcano_plot_purified_pbmc_cd34.html",
               width = 600,height = 600)
volcano_plotly(de,k = "k6",ymax = 200,labels = genes$symbol,
               file = "volcano_plot_purified_pbmc_ribosome.html",
               width = 600,height = 600)
```

You can explore these interactive volcano plots by following these
links:

+ [topic 1 (T cells)](volcano_plots/volcano_plot_purified_pbmc_tcells.html)

+ [topic 2 (CD14+ monocytes)](volcano_plots/volcano_plot_purified_pbmc_cd14.html)

+ [topic 3 (B cells)](volcano_plots/volcano_plot_purified_pbmc_bcells.html)

+ [topic 4 (natural killer cells)](volcano_plots/volcano_plot_purified_pbmc_nkcells.html)

+ [topic 5 (CD34+)](volcano_plots/volcano_plot_purified_pbmc_cd34.html)

+ [topic 6 (ribosomal protein genes)](volcano_plots/volcano_plot_purified_pbmc_ribosome.html)

```{r volcano-plots-for-paper, echo=FALSE}
ggsave("../plots/volcano_plots_purified_pbmc.eps",
       plot_grid(p1,p2,p3,p4,p5,p6,nrow = 2,ncol = 3),
	   height = 7,width = 12.5)
```

[deseq2]: https://github.com/mikelove/DESeq2
