---
title: "DE analysis using a topic model: evaluation using simulated data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

*Add summary of the analysis here.*

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.show = "hold",fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, and some additional
functions for simulating the data.

```{r load-pkgs, message=FALSE, warning=FALSE}
library(Matrix)
library(scran)
library(DESeq2)
library(fastTopics)
library(ggplot2)
library(cowplot)
source("../code/de_eval_functions.R")
```

Simulate UMI counts
-------------------

We begin by simulating counts from a rank-2 Poisson NMF model with
parameters chosen to roughly mimic the UMI counts from a single-cell
RNA sequencing experiment. For now, we focus our attention on the
rank-2 (*i.e.*, two topics) case to simplify the evaluation of the DE
methods; comparing gene expression between three or topics brings some
additional complications which aren't necessary for assessing basic
properties of the new DE methods. For more details on how the data are
simulated, see function `simulate_twotopic_umi_data`.

```{r simulate-counts}
set.seed(1)
dat <- simulate_twotopic_umi_data()
X <- dat$X
F <- dat$F
L <- dat$L
```

Before fitting a topic model and running a DE analysis, we first
inspect some basic properties of the simulated data.

The sample sizes (total counts for each cell) are roughly normal on
the (base-10) log scale:

```{r sizes, fig.height=2, fig.width=3.5}
s <- rowSums(X)
p1 <- ggplot(data.frame(s = log10(s)),aes(x = s)) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "log10 size",y = "cells") +
  theme_cowplot()
print(p1)
```

The expression rates were simulated so that they are normally
distributed on the log scale:

```{r gene-expression-rates, fig.height=2, fig.width=3.5}
p2 <- ggplot(data.frame(f = as.vector(F)),aes(x = log10(f))) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "log10 expression rate",y = "genes") +
  theme_cowplot()
print(p2)
```

About half of the genes have nonzero differences in expression between
the two topics. Among the nonzero gene expression differences, the
log-fold changes (LFCs) were simulated from the normal distribution:

```{r lfc-density, fig.height=2, fig.width=3.5}
nonzero_lfc <- abs(F[,1] - F[,2]) > 1e-8
lfc <- log2(F[,2]/F[,1])
p3 <- ggplot(data.frame(lfc = lfc[nonzero_lfc]),aes(x = lfc)) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "LFC",y = "genes") +
  theme_cowplot()
print(p3)
mean(nonzero_lfc)
```

Fit multinomial topic model to UMI counts
-----------------------------------------

Here we fit a multinomial topic model to the simulated UMI count
data. To simplify evaluation of the DE methods, we assume that the
topic proportions are known, and set them to their ground truth values
(that is, the values used to simulate the data). In this way, the only
error that can arise is in the estimates of the expression rates
$f_{ij}$.

```{r fit-poisson-nmf, warning=FALSE, message=FALSE, cache=TRUE}
fit0 <- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit <- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = "scd",
                       update.loadings = NULL,verbose = "none")
fit <- poisson2multinom(fit)
summary(fit)
```

DE analysis with and without shrinkage
--------------------------------------

First we perform a DE analysis *without* shrinking the LFC estimates:

```{r de-noshrink, message=FALSE, cache=TRUE}
set.seed(1)
de.noshrink <- de_analysis(fit,X,shrink.method = "none",
                           control = list(ns = 1e4,nc = 2))
```

Now we perform a second DE analysis using adaptive shrinkage to
shrink (and hopefully improve accuracy of) the LFC estimates:

```{r de-with-ash, message=FALSE, cache=TRUE}
set.seed(1)
de <- de_analysis(fit,X,shrink.method = "ash",control = list(ns = 1e4,nc = 2))
```

In this scatterplot, we compare the posterior mean LFC estimates with
and without performing the shrinkage step. As expected, the smaller
LFC estimates, as well as the LFC estimates corresponding to genes
with lower expression, tend to be more strongly shrunk toward zero.

```{r shrink-vs-noshrink-scatterplot, fig.height=2.75, fig.width=5}
pdat <- data.frame(noshrink  = de.noshrink$postmean[,2],
                   shrink    = de$postmean[,2],
                   log10mean = log10(de$f0))
p4 <- ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -4) +
  labs(x = "original LFC estimates",
       y = "shrunken LFC estimates") +
  theme_cowplot()
print(p4)
```

These next two bar charts show the overall impact of the shrinkage on
the distribution of the $z$-scores; dark blue bars are for
diifferentially expressed genes, and orange bars are for
non-differentially expressed genes.

```{r shrink-vs-noshrink-zscores, fig.height=2.75, fig.width=8, warning=FALSE}
pdat <- data.frame(noshrink = clamp(de.noshrink$z[,2],-8,+8),
                   shrink   = clamp(de$z[,2],-8,+8),
                   de       = factor(nonzero_lfc))
p5 <- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "z-score",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p6 <- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "z-score",y = "genes",title = "with shrinkage") +
  theme_cowplot()
print(plot_grid(p5,p6))
```

And the next two bar charts show the overall impact of the shrinkage
on the $p$-values.

```{r shrink-vs-noshrink-pvalues, fig.height=2.75, fig.width=8, warning=FALSE}
pdat <- data.frame(noshrink = 10^(-de.noshrink$lpval[,2]),
                   shrink   = 10^(-de$lpval[,2]),
                   de       = factor(nonzero_lfc))
p7 <- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "p-value",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p8 <- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "p-value",y = "genes",title = "with shrinkage") +
  theme_cowplot()
print(plot_grid(p7,p8))
```

Both the $z$-score and $p$-value plots show that the adaptive
shrinkage step more strongly encourages the LFC estimates for the
non-DE genes toward zero, thereby improving the overall accuracy.

Quantifying uncertainty in the LFC estimates
--------------------------------------------

Above, we assessed the usefulness of applying the adaptive shrinkage
method to improve the accuracy of the LFC estimates. This idea of
shrinking the LFC estimates is not new, and is already implemented in
DESeq2, also using adaptive shrinkage.

A more important aspect of the DE analysis methods in fastTopics is
that it quantifies uncertainty in the LFC estimates by computing
posterior HPD intervals. To see why this is important, consider this
plot comparing point estimates (specifically, MLEs) of the LFCs (X
axis) against the shrunken posterior mean estimates (Y axis). What is
striking is that even very large (in magnitude) point estimates are
sometimes shrunk to zero; this happens because although the MLEs can
be far from zero, the HPD intervals are also very large. 

```{r point-vs-shrunken-scatterplot, fig.width=8, fig.height=2.5}
pdat <- data.frame(est       = de$est[,2],
                   shrink    = de$postmean[,2],
				   log10mean = log10(de$f0))
p9 <- ggplot(pdat,aes(x = est,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -4) +
  labs(x = "point estimates",
       y = "shrunken estimates") +
  theme_cowplot()
print(p9)
```

To illustrate the impact of this on the DE analysis, consider the K-L
divergence measure used in Dey, Hsiao & Stephens (2017) to rank genes,
a measure which doesn't fully account for uncertainty in the LFCs. As
a result, sometimes genes with very large LFCs get highly ranked, but
after a full accounting for uncertainty, they are no longer ranked
highly. (Since the K-L divergence is not a signed ranking measure like
the $z$-score, here we focus only on genes that are estimated to have
greater expression in topic 2 compared to topic 1.)

```{r kl-vs-fasttopics-1, fig.width=8, fig.height=2.5}
i     <- which(fit$F[,2] - fit$F[,1] > -1e-8)
D     <- fastTopics:::min_kl_poisson(fit$F)
pdat1 <- data.frame(kl = pmin(D[,2],0.00025),de = factor(nonzero_lfc))
pdat2 <- data.frame(z = clamp(de$z[,2],-4,+4),de = factor(nonzero_lfc))
pdat1 <- pdat1[i,]
pdat2 <- pdat2[i,]
p10 <- ggplot(pdat1,aes(x = kl,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_x_continuous(breaks = c(0,0.0001,0.0002)) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "K-L divergence",y = "genes") +
  theme_cowplot()
p11 <- ggplot(pdat2,aes(x = z,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "z-score",y = "genes") +
  theme_cowplot()
print(plot_grid(p10,p11))
```

To underscore this point, suppose we used the KL-divergences and the
*p*-values returned by `de_analysis(shrink.method = "ash")` to
discover differentially expressed genes. The power vs. FDR curves show
that the rankings accounting for uncertainty in the LFC estimates
(with or without shrinkage) perform better than the ranking based on
the KL divergence measure. We also see that shrinkage leads to a
modest improvement.

```{r kl-vs-fasttopics-2, fig.height, fig.width=4.5, fig.height=3}
pdat1 <- create_fdr_vs_power_curve(-D[,2],nonzero_lfc)
pdat2 <- create_fdr_vs_power_curve(-de.noshrink$lpval[,2],nonzero_lfc)
pdat3 <- create_fdr_vs_power_curve(-de$lpval[,2],nonzero_lfc)
pdat  <- rbind(cbind(pdat1,method = "kl"),
               cbind(pdat2,method = "noshrink"),
               cbind(pdat3,method = "shrink"))
p12 <- ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_point(size = 0.3)  +
  scale_color_manual(values = c("royalblue","limegreen","darkorange")) +
  theme_cowplot()
print(p12)
```

Comparison with DESeq2
----------------------

DESeq2 is a leading method for DE analysis with single-cell RNA
sequencing data. Here we apply DESeq2 to these same UMI count data,
and show that the fastTopics DE results are very similar to
DESeq2. Since DESeq2 assumes discrete cell types or conditions, we
apply DESeq2 to the subset of cells that were are almost entirely
(>99%) simulated from one topic; in this simulation, approximatmely
2/3 of the cells were simulated from one topic (entirely topic 1, or
entirely topic 2), and the remaining 1/3 were simulated from a mixture
of the two topics. There may be a small loss of accuracy in the DESeq2
results since DESeq2 is using only 2/3 of the data, but overall we
expect that the fastTopics DE analysis should closely replicate the
DESeq2 analysis.

```{r deseq-1}
single_topic <- apply(L,1,max) > 0.99
mean(single_topic)
```

These next few lines of code prepare the UMI count data for analysis
with DESeq2:

```{r deseq-2}
i <- which(single_topic)
Y <- t(X[i,])
cluster <- factor(apply(L,1,which.max))
deseq <- DESeqDataSetFromMatrix(Y,data.frame(cluster = cluster[i]),~cluster)
```

Now we perform the DE analysis using DESeq2, using the settings
recommended for single-cell RNA-seq data (see the DESeq2 vignette for
details). To replicate the fastTopics analysis as closely as possible,
we also shrink the LFC estimates using adaptive shrinkage.

```{r deseq-3}
sizeFactors(deseq) <- calculateSumFactors(Y)
deseq <- DESeq(deseq,test = "LRT",reduced=~1,useT = TRUE,minmu = 1e-6,
               minReplicatesForReplace = Inf)
deseq <- lfcShrink(deseq,coef = "cluster_2_vs_1",type = "ashr")
```

The posterior mean estimates from the fastTopics DE analysis closely
align with the DESeq2 estimates:

```{r fasttopics-vs-deseq2-lfc, fig.height=3, fig.width=3}
pdat <- data.frame(DESeq2     = deseq$log2FoldChange,
                   fastTopics = de$postmean[,2])
p14 <- ggplot(pdat,aes(x = DESeq2,y = fastTopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = "skyblue",linetype = "dotted") +
  ggtitle("LFC estimates") +
  theme_cowplot()
print(p14)
```

And likewise for the $z$-scores:

```{r fasttopics-vs-deseq2-zscores, fig.height=3, fig.width=3}
deseq$z <- with(deseq,log2FoldChange/lfcSE)
pdat <- data.frame(DESeq2     = deseq$z,
                   fastTopics = de$z[,2])
p15 <- ggplot(pdat,aes(x = DESeq2,y = fastTopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = "skyblue",linetype = "dotted") +
  ggtitle("z-scores") +
  theme_cowplot()
print(p15)
```

The $p$-value and LFSR distributions are also correspondingly similar
in shape:

```{r fasttopics-vs-deseq2-pvalues, fig.height=2.5, fig.width=8, warning=FALSE}
pdat <- data.frame(deseq2 = deseq$padj,
                   shrink = de$lfsr[,2],
                   de     = factor(nonzero_lfc))
p16 <- ggplot(pdat,aes(x = deseq2,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "p-value",y = "genes",title = "DESeq2") +
  theme_cowplot()
p17 <- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "lfsr",y = "genes",title = "fastTopics") +
  theme_cowplot()
print(plot_grid(p16,p17))
```

And therefore the performance of the two DE methods for discovering
differentially expressed genes is very similar, with the slight
advantage given to fastTopics (probably because it makde use of 1/3
more data):

```{r fasttopics-vs-deseq2-power, fig.height=3, fig.width=4.5}
pdat1 <- create_fdr_vs_power_curve(deseq$padj,nonzero_lfc)
pdat2 <- create_fdr_vs_power_curve(de$lfsr[,2],nonzero_lfc)
pdat  <- rbind(cbind(pdat1,method = "DESeq2"),
               cbind(pdat2,method = "fastTopics"))
p18 <- ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_point(size = 0.3)  +
  scale_color_manual(values = c("darkblue","tomato")) +
  theme_cowplot()
print(p18)
```
