---
title: "Differential expression analysis using a topic model: illustration in mixture of FACS-purified PBMC data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

The aim of this analysis is to understand the differences between a
classical differential expresion analysis comparing expression among
cell types (implemented using DESEq2) and a differential expression
analysis using the topic model, which allows for grades of membership
to cell types or cell states.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Begin by loading the packages and some function definitions used in
the analysis.

```{r load-pkgs, message=FALSE, warning=FALSE}
library(Matrix)
library(DESeq2)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source("../code/de_analysis_functions.R")
```

Load the UMI gene count data.

```{r load-data}
load("../data/pbmc_purified.RData")
```

Load the $K = 6$ Poisson NMF model fit, and convert it to a
topic model.

```{r load-fit}
fit <- readRDS(file.path("../output/pbmc-purified/rds",
                         "fit-pbmc-purified-scd-ex-k=6.rds"))$fit
fit <- poisson2multinom(fit)
```

The cells are subdivided, based on FACS sorting, into 10 "cell types",
almost many of the cell types are virtually indistinguishable based on
their gene expression, so we combine them into a single "T cell"
category. This results in 5 predefined cell types:

```{r cell-types}
set.seed(1)
celltype <- as.character(samples$celltype)
celltype[celltype == "CD4+ T Helper2" |
         celltype == "CD4+/CD45RO+ Memory" |
         celltype == "CD8+/CD45RA+ Naive Cytotoxic" |
         celltype == "CD4+/CD45RA+/CD25- Naive T" |
         celltype == "CD8+ Cytotoxic T" |
         celltype == "CD4+/CD25 T Reg"] <- "T cell"
celltype <- factor(celltype,
                   c("CD19+ B","CD14+ Monocyte","CD34+","CD56+ NK","T cell"))
table(celltype)
```

This Structure plot, in which cells are arranged horizontally
according to their predefined cell type, shows how the estimated topic
proportions relate to the predefined cell types.

```{r structure-plot, fig.width=8, fig.height=1.8}
topic_colors <- c("gold","forestgreen","dodgerblue","gray",
                  "darkmagenta","violet")
topics <- c(5,3,2,4,1,6)
rows <- sort(c(sample(which(celltype == "CD19+ B"),500),
               sample(which(celltype == "CD14+ Monocyte"),250),
               sample(which(celltype == "CD34+"),500),
               sample(which(celltype == "CD56+ NK"),400),
               sample(which(celltype == "T cell"),1000)))
p1 <- structure_plot(select_loadings(fit,loadings = rows),
                     grouping = celltype[rows],
                     topics = topics,colors = topic_colors[topics],
   	  			     perplexity = c(70,30,30,30,70),n = Inf,gap = 30,
					 num_threads = 4,verbose = FALSE)
print(p1)
```

In particular, topics 1 through 4 closely correspond, respectively, to
T cells, CD14+ monocytes (myeloid cells), B cells and natural killer
cells. Topic 5 (violet) also closely corresponds to the "CD34+" FACS
cell type label but also suggests considerable FACS FACS mislabeling
of the CD34+ cells. Topic 6 (magenta) does not correspond to any FACS
cell type and as we will it is capturing a different characteristic of
the cells---specifically, abundance of ribosomal protein genes.
Therefore, we expect the results from the topic-model-based DE
analysis for topics 1--4 to most closely resemble the DESeq2 results
for the corresponding cell types.

Before entering into this comparison, we first assess accuracy of the
MCMC computations in the topic-model-based analysis.

Assessing accuracy of the Monte Carlo estimates
-----------------------------------------------

To assess accuracy of the posterior calculations, we perform the DE
analysis twice (using different seeds), and compare the posterior mean
estimates and *z*-scores returned by the two `de_analysis` runs.

```{r load-de-analysis}
load("../output/pbmc-purified/de-pbmc-purified-seed=1.RData")
de1 <- de
load("../output/pbmc-purified/de-pbmc-purified-seed=2.RData")
de2 <- de
rm(de)
```

The two MCMC estimates of the posterior mean log-fold change (LFC)
estimates are largely consistent:

```{r assess-monte-carlo-1, fig.width=3, fig.height=3}
pdat <- data.frame(postmean1 = as.vector(de1$postmean),
                   postmean2 = as.vector(de2$postmean))
ggplot(pdat,aes(x = postmean1,y = postmean2)) +
  geom_point(shape = 21,color = "white",fill = "black",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dashed") +
  labs(x = "first posterior mean",y = "second posterior mean") +
  theme_cowplot()
```

The $z$-scores on the other hand are estimated less consistently,
presumably because accurately calculating uncertainty in the estimates
is harder. Despite this, the $z$-scores are still are consistent
enough in that it is rare for an LFC to have an lfsr less than 0.05 in
one MCMC simulation and not the other (these are the red points in the
scatterplot).

```{r assess-monte-carlo-3, fig.width=4.5, fig.height=3, warning=FALSE}
pdat <- data.frame(z1 = clamp(as.vector(de1$z),-100,+100),
                   z2 = clamp(as.vector(de2$z),-100,+100),
				   lfsr = factor((de1$lfsr < 0.05) + (de2$lfsr < 0.05)))
ggplot(pdat,aes(x = z1,y = z2,fill = lfsr)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_manual(values = c("darkblue","tomato","dodgerblue"),
                    na.value = "white") +
  labs(x = "first z-score",y = "second z-score",fill = "lfsr < 0.05") +
  theme_cowplot()
```

Note that for better visualization $z$-scores larger than 100 (or
smaller than -100) are shown as 100 (or -100).

*Add text here.*

```{r fix-z-scores}
de <- de1[c("f0","postmean","z","lfsr")]
class(de) <- c("topic_model_de_analysis","list")
i  <- which(abs(de2$z) < abs(de1$z))
de$postmean[i] <- de2$postmean[i]
de$z[i]        <- de2$z[i]
de$lfsr[i]     <- de$lfsr[i]
```

DESeq2 vs. fastTopics
---------------------

*Next:* Compare classical DE analysis (DESeq2) against DE analysis
allowing for grades of membership.

*Add text here.*

```{r load-deseq2}
load("../output/pbmc-purified/deseq2-pbmc-purified.RData")
celltypes <- names(deseq)
n <- length(celltypes)
p <- nrow(genes)
deseq2 <- list(postmean = matrix(0,p,n),
               z        = matrix(0,p,n))
rownames(deseq2$postmean) <- genes$ensembl
rownames(deseq2$z)        <- genes$ensembl
colnames(deseq2$postmean) <- celltypes
colnames(deseq2$z)        <- celltypes
for (i in 1:n) {
  deseq2$postmean[,i] <- deseq[[i]]$log2FoldChange
  deseq2$z[,i]        <- with(deseq[[i]],log2FoldChange/lfcSE)
}
deseq <- deseq2
rm(deseq2)
```

*Add text here.*

```{r align-with-deseq2}
rows        <- match(rownames(deseq$z),rownames(de$z))
de$f0       <- de$f0[rows]
de$postmean <- de$postmean[rows,]
de$z        <- de$z[rows,]
de$lfsr     <- de$lfsr[rows,]
```

*Add text here.*

```{r z-score-hist}
pdat1 <- data.frame(x = clamp(as.vector(deseq$z),-40,+40))
pdat2 <- data.frame(x = clamp(as.vector(de$z),-40,+40))
p1 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(bins = 64,fill = "black") +
  labs(x = "z-score",y = "genes",title = "DESeq2") +
  theme_cowplot()
p2 <- ggplot(pdat1,aes(x = x)) +
  geom_histogram(bins = 64,fill = "black") +
  labs(x = "z-score",y = "genes",title = "fastTopics") +
  theme_cowplot()
plot_grid(p1,p2)
```

*Add text here.*

```{r z-score-scatterplot}
pdat <-
  data.frame(deseq     =quantile(deseq$z,seq(0,1,length.out=1000),na.rm=TRUE),
             fasttopics=quantile(de$z,seq(0,1,length.out=1000),na.rm=TRUE))
pdat <- transform(pdat,
                  deseq      = clamp(deseq,-100,+100),
                  fasttopics = clamp(fasttopics,-100,+100))
ggplot(pdat,aes(x = deseq,y = fasttopics)) +
  geom_point() +
  geom_abline(intercept = 0,slope = 1,color = "magenta",linetype = "dotted") +
  labs(x = "DESeq2",y = "fastTopics") +
  theme_cowplot()
```

*Add text here.*

```{r bcells-z-scores}
pdat <- data.frame(deseq        = deseq$postmean[,"CD19+ B"],
                   fasttopics   = de$postmean[,"k3"],
				   z.deseq      = deseq$z[,"CD19+ B"],
				   z.fasttopics = de$z[,"k3"],
				   lfsr         = -log10(de$lfsr[,"k3"] + 1e-20),
                   f0           = log10(de$f0),
                   gene         = genes$symbol,
				   stringsAsFactors = FALSE)
pdat <- transform(pdat,lfsr = cut(lfsr,c(0,1.3,2,3,Inf)))
i <- which(pdat$fasttopics < 8)
pdat[i,"gene"] <- ""
pdat[genes$symbol == "HOPX","gene"] <- "HOPX"
pdat <- subset(pdat,abs(z.deseq) > 2 | abs(z.fasttopics) > 2)
ggplot(pdat,aes(x = deseq,y = fasttopics,fill = lfsr,label = gene)) +
  geom_point(shape = 21,color = "white") +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  geom_text_repel(color = "darkgray",size = 2.25,fontface = "italic",
                  segment.color = "darkgray",segment.size = 0.25,
                  min.segment.length = 0,max.overlaps = Inf,na.rm = TRUE) +
  scale_fill_manual(values = c("deepskyblue","gold","orange","tomato"),
                    na.value = "gainsboro") +
  labs(x = "DESeq2",y = "fastTopics",
       title = "LFC in B cells") +
  theme_cowplot()
```

*Add text here.*

```{r bcells-volcano-plot}
pdat <- data.frame(gene     = genes$symbol,
                   postmean = de$postmean[,"k3"],
                   z        = abs(de$z[,"k3"]),
				   lfsr     = -log10(de$lfsr[,"k3"]),
				   stringsAsFactors = FALSE)
pdat <- transform(pdat,lfsr = cut(lfsr,c(0,1.3,2,3,Inf)))
i <- which(with(pdat,!(postmean > 8 | z > 75 | (postmean > 5 & z > 20))))
pdat[i,"gene"] <- ""
ggplot(pdat,aes(x = postmean,y = z,fill = lfsr,label = gene)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_text_repel(color = "darkgray",size = 2.25,fontface = "italic",
                  segment.color = "darkgray",segment.size = 0.25,
                  min.segment.length = 0,max.overlaps = Inf,na.rm = TRUE) +
  scale_y_continuous(trans = "sqrt") +
  scale_fill_manual(values = c("deepskyblue","gold","orange","tomato"),
                    na.value = "gainsboro") +
  labs(x = "posterior mean LFC",y = "|z-score|") +
  theme_cowplot()
```

