---
title: "DE analysis using a topic model: evaluation using simulated data"
author: Peter Carbonetto
output: workflowr::wflow_html
---

*Add summary of the analysis here.*

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.show = "hold",fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, and some additional
functions for simulating the data.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(fastTopics)
library(MCMCpack)
library(ggplot2)
library(cowplot)
source("../code/de_eval_functions.R")
```

Simulate UMI counts
-------------------

We begin by simulating counts intended to "mimic" UMI count data from
a single-cell RNA sequencing experiment. In particular, we simulate
counts $x_{ij}$ from a Poisson NMF model $x_{ij} \sim
\mathrm{Poisson}(\lambda_{ij})$, such that $\lambda_{ij} =
\sum_{k=1}^K l_{ik} f_{jk}$. We focus our attention on the $K = 2$
case to simplify the analysis; comparing gene expression between three
or topics brings some additional complications which aren't necessary
for assessing basic properties of the new DE methods. The simulation
of the topic proportions $l_{ik}$ and the gene expression rates
$f_{jk}$ is described in the comments accompanying function
`simulate_twotopic_umi_data`.

```{r simulate-counts}
set.seed(1)
dat <- simulate_twotopic_umi_data()
X <- dat$X
```

The sample sizes (total counts for each cell) in the simulated data
are roughly normally distributed on the (base-10) log scale with
mean 3.2 and standard deviation 0.2:

```{r sizes, fig.height=2.5, fig.width=3}
s <- rowSums(X)
p1 <- ggplot(data.frame(s = log10(s)),aes(x = s)) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "log10 size",y = "cells") +
  theme_cowplot()
print(p1)
mean(log10(s))
sd(log10(s))
```

We have generated the expression rates so that they are roughly
uniformly distributed on the log scale:

```{r gene-expression-rates, fig.height=2.5, fig.width=3}
p2 <- ggplot(data.frame(f = as.vector(dat$F)),aes(x = log10(f))) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "log10 expression rate",y = "genes") +
  theme_cowplot()
print(p2)
```

About half of the genes have differences in expression between the two
topics, and among the nonzero gene expression differences, the LFCs
are roughly *t*-distibuted with 3 degrees of freedom:

```{r lfc-density, fig.height=2.5, fig.width=3}
lfc <- log2(dat$F[,1]/dat$F[,2])
p3 <- ggplot(data.frame(lfc = lfc[lfc != 0]),aes(x = lfc)) +
  geom_histogram(color = "white",fill = "black",bins = 32) +
  labs(x = "LFC",y = "genes") +
  theme_cowplot()
print(p3)
mean(lfc == 0)
```
