---
title: "Evaluation of DE analysis with more than two topics"
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we expand the evaluation of the new topic-model-based DE analysis
methods to consider the case of more than two topics. When comparing
multiple groups, clusters or topics it is less clear what it means for
a gene to be "differentially expressed". So to simplify the evaluation
we simulate the data so that each gene is differentially expressed in
at most one topic.

In this evaluation can no longer compare to DESeq2 because DESeq2 is
designed to compare gene expression across two groups. Still, we can
closely examine the benefit of accounting for uncertainty.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.show = "hold",fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, and some additional
functions for simulating the data.

```{r load-pkgs, message=FALSE, warning=FALSE}
library(Matrix)
library(fastTopics)
library(ggplot2)
library(cowplot)
source("../code/de_analysis_functions.R")
```

Simulate UMI counts
-------------------

We begin by simulating UMI count data from a rank-6 Poisson NMF model
such that roughly half of the genes have the same expression rate
across all topics, and the other half have a different rate of
expression in exactly one out of the six topics. The details of the
simulation are similar to the two-topic case.

```{r simulate-counts}
set.seed(1)
m <- 10000
k <- 6
dat <- simulate_manytopic_umi_data(m = m,k = k)
X <- dat$X
F <- dat$F
L <- dat$L
```

Fit multinomial topic model to UMI counts
-----------------------------------------

Now we fit a multinomial topic model to the simulated UMI count data.
To simplify evaluation, we assume that the topic proportions are
known, and fix them to their ground-truth values. In this way, the
only error that can arise is in the estimates of the expression rates
$f_{ij}$.

```{r fit-poisson-nmf, warning=FALSE, message=FALSE, cache=TRUE}
fit0 <- init_poisson_nmf(X,F = dat$F,L = with(dat,s*L))
fit <- fit_poisson_nmf(X,fit0 = fit0,numiter = 40,method = "scd",
                       update.loadings = NULL,verbose = "none")
fit <- poisson2multinom(fit)
summary(fit)
```

DE analysis with and without shrinkage
--------------------------------------

First we perform a DE analysis *without* shrinking the LFC estimates:

```{r de-noshrink, message=FALSE, cache=TRUE}
set.seed(1)
de.noshrink <- de_analysis(fit,X,shrink.method = "none",
                           control = list(ns = 10000,nc = 2))
```

Next we perform a second DE analysis using adaptive shrinkage to
shrink the LFC estimates:

```{r de-with-ash, message=FALSE, cache=TRUE}
set.seed(1)
de <- de_analysis(fit,X,shrink.method = "ash",control = list(ns = 1e4,nc = 2))
```

To measure differential expression across multiple topics, we use the
"least extreme" log-fold change, which, for topic $k$, is defined as
the LFC between topic $k$ and $j \neq k$ such that the expression
ratio is closest to 1. Then we compare estimates of the "least
extreme" LFCs with and without shrinkage.

As expected, smaller estimates, and estimates corresponding to genes
with lower expression, tend to be more strongly shrunk toward zero.

```{r shrink-vs-noshrink-scatterplot, fig.height=2.75, fig.width=4.5}
pdat <- data.frame(noshrink  = c(de.noshrink$postmean),
                   shrink    = c(de$postmean),
                   log10mean = rep(log10(de$f0),k))
ggplot(pdat,aes(x = noshrink,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -4) +
  labs(x = "original LFC estimates",
       y = "shrunken LFC estimates") +
  theme_cowplot()
```

The next two bar charts show the overall impact of the shrinkage on
the distribution of the $z$-scores; dark blue bars are for
diifferentially expressed genes, and orange bars are for
non-differentially expressed genes. Note that for visualization
purposes the $z$-scores are projected onto the interval $[-4,+4]$.

```{r shrink-vs-noshrink-zscores, fig.height=5, fig.width=8, warning=FALSE}
nonzero_lfc <- matrix(FALSE,m,k)
for (i in 1:m) {
  y <- dat$F[i,]
  if (max(y) - min(y) > 1e-8) {
    j <- which.max((y - mean(y))^2)
    nonzero_lfc[i,j] <- TRUE
  }
}
pdat <- data.frame(noshrink = clamp(c(de.noshrink$z),-4,+4),
                   shrink   = clamp(c(de$z),-4,+4),
                   de       = c(nonzero_lfc))
p1 <- ggplot(subset(pdat,de),aes(x = noshrink,color = de)) +
  geom_histogram(color = "darkblue",fill = "darkblue",bins = 64) +
  coord_cartesian(xlim = c(-4,+4),ylim = c(0,400)) +
  labs(x = "z-score",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p2 <- ggplot(subset(pdat,de),aes(x = shrink,color = de)) +
  geom_histogram(color = "darkblue",fill = "darkblue",bins = 64) +
  coord_cartesian(xlim = c(-4,+4),ylim = c(0,400)) +
  labs(x = "z-score",y = "genes",title = "with shrinkage") +
  theme_cowplot()
p3 <- ggplot(subset(pdat,!de),aes(x = noshrink,color = de)) +
  geom_histogram(color = "darkorange",fill = "darkorange",bins = 64) +
  coord_cartesian(xlim = c(-4,+4)) +
  labs(x = "z-score",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p4 <- ggplot(subset(pdat,!de),aes(x = shrink,color = de)) +
  geom_histogram(color = "darkorange",fill = "darkorange",bins = 64) +
  coord_cartesian(xlim = c(-4,+4)) +
  labs(x = "z-score",y = "genes",title = "with shrinkage") +
  theme_cowplot()
print(plot_grid(p1,p2,p3,p4,nrow = 2,ncol = 2))
```

And the next two bar charts show the overall impact of the shrinkage
on the $p$-values. (The Y axis in the "with shrinkage" plot has been
cut off at 2,500 genes to show in more detail the smaller $p$-values
identifying true positives.)

```{r shrink-vs-noshrink-pvalues, fig.height=2.75, fig.width=8, warning=FALSE}
pdat <- data.frame(noshrink = 10^(-c(de.noshrink$lpval)),
                   shrink   = 10^(-c(de$lpval)),
                   de       = factor(c(nonzero_lfc)))
p1 <- ggplot(pdat,aes(x = noshrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  coord_cartesian(ylim = c(0,2500)) +
  labs(x = "p-value",y = "genes",title = "without shrinkage") +
  theme_cowplot()
p2 <- ggplot(pdat,aes(x = shrink,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  coord_cartesian(ylim = c(0,2500)) +
  labs(x = "p-value",y = "genes",title = "with shrinkage") +
  theme_cowplot()
print(plot_grid(p1,p2))
```

Since most of the LFCs are zero, the adaptive shrinkage strongly
shrinks most of the LFC estimates toward zero. Still, because it is
adaptive, the DE genes with stronger support are shrunk only a little
or not at all.

Quantifying uncertainty in the LFC estimates
--------------------------------------------

Here we briefly assess the benefits of accounting for uncertainty in
the LFC estimates. The benefits are more pronounced here because more
topics brings more uncertainty to the estimates.

First we compare the point estimates to the (shrunk) posterior mean
estimates:

```{r point-vs-shrunken-scatterplot, fig.width=8, fig.height=2.5}
pdat <- data.frame(est       = c(de$est),
                   shrink    = c(de$postmean),
				   log10mean = rep(log10(de$f0),k))
ggplot(pdat,aes(x = est,y = shrink,fill = log10mean)) +
  geom_point(shape = 21,color = "white",size = 2) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_fill_gradient2(low = "deepskyblue",mid = "gold",high = "orangered",
                       midpoint = -4) +
  labs(x = "point estimates",
       y = "shrunken estimates") +
  theme_cowplot()
```

As before, even very large (in magnitude) point estimates are
sometimes shrunk to zero; this happens because although the MLEs can
be far from zero, the HPD intervals can also be very large.

Now we compare a ranking of the LFCs based on the K-L divergence
measure used in Dey, Hsiao & Stephens (2017), which doesn't fully
account for uncertainty in the LFC estimates, to a ranking based on
the $z$-scores. Since the K-L divergence is not a signed ranking, here
we restrict our attention to LFCs estimated to be positive. (Note that
the Y axis in the right-hand plot has been cut off at 2,000 genes to
give more detail to the distribution of the true positives. Also, for
better visualization the K-L divergences are shown on the log scale.)

```{r kl-vs-fasttopics-1, fig.width=8, fig.height=2.5}
D     <- fastTopics:::min_kl_poisson(fit$F)
i     <- which(de$est > 0)
pdat1 <- data.frame(lkl = log10(D[i] + 1e-8),de = factor(nonzero_lfc[i]))
pdat2 <- data.frame(z = clamp(de$z[i],-4,+4),de = factor(nonzero_lfc[i]))
p1 <- ggplot(pdat1,aes(x = lkl,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  labs(x = "log10 K-L divergence",y = "genes") +
  theme_cowplot()
p2 <- ggplot(pdat2,aes(x = z,color = de,fill = de)) +
  geom_histogram(bins = 64) +
  scale_color_manual(values = c("darkorange","darkblue")) +
  scale_fill_manual(values = c("darkorange","darkblue")) +
  coord_cartesian(ylim = c(0,2000)) +
  labs(x = "z-score",y = "genes") +
  theme_cowplot()
print(plot_grid(p1,p2))
```

Sometimes genes with very large LFCs get highly ranked by the K-L
divergence measure (while being uncertain about these estimates),
whereas this happens less often for the $z$-scores. As a result the
fastTopics $z$-scores yield improved power at low false discovery
rates:

```{r kl-vs-fasttopics-2, fig.width=4.5, fig.height=3}
pdat1 <- create_fdr_vs_power_curve(-D[i],nonzero_lfc[i],length.out = 400)
pdat2 <- create_fdr_vs_power_curve(de$lfsr[i],nonzero_lfc[i],length.out = 400)
pdat  <- rbind(cbind(pdat1,method = "kl"),
               cbind(pdat2,method = "fastTopics"))
ggplot(pdat,aes(x = fdr,y = power,color = method)) +
  geom_line(size = 0.65,orientation = "y")  +
  scale_color_manual(values = c("royalblue","darkorange")) +
  theme_cowplot()
```

Alternatively we can compare the two rankings in an ROC plot:

```{r kl-vs-fasttopics-3, fig.width=4.5, fig.height=3}
pdat1 <- create_roc_curve(-log(D[i]),nonzero_lfc[i],length.out = 100)
pdat2 <- create_roc_curve(de$lfsr[i],nonzero_lfc[i],length.out = 100)
pdat  <- rbind(cbind(pdat1,method = "kl"),
               cbind(pdat2,method = "fastTopics"))
ggplot(pdat,aes(x = fpr,y = tpr,color = method)) +
  geom_line(size = 0.75) +
  geom_abline(intercept = 0,slope = 1,color = "black",linetype = "dotted") +
  scale_color_manual(values = c("royalblue","darkorange")) +
  theme_cowplot()
```
