---
title: Perform differential expression analysis on pulse-seq topics and clusters
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we perform a differential expression analysis using the topic
model fit to the pulse-seq data, as well as the clusters identified
from this topic model.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(dplyr)
library(fastTopics)
library(tools)
library(ggplot2)
library(ggrepel)
library(cowplot)
source("../code/plots.R")
```

Load data and results
---------------------

Load the pulse-seq data, the $k = 11$ Poisson NMF model fit, and the
clusters identified in the [clustering analysis](clusters_pulseseq.html).

```{r load-data}
load("../data/pulseseq.RData")
fit <- readRDS("../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds")$fit
samples <- readRDS("../output/pulseseq/clustering-pulseseq.rds")
```

Perform differential expression analysis
----------------------------------------

Perform differential expression analysis using the multinomial topic
model.

```{r diff-count-analysis-topics}
timing <- system.time(
  diff_count_topics <- diff_count_analysis(fit,counts))
cat(sprintf("Computation took %0.2f seconds.\n",timing["elapsed"]))
```

Next, calculate differential expression statistics using the clusters
that were identified from the topic proportion PCs.

```{r diff-count-analysis-clusters}
fit_clusters <- init_poisson_nmf_from_clustering(counts,samples$cluster)
timing <- system.time(
  diff_count_clusters <- diff_count_analysis(fit_clusters,counts))
cat(sprintf("Computation took %0.2f seconds.\n",timing["elapsed"]))
```

Finally, perform a differential expression analysis after removing the
"P" cluster.

```{r diff-count-analysis-topics-hillock}
rows <- which(samples$cluster != "P")
fit_hillock <- select(poisson2multinom(fit),loadings = rows)
timing <- system.time(
  diff_count_hillock <- diff_count_analysis(fit_hillock,counts[rows,]))
cat(sprintf("Computation took %0.2f seconds.\n",timing["elapsed"]))
```

Finally, perform another differential expression analysis after
merging the 5 club cell topics and the 3 basal cell topics, and after
removing the "P" cluster.

```{r diff-count-analysis-topics-basal-club}
fit_bc <- merge_topics(fit_hillock,c("k4","k5","k6","k8","k10"))
fit_bc <- merge_topics(fit_bc,c("k1","k3","k9"))
timing <- system.time(
  diff_count_bc <- diff_count_analysis(fit_bc,counts[rows,]))
cat(sprintf("Computation took %0.2f seconds.\n",timing["elapsed"]))
```

Topics capture shared gene expression patterns
----------------------------------------------

In some cases, the topics capture individual cell-types, and in other
cases they capture gene expression patterns common to multiple
cell-types.

```{r volcano-plots, fig.width=7.5, fig.height=5, message=FALSE}
basal_genes    <- c("Aqp3","Krt5","Dapl1","Hspa1a","Trp63")
ciliated_genes <- c("Ccdc113","Ccdc153","Cdhr3","Foxj1","Lztfl1","Mlf1")
club_genes     <- c("Bpifa1","Cbr2","Cyp2a5","Cyp2f2","Krt15",
                    "Lypd2","Muc5b","Nfia","Scgb1a1","Scgb3a2")
goblet_genes   <- "Gp2"
hillock_genes  <- c("Anxa1","Cldn3","Ecm1","Krt13","Krt4","Lgals3","S100a11")
ionocyte_genes <- c("Ascl3","Asgr1","Atp6v0d2","Atp6v1c2","Cftr","Foxi1",
                    "Moxd1","P2ry14","Stap1")
tuft_neuroendocrine_genes <- c("Ascl1","Ascl2","Ascl3","Chga","Dclk1",
                               "Gnat3","Rgs13")
p1 <- volcano_plot_with_highlighted_genes(diff_count_bc,"k7",
        ciliated_genes,label_above_quantile = Inf) +
		guides(fill = "none") +
		ggtitle("topic 7 (cilitated)")
p2 <- volcano_plot_with_highlighted_genes(diff_count_bc,"k1+k3+k9",
        basal_genes,label_above_quantile = Inf) +
		guides(fill = "none") +
		ggtitle("topics 1, 3, 9 (basal)")
p3 <- volcano_plot_with_highlighted_genes(diff_count_bc,"k4+k5+k6+k8+k10",
        club_genes,label_above_quantile = Inf) +
		guides(fill = "none") +
		ggtitle("topics 4-6, 8, 10 (club)")
p4 <- volcano_plot_with_highlighted_genes(diff_count_bc,"k2",
        c(ionocyte_genes,tuft_neuroendocrine_genes),
        label_above_quantile = Inf) +
		guides(fill = "none") +
		ggtitle("topic 2 (tuft + PNEC + ionocyte)")
p5 <- volcano_plot_with_highlighted_genes(diff_count_hillock,"k1",
        hillock_genes,label_above_quantile = Inf) +
		guides(fill = "none") +
		ggtitle("topic 1 (hillock)")
plot_grid(p1,p2,p3,p4,p5,nrow = 2,ncol = 3)
```

Save results
------------

Save the results of the differential expression analyses to an RData
file.

```{r save-results}
save(list = c("diff_count_topics",
              "diff_count_clusters",
              "diff_count_hillock",
			  "diff_count_bc"),
     file = "diff-count-pulseseq.RData")
resaveRdaFiles("diff-count-pulseseq.RData")
```
