---
title: Visualize and interpret droplet and pulse-seq topics
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we closely examine the topic modeling results for the two
epithelial airway data sets (droplet and pulse-seq), and investigate
the benefits of modeling single cells as *mixtures* of gene expression
programs in these data sets. In particular, we will compare and
contrast differential expression in clusters vs. topics.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, as well as additional
functions that we will use to generate some of the plots.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(dplyr)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source("../code/plots.R")
```

Load data and results
---------------------

Load the smaller "droplet" data set, the $k = 7$ Poisson NMF model fit
for these data, the 8 clusters identified in the
[clustering analysis](clusters_droplet.html), and the results of the
differential expression analysis.

```{r load-droplet-data}
load("../data/droplet.RData")
load("../output/droplet/diff-count-droplet.RData")
counts_droplet <- counts
samples_droplet <- readRDS("../output/droplet/clustering-droplet.rds")
fit_droplet <- readRDS("../output/droplet/rds/fit-droplet-scd-ex-k=7.rds")$fit
diff_count_droplet <- diff_count_topics
diff_count_clusters_droplet <- diff_count_clusters
diff_count_merge_droplet <- diff_count_merge_club
rm(samples,counts)
rm(diff_count_topics,diff_count_clusters,diff_count_merge_club)
```

For reference, we show here the Structure plot from the
[clustering analysis of the droplet data](clusters_droplet.html). This
Structure plot summarizes the topic proportions in each of the 8
subsets (including the background cluster).

![](figure/clusters_droplet.Rmd/structure-plot-1.png)

Next, we load the larger "pulse-seq" data set, the $k = 11$ Poisson
NMF model fit for these data, and the 7 clusters identified in the
[clustering analysis](clusters_pulseseq.html).

```{r load-pulseseq-data}
load("../data/pulseseq.RData")
load("../output/pulseseq/diff-count-pulseseq.RData")
counts_pulseseq  <- counts
samples_pulseseq <- readRDS("../output/pulseseq/clustering-pulseseq.rds")
fit_pulseseq <- readRDS("../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds")$fit
diff_count_pulseseq <- diff_count_topics
diff_count_clusters_pulseseq <- diff_count_clusters
diff_count_merge_pulseseq <- diff_count_merge_bc
rm(samples,counts)
rm(diff_count_topics,diff_count_clusters,diff_count_merge_bc)
```

For reference, we show here the Structure plot from the
[clustering analysis of the pulse-seq data](clusters_pulseseq.html).
This Structure plot summarizes the topic proportions in each of the 7
subsets (including the background cluster):

![](figure/clusters_pulseseq.Rmd/structure-plot-1.png)

Ciliated cells
--------------

We begin with the cluster that captures ciliated cells. This cluster
is one of the most distinctive in both the droplet and pulse-seq data
sets. Ciliated cells are abundant, though not as much as basal and
club cells.

```{r volcano-plot-droplet-cil, fig.width=4.5, fig.height=4}
ciliated_genes <- c("Ccdc113","Ccdc153","Cdhr3","Foxj1","Lztfl1","Mlf1")
p1 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"Cil",
                                          ciliated_genes,
										  label_above_quantile = 0.998)
print(p1)
```

Marker genes and transcription factors identified in Montoro *et al*
(2018) are highlighted in black, and other top differentially
expressed genes are shown with gray labels.

We obtain similar top differentially expressed genes in the pulse-seq
data:

```{r volcano-plot-pulseseq-cil, fig.width=4.5, fig.height=4}
p2 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"Cil",
                                          ciliated_genes,
										  label_above_quantile = 0.998)
print(p2)
```

The topic for the ciliated cell-type ($k = 7$) corresponds very
closely to the "Cil" cluster in the pulse-seq data. In addition, gene
enrichments are considerably stronger in the topic for ciliated genes,
particularly so for some genes with lower expression levels.

TO DO: Show z-scores in scatterplot on sqrt scale.

```{r scatterplots-pulseseq-Cil-vs-k7, fig.width=8, fig.height=3}
p3 <- zscores_scatterplot(diff_count_clusters_pulseseq,
                          diff_count_pulseseq,"Cil","k7",ciliated_genes,
						  label_above_score = Inf,zmax = 750) +
  labs(x = "cluster Cil",y = "topic 7",title = "z-scores")
p4 <- beta_scatterplot(diff_count_clusters_pulseseq,diff_count_pulseseq,
                       "Cil","k7",ciliated_genes,label_above_score = Inf) +
  labs(x = "cluster Cil",y = "topic 7",title = "log-fold change (\u03b2)")
plot_grid(p3,p4)
```

Ionocytes
---------

In the pulse-seq data, we identify a distinctive cluster for the newly
discovered, rare "ionocyte" cell type. Gene expression in these cells
is not fully captured by any single topic, yet the mixture of topics
forms a distinctive cluster.

```{r volcano-plot-pulseseq-I, fig.width=4.4, fig.height=4}
ionocyte_genes <- c("Ascl3","Asgr1","Atp6v0d2","Atp6v1c2","Cftr","Foxi1",
                    "Moxd1","P2ry14","Stap1")
p5 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"I",
                                          ionocyte_genes,
										  label_above_quantile = 0.998)
print(p5)
```

We do not identify a cluster for ionocytes in the droplet
data. Judging by expression of the *Foxi1* ionocyte marker gene, only
a handful of cells in the droplet data are ionocytes:

```{r pca-plot-droplet-Foxi1, fig.width=3.75, fig.height=3}
p6 <- pca_plot_with_counts(fit_droplet,counts_droplet[,"Foxi1"],1:2)
print(p6)
```

Goblet cells
------------

Consistent with Montoro *et al* (2018), we identify a cluster of
Goblet cells in the droplet data.

```{r volcano-plot-droplet-G, fig.width=4.5, fig.height=4}
goblet_genes <- "Gp2"
p7 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"G",
                                          goblet_genes,
										  label_above_quantile = 0.998)
print(p7)
```

Topic $k = 1$ is unique to this cluster suggesting that the this topic
characterizes the goblet cell type. Indeed, there is a very close
correspondence between the topic and cluster, with several
characteristic genes (e.g. *Gp2*) showing somewhat stronger enrichment
in the topic.

```{r scatterplots-droplet-G-vs-k1, fig.width=8, fig.height=3}
p8 <- zscores_scatterplot(diff_count_clusters_droplet,
                          diff_count_droplet,"G","k1",goblet_genes,
						  label_above_score = 200,zmax = 800) +
  labs(x = "cluster G",y = "topic 1",title = "z-scores")
p9 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                       "G","k1",goblet_genes,label_above_score = Inf) +
  labs(x = "cluster G",y = "topic 1",title = "log-fold change (\u03b2)")
plot_grid(p8,p9)
```

Tuft and pulmonary neuroendocrine cells
---------------------------------------

In both data sets, we identify clusters for tuft and pulmonary
neuroendocrine cells. The topic models do not clearly distinguish
between these two rare cell types; we identify them in a single
cluster.

```{r volcano-plot-droplet-T+N, fig.width=4.5, fig.height=4}
tuft_neuroendocrine_genes <- c("Ascl1","Ascl2","Ascl3","Chga","Dclk1","Rgs13")
p8 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"T+N",
                                          tuft_neuroendocrine_genes,
										  label_above_quantile = 0.998)
print(p8)
```

Here is the volcano plot from the pulse-seq data:

```{r volcano-plot-pulseseq-T+N, fig.width=4.5, fig.height=4}
p9 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"T+N",
                                          tuft_neuroendocrine_genes,
										  label_above_quantile = 0.998)
print(p9)
```

Basal cells
-----------

We end our exploration with the large subset of cells that does not
decompose to form distinct clusters.

Abundant basal cells in droplet data:

```{r, fig.width=4.5, fig.height=4}
basal_genes <- c("Aqp3","Krt5","Dapl1","Hspa1a","Trp63")
p10 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"B",
                                           basal_genes,
	 								 	   label_above_quantile = 0.995)
print(p10)
```

Compare basal topic ($k = 2$) against basal cluster in droplet data:

```{r, fig.width=4, fig.height=3}
p11 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                        "B","k2",basal_genes,label_above_score = Inf) +
  labs(x = "cluster B",y = "topic 2",title = "log-fold change (\u03b2)")
print(p11)
```

Basal cells in pulse-seq data:

```{r, fig.width=4.5, fig.height=4}
p12 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"B",
                                           basal_genes,
	 								 	   label_above_quantile = 0.995)
print(p12)
```

Compare combined basal topics ($k = 1, 3, 9$) against cluster in
pulse-seq data:

```{r, fig.width=4, fig.height=3}
p13 <- beta_scatterplot(diff_count_clusters_pulseseq,
                        diff_count_merge_pulseseq,
                        "B","k1+k3+k9",basal_genes,label_above_score = Inf) +
  labs(x = "cluster B",y = "topics 1, 3 and 9",
 	   title = "log-fold change (\u03b2)")
print(p13)
```

Club cells
----------

Club cells in droplet data:

```{r, fig.width=4.5, fig.height=4}
club_genes <- c("Nfia","Cbr2","Krt15","Cyp2f2","Lypd2","Scgb1a1")
p16 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"C",
                                           club_genes,
	 								 	   label_above_quantile = 0.995)
print(p16)
```

Club cells in pulse-seq data:

```{r, fig.width=4.5, fig.height=4}
p17 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"C",
                                           club_genes,
	 								 	   label_above_quantile = 0.995)
print(p17)
```

Club topic in droplet data:

```{r}
p18 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_merge_droplet,
                        "C","k5+k7",club_genes,label_above_score = Inf)
print(p18)
```

Club topic in pulse-seq data:

```{r}
p19 <- beta_scatterplot(diff_count_clusters_pulseseq,diff_count_merge_pulseseq,
                        "C","k4+k5+k6+k8+k10",club_genes,
                        label_above_score = Inf)
print(p19)
```

Proliferating cells
-------------------

Proliferating cells in pulse-seq data:

```{r, fig.width=4.5, fig.height=4}
cell_cycle_genes <- c("Cdk1","Ube2c","Top2a")
p14 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"P",
                                           cell_cycle_genes,
	 								 	   label_above_quantile = 0.998)
print(p14)
```

```{r, fig.width=4, fig.height=3}
qt_random_tie <- function (x) {
  y = rank(x,ties.method = "random")
  return(qqnorm(y,plot.it = FALSE)$x)
}
rows <- with(samples_pulseseq,which(cluster == "B" | cluster == "P"))
fit2 <- select(poisson2multinom(fit_pulseseq),loadings = rows)
X <- counts_pulseseq[rows,cell_cycle_genes]
Y <- apply(X,2,qt_random_tie)
rownames(Y) <- rownames(X)
score <- rowSums(Y)
pdat <- as.data.frame(prcomp(fit2$L)$x)
pdat <- cbind(pdat,data.frame(score = score))
p15 <- ggplot(pdat,aes(x = PC5,y = PC6,fill = score)) +
  geom_point(shape = 21,color = "white",size = 1.25) +
  scale_fill_gradientn(colors = c("darkblue","royalblue",
                       "lightskyblue","darkorange","firebrick")) +
  theme_cowplot(font_size = 10)
```

Hillock cells
-------------

Hillock cells in droplet data:

```{r, fig.width=4.5, fig.height=4}
hillock_genes <- c("Anxa1","Cldn3","Ecm1","Krt13","Krt4","Lgals3","S100a11")
p14 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"H",
                                           hillock_genes,
	 								 	   label_above_quantile = 0.998)
print(p14)
```

Compare Hillock topic ($k = 4$) against cluster in droplet data:

```{r, fig.width=4, fig.height=3}
p15 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                        "H","k4",hillock_genes,label_above_score = Inf) +
  labs(x = "cluster H",y = "topic 4",title = "log-fold change (\u03b2)")
print(p15)
p15 <- zscores_scatterplot(diff_count_clusters_droplet$Z[,"H"],
                           diff_count_droplet$Z[,4],
 						   diff_count_droplet$colmeans,
 						   colnames(counts_droplet),
 						   label_above_score = 100,
						   zmax = 400) +
  labs(x = "cluster H",y = "topic 4",title = "z-scores")
```

TO DO: Hillock topic ($k = 1$) in pulse-seq data:

```{r}
```

