---
title: Visualize and interpret topics in PBMC data
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we examine and compare the topic modeling results for the two
closely related data sets from [Zheng *et al* (2017)][zheng-2017], the
mixture of FACS-purified PBMC data sets, and the "unsorted" 68k PBMC
data. The goal of this analysis is to illustrate how the topic models
fitted to these data sets can be used to learn about structure in the
data. In particular, we would like to identify clusters, and interpret
clusters and topics as "cell types" or "gene expression programs".

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, as well as additional
functions that will be used to generate some of the plots.

```{r load-pkgs, message=FALSE}
library(dplyr)
library(fastTopics)
library(ggplot2)
library(cowplot)
source("../code/plots.R")
```

Mixture of FACS-purified PBMC data
----------------------------------

We begin with the mixture of FACS-purified PBMC data.

```{r load-purified-data}
load("../data/pbmc_purified.RData")
samples_purified <- samples
rm(samples,genes)
```

Load the $k = 6$ Poisson NMF model fit.

```{r load-purified-fit}
fit_purified <-
  readRDS("../output/pbmc-purified/rds/fit-pbmc-purified-scd-ex-k=6.rds")$fit
```

Here, we explore the structure of the single-cell data as inferred by
the topic model. Specifically, we use PCA to uncover structure in the
estimated topic proportions of the multinomial topic model. Although
PCA is simple, we will see that it works well, both for visualization
and identifying clusters, and avoids the complications of the popular
*t*-SNE and UMAP nonlinear dimensionality reduction methods. (Note
that, since the topic proportions sum to 1, there are only 5 PCs to
examine, not 6.)

```{r pbmc-purified-clustering-1}
fit <- poisson2multinom(fit_purified)
pca <- prcomp(fit$L)$x
```

Three large clusters are evident from first two PCs.  We label the
three large clusters as "A", "B" and "C".

Since there are so many samples, the scatterplot suffers from
"overplotting". So it also helpful to view this PC projection as a
density plot.

```{r pbmc-purified-clustering-2, fig.width=8, fig.height=3}
n   <- nrow(pca)
x   <- rep("C",n)
pc1 <- pca[,"PC1"]
pc2 <- pca[,"PC2"]
x[pc1 + 0.2 > pc2] <- "A"
x[pc2 > 0.25] <- "B"
x[(pc1 + 0.4)^2 + (pc2 + 0.1)^2 < 0.07] <- "C"
samples_purified$cluster <- x
p1 <- pca_plot_with_labels(fit_purified,c("PC1","PC2"),
                           samples_purified$cluster) +
      labs(fill = "cluster")
p2 <- pca_hex_plot(fit_purified,c("PC1","PC2"))
plot_grid(p1,p2,rel_widths = c(9,10))
```

Most of the samples are in cluster A:

```{r pbmc-purified-clustering-3}
table(x)
```

A small number of outlying data points do not seem to belong to any
the three clusters, or they fall in between the clusters. For these
data points, we assign them (rather arbitrarily) to one of the three
clusters.

From these plots, there also also appears to be finer scale structure.
For example, judging by the density plot, cluster A appears to split
into two subclusters. We will examine this finer scale structure
below.

Also note that other PCs beyond the first two may also sometimes
reveal additional clustering, and we will see examples of this in the
68k PBMC data.

Within cluster C, there are two mostly well-defined subclusters
(labeled "C1" and "C2"). There appear to be at least a couple other
smaller, less well-defined subclusters, but in this analysis we focus
on the largest, most obvious clusters.

```{r pbmc-purified-clustering-4, fig.width=8, fig.height=3}
rows <- which(samples_purified$cluster == "C")
fit  <- select(poisson2multinom(fit_purified),loadings = rows)
pca  <- prcomp(fit$L)$x
n    <- nrow(pca)
x    <- rep("C3",n)
pc1  <- pca[,1]
pc2  <- pca[,2]
x[pc1 < 0 & pc2 < 0.4] <- "C1"
x[pc1 > 0.5 & pc2 < 0.15] <- "C2"
samples_purified[rows,"cluster"] <- x
p3 <- pca_plot_with_labels(fit,c("PC1","PC2"),x) +
      labs(fill = "cluster")
p4 <- pca_hex_plot(fit,c("PC1","PC2"),bins = c(0,1,5,10,100,Inf))
plot_grid(p3,p4,rel_widths = c(9,10))
```

The two subclusters, C1 and C2, account for most of the samples in
cluster C. We also define a third subset, C3---a "background
cluster"---containing all the samples that were not assigned to C1 or
C2.

```{r pbmc-purified-clustering-5}
table(x)
```

Now we turn to cluster A. Within this cluster, there is a large
subcluster, which we label as "A1"; the subset of samples that are not
assigned to this cluster are labeled "A2". (The A1 subcluster is much
less distinct than the other clusters we have seen so far, and may not
show up clearly in this scatterplot---it is more apparent from the
density plot.) Otherwise, there is no obvious additional clustering of
the samples within cluster A.

```{r pbmc-purified-clustering-6, fig.width=8, fig.height=3}
rows <- which(samples_purified$cluster == "A")
fit  <- select(poisson2multinom(fit_purified),loadings = rows)
pca  <- prcomp(fit$L)$x
n    <- nrow(fit$L)
x    <- rep("A2",n)
pc1  <- pca[,1]
pc2  <- pca[,2]
x[pc1 > 0.55 - pc2 | pc1 > 0.65] <- "A1"
samples_purified[rows,"cluster"] <- x
p5 <- pca_plot_with_labels(fit,c("PC1","PC2"),x) +
      labs(fill = "cluster")
p6 <- pca_hex_plot(fit,c("PC1","PC2"))
plot_grid(p5,p6,rel_widths = c(9,10))
```

In summary, we have subdivided the data into 6 subsets:

```{r pbmc-purified-clustering-7}
samples_purified$cluster <- factor(samples_purified$cluster)
table(samples_purified$cluster)
```

We also inspected principal components individually in each of these 6
clusters and we did not find any of clear-cut examples of subclustering
withing these clusters.

The structure plot summarizes the topic proportions in each of these 6
subsets:

```{r pbmc-purified-structure-plot, fig.width=7, fig.height=1.5}
set.seed(1)
pbmc_purified_topic_colors <- c("gold","forestgreen","dodgerblue",
                                "gray","greenyellow","magenta")
pbmc_purified_topics <- c(2,5,3,1,4,6)
rows <- sort(c(sample(which(samples_purified$cluster == "A1"),250),
               sample(which(samples_purified$cluster == "A2"),1200),
               sample(which(samples_purified$cluster == "B"),250),
               sample(which(samples_purified$cluster == "C1"),250),
               sample(which(samples_purified$cluster == "C2"),200),
               sample(which(samples_purified$cluster == "C3"),200)))
p7 <- structure_plot(select(poisson2multinom(fit_purified),loadings = rows),
                     grouping = samples_purified[rows,"cluster"],
                     topics = pbmc_purified_topics,
                     colors = pbmc_purified_topic_colors[pbmc_purified_topics],
                     n = Inf,perplexity = c(70,100,70,70,50,50),
					 gap = 40,num_threads = 4,verbose = FALSE)
print(p7)
```

Out of the 6 topics, 4 of them ($k = 2, 3, 4, 5$) align closely with
clusters (labeled A1, B, C1, C2). And, indeed, they align closely with
their inclusion in the individual FACS-purified data sets:

```{r pbmc-purified-clusters-vs-celltypes}
with(samples_purified,table(celltype,cluster))
```

Based on the above results, we make a few observations:

+ Because of their close correspondence, subsequent analysis of topics
  2, 3, 4 and 5 should yield similar results to analyzing the clusters
  A1, B, C1, C2. For example, cluster B corresponds almost exactly to
  the B-cell data set. The largest cluster, cluster A2, is mostly
  comprised of T-cells.

+ Cluster A2---see also the PCA plot above---is an example where
  analyzing the most prevalent topics ($k = 1, 6$) will yield
  different results than a cluster-based analysis.

+ Many samples labeled as "CD34+" are not assigned to the CD34+
  cluster (C1). This reflects the fact that this population was much
  less pure (45%) than the others, so we would expect some cells
  labeled as "CD34+" to not necessarily be CD34+ cells.

+ Cluster C3 is a heterogeneous cluster with a relatively small number
  of samples (790) that could potentially contain additional clusters
  of biological relevance, but will likely be more challenging to
  analyze and interpret than the other clusters, so we do not
  investigate this cluster further.

In summary, a cluster-based analysis and topic-based analysis should
yield mostly similar results, except for the analysis of cluster A2,
which should benefit from a topic-based analysis.

Unsorted 68k PBMC data
----------------------

Next, we turn to the 68k data set. One feature of this data set is
that it is not biased by the FACS purification, so we expect to
observe a greater variety---or more continuous range---of cells
states.

```{r load-68k-data}
load("../data/pbmc_68k.RData")
samples_68k <- samples
rm(samples,genes,counts)
```

Load the $k = 6$ Poisson NMF model fit, and compute PCs from the topic
proportions.

```{r load-68k-fit}
fit_68k <- readRDS("../output/pbmc-68k/rds/fit-pbmc-68k-scd-ex-k=6.rds")$fit
fit <- poisson2multinom(fit_68k)
pca <- prcomp(fit$L)$x
```

From the $k = 6$ fit, we find least three distinct clusters in the
projection onto PCs 3 and 4. We label these clusters "A", "B" and "C",
as above, while cautioning that this labeling does not imply a
connection with the purified PBMC clusters above.

```{r pbmc-68k-clustering-1, fig.width=8, fig.height=3}
n   <- nrow(pca)
x   <- rep("A",n)
pc3 <- pca[,"PC3"]
pc4 <- pca[,"PC4"]
x[pc4 < -0.12 | pc3/1.9 - 0.17 > pc4] <- "B"
x[pc4 < -0.75] <- "C"
samples_68k$cluster <- x
p8 <- pca_plot_with_labels(fit_68k,c("PC3","PC4"),x) +
      labs(fill = "cluster")
p9 <- pca_hex_plot(fit_68k,c("PC3","PC4"))
plot_grid(p8,p9,rel_widths = c(9,10))
```

The vast majority of the cells are in cluster A.

```{r pbmc-68k-clustering-2}
table(samples_68k$cluster)
```

The wide range in the sizes of these clusters is striking; the
smallest cluster (C) is less than 1% the size of the largest (A). By
contrast, community detection methods such as the Louvain algorithm
are biased toward more uniformly sized clusters (this is a known
limitation of community detection methods).

Examine the top two PCs in cluster B, we identify two large clusters,
with the remaining assigned to a "background cluster", B3.

```{r pbmc-68k-clustering-3, fig.width=8, fig.height=3}
rows <- which(samples_68k$cluster == "B")
fit  <- select(poisson2multinom(fit_68k),loadings = rows)
pca  <- prcomp(fit$L)$x
n    <- nrow(pca)
x    <- rep("B3",n)
pc1  <- pca[,"PC1"]
x[pc1 > -0.12] <- "B1"
x[pc1 < -0.3]  <- "B2"
samples_68k[rows,"cluster"] <- x
p10 <- pca_plot_with_labels(fit,c("PC1","PC2"),x) +
       labs(fill = "cluster")
p11 <- pca_hex_plot(fit,c("PC1","PC2"),bins = c(0,1,5,10,20,Inf))
plot_grid(p10,p11)
```

Cluster A subdivides into two large clusters, labeled as A1 and
A2. The remaining samples are assigned to a subset labeled "A3".

```{r pbmc-68k-clustering-4, fig.width=8, fig.height=3}
rows <- which(samples_68k$cluster == "A")
fit  <- select(poisson2multinom(fit_68k),loadings = rows)
pca  <- prcomp(fit$L)$x
n    <- nrow(pca)
x    <- rep("A3",n)
pc2  <- pca[,"PC2"]
pc3  <- pca[,"PC3"]
x[2.5*pc3 < 0.4 - pc2] <- "A1"
x[pc3 > 0.75 - pc2] <- "A2"
samples_68k[rows,"cluster"] <- x
p12 <- pca_plot_with_labels(fit,c("PC2","PC3"),x) +
       labs(fill = "cluster")
p13 <- pca_hex_plot(fit,c("PC2","PC3"),bins = c(0,1,5,10,100,Inf))
plot_grid(p12,p13)
```

Within cluster A, the vast majority of the samples are assigned to the
A1 subcluster:

```{r pbmc-68k-clustering-5}
table(x)
```

Although we do not find additional clusters within the large A1
subset, the continuous structure is nonetheless quite interesting, and
worth investigating. 

```{r pbmc-68k-clustering-6, fig.width=7.5, fig.height=3}
rows <- which(samples_68k$cluster == "A1")
fit  <- select(poisson2multinom(fit_68k),loadings = rows)
p14 <- pca_plot(fit,k = 3:4)
print(p14)
```

From these two plots, we observe that topics 3 and 4 exist on a
continuous spectrum, but that *mixtures of topics 3 and 4 are
relatively rare.* This is particularly evident from a density plot:

```{r pbmc-68k-clustering-7, fig.width=4, fig.height=3}
p15 <- pca_hex_plot(fit,c("PC1","PC2"),bins = c(0,1,10,20,100,Inf))
print(p15)
```

Topic 3 characterizes natural killer cells, and topic 4 has yet to be
characterized, but may represent some subset of naive T-cells. This is
an example with interesting substructure that can't be captured by
clusters. This is also an example of substructure that is much better
captured by linear representations such as PCA as opposed to nonlinear
dimensionality reduction methods such as *t*-SNP and UMAP.

In summary, we have subdivided these data into 7 subsets:

```{r pbmc-68k-clustering-8}
samples_68k$cluster <- factor(samples_68k$cluster)
table(samples_68k$cluster)
```

Again, the wide range in cluster sizes is striking.

The structure plot summarizes the topic proportions in each of these 7
subsets:

```{r pbmc-68k-structure-plot, fig.width=7.5, fig.height=1.5}
set.seed(1)
pbmc_68k_topic_colors <- c("yellow","lightskyblue","salmon",
                           "firebrick","royalblue","olivedrab")
pbmc_68k_topics <- c(2,5,1,3,4,6)
rows <- sort(c(sample(which(samples_68k$cluster == "A1"),1200),
               sample(which(samples_68k$cluster == "A2"),500),
               sample(which(samples_68k$cluster == "A3"),300),
               sample(which(samples_68k$cluster == "B1"),500),
               sample(which(samples_68k$cluster == "B2"),300),
               which(samples_68k$cluster == "B3"),
               which(samples_68k$cluster == "C")))
p16 <- structure_plot(select(poisson2multinom(fit_68k),loadings = rows),
                      grouping = samples_68k[rows,"cluster"],
                      topics = pbmc_68k_topics,
   	 			      colors = pbmc_68k_topic_colors[pbmc_68k_topics],
                      perplexity = c(160,100,70,80,50,50,50),
                      n = Inf,gap = 40,num_threads = 4,verbose = FALSE)
print(p16)
```

These subsets do not align as closely with the cell-type labeling
inferred by Zheng *et al* (2017). This is not surprising considering
that the Zheng *et al* labeling is based on the FACS-purified data set.

```{r pbmc-68k-clusters-vs-celltypes}
with(samples_68k,table(celltype,cluster))
```

A few more notes about these results:

+ As in the purified fit, here we identify a B-cells cluster (B) and
  topic (5) that closely match. Other close matches include CD34+
  cells (C), CD14+ monocytes (B1) and dendritic cells (B2).

+ As we found above, unlike the FACS-purified data, we don't identify
  a clear-cut cluster for NK cells from the 68k data; the NK cells are
  mixed in with the T-cells (subset A1). NK cells will therefore only
  emerge only after subsequent analysis of topic 3.

+ The small number (~1%) in the A3 and B3 subsets aren't distinct
  subpopulations *per se* but rather appear to be in some intermediate
  developmental state, and the mixture of topics in these subsets
  captures this.

In summary, the topics and clusters seem to offer very much
complementary biological insights, although subsequent analysis is
needed to determine what these insights are.

[zheng-2017]: https://doi.org/10.1038/ncomms14049
