---
title: Visualize and interpret droplet and pulse-seq topics
author: Peter Carbonetto
output: workflowr::wflow_html
---

Here we closely examine the topic modeling results for the two
epithelial airway data sets (droplet and pulse-seq), and investigate
the benefits of modeling single cells as *mixtures* of gene expression
programs in these data sets. In particular, we will compare and
contrast differential expression in clusters vs. topics.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load the packages used in the analysis below, as well as additional
functions that we will use to generate some of the plots.

```{r load-pkgs, message=FALSE}
library(Matrix)
library(dplyr)
library(fastTopics)
library(ggplot2)
library(ggrepel)
library(cowplot)
source("../code/plots.R")
```

Droplet data
------------

Load the smaller "droplet" data set, the $k = 7$ Poisson NMF model fit
for these data, the 8 clusters identified in the
[clustering analysis](clusters_droplet.html), and the results of the
differential expression analysis.

```{r load-droplet-data}
load("../data/droplet.RData")
load("../output/droplet/diff-count-droplet.RData")
counts_droplet <- counts
samples_droplet <- readRDS("../output/droplet/clustering-droplet.rds")
fit_droplet <- readRDS("../output/droplet/rds/fit-droplet-scd-ex-k=7.rds")$fit
diff_count_droplet <- diff_count_topics
diff_count_clusters_droplet <- diff_count_clusters
diff_count_club_droplet <- diff_count_club
diff_count_basal_droplet <- diff_count_basal
rm(samples,counts)
rm(diff_count_topics,diff_count_clusters,diff_count_club,diff_count_basal)
```

For reference, we show here the Structure plot from the
[clustering analysis of the droplet data](clusters_droplet.html). This
Structure plot summarizes the topic proportions in each of the 8
subsets (including the background cluster).

![](figure/clusters_droplet.Rmd/structure-plot-1.png)

*Add text here.*

![](figure/clusters_droplet.Rmd/clustering-8-1.png)

*Add text here.*

```{r pca-plots-with-expression, fig.width=8, fig.height=3}
pp1 <- pca_plot(poisson2multinom(fit_droplet),pcs = 1:2,
                fill = log10(counts_droplet[,"Cdhr3"])) +
       labs(title = "Cdhr3 (ciliated)",fill = "log10(count)")
pp2 <- pca_plot(poisson2multinom(fit_droplet),pcs = 1:2,
                fill = log10(counts_droplet[,"Krt5"])) +
       labs(title = "Krt5 (basal)",fill = "log10(count)")
pp3 <- pca_plot(poisson2multinom(fit_droplet),pcs = 1:2,
                fill = log10(counts_droplet[,"Scgb1a1"])) +
       labs(title = "Scgb1a1 (club)",fill = "log10(count)")
pp4 <- pca_plot(poisson2multinom(fit_droplet),pcs = 1:2,
                fill = log10(counts_droplet[,"Chga"])) +
       labs(title = "Chga (neuroendocrine)",fill = "log10(count)")
pp5 <- pca_plot(poisson2multinom(fit_droplet),pcs = 1:2,
                fill = log10(counts_droplet[,"Gnat3"])) +
       labs(title = "Gnat3 (tuft)",fill = "log10(count)")
pp6 <- pca_plot(poisson2multinom(fit_droplet),pcs = 3:4,
                fill = log10(counts_droplet[,"Krt13"])) +
       labs(title = "Krt13 (hillock)",fill = "log10(count)")
plot_grid(pp1,pp2,pp3,pp4,pp5,pp6,nrow = 2,ncol = 3)
```

Pulse-seq data
--------------

Next, we load the larger "pulse-seq" data set, the $k = 11$ Poisson
NMF model fit for these data, and the 7 clusters identified in the
[clustering analysis](clusters_pulseseq.html).

```{r load-pulseseq-data}
load("../data/pulseseq.RData")
load("../output/pulseseq/diff-count-pulseseq.RData")
counts_pulseseq  <- counts
samples_pulseseq <- readRDS("../output/pulseseq/clustering-pulseseq.rds")
fit_pulseseq <- readRDS("../output/pulseseq/rds/fit-pulseseq-scd-ex-k=11.rds")$fit
diff_count_pulseseq <- diff_count_topics
diff_count_clusters_pulseseq <- diff_count_clusters
diff_count_hillock_pulseseq <- diff_count_hillock
diff_count_bc_pulseseq <- diff_count_bc
rm(samples,counts)
rm(diff_count_topics,diff_count_clusters,diff_count_hillock,diff_count_bc)
```

For reference, we show here the Structure plot from the
[clustering analysis of the pulse-seq data](clusters_pulseseq.html).
This Structure plot summarizes the topic proportions in each of the 7
subsets (including the background cluster):

![](figure/clusters_pulseseq.Rmd/structure-plot-1.png)

Ciliated cells
--------------

We begin with the cluster that captures ciliated cells. This cluster
is one of the most distinctive in both the droplet and pulse-seq data
sets. Ciliated cells are abundant, though not as much as basal and
club cells.

```{r volcano-plot-droplet-cil, fig.width=4.5, fig.height=4}
ciliated_genes <- c("Ccdc113","Ccdc153","Cdhr3","Foxj1","Lztfl1","Mlf1")
p1 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"Cil",
                                          ciliated_genes,
										  label_above_quantile = 0.998)
print(p1)
```

Marker genes and transcription factors identified in Montoro *et al*
(2018) are highlighted in black, and other top differentially
expressed genes are shown with gray labels.

We obtain similar top differentially expressed genes in the pulse-seq
data:

```{r volcano-plot-pulseseq-cil, fig.width=4.5, fig.height=4}
p2 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"Cil",
                                          ciliated_genes,
										  label_above_quantile = 0.998)
print(p2)
```

The topic for the ciliated cell-type ($k = 7$) corresponds very
closely to the "Cil" cluster in the pulse-seq data. In addition, gene
enrichments are considerably stronger in the topic for ciliated genes,
particularly so for some genes with lower expression levels.

```{r scatterplots-pulseseq-Cil-vs-k7, fig.width=8, fig.height=3}
p3 <- zscores_scatterplot(diff_count_clusters_pulseseq,
                          diff_count_pulseseq,"Cil","k7",ciliated_genes,
                          xlab = "cluster Cil",ylab = "topic 7")
p4 <- beta_scatterplot(diff_count_clusters_pulseseq,diff_count_pulseseq,
                       "Cil","k7",ciliated_genes,
                       xlab = "cluster Cil",ylab = "topic 7")
plot_grid(p3,p4)
```

Ionocytes
---------

In the pulse-seq data, we identify a distinctive cluster for the newly
discovered, rare "ionocyte" cell type. Gene expression in these cells
is not fully captured by any single topic, yet the mixture of topics
forms a distinctive cluster.

```{r volcano-plot-pulseseq-I, fig.width=4.4, fig.height=4}
ionocyte_genes <- c("Ascl3","Asgr1","Atp6v0d2","Atp6v1c2","Cftr","Foxi1",
                    "Moxd1","P2ry14","Stap1")
p5 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"I",
                                          ionocyte_genes,
										  label_above_quantile = 0.998)
print(p5)
```

We do not identify a topic or cluster for ionocytes in the droplet
data. Judging by expression of the *Foxi1* ionocyte marker gene, only
a handful of cells in the droplet data are ionocytes:

```{r pca-plot-droplet-Foxi1, fig.width=3.75, fig.height=3}
p6 <- pca_plot(poisson2multinom(fit_droplet),fill = counts_droplet[,"Foxi1"]) +
      labs(fill = "count")
print(p6)
```

Goblet cells
------------

Consistent with Montoro *et al* (2018), we identify a cluster of
Goblet cells in the droplet data.

```{r volcano-plot-droplet-G, fig.width=4.5, fig.height=4}
goblet_genes <- "Gp2"
p7 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"G",
                                          goblet_genes,
										  label_above_quantile = 0.998)
print(p7)
```

Topic $k = 1$ is unique to this cluster, suggesting that the this topic
characterizes the goblet cell type. Indeed, there is a very close
correspondence between the topic and cluster, with several
characteristic genes (e.g. *Gp2*) showing somewhat stronger enrichment
in the topic.

```{r scatterplots-droplet-G-vs-k1, fig.width=8, fig.height=3}
p8 <- zscores_scatterplot(diff_count_clusters_droplet,
                          diff_count_droplet,"G","k1",goblet_genes,
						  label_above_score = 200,
						  xlab = "cluster G",ylab = "topic 1")
p9 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                       "G","k1",goblet_genes,xlab = "cluster G",
                       ylab = "topic 1")
plot_grid(p8,p9)
```

Tuft and pulmonary neuroendocrine cells
---------------------------------------

In both data sets, we identify clusters for tuft and pulmonary
neuroendocrine cells. The fitted topic models do not distinguish
between these two rare cell types; we identify these cell types as a
single cluster.

```{r volcano-plot-droplet-T+N, fig.width=4.5, fig.height=4}
tuft_neuroendocrine_genes <- c("Ascl1","Ascl2","Ascl3","Chga","Dclk1",
                               "Gnat3","Rgs13")
p8 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"T+N",
                                          tuft_neuroendocrine_genes,
										  label_above_quantile = 0.998)
print(p8)
```

Here is the volcano plot from the pulse-seq data:

```{r volcano-plot-pulseseq-T+N, fig.width=4.5, fig.height=4}
p9 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"T+N",
                                          tuft_neuroendocrine_genes,
										  label_above_quantile = 0.998)
print(p9)
```

TO DO: Add volcano plots showing that droplet and pulse-seq topics
within "T+N" cluster distinguish tuft and neuroendocrine cells.

Basal cells
-----------

We now move on to the large majority of cells in each of the
epithelial airway data sets (over 90% in droplet and over 92% in
pulseseq) that do not break down into distinct clusters.

Although we do not obtain a distinct basal cells cluster, attempting
to form a cluster does indeed reasonably distinguish basal cells:

```{r volcano-plot-droplet-basal, fig.width=4.5, fig.height=4}
basal_genes <- c("Aqp3","Krt5","Dapl1","Hspa1a","Trp63")
p10 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"B",
                                           c(basal_genes,
                                             tuft_neuroendocrine_genes),
	 								 	   label_above_quantile = 0.995)
print(p10)
```

Comparing the basal cells topic ($k = 2$) against the basal cells
cluster in the droplet data, there is a close correspondence between
the two, with the topic showing much stronger enrichment of
characteristic basal genes:

```{r scatterplots-droplet-B-vs-k2, fig.width=8, fig.height=3}
p11 <- zscores_scatterplot(diff_count_clusters_droplet,
                           diff_count_basal_droplet,"B","k2",
                           basal_genes,zmax = 400,
						   xlab = "cluster B",ylab = "topic 2")
p12 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_basal_droplet,
                        "B","k2",basal_genes,zmin = 2,
                        xlab = "cluster B",y = "topic 2")
plot_grid(p11,p12)
```

We obtain similar results with a cluster identified in the pulse-seq
data,

```{r volcano-plot-pulseseq-basal, fig.width=4.5, fig.height=4}
p13 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"B",
                                           basal_genes,
	 								 	   label_above_quantile = 0.995)
print(p13)
```

and we achieve much stronger enrichment of basal genes in the combined
basal topics ($k = 1, 3, 9$) when compared to the basal cluster in the
pulse-seq data:

```{r scatterplots-pulseseq-B-vs-k1+3+9, fig.width=8, fig.height=3}
p14 <- zscores_scatterplot(diff_count_clusters_pulseseq,
                           diff_count_bc_pulseseq,"B","k1+k3+k9",
                           basal_genes,zmax = 2000,
						   xlab = "cluster B",ylab = "topic 2")
p15 <- beta_scatterplot(diff_count_clusters_pulseseq,
                        diff_count_bc_pulseseq,
                        "B","k1+k3+k9",basal_genes,
						xlab = "cluster B",ylab = "topics 1, 3, 9")
plot_grid(p14,p15)
```

Proliferating cells
-------------------

There is an interesting subset of basal cells, uniquely in the
pulse-seq set set, that isn't captured well by any single topic, but it
stands out in PCA plots of the topic proportions. Forming a cluster
from this subset, we obtain strong enrichment of cell-cycle genes:

```{r volcano-plot-pulseseq-proliferating, fig.width=4.5, fig.height=4}
cell_cycle_genes <- c("Cdk1","Ube2c","Top2a")
p16 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"P",
                                           cell_cycle_genes,
	 								 	   label_above_quantile = 0.998)
print(p16)
```

Indeed, if we formulate a simple a "cell-cycle score", we see that
the signal is quite visible in PCs 5 and 6 of the topic proportions:

```{r pca-cell-cycle-pulseseq-proliferating, fig.width=3.75, fig.height=3}
rows <- with(samples_pulseseq,which(cluster == "B" | cluster == "P"))
fit2 <- select(poisson2multinom(fit_pulseseq),loadings = rows)
p17  <- cellcycle_pca_plot(fit2,counts_pulseseq[rows,],pcs = 5:6)
print(p17)
```

Hillock cells
-------------

The Montoro *et al* (2018) paper identifies a rare transitional cell
though a diffusion maps analysis of the droplet data. They call these
rare transition cells uniquely expressing *Krt13* as "Hillock
cells". Indeed, we do not identify a distinct cluster for Hillock
celels in the droplet data, but we nonetheless formed a small ($n =
197$) cluster in the clustering analysis that shows strong enrichment
of the top hillock genes (e.g., *Krt4*, *Krt13*):

```{r volcano-plot-droplet-hillock, fig.width=4.5, fig.height=4}
hillock_genes <- c("Anxa1","Cldn3","Ecm1","Krt13","Krt4","Lgals3","S100a11")
p18 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"H",
                                           hillock_genes,
	 								 	   label_above_quantile = 0.998)
print(p18)
```

Given the transitional nature of these cells, one would expect that
mixtures of gene programs would better characterize the continuous
variation in the hillock-specific gene program, and indeed topic 4,
which is the greatest contributor to this cluster of hillock cells,
picks up a more distinct enrichment signal for hillock genes:

```{r scatterplots-droplet-H-vs-k4, fig.width=8, fig.height=3}
p19 <- zscores_scatterplot(diff_count_clusters_droplet,
                           diff_count_droplet,"H","k4",
                           hillock_genes,zmax = 400,
						   xlab = "cluster H",ylab = "topic 4")
p20 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_droplet,
                        "H","k4",hillock_genes,
                        xlab = "cluster H",ylab = "topic 4")
plot_grid(p19,p20)
```

For example, we estimate roughly a 68-fold enrichment of *Krt13*
expression in the hillock cluster, and over *1,000*-fold enrichment in
the hillock topic (essentially because there is no expression of
*Krt13* outside this topic).

The presence of hillock cells in the pulse-seq data is more subtle,
but it is nonetheless well-captured by a single topic, $k = 1$:

```{r volcano-plot-pulseseq-hillock, fig.width=4.5, fig.height=4}
p21 <- volcano_plot_with_highlighted_genes(diff_count_hillock_pulseseq,"k1",
                                           hillock_genes,
	 								 	   label_above_quantile = 0.995)
print(p21)
```

Club cells
----------

The remaining cluster in the droplet captures club cells:

```{r volcano-plot-droplet-club, fig.width=4.5, fig.height=4}
club_genes <- c("Bpifa1","Cbr2","Cyp2a5","Cyp2f2","Krt15",
                "Lypd2","Muc5b","Nfia","Scgb1a1","Scgb3a2")
p22 <- volcano_plot_with_highlighted_genes(diff_count_clusters_droplet,"C",
                                           club_genes,
	 								 	   label_above_quantile = 0.995)
print(p22)
```

And likewise in the pulse-seq data:

```{r volcano-plot-pulseseq-club, fig.width=4.5, fig.height=4}
p23 <- volcano_plot_with_highlighted_genes(diff_count_clusters_pulseseq,"C",
                                           club_genes,
	 								 	   label_above_quantile = 0.995)
print(p23)
```

Both the droplet and pulse-seq clusters exhibit a high level of
heterogeneity in the topic proportions; in particular, 2 topics are
unique to club cluster in the droplet data, and 5 topics in the
pulse-seq club cluster. This heterogeneity will require further
exploration; for now we focus on the combined topic proportions
capturing the club cell-type-specific gene program. In the droplet
data, compare cluster C to the combined topic $k = 5 + 7$,

```{r scatterplots-club-droplet-C-vs-k5+k7, fig.width=8, fig.height=3}
p24 <- zscores_scatterplot(diff_count_clusters_droplet,
                           diff_count_club_droplet,"C","k5+k7",club_genes,
                           zmax = 500,xlab = "cluster C",ylab = "topics 5 + 7")
p25 <- beta_scatterplot(diff_count_clusters_droplet,diff_count_club_droplet,
                        "C","k5+k7",club_genes,
                        xlab = "cluster C",ylab = "topics 5 + 7")
plot_grid(p24,p25)
```

and in the pulse-seq data compare cluster C to the combined topic $k
= 4 + 5 + 6 + 8 + 10$,

```{r scatterplots-club-pulseseq-C-vs-k4+5+6+8+9+10, fig.width=8, fig.height=3}
p26 <- zscores_scatterplot(diff_count_clusters_pulseseq,
                           diff_count_bc_pulseseq,"C","k4+k5+k6+k8+k10",
                           club_genes,zmax = 1000,
                           xlab = "cluster C",
						   ylab = "topics 4-6, 8, 10")
p27 <- beta_scatterplot(diff_count_clusters_pulseseq,diff_count_bc_pulseseq,
                        "C","k4+k5+k6+k8+k10",club_genes,
                        xlab = "cluster C",ylab = "topics 4-6, 8, 10")
plot_grid(p26,p27)
```

In both data sets, we observe a much stronger enrichment of
characteristic club genes in the combined topics.
