---
title: Prepare mixture of FACS-purified PBMCs for topic modeling analysis
author: Peter Carbonetto
output: workflowr::wflow_html
---

These are the steps taken to prepare the mixture of FACS-purified PBMC
data for topic modeling analysis. These data are the reference
transcriptome profiles generated via single-cell RNA-seq of 10
bead-enriched subpopulations described in [Zheng *et al*
(2017)][zheng-2017].

*Note:* In Zheng *et al* (2017), the clustering analysis suggests that
the subpopulations are mostly "homogenous", with a couple exceptions,
the CD14+ and CD34+ subpopulations. For the CD14+ monocytes, two
clusters were observed, and they were identified as "CD14+ monocytes"
and "dendritic cells". See Supplementary Figures 6 and 7 in that
paper.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center")
```

I downloaded the "Gene/cell matrix (filtered)" tar.gz file from the
[10x Genomics website][10x-data] for each of the following 10 data
sets: CD14+ Monocytes, CD19+ B Cells, CD34+ Cells, CD4+ Helper T
Cells, CD4+/CD25+ Regulatory T Cells, CD4+/CD45RA+/CD25- Naive T
cells, CD4+/CD45RO+ Memory T Cells, CD56+ Natural Killer Cells, CD8+
Cytotoxic T cells and CD8+/CD45RA+ Naive Cytotoxic T Cells. I unpacked
the individual tar archives and re-organized the files into 10
subdirectories, each containing three (compressed) files,
`barcodes.tsv.gz`, `genes.tsv.gz` and `matrix.mtx.gz`. The "datasets"
variable lists the names of the subdirectories I used, along with the
"cell type" labels that Zheng *et al* (2017) assigned to each of the
10 data sets:

```{r data-sets}
datasets <- list("CD19+ B"                      = "b_cells",
                 "CD14+ Monocyte"               = "cd14_monocytes",
                 "CD34+"                        = "cd34_filtered",
                 "CD4+ T Helper2"               = "cd4_t_helper",
                 "CD56+ NK"                     = "cd56_nk",
                 "CD8+ Cytotoxic T"             = "cytotoxic_t",
                 "CD4+/CD45RO+ Memory"          = "memory_t",
                 "CD8+/CD45RA+ Naive Cytotoxic" = "naive_cytotoxic",
                 "CD4+/CD45RA+/CD25- Naive T"   = "naive_t",
                 "CD4+/CD25 T Reg"              = "regulatory_t")
```

Load the packages used in the analysis below, and some functions
written specifically for loading and preparing these data.

```{r load-pkgs}
library(tools)
library(Matrix)
library(readr)
source("../code/purified_pbmc.R")
```

Call `read_purified_pbmc_data`, which imports the individual data sets
into R, and combines them:

```{r load-data}
out     <- read_purified_pbmc_data("../data",datasets)
samples <- out$samples
genes   <- out$genes
counts  <- out$counts
rm(out)
```

Remove genes that are not expressed in any of the cells.

```{r filter-genes}
j      <- which(colSums(counts > 0) >= 1)
genes  <- genes[j,]
counts <- counts[,j]
```

After filtering out genes with no expression, the count data are
stored in a 94,655 x 21,952 sparse matrix, with rows corresponding to
samples and columns corresponding to genes. This matrix is very
sparse---less than 3% of the counts are greater than zero:

```{r summarize-data-1}
cat(sprintf("Number of samples: %d\n",nrow(counts)))
cat(sprintf("Number of genes: %d\n",ncol(counts)))
cat(sprintf("Proportion of counts that are non-zero: %0.1f%%.\n",
            100*mean(counts > 0)))
```

This is the breakdown of samples by data set, or "cell type":

```{r summarize-data-2}
summary(samples["celltype"],maxsum = 10)
```

Save the processed data.

```{r save-data}
save(list = c("samples","genes","counts"),
     file = "pbmc_purified.RData")
resaveRdaFiles("pbmc_purified.RData")
```

[zheng-2017]: https://doi.org/10.1038/ncomms14049
[10x-data]: https://support.10xgenomics.com/single-cell-gene-expression/datasets
