---
title: Prepare 68k PBMC data for topic modeling analysis
author: Peter Carbonetto
output: workflowr::wflow_html
---

These are the steps taken to prepare the unsorted 68k PBMC data from
[Zheng *et al* (2017)][zheng-2017] for topic modeling analysis.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center")
```

Load the packages used in the analysis below, and some functions
written specifically for loading and preparing these data.

```{r load-pkgs}
library(tools)
library(Matrix)
library(readr)
source("../code/pbmc.R")
```

Before loading the data, download the "Gene/cell matrix (filtered)"
tar.gz file for the "Fresh 68k PBMCs (Donor A) data set from
[10x Genomics website][10x-data], download
`68k_pbmc_barcodes_annotation.tsv` from the
[companion repository on GitHub][zheng-2017-github], and then save and
compress these files to the `data/pbmc_68k` directory.

```{r load-data, message=FALSE}
samples <- read_barcodes_with_annotations(file.path("../data/pbmc_68k",
             "68k_pbmc_barcodes_annotation.tsv.gz"))
genes   <- read_genes("../data/pbmc_68k/genes.tsv.gz")
counts  <- create_counts_matrix("../data/pbmc_68k/matrix.mtx.gz",
                                sample.names = samples$barcode,
								gene.names = genes$ensembl)
```

Remove genes that are not expressed in any of the cells.

```{r filter-genes}
j      <- which(colSums(counts > 0) >= 1)
genes  <- genes[j,]
counts <- counts[,j]
```

After filtering out genes with no expression, the count data are
stored in a 68,579 x 20,387 sparse matrix, with rows corresponding to
samples, and columns corresponding to genes. This matrix is very
sparse---less than 3% of the counts are greater than zero:

```{r summarize-data-1}
cat(sprintf("Number of samples: %d\n",nrow(counts)))
cat(sprintf("Number of genes: %d\n",ncol(counts)))
cat(sprintf("Proportion of counts that are non-zero: %0.1f%%.\n",
            100*mean(counts > 0)))
```

Save the processed data.

```{r save-data}
save(list = c("samples","genes","counts"),
     file = "pbmc_68k.RData")
resaveRdaFiles("pbmc_68k.RData")
```

[zheng-2017]: https://doi.org/10.1038/ncomms14049
[zheng-2017-github]: https://github.com/10XGenomics/single-cell-3prime-paper
[10x-data]: https://support.10xgenomics.com/single-cell-gene-expression/datasets
