---
title: "Topic modeling vs. clustering of gene expression data: an illustration"
author: Peter Carbonetto
output: workflowr::wflow_html
---

Clustering methods are widely used to discover interesting
substructure in bulk or single-cell RNA sequencing ("RNA-seq")
data. An alternative is to use a topic model, which explains each
sample (an expression profile) as a *mixture* of "gene programs" that
are estimated from the data. These gene expression programs are the
topics.

In this short vignette, we show that a topic model picks up a very
different kind of substructure that cannot be identified by (hard)
clustering methods.

```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE,results = "hold",
                      fig.align = "center",dpi = 120)
```

Load packages used in the analysis below, as well as some functions
written for this analysis.

```{r load-pkgs, message=FALSE}
library(ggplot2)
library(cowplot)
library(Ternary)
library(Rtsne)
library(fastTopics)
source("../code/functions_for_topics_vs_clusters.R")
```

Simulate a "toy" gene expression data set from the multinomial topic
model. Each sample---a vector of 400 counts---is generated as a
mixture of three "gene programs" (these are the topics in the topic
model). In particular, samples are either mostly generated by a single
gene program or an approximately equal mixture of the three gene
programs.

```{r simulate-data}
set.seed(1)
X <- simulate_toy_gene_data(n = 400,m = 40,k = 3,s = 1000)$X
```

Topic modeling analysis
-----------------------

To illustrate a topic modeling analysis, we fit a multinomial topic
model to these data,

```{r fit-multinom-topic-model, message=FALSE}
fit <- fit_poisson_nmf(X,k = 3,numiter = 100,verbose = FALSE,
                       control = list(extrapolate = TRUE))
fit <- poisson2multinom(fit)
```

then we plot the estimated topic proportions.

```{r plot-topic-proportions, fig.height=3, fig.width=3}
par(mar = c(0,0,0,0))
create_ternary_plot(fit$L)
```

We observe in this plot that the estimated topic model recaptures the
simulation: the samples are estimated to be either primarily explained
by a single gene program or an approximately equal mixture of the
three gene programs.

Clustering analysis
-------------------

To illustrate clustering, we use the *t*-SNE algorithm to project the
samples onto a 2-d embedding.

```{r tsne-from-counts-1, fig.height=3, fig.width=3}
tsne1 <- Rtsne(X,2,pca = FALSE,normalize = FALSE)
plot_tsne(tsne1)
```

This 2-d projection suggests that there are four clusters in the
data. (In practice, one could automate the clustering with a community
detection algorithm or by applying $k$-means to the $t$-SNE
projection.)

This clustering fails to reveal any insight into the relationship
between these clusters---in particular, the distribution of genes in
the middle cluster is an equal mixture of the gene distributions in
the other clusters. To make this more clear, we label each sample by
the topic with the highest estimated weight:

```{r tsne-from-counts-2, fig.height=3, fig.width=3}
plot_tsne_with_topics(tsne1,fit$L)
```

The clustering will align closely with the topic modeling whenever the
samples can be largely explained by a single topic or gene program.
Outside this setting, topic modeling and clustering can diverge.

Inferring clusters from the topic model
---------------------------------------

Despite the limitation of the pure clustering, it nonetheless might be
useful to identify the four clusters exhibiting different gene
expression patterns.

Now use t-SNE in the same way to project the topic proportions onto a
2-d embedding.

```{r tsne-from-loadings, fig.height=3, fig.width=3}
tsne2 <- Rtsne(fit$L,2,pca = FALSE,normalize = FALSE)
plot_tsne_with_topics(tsne2,fit$L)
```

We can achieve something similar with PCA.

```{r pca-from-loadings, fig.height=3, fig.width=3}
pca <- prcomp(fit$L,center = TRUE,scale = FALSE)
plot_pca(pca,fit$L)
```
